<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Codebook | RNA-seq-analysis</title>
  <meta name="description" content="RNA-seq analysis workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Codebook | RNA-seq-analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bookdown.org/ggiaever/RNA-seq-analysis/img/football22.png" />
  <meta property="og:description" content="RNA-seq analysis workshop" />
  <meta name="github-repo" content="gurinina/RNA-seq-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Codebook | RNA-seq-analysis" />
  
  <meta name="twitter:description" content="RNA-seq analysis workshop" />
  <meta name="twitter:image" content="https://bookdown.org/ggiaever/RNA-seq-analysis/img/football22.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="functional-analysis-of-rnaseq-data.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet" />
<script src="libs/vis-9.1.0/vis-network.min.js"></script>
<script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="css/git.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>RNA-seq analysis workshop</a></li>
<li class="chapter" data-level="1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html"><i class="fa fa-check"></i><b>1</b> Differential gene expression (DGE) analysis overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#review-of-the-dataset"><i class="fa fa-check"></i><b>1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="1.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#setting-up"><i class="fa fa-check"></i><b>1.2</b> Setting up</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-libraries"><i class="fa fa-check"></i><b>1.2.1</b> Loading libraries</a></li>
<li class="chapter" data-level="1.2.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-data"><i class="fa fa-check"></i><b>1.2.2</b> Loading data</a></li>
<li class="chapter" data-level="1.2.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#viewing-data"><i class="fa fa-check"></i><b>1.2.3</b> Viewing data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-analysis-overview"><i class="fa fa-check"></i><b>1.3</b> Differential gene expression analysis overview</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#rna-seq-count-distribution"><i class="fa fa-check"></i><b>1.3.1</b> RNA-seq count distribution</a></li>
<li class="chapter" data-level="1.3.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#modeling-count-data"><i class="fa fa-check"></i><b>1.3.2</b> Modeling count data</a></li>
<li class="chapter" data-level="1.3.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates"><i class="fa fa-check"></i><b>1.3.3</b> Improving mean estimates (i.e.Â reducing variance) with biological replicates</a></li>
<li class="chapter" data-level="1.3.4" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>1.3.4</b> Differential expression analysis workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="count-normalization.html"><a href="count-normalization.html"><i class="fa fa-check"></i><b>2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="2.1" data-path="count-normalization.html"><a href="count-normalization.html#normalization"><i class="fa fa-check"></i><b>2.1</b> Normalization</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="count-normalization.html"><a href="count-normalization.html#common-normalization-methods"><i class="fa fa-check"></i><b>2.1.1</b> Common normalization methods</a></li>
<li class="chapter" data-level="2.1.2" data-path="count-normalization.html"><a href="count-normalization.html#rpkmfpkm-not-recommended"><i class="fa fa-check"></i><b>2.1.2</b> RPKM/FPKM (not recommended)</a></li>
<li class="chapter" data-level="2.1.3" data-path="count-normalization.html"><a href="count-normalization.html#deseq2-normalized-counts-median-of-ratios-method"><i class="fa fa-check"></i><b>2.1.3</b> DESeq2-normalized counts: Median of ratios method</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="count-normalization.html"><a href="count-normalization.html#geometric-mean"><i class="fa fa-check"></i><b>2.2</b> Geometric Mean</a></li>
<li class="chapter" data-level="2.3" data-path="count-normalization.html"><a href="count-normalization.html#count-normalization-of-mov10-dataset"><i class="fa fa-check"></i><b>2.3</b> Count normalization of Mov10 dataset</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="count-normalization.html"><a href="count-normalization.html#match-the-metadata-and-counts-data"><i class="fa fa-check"></i><b>2.3.1</b> 1. Match the metadata and counts data</a></li>
<li class="chapter" data-level="2.3.2" data-path="count-normalization.html"><a href="count-normalization.html#create-deseq2-object"><i class="fa fa-check"></i><b>2.3.2</b> 2. Create DESEq2 object</a></li>
<li class="chapter" data-level="2.3.3" data-path="count-normalization.html"><a href="count-normalization.html#generate-the-mov10-normalized-counts"><i class="fa fa-check"></i><b>2.3.3</b> 3. Generate the Mov10 normalized counts</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#sample-level-qc"><i class="fa fa-check"></i><b>3.1</b> Sample-level QC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quality-control.html"><a href="quality-control.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.1.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.1.2" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>3.1.2</b> Hierarchical Clustering Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#gene-level-qc"><i class="fa fa-check"></i><b>3.2</b> Gene-level QC</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#mov10-quality-assessment-and-exploratory-analysis-using-deseq2"><i class="fa fa-check"></i><b>3.3</b> Mov10 quality assessment and exploratory analysis using DESeq2</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="quality-control.html"><a href="quality-control.html#transform-normalized-counts-using-the-rlog-transformation"><i class="fa fa-check"></i><b>3.3.1</b> Transform normalized counts using the rlog transformation</a></li>
<li class="chapter" data-level="3.3.2" data-path="quality-control.html"><a href="quality-control.html#principal-components-analysis-pca"><i class="fa fa-check"></i><b>3.3.2</b> Principal components analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.3" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.3</b> Hierarchical Clustering</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html"><i class="fa fa-check"></i><b>4</b> DGE analysis workflow</a>
<ul>
<li class="chapter" data-level="4.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#running-deseq2"><i class="fa fa-check"></i><b>4.1</b> Running DESeq2</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#design-formula"><i class="fa fa-check"></i><b>4.1.1</b> Design formula</a></li>
<li class="chapter" data-level="4.1.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-de-analysis"><i class="fa fa-check"></i><b>4.1.2</b> MOV10 DE analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#deseq2-differential-gene-expression-analysis-workflow"><i class="fa fa-check"></i><b>4.2</b> DESeq2 differential gene expression analysis workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-1-estimate-size-factors"><i class="fa fa-check"></i><b>4.2.1</b> Step 1: Estimate size factors</a></li>
<li class="chapter" data-level="4.2.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-2-estimate-gene-wise-dispersion"><i class="fa fa-check"></i><b>4.2.2</b> Step 2: Estimate gene-wise dispersion</a></li>
<li class="chapter" data-level="4.2.3" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-3-fit-curve-to-gene-wise-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.3</b> Step 3: Fit curve to gene-wise dispersion estimates</a></li>
<li class="chapter" data-level="4.2.4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-4-shrink-gene-wise-dispersion-estimates-toward-the-values-predicted-by-the-curve"><i class="fa fa-check"></i><b>4.2.4</b> Step 4: Shrink gene-wise dispersion estimates toward the values predicted by the curve</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html"><i class="fa fa-check"></i><b>5</b> Model and hypothesis testing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#generalized-linear-model-fit-for-each-gene"><i class="fa fa-check"></i><b>5.1</b> Generalized Linear Model fit for each gene</a></li>
<li class="chapter" data-level="5.2" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#shrunken-log2-foldchanges-lfc"><i class="fa fa-check"></i><b>5.2</b> Shrunken log2 foldchanges (LFC)</a></li>
<li class="chapter" data-level="5.3" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#statistical-test-for-lfc-estimates-wald-test"><i class="fa fa-check"></i><b>5.3</b> Statistical test for LFC estimates: Wald test</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#overall-deseq2-workflow"><i class="fa fa-check"></i><b>5.3.1</b> Overall DESeq2 workflow:</a></li>
<li class="chapter" data-level="5.3.2" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#creating-contrasts"><i class="fa fa-check"></i><b>5.3.2</b> Creating contrasts</a></li>
<li class="chapter" data-level="5.3.3" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#building-the-results-table"><i class="fa fa-check"></i><b>5.3.3</b> Building the results table</a></li>
<li class="chapter" data-level="5.3.4" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#ma-plot"><i class="fa fa-check"></i><b>5.3.4</b> MA Plot</a></li>
<li class="chapter" data-level="5.3.5" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#mov10-de-analysis-results-exploration"><i class="fa fa-check"></i><b>5.3.5</b> MOV10 DE analysis: results exploration</a></li>
<li class="chapter" data-level="5.3.6" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.3.6</b> Multiple test correction</a></li>
<li class="chapter" data-level="5.3.7" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#mov10-de-analysis-control-versus-knockdown"><i class="fa fa-check"></i><b>5.3.7</b> MOV10 DE analysis: Control versus Knockdown</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#summarizing-results"><i class="fa fa-check"></i><b>5.4</b> Summarizing results</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#extracting-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.4.1</b> Extracting significant differentially expressed genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html"><i class="fa fa-check"></i><b>6</b> Visualizing RNA-seq results</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#plotting-signicant-de-genes"><i class="fa fa-check"></i><b>6.0.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="6.0.2" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#heatmap"><i class="fa fa-check"></i><b>6.0.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.0.3" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#volcano-plot"><i class="fa fa-check"></i><b>6.0.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html"><i class="fa fa-check"></i><b>7</b> Summary of DGE workflow</a></li>
<li class="chapter" data-level="8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html"><i class="fa fa-check"></i><b>8</b> Functional analysis of RNAseq data</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#over-representation-analysis"><i class="fa fa-check"></i><b>8.0.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="8.0.2" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#clusterprofiler"><i class="fa fa-check"></i><b>8.0.2</b> clusterProfiler</a></li>
<li class="chapter" data-level="8.0.3" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>8.0.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="8.0.4" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#other-tools-and-resources"><i class="fa fa-check"></i><b>8.0.4</b> Other tools and resources</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="codebook.html"><a href="codebook.html"><i class="fa fa-check"></i><b>9</b> Codebook</a>
<ul>
<li class="chapter" data-level="9.1" data-path="codebook.html"><a href="codebook.html#dge-analysis-overview"><i class="fa fa-check"></i><b>9.1</b> DGE analysis overview</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="codebook.html"><a href="codebook.html#review-of-the-dataset-1"><i class="fa fa-check"></i><b>9.1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="9.1.2" data-path="codebook.html"><a href="codebook.html#setting-up-1"><i class="fa fa-check"></i><b>9.1.2</b> Setting up</a></li>
<li class="chapter" data-level="9.1.3" data-path="codebook.html"><a href="codebook.html#dge-analysis-workflow-1"><i class="fa fa-check"></i><b>9.1.3</b> DGE analysis workflow</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="codebook.html"><a href="codebook.html#count-normalization-1"><i class="fa fa-check"></i><b>9.2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="codebook.html"><a href="codebook.html#normalization-1"><i class="fa fa-check"></i><b>9.2.1</b> Normalization</a></li>
<li class="chapter" data-level="9.2.2" data-path="codebook.html"><a href="codebook.html#count-normalization-of-mov10-dataset-1"><i class="fa fa-check"></i><b>9.2.2</b> Count normalization of Mov10 dataset</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="codebook.html"><a href="codebook.html#dge-qc-analysis"><i class="fa fa-check"></i><b>9.3</b> DGE QC analysis</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="codebook.html"><a href="codebook.html#sample-level-qc-1"><i class="fa fa-check"></i><b>9.3.1</b> Sample-level QC</a></li>
<li class="chapter" data-level="9.3.2" data-path="codebook.html"><a href="codebook.html#mov10-quality-assessment-and-exploratory-analysis"><i class="fa fa-check"></i><b>9.3.2</b> Mov10 quality assessment and exploratory analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="codebook.html"><a href="codebook.html#deg-analysis-workflow"><i class="fa fa-check"></i><b>9.4</b> DEG analysis workflow</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="codebook.html"><a href="codebook.html#running-deseq2-1"><i class="fa fa-check"></i><b>9.4.1</b> Running DESeq2</a></li>
<li class="chapter" data-level="9.4.2" data-path="codebook.html"><a href="codebook.html#deseq2-differential-gene-expression-analysis-workflow-1"><i class="fa fa-check"></i><b>9.4.2</b> DESeq2 differential gene expression analysis workflow</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="codebook.html"><a href="codebook.html#model-fitting"><i class="fa fa-check"></i><b>9.5</b> Model fitting</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="codebook.html"><a href="codebook.html#generalized-linear-model-fit-for-each-gene-1"><i class="fa fa-check"></i><b>9.5.1</b> Generalized Linear Model fit for each gene</a></li>
<li class="chapter" data-level="9.5.2" data-path="codebook.html"><a href="codebook.html#shrunken-log2-foldchanges-lfc-1"><i class="fa fa-check"></i><b>9.5.2</b> Shrunken log2 foldchanges (LFC)</a></li>
<li class="chapter" data-level="9.5.3" data-path="codebook.html"><a href="codebook.html#hypothesis-testing-using-the-wald-test"><i class="fa fa-check"></i><b>9.5.3</b> Hypothesis testing using the Wald test</a></li>
<li class="chapter" data-level="9.5.4" data-path="codebook.html"><a href="codebook.html#summarizing-results-1"><i class="fa fa-check"></i><b>9.5.4</b> Summarizing results</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="codebook.html"><a href="codebook.html#visualizing-rna-seq-results-1"><i class="fa fa-check"></i><b>9.6</b> Visualizing rna-seq results</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="codebook.html"><a href="codebook.html#plotting-signicant-de-genes-1"><i class="fa fa-check"></i><b>9.6.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="9.6.2" data-path="codebook.html"><a href="codebook.html#heatmap-1"><i class="fa fa-check"></i><b>9.6.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="codebook.html"><a href="codebook.html#summary-of-differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>9.7</b> Summary of differential expression analysis workflow</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="codebook.html"><a href="codebook.html#import-data-into-dds-object"><i class="fa fa-check"></i><b>9.7.1</b> 1. Import data into dds object:</a></li>
<li class="chapter" data-level="9.7.2" data-path="codebook.html"><a href="codebook.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data"><i class="fa fa-check"></i><b>9.7.2</b> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:</a></li>
<li class="chapter" data-level="9.7.3" data-path="codebook.html"><a href="codebook.html#run-deseq2"><i class="fa fa-check"></i><b>9.7.3</b> 3. Run DESeq2:</a></li>
<li class="chapter" data-level="9.7.4" data-path="codebook.html"><a href="codebook.html#check-the-fit-of-the-dispersion-estimates"><i class="fa fa-check"></i><b>9.7.4</b> 4. Check the fit of the dispersion estimates:</a></li>
<li class="chapter" data-level="9.7.5" data-path="codebook.html"><a href="codebook.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions"><i class="fa fa-check"></i><b>9.7.5</b> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:</a></li>
<li class="chapter" data-level="9.7.6" data-path="codebook.html"><a href="codebook.html#output-significant-results"><i class="fa fa-check"></i><b>9.7.6</b> 6. Output significant results:</a></li>
<li class="chapter" data-level="9.7.7" data-path="codebook.html"><a href="codebook.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc."><i class="fa fa-check"></i><b>9.7.7</b> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.</a></li>
<li class="chapter" data-level="9.7.8" data-path="codebook.html"><a href="codebook.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis"><i class="fa fa-check"></i><b>9.7.8</b> 8. Make sure to output the versions of all tools used in the DE analysis:</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="codebook.html"><a href="codebook.html#functional-analysis-of-rna-seq-data"><i class="fa fa-check"></i><b>9.8</b> Functional analysis of RNA-seq data</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="codebook.html"><a href="codebook.html#over-representation-analysis-1"><i class="fa fa-check"></i><b>9.8.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="9.8.2" data-path="codebook.html"><a href="codebook.html#clusterprofiler-1"><i class="fa fa-check"></i><b>9.8.2</b> clusterProfiler</a></li>
<li class="chapter" data-level="9.8.3" data-path="codebook.html"><a href="codebook.html#gene-set-enrichment-analysis-gsea-1"><i class="fa fa-check"></i><b>9.8.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="9.8.4" data-path="codebook.html"><a href="codebook.html#other-tools-and-resources-1"><i class="fa fa-check"></i><b>9.8.4</b> Other tools and resources</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">RNA-seq-analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="codebook" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Codebook<a href="codebook.html#codebook" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>20 total points</p>
<div id="dge-analysis-overview" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> DGE analysis overview<a href="codebook.html#dge-analysis-overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 01-DGE_setup_and_overview.md</strong></p>
<div id="review-of-the-dataset-1" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Review of the dataset<a href="codebook.html#review-of-the-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>RNA-Seq was performed on HEK293F cells that were either transfected with a <strong>MOV10</strong> transgene, or siRNA to knock down Mov10 expression, or non-specific (irrelevant) siRNA. This resulted in 3 conditions <strong>Mov10 oe</strong> (over expression), <strong>Mov10 kd</strong> (knock down) and <strong>Irrelevant kd</strong>, respectively. The number of replicates is as shown below.</p>
<p><img src="img/dataset.png" width="400"></p>
<p>Using these data, we will evaluate transcriptional patterns associated with perturbation of <strong>MOV10</strong> expression. Please note that the irrelevant siRNA will be treated as our control condition.</p>
<p><strong>Our questions:</strong>
* What patterns of expression can we identify with the loss or gain of MOV10?
* Are there any genes shared between the two conditions?</p>
</div>
<div id="setting-up-1" class="section level3 hasAnchor" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> Setting up<a href="codebook.html#setting-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="loading-libraries-1" class="section level4 hasAnchor" number="9.1.2.1">
<h4><span class="header-section-number">9.1.2.1</span> Loading libraries<a href="codebook.html#loading-libraries-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="codebook.html#cb112-1" tabindex="-1"></a><span class="do">### Setup: Use BiocManager::install if you haven&#39;t already installed these libraries</span></span>
<span id="cb112-2"><a href="codebook.html#cb112-2" tabindex="-1"></a><span class="do">### </span></span>
<span id="cb112-3"><a href="codebook.html#cb112-3" tabindex="-1"></a><span class="do">### BiocManager::install(c(&quot;tidyverse&quot;,&quot;RColorBrewer&quot;,&quot;DESeq2&quot;,&quot;pheatmap&quot;,&quot;ggplot2&quot;,&quot;ggrepel&quot;))</span></span>
<span id="cb112-4"><a href="codebook.html#cb112-4" tabindex="-1"></a></span>
<span id="cb112-5"><a href="codebook.html#cb112-5" tabindex="-1"></a><span class="do">### Bioconductor and CRAN libraries used</span></span>
<span id="cb112-6"><a href="codebook.html#cb112-6" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb112-7"><a href="codebook.html#cb112-7" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb112-8"><a href="codebook.html#cb112-8" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb112-9"><a href="codebook.html#cb112-9" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb112-10"><a href="codebook.html#cb112-10" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb112-11"><a href="codebook.html#cb112-11" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span></code></pre></div>
</div>
<div id="loading-data-1" class="section level4 hasAnchor" number="9.1.2.2">
<h4><span class="header-section-number">9.1.2.2</span> Loading data<a href="codebook.html#loading-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb113-1"><a href="codebook.html#cb113-1" tabindex="-1"></a><span class="do">## Load in data</span></span>
<span id="cb113-2"><a href="codebook.html#cb113-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/Mov10_full_counts.txt&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb113-3"><a href="codebook.html#cb113-3" tabindex="-1"></a></span>
<span id="cb113-4"><a href="codebook.html#cb113-4" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;meta/Mov10_full_meta.txt&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span></code></pre></div>
<p>Use class() to inspect our data and make sure we are working with data frames:</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="codebook.html#cb114-1" tabindex="-1"></a><span class="do">### Check classes of the data we just brought in</span></span>
<span id="cb114-2"><a href="codebook.html#cb114-2" tabindex="-1"></a><span class="fu">class</span>(meta)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="codebook.html#cb116-1" tabindex="-1"></a><span class="fu">class</span>(data)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
</div>
<div id="viewing-data-1" class="section level4 hasAnchor" number="9.1.2.3">
<h4><span class="header-section-number">9.1.2.3</span> Viewing data<a href="codebook.html#viewing-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="codebook.html#cb118-1" tabindex="-1"></a><span class="fu">head</span>(meta)</span></code></pre></div>
<div class="kable-table">
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sampletype</th>
<th align="left">MOVexpr</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Irrel_kd_1</td>
<td align="left">control</td>
<td align="left">normal</td>
</tr>
<tr class="even">
<td align="left">Irrel_kd_2</td>
<td align="left">control</td>
<td align="left">normal</td>
</tr>
<tr class="odd">
<td align="left">Irrel_kd_3</td>
<td align="left">control</td>
<td align="left">normal</td>
</tr>
<tr class="even">
<td align="left">Mov10_kd_2</td>
<td align="left">MOV10_knockdown</td>
<td align="left">low</td>
</tr>
<tr class="odd">
<td align="left">Mov10_kd_3</td>
<td align="left">MOV10_knockdown</td>
<td align="left">low</td>
</tr>
<tr class="even">
<td align="left">Mov10_oe_1</td>
<td align="left">MOV10_overexpression</td>
<td align="left">high</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb119-1"><a href="codebook.html#cb119-1" tabindex="-1"></a><span class="fu">head</span>(data[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="17%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Mov10_kd_2</th>
<th align="right">Mov10_kd_3</th>
<th align="right">Mov10_oe_1</th>
<th align="right">Mov10_oe_2</th>
<th align="right">Mov10_oe_3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1/2-SBSRNA4</td>
<td align="right">57</td>
<td align="right">41</td>
<td align="right">64</td>
<td align="right">55</td>
<td align="right">38</td>
</tr>
<tr class="even">
<td align="left">A1BG</td>
<td align="right">71</td>
<td align="right">40</td>
<td align="right">100</td>
<td align="right">81</td>
<td align="right">41</td>
</tr>
<tr class="odd">
<td align="left">A1BG-AS1</td>
<td align="right">256</td>
<td align="right">177</td>
<td align="right">220</td>
<td align="right">189</td>
<td align="right">107</td>
</tr>
<tr class="even">
<td align="left">A1CF</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">A2LD1</td>
<td align="right">146</td>
<td align="right">81</td>
<td align="right">138</td>
<td align="right">125</td>
<td align="right">52</td>
</tr>
<tr class="even">
<td align="left">A2M</td>
<td align="right">10</td>
<td align="right">9</td>
<td align="right">2</td>
<td align="right">5</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="dge-analysis-workflow-1" class="section level3 hasAnchor" number="9.1.3">
<h3><span class="header-section-number">9.1.3</span> DGE analysis workflow<a href="codebook.html#dge-analysis-workflow-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="rna-seq-count-distribution-1" class="section level4 hasAnchor" number="9.1.3.1">
<h4><span class="header-section-number">9.1.3.1</span> RNA-seq count distribution<a href="codebook.html#rna-seq-count-distribution-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To determine the appropriate statistical model, we need information about the distribution of counts. To get an idea about how RNA-seq counts are distributed, letâs plot the counts for a single sample, âMov10_oe_1â:</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="codebook.html#cb120-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb120-2"><a href="codebook.html#cb120-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb120-3"><a href="codebook.html#cb120-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb120-4"><a href="codebook.html#cb120-4" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p>If we zoom in close to zero, we can see a large number of genes with counts of zero:</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="codebook.html#cb121-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb121-2"><a href="codebook.html#cb121-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb121-3"><a href="codebook.html#cb121-3" tabindex="-1"></a>   <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">500</span>)  <span class="sc">+</span></span>
<span id="cb121-4"><a href="codebook.html#cb121-4" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb121-5"><a href="codebook.html#cb121-5" tabindex="-1"></a>   <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Removed 9121 rows containing non-finite outside the scale range
## (`stat_bin()`).</code></pre>
<pre><code>## Warning: Removed 2 rows containing missing values or values outside the scale range
## (`geom_bar()`).</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>These images illustrate some common features of RNA-seq count data, including a low number of counts associated with a large proportion of genes, and a long right tail due to the lack of any upper limit for expression.</p>
</div>
<div id="modeling-count-data-1" class="section level4 hasAnchor" number="9.1.3.2">
<h4><span class="header-section-number">9.1.3.2</span> Modeling count data<a href="codebook.html#modeling-count-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>By plotting the <em>mean versus the variance</em> of our data we should be able to see that the variance &gt; mean and therefore it does not fit the Poisson distribution and is better suited to the Negative Binomial (NB) model.</p>
<p>Run the following code to plot the <em>mean versus variance</em> for the âMov10 overexpressionâ replicates:</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="codebook.html#cb124-1" tabindex="-1"></a><span class="co"># apply applies a function to the margins of a matrix</span></span>
<span id="cb124-2"><a href="codebook.html#cb124-2" tabindex="-1"></a><span class="co"># apply(X, MARGIN, FUN)</span></span>
<span id="cb124-3"><a href="codebook.html#cb124-3" tabindex="-1"></a><span class="co"># where X is a matrix</span></span>
<span id="cb124-4"><a href="codebook.html#cb124-4" tabindex="-1"></a><span class="co"># to apply a function to each row of a matrix, use MARGIN = 1</span></span>
<span id="cb124-5"><a href="codebook.html#cb124-5" tabindex="-1"></a><span class="co"># to apply a function to each column of a matrix, use MARGIN = 2</span></span>
<span id="cb124-6"><a href="codebook.html#cb124-6" tabindex="-1"></a><span class="co"># FUN can be any function</span></span>
<span id="cb124-7"><a href="codebook.html#cb124-7" tabindex="-1"></a></span>
<span id="cb124-8"><a href="codebook.html#cb124-8" tabindex="-1"></a>mean_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, mean)</span>
<span id="cb124-9"><a href="codebook.html#cb124-9" tabindex="-1"></a>variance_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, var)</span>
<span id="cb124-10"><a href="codebook.html#cb124-10" tabindex="-1"></a><span class="co"># for ggplot we need the data to be in a data.frame</span></span>
<span id="cb124-11"><a href="codebook.html#cb124-11" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mean_counts, variance_counts)</span>
<span id="cb124-12"><a href="codebook.html#cb124-12" tabindex="-1"></a></span>
<span id="cb124-13"><a href="codebook.html#cb124-13" tabindex="-1"></a><span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb124-14"><a href="codebook.html#cb124-14" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_counts, <span class="at">y=</span>variance_counts)) <span class="sc">+</span></span>
<span id="cb124-15"><a href="codebook.html#cb124-15" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>mean_counts, <span class="at">y=</span>mean_counts, <span class="at">color=</span><span class="st">&quot;red&quot;</span>)) <span class="sc">+</span></span>
<span id="cb124-16"><a href="codebook.html#cb124-16" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb124-17"><a href="codebook.html#cb124-17" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<pre><code>## Warning in scale_y_log10(): log-10 transformation introduced infinite values.</code></pre>
<pre><code>## Warning in scale_x_log10(): log-10 transformation introduced infinite values.</code></pre>
<pre><code>## Warning in scale_y_log10(): log-10 transformation introduced infinite values.</code></pre>
<pre><code>## Warning in scale_x_log10(): log-10 transformation introduced infinite values.</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>Note that in the above figure, the variance across replicates tends to be greater than the mean (slope &gt; 1, red line), especially for genes with large mean expression levels. This is a good indication that our data do not fit the Poisson distribution and we need to account for this increase in variance using the Negative Binomial model.</p>
</div>
</div>
</div>
<div id="count-normalization-1" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Count normalization<a href="codebook.html#count-normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 02-DGE_count_normalization.md</strong></p>
<div id="normalization-1" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> Normalization<a href="codebook.html#normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The steps fpr DESeq2 median of ratios method:</p>
<ol style="list-style-type: decimal">
<li><p>calculate the geometric mean of each row in the count matrix</p></li>
<li><p>divide each row by the geometric mean for that row</p></li>
<li><p>take the median value for each column â these are your size factors, one for each sample</p></li>
<li><p>divide each sample by its size factor</p></li>
</ol>
<p><strong>Normalized Counts</strong></p>
<p><strong>Exercise</strong></p>
<p>Determine the normalized counts for your gene of interest, PD1, given the raw counts and size factors below.</p>
<p>NOTE: You will need to run the code below to generate the raw counts dataframe (PD1) and the size factor vector (size_factors), then use these objects to determine the normalized counts values:</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="codebook.html#cb129-1" tabindex="-1"></a><span class="co"># Raw counts for PD1</span></span>
<span id="cb129-2"><a href="codebook.html#cb129-2" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">58</span>, <span class="dv">17</span>, <span class="dv">97</span>, <span class="dv">83</span>, <span class="dv">10</span>)</span>
<span id="cb129-3"><a href="codebook.html#cb129-3" tabindex="-1"></a><span class="fu">names</span>(PD1) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Sample&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)</span>
<span id="cb129-4"><a href="codebook.html#cb129-4" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(PD1)</span>
<span id="cb129-5"><a href="codebook.html#cb129-5" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">t</span>(PD1)</span>
<span id="cb129-6"><a href="codebook.html#cb129-6" tabindex="-1"></a></span>
<span id="cb129-7"><a href="codebook.html#cb129-7" tabindex="-1"></a><span class="co"># Size factors for each sample</span></span>
<span id="cb129-8"><a href="codebook.html#cb129-8" tabindex="-1"></a>size_factors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.32</span>, <span class="fl">0.70</span>, <span class="fl">1.04</span>, <span class="fl">1.27</span>, <span class="fl">1.11</span>, <span class="fl">0.85</span>)</span></code></pre></div>
<p>Normalized counts:</p>
</div>
<div id="count-normalization-of-mov10-dataset-1" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> Count normalization of Mov10 dataset<a href="codebook.html#count-normalization-of-mov10-dataset-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="match-the-metadata-and-counts-data-1" class="section level4 hasAnchor" number="9.2.2.1">
<h4><span class="header-section-number">9.2.2.1</span> 1. Match the metadata and counts data<a href="codebook.html#match-the-metadata-and-counts-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb130-1"><a href="codebook.html#cb130-1" tabindex="-1"></a><span class="do">### Check that sample names match in both files</span></span>
<span id="cb130-2"><a href="codebook.html#cb130-2" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb132-1"><a href="codebook.html#cb132-1" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">==</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>The colnames of our data donât match the rownames of our metadata so we need to reorder them. We can use the <code>match</code> function:</p>
<div class="sourceCode" id="cb134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb134-1"><a href="codebook.html#cb134-1" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">rownames</span>(meta),<span class="fu">colnames</span>(data))</span>
<span id="cb134-2"><a href="codebook.html#cb134-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[,idx]</span>
<span id="cb134-3"><a href="codebook.html#cb134-3" tabindex="-1"></a></span>
<span id="cb134-4"><a href="codebook.html#cb134-4" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">==</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p><strong>Exercise</strong> points = +2</p>
<p>Suppose we had sample names matching in the counts matrix and metadata file, but they were out of order. Write the line(s) of code required to create a new matrix with columns ordered such that they were identical to the row names of the metadata.</p>
</div>
<div id="create-deseq2-object-1" class="section level4 hasAnchor" number="9.2.2.2">
<h4><span class="header-section-number">9.2.2.2</span> 2. Create DESEq2 object<a href="codebook.html#create-deseq2-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb136-1"><a href="codebook.html#cb136-1" tabindex="-1"></a><span class="do">## Create DESeq2Dataset object</span></span>
<span id="cb136-2"><a href="codebook.html#cb136-2" tabindex="-1"></a></span>
<span id="cb136-3"><a href="codebook.html#cb136-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data, <span class="at">colData =</span> meta, <span class="at">design =</span> <span class="sc">~</span> sampletype)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in design
## formula are characters, converting to factors</code></pre>
<p>You can use DESeq-specific functions to access the different slots and retrieve information, if you wish. For example, suppose we wanted the original count matrix we would use counts():</p>
<div class="sourceCode" id="cb138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb138-1"><a href="codebook.html#cb138-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">counts</span>(dds[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]))</span></code></pre></div>
<pre><code>##             Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3
## 1/2-SBSRNA4         45         31         39         57         41
## A1BG                77         58         40         71         40
## A1BG-AS1           213        172        126        256        177
## A1CF                 0          0          0          0          1
## A2LD1               91         80         50        146         81
## A2M                  9          8          4         10          9</code></pre>
<div class="sourceCode" id="cb140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb140-1"><a href="codebook.html#cb140-1" tabindex="-1"></a><span class="fu">colData</span>(dds)</span></code></pre></div>
<pre><code>## DataFrame with 8 rows and 2 columns
##                      sampletype     MOVexpr
##                        &lt;factor&gt; &lt;character&gt;
## Irrel_kd_1 control                   normal
## Irrel_kd_2 control                   normal
## Irrel_kd_3 control                   normal
## Mov10_kd_2 MOV10_knockdown              low
## Mov10_kd_3 MOV10_knockdown              low
## Mov10_oe_1 MOV10_overexpression        high
## Mov10_oe_2 MOV10_overexpression        high
## Mov10_oe_3 MOV10_overexpression        high</code></pre>
<div class="sourceCode" id="cb142"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb142-1"><a href="codebook.html#cb142-1" tabindex="-1"></a><span class="fu">colData</span>(dds)<span class="sc">$</span>sampletype</span></code></pre></div>
<pre><code>## [1] control              control              control             
## [4] MOV10_knockdown      MOV10_knockdown      MOV10_overexpression
## [7] MOV10_overexpression MOV10_overexpression
## Levels: control MOV10_knockdown MOV10_overexpression</code></pre>
<div class="sourceCode" id="cb144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb144-1"><a href="codebook.html#cb144-1" tabindex="-1"></a><span class="fu">levels</span>(<span class="fu">colData</span>(dds)<span class="sc">$</span>sampletype)</span></code></pre></div>
<pre><code>## [1] &quot;control&quot;              &quot;MOV10_knockdown&quot;      &quot;MOV10_overexpression&quot;</code></pre>
<p>As we go through the workflow we will use the relevant functions to check what information gets stored inside our object. We can also run:</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="codebook.html#cb146-1" tabindex="-1"></a><span class="fu">slotNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;design&quot;             &quot;dispersionFunction&quot; &quot;rowRanges&quot;         
## [4] &quot;colData&quot;            &quot;assays&quot;             &quot;NAMES&quot;             
## [7] &quot;elementMetadata&quot;    &quot;metadata&quot;</code></pre>
</div>
<div id="generate-the-mov10-normalized-counts-1" class="section level4 hasAnchor" number="9.2.2.3">
<h4><span class="header-section-number">9.2.2.3</span> 3. Generate the Mov10 normalized counts<a href="codebook.html#generate-the-mov10-normalized-counts-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The next step is to normalize the count data in order to be able to make fair gene comparisons between samples.</p>
<p><img src="img/deseq_workflow_normalization_2018.png" width="400"></p>
<p>To perform the <strong>median of ratios method</strong> of normalization, DESeq2 has a single <code>estimateSizeFactors()</code> function that will generate size factors for us. We will use the function in the example below, but <strong>in a typical RNA-seq analysis this step is automatically performed by the <code>DESeq()</code> function</strong>, which we will see later.</p>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="codebook.html#cb148-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span></code></pre></div>
<p>By assigning the results back to the dds object we are filling in the slots of the DESeqDataSet object with the appropriate information. We can take a look at the normalization factor applied to each sample using:</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="codebook.html#cb149-1" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 Mov10_oe_2 
##  1.1224020  0.9625632  0.7477715  1.5646728  0.9351760  1.2016082  1.1205912 
## Mov10_oe_3 
##  0.6534987</code></pre>
<p>Now, to retrieve the normalized counts matrix from dds, we use the counts() function and add the argument normalized=TRUE.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="codebook.html#cb151-1" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can save this normalized data matrix to file for later use:</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="codebook.html#cb152-1" tabindex="-1"></a><span class="fu">write.table</span>(normalized_counts, <span class="at">file=</span><span class="st">&quot;data/normalized_counts.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb152-2"><a href="codebook.html#cb152-2" tabindex="-1"></a></span>
<span id="cb152-3"><a href="codebook.html#cb152-3" tabindex="-1"></a><span class="co"># it&#39;s always good to save our results, in case we forget to save objects</span></span>
<span id="cb152-4"><a href="codebook.html#cb152-4" tabindex="-1"></a><span class="fu">save.image</span>()</span>
<span id="cb152-5"><a href="codebook.html#cb152-5" tabindex="-1"></a><span class="co"># saves all objects in session/environment</span></span></code></pre></div>
</div>
</div>
</div>
<div id="dge-qc-analysis" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> DGE QC analysis<a href="codebook.html#dge-qc-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 03-DGE_QC_analysis.md</strong></p>
<p>The next step in the DESeq2 workflow is QC, which includes sample-level and gene-level steps to perform QC checks on the count data to help us ensure that the samples/replicates look good.</p>
<p><img src="img/deseq_workflow_qc_2018.png" width="400"></p>
<div id="sample-level-qc-1" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Sample-level QC<a href="codebook.html#sample-level-qc-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>When using these unsupervised clustering methods, log2-transformation of the normalized counts improves the distances/clustering for visualization. DESeq2 uses a regularized log transform (rlog) of the normalized counts for sample-level QC as it moderates the variance across the mean, improving the clustering.</p>
</div>
<div id="mov10-quality-assessment-and-exploratory-analysis" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Mov10 quality assessment and exploratory analysis<a href="codebook.html#mov10-quality-assessment-and-exploratory-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="transform-normalized-counts-using-the-rlog-transformation-1" class="section level4 hasAnchor" number="9.3.2.1">
<h4><span class="header-section-number">9.3.2.1</span> Transform normalized counts using the rlog transformation<a href="codebook.html#transform-normalized-counts-using-the-rlog-transformation-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To improve the distances/clustering for the PCA and heirarchical clustering visualization methods, we need to moderate the variance across the mean by applying the rlog transformation to the normalized counts.</p>
<p>The rlog transformation of the normalized counts is only necessary for these visualization methods during this quality assessment. We will not be using these tranformed counts downstream:</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="codebook.html#cb153-1" tabindex="-1"></a><span class="do">### Transform counts for data visualization</span></span>
<span id="cb153-2"><a href="codebook.html#cb153-2" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>We use this object to plot the PCA and heirarchical clustering figures for quality assessment.</p>
</div>
<div id="principal-components-analysis-pca-1" class="section level4 hasAnchor" number="9.3.2.2">
<h4><span class="header-section-number">9.3.2.2</span> Principal components analysis (PCA)<a href="codebook.html#principal-components-analysis-pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Principal Component Analysis (PCA) is a technique used to emphasize variation and bring out strong patterns in a dataset (dimensionality reduction). The take home message for our purposes is that <strong>if two samples have similar levels of expression for the genes that contribute significantly to the variation represented by PC1, they will be plotted close together on the PC1 axis.</strong> Therefore, we would expect that biological replicates to have similar scores (since the same genes are changing) and cluster together on PC1 and/or PC2, and the samples from different treatment groups to have different score. This is easiest to understand by visualizing example PCA plots.</p>
<p><strong>Exercise</strong> points = +8</p>
<p>The figure below was generated from a time course experiment with sample groups âCtrlâ and âSciâ and the following timepoints: 0h, 2h, 8h, and 16h.</p>
<p><img src="img/PCA_example3.png" width="600"></p>
<p><strong>Determine the sources explaining the variation represented by PC1 and PC2.</strong> points = +1</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Do the sample groups separate well?</strong> points = +1</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Do the replicates cluster together for each sample group?</strong> points = +1</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Are there any outliers in the data?</strong> points = +2</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Should we have any other concerns regarding the samples in the dataset?</strong></p>
<ul>
<li>Ans:</li>
</ul>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="codebook.html#cb154-1" tabindex="-1"></a><span class="do">### Plot PCA</span></span>
<span id="cb154-2"><a href="codebook.html#cb154-2" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>)</span></code></pre></div>
<pre><code>## using ntop=500 top features by variance</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
<p><strong>What does this plot tell you about the similarity of samples?</strong> points = +1</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Does it fit the expectation from the experimental design?</strong> points = +1</p>
<ul>
<li>Ans:</li>
</ul>
<p>By default the function uses the top 500 most variable genes. You can change this by adding the <code>ntop</code> argument and specifying how many genes you want to use to draw the plot.</p>
</div>
<div id="hierarchical-clustering-1" class="section level4 hasAnchor" number="9.3.2.3">
<h4><span class="header-section-number">9.3.2.3</span> Hierarchical Clustering<a href="codebook.html#hierarchical-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Similar to PCA, hierarchical clustering is another, complementary method for identifying strong patterns in a dataset and potential outliers. The heatmap displays the correlation of gene expression for all pairwise combinations of samples in the dataset. Since the majority of genes are not differentially expressed, samples generally have high correlations with each other (values higher than 0.80). Samples below 0.80 may indicate an outlier in your data and/or sample contamination.</p>
<p>The heatmap displays the correlation of gene expression for all pairwise combinations of samples in the dataset. Since the majority of genes are not differentially expressed, samples generally have high correlations with each other (values higher than 0.80). Samples below 0.80 may indicate an outlier in your data and/or sample contamination. We expect to see samples clustered similar to the groupings observed in a PCA plot.</p>
<p>We will be using the <code>pheatmap()</code> function from the pheatmap package for heatmaps. This function requires a matrix/dataframe of numeric values as input, and so the first thing we need to is retrieve that information from the rld object:</p>
<div class="sourceCode" id="cb156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb156-1"><a href="codebook.html#cb156-1" tabindex="-1"></a><span class="do">### let&#39;s look at the structure of rld.</span></span>
<span id="cb156-2"><a href="codebook.html#cb156-2" tabindex="-1"></a><span class="fu">class</span>(rld)</span></code></pre></div>
<pre><code>## [1] &quot;DESeqTransform&quot;
## attr(,&quot;package&quot;)
## [1] &quot;DESeq2&quot;</code></pre>
<div class="sourceCode" id="cb158"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb158-1"><a href="codebook.html#cb158-1" tabindex="-1"></a><span class="fu">slotNames</span>(rld)</span></code></pre></div>
<pre><code>## [1] &quot;rowRanges&quot;       &quot;colData&quot;         &quot;assays&quot;          &quot;NAMES&quot;          
## [5] &quot;elementMetadata&quot; &quot;metadata&quot;</code></pre>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="codebook.html#cb160-1" tabindex="-1"></a><span class="do">### we can extract the rlog matrix from the object using the `assay` function:</span></span>
<span id="cb160-2"><a href="codebook.html#cb160-2" tabindex="-1"></a><span class="do">###</span></span>
<span id="cb160-3"><a href="codebook.html#cb160-3" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span></code></pre></div>
<p>Then we need to compute the pairwise correlation values for samples. We can do this using the <code>cor()</code> function:</p>
<div class="sourceCode" id="cb161"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb161-1"><a href="codebook.html#cb161-1" tabindex="-1"></a><span class="do">### Compute pairwise correlation values</span></span>
<span id="cb161-2"><a href="codebook.html#cb161-2" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)    <span class="do">## cor() is a base R function</span></span>
<span id="cb161-3"><a href="codebook.html#cb161-3" tabindex="-1"></a></span>
<span id="cb161-4"><a href="codebook.html#cb161-4" tabindex="-1"></a><span class="fu">head</span>(rld_cor[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])   <span class="do">## check the output of cor(), make note of the rownames and colnames</span></span></code></pre></div>
<pre><code>##            Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3
## Irrel_kd_1  1.0000000  0.9999614  0.9999532  0.9997202  0.9997748
## Irrel_kd_2  0.9999614  1.0000000  0.9999544  0.9996918  0.9997568
## Irrel_kd_3  0.9999532  0.9999544  1.0000000  0.9996816  0.9997574
## Mov10_kd_2  0.9997202  0.9996918  0.9996816  1.0000000  0.9999492
## Mov10_kd_3  0.9997748  0.9997568  0.9997574  0.9999492  1.0000000
## Mov10_oe_1  0.9996700  0.9996984  0.9997067  0.9994868  0.9996154</code></pre>
<div class="sourceCode" id="cb163"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb163-1"><a href="codebook.html#cb163-1" tabindex="-1"></a><span class="fu">min</span>(rld_cor) <span class="do">## we can see our samples are highly correlated</span></span></code></pre></div>
<pre><code>## [1] 0.9993869</code></pre>
<p>And now to plot the correlation values as a heatmap:</p>
<div class="sourceCode" id="cb165"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb165-1"><a href="codebook.html#cb165-1" tabindex="-1"></a><span class="do">### Plot heatmap</span></span>
<span id="cb165-2"><a href="codebook.html#cb165-2" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<p><strong>Exercise</strong> The pheatmap function has a number of different arguments that we can alter from default values to enhance the aesthetics of the plot. If you are curious and want to explore more, try running the code below. How does your plot change? Take a look through the help pages (?pheatmap) and identify what each of the added arguments is contributing to the plot.</p>
<div class="sourceCode" id="cb166"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb166-1"><a href="codebook.html#cb166-1" tabindex="-1"></a><span class="fu">display.brewer.all</span>()</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-43-1.png" width="672" /></p>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="codebook.html#cb167-1" tabindex="-1"></a>heat.colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)</span>
<span id="cb167-2"><a href="codebook.html#cb167-2" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor, <span class="at">color =</span> heat.colors, <span class="at">border_color=</span><span class="st">&quot;black&quot;</span>, <span class="at">fontsize =</span> <span class="dv">10</span>, <span class="at">fontsize_row =</span> <span class="dv">12</span>, <span class="at">show_rownames =</span> T, <span class="at">show_colnames =</span> T)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-43-2.png" width="672" /></p>
<ul>
<li>Ans:</li>
</ul>
</div>
</div>
</div>
<div id="deg-analysis-workflow" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> DEG analysis workflow<a href="codebook.html#deg-analysis-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 04_DGE_DESeq2_analysis.md</strong></p>
<p><strong>Differential expression analysis with DESeq2</strong></p>
<p>Differential expression analysis with DESeq2 involves multiple steps as displayed in the flowchart below in blue. Briefly:</p>
<ul>
<li><p>DESeq2 will model the raw counts, using normalization factors (size factors) to account for differences in library depth.</p></li>
<li><p>estimate the gene-wise dispersion</p></li>
<li><p>estimate the expected dispersion</p></li>
<li><p>shrink gene-wise dispersion estimates to generate more accurate estimates to model the counts</p></li>
<li><p>fit the negative binomial model and perform hypothesis testing using the Wald test</p></li>
</ul>
<p><img src="img/DESeq2_workflow_2018.png" width="500"></p>
<div id="running-deseq2-1" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Running DESeq2<a href="codebook.html#running-deseq2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="design-formula-1" class="section level4 hasAnchor" number="9.4.1.1">
<h4><span class="header-section-number">9.4.1.1</span> Design formula<a href="codebook.html#design-formula-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A design formula tells the statistical software the known sources of variation to control for, as well as, the factor of interest to test for during differential expression testing. For example, letâs look at the design formula for our count matrix:</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="codebook.html#cb168-1" tabindex="-1"></a><span class="fu">design</span>(dds)</span></code></pre></div>
<pre><code>## ~sampletype</code></pre>
<p>As another example, suppose you have the following metadata:</p>
<p><img src="img/meta_example.png" width="300"></p>
<p><strong>Exercise</strong> points = +5</p>
<ol style="list-style-type: decimal">
<li><p>Suppose you wanted to study the expression differences between the two age groups in the metadata shown above, and major sources of variation were <code>sex</code> and <code>treatment</code>, how would the design formula be written?</p></li>
<li><p>Based on our Mov10 metadata dataframe, which factors could we include in our design formula?</p></li>
</ol>
<ul>
<li>Ans:</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>What would you do if you wanted to include a factor in your design formula that is not in your metadata?</li>
</ol>
<ul>
<li>Ans:</li>
</ul>
</div>
<div id="mov10-de-analysis-1" class="section level4 hasAnchor" number="9.4.1.2">
<h4><span class="header-section-number">9.4.1.2</span> MOV10 DE analysis<a href="codebook.html#mov10-de-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To get our differential expression results from our raw count data, we only need to run 2 lines of code!</p>
<div class="sourceCode" id="cb170"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb170-1"><a href="codebook.html#cb170-1" tabindex="-1"></a><span class="do">## Create DESeq object</span></span>
<span id="cb170-2"><a href="codebook.html#cb170-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data, <span class="at">colData =</span> meta, <span class="at">design =</span> <span class="sc">~</span> sampletype)</span></code></pre></div>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in design
## formula are characters, converting to factors</code></pre>
<p>Then, to run the actual differential expression analysis, we use a single call to the function DESeq().</p>
<div class="sourceCode" id="cb172"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb172-1"><a href="codebook.html#cb172-1" tabindex="-1"></a><span class="do">## Run analysis</span></span>
<span id="cb172-2"><a href="codebook.html#cb172-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
</div>
</div>
<div id="deseq2-differential-gene-expression-analysis-workflow-1" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> DESeq2 differential gene expression analysis workflow<a href="codebook.html#deseq2-differential-gene-expression-analysis-workflow-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Everything from normalization to linear modeling was carried out by the use of a single function!</p>
<p>With the 2 lines of code above, we just completed the workflow for the differential gene expression analysis with DESeq2. The steps in the analysis are output below:</p>
<p><img src="img/deseq2_workflow_separate.png" width="200"></p>
<div id="step-1-estimate-size-factors-1" class="section level4 hasAnchor" number="9.4.2.1">
<h4><span class="header-section-number">9.4.2.1</span> Step 1: Estimate size factors<a href="codebook.html#step-1-estimate-size-factors-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>MOV10 DE analysis: examining the size factors</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="codebook.html#cb179-1" tabindex="-1"></a><span class="do">## Check the size factors</span></span>
<span id="cb179-2"><a href="codebook.html#cb179-2" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 Mov10_oe_2 
##  1.1224020  0.9625632  0.7477715  1.5646728  0.9351760  1.2016082  1.1205912 
## Mov10_oe_3 
##  0.6534987</code></pre>
<p>Take a look at the total number of reads for each sample:</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="codebook.html#cb181-1" tabindex="-1"></a><span class="do">## Total number of raw counts per sample</span></span>
<span id="cb181-2"><a href="codebook.html#cb181-2" tabindex="-1"></a><span class="fu">colSums</span>(<span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 Mov10_oe_2 
##   22687366   19381680   14962754   32826936   19360003   23447317   21713289 
## Mov10_oe_3 
##   12737889</code></pre>
<p><strong>How do the numbers correlate with the size factor?</strong></p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="codebook.html#cb183-1" tabindex="-1"></a><span class="fu">cor</span>(<span class="fu">colSums</span>(<span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">FALSE</span>)),<span class="fu">sizeFactors</span>(dds))</span></code></pre></div>
<pre><code>## [1] 0.9946543</code></pre>
<ul>
<li>Ans: they are highly correlated</li>
</ul>
</div>
<div id="step-2-estimate-gene-wise-dispersion-1" class="section level4 hasAnchor" number="9.4.2.2">
<h4><span class="header-section-number">9.4.2.2</span> Step 2: Estimate gene-wise dispersion**<a href="codebook.html#step-2-estimate-gene-wise-dispersion-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Dispersion is a measure of spread or variability in the data. DESeq2 uses a specific measure of dispersion (Î±) related to the mean (Î¼) and variance of the data: <span class="math inline">\(Var = Î¼ + Î±*Î¼^2\)</span>. For genes with moderate to high count values, the square root of dispersion will be equal to the coefficient of variation (<code>Ï/ Î¼</code>). So 0.01 dispersion means 10% variation around the mean expected across biological replicates.</p>
<p>We can make a rough estimate of the gene-wise dispersions using this formula.
We know that:</p>
<p><span class="math display">\[\sigma^2 = {\mu + \alpha*}{\mu^2}\]</span></p>
<p>So the dispersion is equal to:
<span class="math display">\[\alpha = \frac{\sigma^2 - \mu}{\mu^2}\]</span></p>
<div class="sourceCode" id="cb185"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb185-1"><a href="codebook.html#cb185-1" tabindex="-1"></a>norm.cts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds,<span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb185-2"><a href="codebook.html#cb185-2" tabindex="-1"></a>mn <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(norm.cts)</span>
<span id="cb185-3"><a href="codebook.html#cb185-3" tabindex="-1"></a>mn<span class="ot">=</span>mn[<span class="fu">order</span>(mn)]</span>
<span id="cb185-4"><a href="codebook.html#cb185-4" tabindex="-1"></a>mn <span class="ot">&lt;-</span> mn[mn<span class="sc">!=</span><span class="dv">0</span>]</span>
<span id="cb185-5"><a href="codebook.html#cb185-5" tabindex="-1"></a></span>
<span id="cb185-6"><a href="codebook.html#cb185-6" tabindex="-1"></a>v <span class="ot">&lt;-</span> <span class="fu">rowVars</span>(norm.cts)</span>
<span id="cb185-7"><a href="codebook.html#cb185-7" tabindex="-1"></a><span class="fu">names</span>(v) <span class="ot">=</span> <span class="fu">rownames</span>(norm.cts)</span>
<span id="cb185-8"><a href="codebook.html#cb185-8" tabindex="-1"></a></span>
<span id="cb185-9"><a href="codebook.html#cb185-9" tabindex="-1"></a>v <span class="ot">=</span> v[<span class="fu">names</span>(mn)]</span>
<span id="cb185-10"><a href="codebook.html#cb185-10" tabindex="-1"></a></span>
<span id="cb185-11"><a href="codebook.html#cb185-11" tabindex="-1"></a></span>
<span id="cb185-12"><a href="codebook.html#cb185-12" tabindex="-1"></a><span class="co"># For genes with moderate to high count values, the square root of dispersion will be equal to the coefficient of variation (`Ï / Î¼`).</span></span>
<span id="cb185-13"><a href="codebook.html#cb185-13" tabindex="-1"></a><span class="co"># Let&#39;s try a relatively low mean</span></span>
<span id="cb185-14"><a href="codebook.html#cb185-14" tabindex="-1"></a></span>
<span id="cb185-15"><a href="codebook.html#cb185-15" tabindex="-1"></a><span class="co"># to calculate the dispersion, v has to be greater than mu otherwise the disp will be negative</span></span>
<span id="cb185-16"><a href="codebook.html#cb185-16" tabindex="-1"></a>wv <span class="ot">=</span> <span class="fu">which</span>(v <span class="sc">&gt;</span> mn)</span>
<span id="cb185-17"><a href="codebook.html#cb185-17" tabindex="-1"></a>var <span class="ot">&lt;-</span> v[wv][<span class="dv">1</span>]</span>
<span id="cb185-18"><a href="codebook.html#cb185-18" tabindex="-1"></a>mu <span class="ot">&lt;-</span> mn[wv][<span class="dv">1</span>]</span>
<span id="cb185-19"><a href="codebook.html#cb185-19" tabindex="-1"></a></span>
<span id="cb185-20"><a href="codebook.html#cb185-20" tabindex="-1"></a>disp <span class="ot">&lt;-</span>  (var <span class="sc">-</span> mu)<span class="sc">/</span>mu<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb185-21"><a href="codebook.html#cb185-21" tabindex="-1"></a></span>
<span id="cb185-22"><a href="codebook.html#cb185-22" tabindex="-1"></a><span class="fu">sqrt</span>(var)<span class="sc">/</span>mu</span></code></pre></div>
<pre><code>##     ACY3 
## 2.828427</code></pre>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="codebook.html#cb187-1" tabindex="-1"></a><span class="fu">sqrt</span>(disp)</span></code></pre></div>
<pre><code>##      ACY3 
## 0.5472605</code></pre>
<p>So the approximation doesnât work very well here. Now letâs try a high mean:</p>
<div class="sourceCode" id="cb189"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb189-1"><a href="codebook.html#cb189-1" tabindex="-1"></a>var <span class="ot">&lt;-</span> v[wv][<span class="fu">length</span>(wv)]</span>
<span id="cb189-2"><a href="codebook.html#cb189-2" tabindex="-1"></a>mu <span class="ot">&lt;-</span> mn[wv][<span class="fu">length</span>(wv)]</span>
<span id="cb189-3"><a href="codebook.html#cb189-3" tabindex="-1"></a></span>
<span id="cb189-4"><a href="codebook.html#cb189-4" tabindex="-1"></a>disp <span class="ot">&lt;-</span> (var <span class="sc">-</span> mu)<span class="sc">/</span>mu<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb189-5"><a href="codebook.html#cb189-5" tabindex="-1"></a></span>
<span id="cb189-6"><a href="codebook.html#cb189-6" tabindex="-1"></a><span class="co"># approximation for the square root of the dispersion</span></span>
<span id="cb189-7"><a href="codebook.html#cb189-7" tabindex="-1"></a><span class="fu">sqrt</span>(var)<span class="sc">/</span>mu</span></code></pre></div>
<pre><code>##       EEF2 
## 0.08296104</code></pre>
<div class="sourceCode" id="cb191"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb191-1"><a href="codebook.html#cb191-1" tabindex="-1"></a><span class="co"># actual sqrt disp</span></span>
<span id="cb191-2"><a href="codebook.html#cb191-2" tabindex="-1"></a><span class="fu">sqrt</span>(disp)</span></code></pre></div>
<pre><code>##       EEF2 
## 0.08288016</code></pre>
<p>Now the approximation is pretty good. The larger the value of the mean, the better the approximation.</p>
<p>What does the DESeq2 dispersion represent?</p>
<p>The DESeq2 dispersion estimates are inversely related to the mean and directly related to variance. Based on this relationship, the dispersion is higher for small mean counts and lower for large mean counts.</p>
<p>How does the dispersion relate to our model?</p>
<p>To accurately model sequencing counts, we need to generate accurate estimates of within-group variation (variation between replicates of the same sample group) for each gene. With only a few (3-6) replicates per group, the estimates of variation for each gene are often unreliable (due to the large differences in dispersion for genes with similar means).</p>
<p>To address this problem, DESeq2 shares information across genes to generate more accurate estimates of variation based on the mean expression level of the gene using a method called âshrinkageâ. DESeq2 assumes that genes with similar expression levels have similar dispersion.</p>
<p>Estimating the dispersion for each gene separately:</p>
<p>To model the dispersion based on expression level (mean counts of replicates), the dispersion for each gene is estimated using maximum likelihood estimation. In statistics, maximum likelihood estimation (MLE) is a method of estimating the parameters of an assumed probability distribution, given some observed data. This is achieved by maximizing a likelihood function so that, under the assumed statistical model, the observed data is most probable. In other words, given the count values of the replicates, the most likely estimate of dispersion is calculated.</p>
</div>
<div id="step-3-fit-curve-to-gene-wise-dispersion-estimates-1" class="section level4 hasAnchor" number="9.4.2.3">
<h4><span class="header-section-number">9.4.2.3</span> Step 3: Fit curve to gene-wise dispersion estimates<a href="codebook.html#step-3-fit-curve-to-gene-wise-dispersion-estimates-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The next step in the workflow is to fit a curve to the dispersion estimates for each gene. The idea behind fitting a curve to the data is that different genes will have different scales of biological variability, but, over all genes, there will be a distribution of reasonable estimates of dispersion.</p>
</div>
<div id="step-4-shrink-gene-wise-dispersion-estimates-toward-the-values-predicted-by-the-curve-1" class="section level4 hasAnchor" number="9.4.2.4">
<h4><span class="header-section-number">9.4.2.4</span> Step 4: Shrink gene-wise dispersion estimates toward the values predicted by the curve<a href="codebook.html#step-4-shrink-gene-wise-dispersion-estimates-toward-the-values-predicted-by-the-curve-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The curve allows for more accurate identification of differentially expressed genes when sample sizes are small, and the strength of the shrinkage for each gene depends on:</p>
<ul>
<li>how close gene dispersions are from the curve</li>
<li>sample size (more samples = less shrinkage)</li>
</ul>
<p>Dispersion plots are a good way to examine your data to ensure it is a good fit for the DESeq2 model.</p>
<div id="mov10-de-analysis-exploring-the-dispersion-estimates-and-assessing-model" class="section level5 hasAnchor" number="9.4.2.4.1">
<h5><span class="header-section-number">9.4.2.4.1</span> MOV10 DE analysis: exploring the dispersion estimates and assessing model<a href="codebook.html#mov10-de-analysis-exploring-the-dispersion-estimates-and-assessing-model" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Letâs take a look at the dispersion estimates for our MOV10 data:</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="codebook.html#cb193-1" tabindex="-1"></a><span class="do">## Plot dispersion estimates</span></span>
<span id="cb193-2"><a href="codebook.html#cb193-2" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
<p>Since we have a small sample size, for many genes we see quite a bit of shrinkage.</p>
<p><strong>Do you think our data are a good fit for the model?</strong></p>
<ul>
<li>Ans: Yes</li>
</ul>
</div>
</div>
</div>
</div>
<div id="model-fitting" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Model fitting<a href="codebook.html#model-fitting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 05_DGE_DESeq2_analysis2</strong></p>
<div id="generalized-linear-model-fit-for-each-gene-1" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Generalized Linear Model fit for each gene<a href="codebook.html#generalized-linear-model-fit-for-each-gene-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The final step in the DESeq2 workflow is fitting the Negative Binomial model for each gene and performing differential expression testing.</p>
<p><img src="img/deseq2_workflow_separate_2018.png" width="400"></p>
</div>
<div id="shrunken-log2-foldchanges-lfc-1" class="section level3 hasAnchor" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Shrunken log2 foldchanges (LFC)<a href="codebook.html#shrunken-log2-foldchanges-lfc-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To generate more accurate log2 foldchange estimates, DESeq2 allows for the <strong>shrinkage of the LFC estimates toward zero</strong> when the information for a gene is low, which could include:</p>
<ul>
<li>Low counts</li>
<li>High dispersion values</li>
</ul>
<p>As with the shrinkage of dispersion estimates, LFC shrinkage uses <strong>information from all genes</strong> to generate more accurate estimates. Specifically, the distribution of LFC estimates for all genes is used (as a prior) to shrink the LFC estimates of genes with little information or high dispersion toward more likely (lower) LFC estimates.</p>
</div>
<div id="hypothesis-testing-using-the-wald-test" class="section level3 hasAnchor" number="9.5.3">
<h3><span class="header-section-number">9.5.3</span> Hypothesis testing using the Wald test<a href="codebook.html#hypothesis-testing-using-the-wald-test" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>With DESeq2, the Wald test is commonly used for hypothesis testing when comparing two groups. A Wald test statistic is computed along with a probability that a test statistic at least as extreme as the observed value were selected at random. This probability is called the p-value of the test. If the p-value is small we reject the null hypothesis and state that there is evidence against the null (i.e.Â the gene is differentially expressed).</p>
<div id="creating-contrasts-1" class="section level4 hasAnchor" number="9.5.3.1">
<h4><span class="header-section-number">9.5.3.1</span> Creating contrasts<a href="codebook.html#creating-contrasts-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To indicate to DESeq2 the two groups we want to compare, we can use contrasts. Contrasts are then provided to DESeq2 to perform differential expression testing using the Wald test. Contrasts can be provided to DESeq2 a couple of different ways:</p>
<ol style="list-style-type: decimal">
<li><p>Do nothing. Automatically DESeq2 will use the base factor level of the condition of interest as the base for statistical testing. The base level is chosen based on alphabetical order of the levels.</p></li>
<li><p>In the results() function you can specify the comparison of interest, and the levels to compare. The level given last is the base level for the comparison. The syntax is given below:</p></li>
</ol>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="codebook.html#cb194-1" tabindex="-1"></a><span class="co"># DO NOT RUN!</span></span>
<span id="cb194-2"><a href="codebook.html#cb194-2" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;level_to_compare&quot;</span>, <span class="st">&quot;base_level&quot;</span>)</span>
<span id="cb194-3"><a href="codebook.html#cb194-3" tabindex="-1"></a><span class="fu">results</span>(dds, <span class="at">contrast =</span> contrast, <span class="at">alpha =</span> alpha_threshold)</span></code></pre></div>
<div id="mov10-de-analysis-contrasts-and-wald-tests-1" class="section level5 hasAnchor" number="9.5.3.1.1">
<h5><span class="header-section-number">9.5.3.1.1</span> MOV10 DE analysis: contrasts and Wald tests<a href="codebook.html#mov10-de-analysis-contrasts-and-wald-tests-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We have three sample classes so we can make three possible pairwise comparisons:</p>
<ol style="list-style-type: decimal">
<li>Control vs.Â Mov10 overexpression</li>
<li>Control vs.Â Mov10 knockdown</li>
<li>Mov10 knockdown vs.Â Mov10 overexpression</li>
</ol>
<p><strong>We are really only interested in #1 and #2 from above</strong>. Using the design formula we provided <code>~ sampletype</code>, indicating that this is our main factor of interest.</p>
</div>
</div>
<div id="building-the-results-table-1" class="section level4 hasAnchor" number="9.5.3.2">
<h4><span class="header-section-number">9.5.3.2</span> Building the results table<a href="codebook.html#building-the-results-table-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="codebook.html#cb195-1" tabindex="-1"></a><span class="do">## Define contrasts, extract results table, and shrink the log2 fold changes</span></span>
<span id="cb195-2"><a href="codebook.html#cb195-2" tabindex="-1"></a></span>
<span id="cb195-3"><a href="codebook.html#cb195-3" tabindex="-1"></a>contrast_oe <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sampletype&quot;</span>, <span class="st">&quot;MOV10_overexpression&quot;</span>, <span class="st">&quot;control&quot;</span>)</span>
<span id="cb195-4"><a href="codebook.html#cb195-4" tabindex="-1"></a></span>
<span id="cb195-5"><a href="codebook.html#cb195-5" tabindex="-1"></a>res_tableOE_unshrunken <span class="ot">&lt;-</span> <span class="fu">results</span>(<span class="at">object =</span> dds, <span class="at">contrast=</span>contrast_oe, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb195-6"><a href="codebook.html#cb195-6" tabindex="-1"></a></span>
<span id="cb195-7"><a href="codebook.html#cb195-7" tabindex="-1"></a><span class="co"># for the coef argument of lfcShrink we have to use coefficient names from the `resultsNames()` </span></span>
<span id="cb195-8"><a href="codebook.html#cb195-8" tabindex="-1"></a></span>
<span id="cb195-9"><a href="codebook.html#cb195-9" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                                 
## [2] &quot;sampletype_MOV10_knockdown_vs_control&quot;     
## [3] &quot;sampletype_MOV10_overexpression_vs_control&quot;</code></pre>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="codebook.html#cb197-1" tabindex="-1"></a>res_tableOE <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef =</span> <span class="st">&quot;sampletype_MOV10_overexpression_vs_control&quot;</span>, <span class="at">res =</span> res_tableOE_unshrunken, <span class="at">type =</span> <span class="st">&quot;apeglm&quot;</span>)</span></code></pre></div>
<pre><code>## using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="codebook.html#cb199-1" tabindex="-1"></a><span class="co"># We will save these results for later use in the data directory using the following command:</span></span>
<span id="cb199-2"><a href="codebook.html#cb199-2" tabindex="-1"></a></span>
<span id="cb199-3"><a href="codebook.html#cb199-3" tabindex="-1"></a><span class="fu">saveRDS</span>(res_tableOE, <span class="at">file =</span> <span class="st">&quot;data/res_tableOE.rds&quot;</span>)</span></code></pre></div>
<p><strong>The order of the names determines the direction of fold change that is reported.</strong> The name provided in the second element is the level that is used as baseline. So for example, if we observe a log2 fold change of -2 this would mean the gene expression is lower in Mov10_oe relative to the control.</p>
</div>
<div id="ma-plot-1" class="section level4 hasAnchor" number="9.5.3.3">
<h4><span class="header-section-number">9.5.3.3</span> MA Plot<a href="codebook.html#ma-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A plot that can be useful to exploring our results is the MA plot. The MA plot shows the mean of the normalized counts versus the log2 fold changes for all genes tested. The genes that are significantly DE are colored to be easily identified. This is also a great way to illustrate the effect of LFC shrinkage. The DESeq2 package offers a simple function to generate an MA plot.</p>
<p><strong>Letâs start with the unshrunken results:</strong></p>
<p>Viewing the results:</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="codebook.html#cb200-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb200-2"><a href="codebook.html#cb200-2" tabindex="-1"></a><span class="fu">plotMA</span>(res_tableOE_unshrunken, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-56-1.png" width="672" /></p>
<p><strong>And now the shrunken results:</strong></p>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="codebook.html#cb201-1" tabindex="-1"></a><span class="fu">plotMA</span>(res_tableOE, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-57-1.png" width="672" /></p>
<div class="sourceCode" id="cb202"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb202-1"><a href="codebook.html#cb202-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>This plot allows us to evaluate the magnitude of fold changes and how they are distributed relative to mean expression.</p>
<div id="mov10-de-analysis-results-exploration-1" class="section level5 hasAnchor" number="9.5.3.3.1">
<h5><span class="header-section-number">9.5.3.3.1</span> MOV10 DE analysis: results exploration<a href="codebook.html#mov10-de-analysis-results-exploration-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The results table looks very much like a dataframe and in many ways it can be treated like one (i.e when accessing/subsetting data). However, it is important to recognize that it is actually stored in a <code>DESeqResults</code> object. When we start visualizing our data, this information will be helpful.</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="codebook.html#cb203-1" tabindex="-1"></a><span class="fu">class</span>(res_tableOE)</span></code></pre></div>
<pre><code>## [1] &quot;DESeqResults&quot;
## attr(,&quot;package&quot;)
## [1] &quot;DESeq2&quot;</code></pre>
<p>Letâs go through some of the columns in the results table to get a better idea of what we are looking at. To extract information regarding the meaning of each column we can use mcols():</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="codebook.html#cb205-1" tabindex="-1"></a><span class="fu">mcols</span>(res_tableOE, <span class="at">use.names=</span>T)</span></code></pre></div>
<pre><code>## DataFrame with 5 rows and 2 columns
##                        type            description
##                 &lt;character&gt;            &lt;character&gt;
## baseMean       intermediate mean of normalized c..
## log2FoldChange      results log2 fold change (MA..
## lfcSE               results posterior SD: sample..
## pvalue              results Wald test p-value: s..
## padj                results   BH adjusted p-values</code></pre>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="codebook.html#cb207-1" tabindex="-1"></a><span class="fu">mcols</span>(res_tableOE, <span class="at">use.names=</span>T)<span class="sc">$</span>description</span></code></pre></div>
<pre><code>## [1] &quot;mean of normalized counts for all samples&quot;                         
## [2] &quot;log2 fold change (MAP): sampletype MOV10_overexpression vs control&quot;
## [3] &quot;posterior SD: sampletype MOV10 overexpression vs control&quot;          
## [4] &quot;Wald test p-value: sampletype MOV10 overexpression vs control&quot;     
## [5] &quot;BH adjusted p-values&quot;</code></pre>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="codebook.html#cb209-1" tabindex="-1"></a>res_tableOE <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">head</span>()</span></code></pre></div>
<div class="kable-table">
<table style="width:100%;">
<colgroup>
<col width="17%" />
<col width="17%" />
<col width="21%" />
<col width="14%" />
<col width="14%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1/2-SBSRNA4</td>
<td align="right">45.6520399</td>
<td align="right">0.1682023</td>
<td align="right">0.2091004</td>
<td align="right">0.1610752</td>
<td align="right">0.2655661</td>
</tr>
<tr class="even">
<td align="left">A1BG</td>
<td align="right">61.0931017</td>
<td align="right">0.1364590</td>
<td align="right">0.1767790</td>
<td align="right">0.2401909</td>
<td align="right">0.3603094</td>
</tr>
<tr class="odd">
<td align="left">A1BG-AS1</td>
<td align="right">175.6658069</td>
<td align="right">-0.0404612</td>
<td align="right">0.1169284</td>
<td align="right">0.6789312</td>
<td align="right">0.7748342</td>
</tr>
<tr class="even">
<td align="left">A1CF</td>
<td align="right">0.2376919</td>
<td align="right">0.0062675</td>
<td align="right">0.2100570</td>
<td align="right">0.7945932</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">A2LD1</td>
<td align="right">89.6179845</td>
<td align="right">0.2901772</td>
<td align="right">0.1953954</td>
<td align="right">0.0333343</td>
<td align="right">0.0741149</td>
</tr>
<tr class="even">
<td align="left">A2M</td>
<td align="right">5.8600841</td>
<td align="right">-0.0961972</td>
<td align="right">0.2335610</td>
<td align="right">0.1057823</td>
<td align="right">0.1900466</td>
</tr>
</tbody>
</table>
</div>
<p>If we used the <code>p-value</code> directly from the Wald test with a significance cut-off of p &lt; 0.05, that means there is a 5% chance it is a false positives. Each p-value is the result of a single test (single gene). The more genes we test, the more we inflate the false positive rate. <strong>This is the multiple testing problem.</strong></p>
<p>In DESeq2, the p-values attained by the Wald test are corrected for multiple testing using the Benjamini and Hochberg method by default. There are options to use other methods in the <code>results()</code> function. The p-adjusted values should be used to determine significant genes.</p>
</div>
<div id="mov10-de-analysis-control-versus-knockdown-1" class="section level5 hasAnchor" number="9.5.3.3.2">
<h5><span class="header-section-number">9.5.3.3.2</span> MOV10 DE analysis: Control versus Knockdown<a href="codebook.html#mov10-de-analysis-control-versus-knockdown-1" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Now that we have results for the overexpression results, letâs do the same for the <strong>Control vs.Â Knockdown samples</strong>. Use contrasts in the <code>results()</code> to extract a results table and store that to a variable called <code>res_tableKD</code>.</p>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="codebook.html#cb210-1" tabindex="-1"></a><span class="do">## Define contrasts, extract results table and shrink log2 fold changes</span></span>
<span id="cb210-2"><a href="codebook.html#cb210-2" tabindex="-1"></a></span>
<span id="cb210-3"><a href="codebook.html#cb210-3" tabindex="-1"></a>contrast_kd <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">&quot;sampletype&quot;</span>, <span class="st">&quot;MOV10_knockdown&quot;</span>, <span class="st">&quot;control&quot;</span>)</span>
<span id="cb210-4"><a href="codebook.html#cb210-4" tabindex="-1"></a></span>
<span id="cb210-5"><a href="codebook.html#cb210-5" tabindex="-1"></a>res_tableKD_unshrunken <span class="ot">&lt;-</span> <span class="fu">results</span>(<span class="at">object =</span> dds, <span class="at">contrast=</span>contrast_kd, <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb210-6"><a href="codebook.html#cb210-6" tabindex="-1"></a></span>
<span id="cb210-7"><a href="codebook.html#cb210-7" tabindex="-1"></a><span class="co"># for the coef argument in &#39;lfcShrink&#39; you can also provide a number based on the order in `resultsNames()`</span></span>
<span id="cb210-8"><a href="codebook.html#cb210-8" tabindex="-1"></a></span>
<span id="cb210-9"><a href="codebook.html#cb210-9" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                                 
## [2] &quot;sampletype_MOV10_knockdown_vs_control&quot;     
## [3] &quot;sampletype_MOV10_overexpression_vs_control&quot;</code></pre>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="codebook.html#cb212-1" tabindex="-1"></a>res_tableKD <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef =</span> <span class="dv">2</span>, <span class="at">res=</span>res_tableKD_unshrunken)</span></code></pre></div>
<pre><code>## using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="codebook.html#cb214-1" tabindex="-1"></a><span class="co"># We will save these results for later use in the data directory using the following command:</span></span>
<span id="cb214-2"><a href="codebook.html#cb214-2" tabindex="-1"></a></span>
<span id="cb214-3"><a href="codebook.html#cb214-3" tabindex="-1"></a><span class="fu">saveRDS</span>(res_tableKD, <span class="at">file =</span> <span class="st">&quot;data/res_tableKD.rds&quot;</span>)</span>
<span id="cb214-4"><a href="codebook.html#cb214-4" tabindex="-1"></a></span>
<span id="cb214-5"><a href="codebook.html#cb214-5" tabindex="-1"></a><span class="co"># just for security, save all our objects in &quot;.RData&quot; again:</span></span>
<span id="cb214-6"><a href="codebook.html#cb214-6" tabindex="-1"></a><span class="fu">save.image</span>()</span></code></pre></div>
<p>Letâs look at the number of genes below the standard p-adjusted value 0.05:</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="codebook.html#cb215-1" tabindex="-1"></a><span class="fu">table</span>(res_tableKD<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
## 10668  4909</code></pre>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="codebook.html#cb217-1" tabindex="-1"></a><span class="fu">table</span>(res_tableOE<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##  9066  6511</code></pre>
</div>
</div>
</div>
<div id="summarizing-results-1" class="section level3 hasAnchor" number="9.5.4">
<h3><span class="header-section-number">9.5.4</span> Summarizing results<a href="codebook.html#summarizing-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To summarize the results table, a handy function in DESeq2 is <code>summary()</code>. Confusingly it has the same name as the function used to inspect data frames. This function when called with a DESeq results table as input, will summarize the results using the alpha threshold: FDR &lt; 0.05 (padj/FDR is used even though the output says <code>p-value &lt; 0.05</code>).</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="codebook.html#cb219-1" tabindex="-1"></a><span class="fu">summary</span>(res_tableOE)</span></code></pre></div>
<pre><code>## 
## out of 19748 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 3103, 16%
## LFC &lt; 0 (down)     : 3408, 17%
## outliers [1]       : 0, 0%
## low counts [2]     : 4171, 21%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<div id="extracting-significant-differentially-expressed-genes-1" class="section level4 hasAnchor" number="9.5.4.1">
<h4><span class="header-section-number">9.5.4.1</span> Extracting significant differentially expressed genes<a href="codebook.html#extracting-significant-differentially-expressed-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>What we noticed is that the FDR threshold on itâs own doesnât appear to be reducing the number of significant genes. With large significant gene lists it can be hard to extract meaningful biological relevance. To help increase stringency, one can also <strong>add a fold change threshold</strong>.</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="codebook.html#cb221-1" tabindex="-1"></a><span class="do">### Set thresholds</span></span>
<span id="cb221-2"><a href="codebook.html#cb221-2" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb221-3"><a href="codebook.html#cb221-3" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fl">0.58</span> <span class="do">## change in expression of 1.5</span></span></code></pre></div>
<p>The lfc.cutoff is set to 0.58; remember that we are working with log2 fold changes so this translates to an actual fold change of 1.5 which is pretty reasonable.</p>
<p>We can easily subset the results table to only include those that are significant using the filter() function, but first we will convert the results table into a tibble:</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="codebook.html#cb222-1" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span></span>
<span id="cb222-2"><a href="codebook.html#cb222-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb222-3"><a href="codebook.html#cb222-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb222-4"><a href="codebook.html#cb222-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>Now we can subset that table to only keep the significant genes using our pre-defined thresholds:</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="codebook.html#cb223-1" tabindex="-1"></a>sigOE <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span></span>
<span id="cb223-2"><a href="codebook.html#cb223-2" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span></code></pre></div>
<p><strong>Exercise</strong></p>
<p><strong>How many genes are differentially expressed in the Overexpression compared to Control, given our criteria specified above? Does this reduce our results?</strong> points = +2</p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Does this reduce our results?</strong></p>
<p>-Ans:</p>
<p>Using the same thresholds as above (<code>padj.cutoff &lt; 0.05</code> and <code>lfc.cutoff = 0.58)</code>, subset <code>res_tableKD</code> to report the number of genes that are up- and down-regulated in Mov10_knockdown compared to control.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="codebook.html#cb224-1" tabindex="-1"></a>res_tableKD_tb <span class="ot">&lt;-</span> res_tableKD <span class="sc">%&gt;%</span></span>
<span id="cb224-2"><a href="codebook.html#cb224-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb224-3"><a href="codebook.html#cb224-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb224-4"><a href="codebook.html#cb224-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb224-5"><a href="codebook.html#cb224-5" tabindex="-1"></a></span>
<span id="cb224-6"><a href="codebook.html#cb224-6" tabindex="-1"></a>sigKD <span class="ot">&lt;-</span> res_tableKD_tb <span class="sc">%&gt;%</span></span>
<span id="cb224-7"><a href="codebook.html#cb224-7" tabindex="-1"></a>         dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span> </span>
<span id="cb224-8"><a href="codebook.html#cb224-8" tabindex="-1"></a>         <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span>
<span id="cb224-9"><a href="codebook.html#cb224-9" tabindex="-1"></a></span>
<span id="cb224-10"><a href="codebook.html#cb224-10" tabindex="-1"></a><span class="co"># We&#39;ll save this object for use in the homework</span></span>
<span id="cb224-11"><a href="codebook.html#cb224-11" tabindex="-1"></a><span class="fu">saveRDS</span>(sigKD,<span class="st">&quot;data/sigKD.rds&quot;</span>)</span></code></pre></div>
<p><strong>Exercise</strong></p>
<p><strong>How many genes are differentially expressed in the Knockdown compared to Control?</strong></p>
<ul>
<li>Ans:</li>
</ul>
</div>
</div>
</div>
<div id="visualizing-rna-seq-results-1" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Visualizing rna-seq results<a href="codebook.html#visualizing-rna-seq-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 06_DGE_visualizing_results.md</strong></p>
<p>Letâs start by loading a few libraries (if not already loaded):</p>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="codebook.html#cb225-1" tabindex="-1"></a><span class="co"># load libraries</span></span>
<span id="cb225-2"><a href="codebook.html#cb225-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb225-3"><a href="codebook.html#cb225-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb225-4"><a href="codebook.html#cb225-4" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb225-5"><a href="codebook.html#cb225-5" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb225-6"><a href="codebook.html#cb225-6" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb225-7"><a href="codebook.html#cb225-7" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb225-8"><a href="codebook.html#cb225-8" tabindex="-1"></a></span>
<span id="cb225-9"><a href="codebook.html#cb225-9" tabindex="-1"></a><span class="co"># we may want to load our objects again. here:</span></span>
<span id="cb225-10"><a href="codebook.html#cb225-10" tabindex="-1"></a><span class="co"># load(&quot;.RData&quot;)</span></span></code></pre></div>
<p>When we are working with large amounts of data it can be useful to display that information graphically to gain more insight.</p>
<p>Letâs create tibble objects from the meta and normalized_counts data frames before we start plotting. This will enable us to use the tidyverse functionality more easily.</p>
<p>Basically, we are taking the rownames and adding them as a field in the tibble/data.frame.</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="codebook.html#cb226-1" tabindex="-1"></a><span class="co"># Create a tibble for meta data</span></span>
<span id="cb226-2"><a href="codebook.html#cb226-2" tabindex="-1"></a>mov10_meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb226-3"><a href="codebook.html#cb226-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;samplename&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb226-4"><a href="codebook.html#cb226-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb226-5"><a href="codebook.html#cb226-5" tabindex="-1"></a>        </span>
<span id="cb226-6"><a href="codebook.html#cb226-6" tabindex="-1"></a><span class="co"># you might to read in normalized_counts if it is not in your current session:</span></span>
<span id="cb226-7"><a href="codebook.html#cb226-7" tabindex="-1"></a></span>
<span id="cb226-8"><a href="codebook.html#cb226-8" tabindex="-1"></a><span class="co"># normalized_counts &lt;- read.delim(&quot;data/normalized_counts.txt&quot;, row.names=1)</span></span>
<span id="cb226-9"><a href="codebook.html#cb226-9" tabindex="-1"></a></span>
<span id="cb226-10"><a href="codebook.html#cb226-10" tabindex="-1"></a><span class="co"># then make sure the colnames of normalized_counts are the same as the mov10_meta$sampname</span></span>
<span id="cb226-11"><a href="codebook.html#cb226-11" tabindex="-1"></a></span>
<span id="cb226-12"><a href="codebook.html#cb226-12" tabindex="-1"></a><span class="fu">all</span>(mov10_meta<span class="sc">$</span>samplename <span class="sc">==</span> <span class="fu">colnames</span>(normalized_counts))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="codebook.html#cb228-1" tabindex="-1"></a><span class="co"># if they aren&#39;t we need to reorder them</span></span>
<span id="cb228-2"><a href="codebook.html#cb228-2" tabindex="-1"></a><span class="co"># mtch = match(mov10_meta$samplename, colnames(normalized_counts))</span></span>
<span id="cb228-3"><a href="codebook.html#cb228-3" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb228-4"><a href="codebook.html#cb228-4" tabindex="-1"></a><span class="co"># normalized_counts &lt;- normalized_counts[,mtch]</span></span>
<span id="cb228-5"><a href="codebook.html#cb228-5" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb228-6"><a href="codebook.html#cb228-6" tabindex="-1"></a><span class="co"># all(mov10_meta$samplename == colnames(normalized_counts))</span></span>
<span id="cb228-7"><a href="codebook.html#cb228-7" tabindex="-1"></a></span>
<span id="cb228-8"><a href="codebook.html#cb228-8" tabindex="-1"></a><span class="co"># Create a tibble for normalized_counts</span></span>
<span id="cb228-9"><a href="codebook.html#cb228-9" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> normalized_counts <span class="sc">%&gt;%</span> </span>
<span id="cb228-10"><a href="codebook.html#cb228-10" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb228-11"><a href="codebook.html#cb228-11" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb228-12"><a href="codebook.html#cb228-12" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span></code></pre></div>
<div id="plotting-signicant-de-genes-1" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Plotting signicant DE genes<a href="codebook.html#plotting-signicant-de-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One way to visualize results would be to simply plot the expression data for a handful of genes. We could do that by picking out specific genes of interest or selecting a range of genes.</p>
<div id="using-deseq2-plotcounts-to-plot-expression-of-a-single-gene-1" class="section level4 hasAnchor" number="9.6.1.1">
<h4><span class="header-section-number">9.6.1.1</span> Using DESeq2 <code>plotCounts()</code> to plot expression of a single gene<a href="codebook.html#using-deseq2-plotcounts-to-plot-expression-of-a-single-gene-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To pick out a specific gene of interest to plot, for example Mov10, we can use the <code>plotCounts()</code> from DESeq2:</p>
<div class="sourceCode" id="cb229"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb229-1"><a href="codebook.html#cb229-1" tabindex="-1"></a><span class="co"># Plot expression for single gene</span></span>
<span id="cb229-2"><a href="codebook.html#cb229-2" tabindex="-1"></a></span>
<span id="cb229-3"><a href="codebook.html#cb229-3" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;MOV10&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>) </span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-73-1.png" width="672" /></p>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="codebook.html#cb230-1" tabindex="-1"></a><span class="co"># we can give it some color and filled points:</span></span>
<span id="cb230-2"><a href="codebook.html#cb230-2" tabindex="-1"></a></span>
<span id="cb230-3"><a href="codebook.html#cb230-3" tabindex="-1"></a>sampletype <span class="ot">=</span> <span class="fu">as.factor</span>(mov10_meta<span class="sc">$</span>sampletype)</span>
<span id="cb230-4"><a href="codebook.html#cb230-4" tabindex="-1"></a></span>
<span id="cb230-5"><a href="codebook.html#cb230-5" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb230-6"><a href="codebook.html#cb230-6" tabindex="-1"></a></span>
<span id="cb230-7"><a href="codebook.html#cb230-7" tabindex="-1"></a><span class="fu">display.brewer.all</span>()</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-73-2.png" width="672" /></p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="codebook.html#cb231-1" tabindex="-1"></a>col <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">&quot;Dark2&quot;</span>)</span>
<span id="cb231-2"><a href="codebook.html#cb231-2" tabindex="-1"></a><span class="fu">palette</span>(col)</span>
<span id="cb231-3"><a href="codebook.html#cb231-3" tabindex="-1"></a></span>
<span id="cb231-4"><a href="codebook.html#cb231-4" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;MOV10&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>,<span class="at">col =</span> <span class="fu">as.numeric</span>(sampletype),<span class="at">pch =</span> <span class="dv">19</span>) </span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-73-3.png" width="672" /></p>
</div>
<div id="using-ggplot2-to-plot-expression-of-a-single-gene-1" class="section level4 hasAnchor" number="9.6.1.2">
<h4><span class="header-section-number">9.6.1.2</span> Using ggplot2 to plot expression of a single gene<a href="codebook.html#using-ggplot2-to-plot-expression-of-a-single-gene-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you wish to change the appearance of this plot, we can save the output of <code>plotCounts()</code> to a variable specifying the <code>returnData=TRUE</code> argument, then use <code>ggplot()</code>:</p>
<div class="sourceCode" id="cb232"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb232-1"><a href="codebook.html#cb232-1" tabindex="-1"></a><span class="co"># Save plotcounts to a data frame object</span></span>
<span id="cb232-2"><a href="codebook.html#cb232-2" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;MOV10&quot;</span>, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb232-3"><a href="codebook.html#cb232-3" tabindex="-1"></a></span>
<span id="cb232-4"><a href="codebook.html#cb232-4" tabindex="-1"></a><span class="co"># Plotting the MOV10 normalized counts, using the samplenames (rownames of d as labels)</span></span>
<span id="cb232-5"><a href="codebook.html#cb232-5" tabindex="-1"></a><span class="fu">ggplot</span>(d, <span class="fu">aes</span>(<span class="at">x =</span> sampletype, <span class="at">y =</span> count, <span class="at">color =</span> sampletype)) <span class="sc">+</span> </span>
<span id="cb232-6"><a href="codebook.html#cb232-6" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">w =</span> <span class="fl">0.1</span>,<span class="at">h =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb232-7"><a href="codebook.html#cb232-7" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">rownames</span>(d))) <span class="sc">+</span> </span>
<span id="cb232-8"><a href="codebook.html#cb232-8" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb232-9"><a href="codebook.html#cb232-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;MOV10&quot;</span>) <span class="sc">+</span></span>
<span id="cb232-10"><a href="codebook.html#cb232-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-74-1.png" width="672" /></p>
</div>
<div id="using-ggplot2-to-plot-multiple-genes-e.g.-top-20-1" class="section level4 hasAnchor" number="9.6.1.3">
<h4><span class="header-section-number">9.6.1.3</span> Using <code>ggplot2</code> to plot multiple genes (e.g.Â top 20)<a href="codebook.html#using-ggplot2-to-plot-multiple-genes-e.g.-top-20-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Often it is helpful to check the expression of multiple genes of interest at the same time. This often first requires some data wrangling.</p>
<p>We are going to plot the normalized count values for the <strong>top 20 differentially expressed genes (by padj values)</strong>.</p>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="codebook.html#cb233-1" tabindex="-1"></a><span class="do">## Order results by padj values</span></span>
<span id="cb233-2"><a href="codebook.html#cb233-2" tabindex="-1"></a>top20_sigOE_genes <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> </span>
<span id="cb233-3"><a href="codebook.html#cb233-3" tabindex="-1"></a>        <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span>   </span>
<span id="cb233-4"><a href="codebook.html#cb233-4" tabindex="-1"></a><span class="co">#Arrange rows by padj values</span></span>
<span id="cb233-5"><a href="codebook.html#cb233-5" tabindex="-1"></a>        <span class="fu">pull</span>(gene) <span class="sc">%&gt;%</span>      </span>
<span id="cb233-6"><a href="codebook.html#cb233-6" tabindex="-1"></a><span class="co">#Extract character vector of ordered genes</span></span>
<span id="cb233-7"><a href="codebook.html#cb233-7" tabindex="-1"></a>        <span class="fu">head</span>(<span class="at">n=</span><span class="dv">20</span>)      </span>
<span id="cb233-8"><a href="codebook.html#cb233-8" tabindex="-1"></a><span class="co">#Extract the first 20 genes</span></span></code></pre></div>
<p>Then, we can extract the normalized count values for these top 20 genes:</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="codebook.html#cb234-1" tabindex="-1"></a><span class="do">## normalized counts for top 20 significant genes</span></span>
<span id="cb234-2"><a href="codebook.html#cb234-2" tabindex="-1"></a>top20_sigOE_norm <span class="ot">&lt;-</span> normalized_counts <span class="sc">%&gt;%</span></span>
<span id="cb234-3"><a href="codebook.html#cb234-3" tabindex="-1"></a>        dplyr<span class="sc">::</span><span class="fu">filter</span>(gene <span class="sc">%in%</span> top20_sigOE_genes)</span></code></pre></div>
<p>Now that we have the normalized counts for each of the top 20 genes for all 8 samples, to plot using <code>ggplot()</code>, we need to <code>pivot_longer</code> top20_sigOE_norm from a wide format to a long format so the counts for all samples will be in a single column to allow us to give ggplot the one column with the values we want it to plot.</p>
<p>The <code>pivot_longer()</code> function in the <strong>tidyr</strong> package will perform this operation and will output the normalized counts for all genes for <em>Mov10_oe_1</em> listed in the first 20 rows, followed by the normalized counts for <em>Mov10_oe_2</em> in the next 20 rows, so on and so forth.</p>
<p><img src="img/melt_wide_to_long_format.png" width="800"></p>
<div class="sourceCode" id="cb235"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb235-1"><a href="codebook.html#cb235-1" tabindex="-1"></a><span class="co"># Pivoting the columns to have normalized counts to a single column</span></span>
<span id="cb235-2"><a href="codebook.html#cb235-2" tabindex="-1"></a></span>
<span id="cb235-3"><a href="codebook.html#cb235-3" tabindex="-1"></a>pivoted_top20_sigOE <span class="ot">&lt;-</span> top20_sigOE_norm <span class="sc">%&gt;%</span></span>
<span id="cb235-4"><a href="codebook.html#cb235-4" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">colnames</span>(top20_sigOE_norm)[<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>], <span class="at">names_to =</span> <span class="st">&quot;samplename&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;normalized_counts&quot;</span>)</span>
<span id="cb235-5"><a href="codebook.html#cb235-5" tabindex="-1"></a></span>
<span id="cb235-6"><a href="codebook.html#cb235-6" tabindex="-1"></a><span class="do">## check the column header in the &quot;pivoted&quot; data frame</span></span>
<span id="cb235-7"><a href="codebook.html#cb235-7" tabindex="-1"></a><span class="fu">head</span>(pivoted_top20_sigOE)</span></code></pre></div>
<div class="kable-table">
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="left">samplename</th>
<th align="right">normalized_counts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_1</td>
<td align="right">6717.735</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_2</td>
<td align="right">6454.641</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_3</td>
<td align="right">6801.543</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_kd_2</td>
<td align="right">6869.168</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_kd_3</td>
<td align="right">8730.977</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_oe_1</td>
<td align="right">13252.239</td>
</tr>
</tbody>
</table>
</div>
<p>Now, if we want our counts colored by sample group, then we need to combine the metadata information with the melted normalized counts data into the same data frame for input to <code>ggplot()</code>:</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="codebook.html#cb236-1" tabindex="-1"></a><span class="co"># `inner_join(x,y)` will merge 2 data frames by the colname in x that matches a column name in y. </span></span>
<span id="cb236-2"><a href="codebook.html#cb236-2" tabindex="-1"></a><span class="co"># it is similar to doing a match(pivoted_top20_sigOE$samplename, mov10_meta$samplename) and then adding sampletype and MOV10exr to pivoted_top20_sigOE, but it is more efficient because the columns from mov10_meta are added in a single command.</span></span>
<span id="cb236-3"><a href="codebook.html#cb236-3" tabindex="-1"></a></span>
<span id="cb236-4"><a href="codebook.html#cb236-4" tabindex="-1"></a>pivoted_top20_sigOE <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(mov10_meta, pivoted_top20_sigOE, <span class="at">by =</span> <span class="st">&quot;samplename&quot;</span>)</span></code></pre></div>
<p>Now that we have a data frame in a format that can be utilized by ggplot easily, letâs plot!</p>
<div class="sourceCode" id="cb237"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb237-1"><a href="codebook.html#cb237-1" tabindex="-1"></a><span class="do">## plot using ggplot2</span></span>
<span id="cb237-2"><a href="codebook.html#cb237-2" tabindex="-1"></a><span class="fu">ggplot</span>(pivoted_top20_sigOE) <span class="sc">+</span></span>
<span id="cb237-3"><a href="codebook.html#cb237-3" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> normalized_counts, <span class="at">color =</span> sampletype)) <span class="sc">+</span></span>
<span id="cb237-4"><a href="codebook.html#cb237-4" tabindex="-1"></a>        <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb237-5"><a href="codebook.html#cb237-5" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb237-6"><a href="codebook.html#cb237-6" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;log10 Normalized Counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb237-7"><a href="codebook.html#cb237-7" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Top 20 Significant DE Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb237-8"><a href="codebook.html#cb237-8" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb237-9"><a href="codebook.html#cb237-9" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb237-10"><a href="codebook.html#cb237-10" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-79-1.png" width="672" /></p>
</div>
</div>
<div id="heatmap-1" class="section level3 hasAnchor" number="9.6.2">
<h3><span class="header-section-number">9.6.2</span> Heatmap<a href="codebook.html#heatmap-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In addition to plotting subsets, we could also extract the normalized values of <em>all</em> the significant genes and plot a heatmap of their expression using <code>pheatmap()</code>.</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="codebook.html#cb238-1" tabindex="-1"></a><span class="do">### Extract normalized expression for significant genes from the OE and control samples c(2:4,7:9), and set the gene column (1) to row names</span></span>
<span id="cb238-2"><a href="codebook.html#cb238-2" tabindex="-1"></a></span>
<span id="cb238-3"><a href="codebook.html#cb238-3" tabindex="-1"></a>norm_OEsig <span class="ot">&lt;-</span> normalized_counts[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb238-4"><a href="codebook.html#cb238-4" tabindex="-1"></a>              <span class="fu">filter</span>(gene <span class="sc">%in%</span> sigOE<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> </span>
<span id="cb238-5"><a href="codebook.html#cb238-5" tabindex="-1"></a>          <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb238-6"><a href="codebook.html#cb238-6" tabindex="-1"></a>          <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) </span></code></pre></div>
<p>Now letâs draw the heatmap using pheatmap:</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="codebook.html#cb239-1" tabindex="-1"></a><span class="do">### Annotate our heatmap (optional)</span></span>
<span id="cb239-2"><a href="codebook.html#cb239-2" tabindex="-1"></a></span>
<span id="cb239-3"><a href="codebook.html#cb239-3" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> mov10_meta <span class="sc">%&gt;%</span> </span>
<span id="cb239-4"><a href="codebook.html#cb239-4" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(samplename, sampletype) <span class="sc">%&gt;%</span> </span>
<span id="cb239-5"><a href="codebook.html#cb239-5" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">row.names =</span> <span class="st">&quot;samplename&quot;</span>)</span>
<span id="cb239-6"><a href="codebook.html#cb239-6" tabindex="-1"></a></span>
<span id="cb239-7"><a href="codebook.html#cb239-7" tabindex="-1"></a><span class="do">### Set a color palette</span></span>
<span id="cb239-8"><a href="codebook.html#cb239-8" tabindex="-1"></a>heat_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">6</span>, <span class="st">&quot;YlOrRd&quot;</span>)</span>
<span id="cb239-9"><a href="codebook.html#cb239-9" tabindex="-1"></a></span>
<span id="cb239-10"><a href="codebook.html#cb239-10" tabindex="-1"></a><span class="do">### Run pheatmap</span></span>
<span id="cb239-11"><a href="codebook.html#cb239-11" tabindex="-1"></a><span class="fu">pheatmap</span>(norm_OEsig, </span>
<span id="cb239-12"><a href="codebook.html#cb239-12" tabindex="-1"></a>         <span class="at">color =</span> heat_colors, </span>
<span id="cb239-13"><a href="codebook.html#cb239-13" tabindex="-1"></a>         <span class="at">cluster_rows =</span> T, </span>
<span id="cb239-14"><a href="codebook.html#cb239-14" tabindex="-1"></a>         <span class="at">show_rownames =</span> F,</span>
<span id="cb239-15"><a href="codebook.html#cb239-15" tabindex="-1"></a>         <span class="at">annotation =</span> annotation, </span>
<span id="cb239-16"><a href="codebook.html#cb239-16" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="cn">NA</span>, </span>
<span id="cb239-17"><a href="codebook.html#cb239-17" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">10</span>, </span>
<span id="cb239-18"><a href="codebook.html#cb239-18" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, </span>
<span id="cb239-19"><a href="codebook.html#cb239-19" tabindex="-1"></a>         <span class="at">fontsize_row =</span> <span class="dv">10</span>, </span>
<span id="cb239-20"><a href="codebook.html#cb239-20" tabindex="-1"></a>         <span class="at">height =</span> <span class="dv">20</span>)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-81-1.png" width="672" /></p>
<div id="volcano-plot-1" class="section level4 hasAnchor" number="9.6.2.1">
<h4><span class="header-section-number">9.6.2.1</span> Volcano plot<a href="codebook.html#volcano-plot-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The above plot would be great to look at the expression levels of a good number of genes, but for more of a global view there are other plots we can draw. A commonly used one is a volcano plot; in which you have the log transformed adjusted p-values plotted on the y-axis and log2 fold change values on the x-axis.</p>
<p>To generate a volcano plot (which you have the log transformed adjusted p-values plotted on the y-axis and log2 fold change values on the x-axis), we first need to have a column in our results data indicating whether or not the gene is considered differentially expressed based on p-adjusted values.</p>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="codebook.html#cb240-1" tabindex="-1"></a><span class="do">## Obtain logical vector where TRUE values denote padj values &lt; 0.05 and fold change &gt; 1.5 in either direction</span></span>
<span id="cb240-2"><a href="codebook.html#cb240-2" tabindex="-1"></a></span>
<span id="cb240-3"><a href="codebook.html#cb240-3" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> </span>
<span id="cb240-4"><a href="codebook.html#cb240-4" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">threshold_OE =</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span>                    <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;=</span> <span class="fl">0.58</span>)</span></code></pre></div>
<p>Now we can start plotting:</p>
<div class="sourceCode" id="cb241"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb241-1"><a href="codebook.html#cb241-1" tabindex="-1"></a><span class="do">## Volcano plot</span></span>
<span id="cb241-2"><a href="codebook.html#cb241-2" tabindex="-1"></a></span>
<span id="cb241-3"><a href="codebook.html#cb241-3" tabindex="-1"></a><span class="fu">ggplot</span>(res_tableOE_tb) <span class="sc">+</span></span>
<span id="cb241-4"><a href="codebook.html#cb241-4" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, </span>
<span id="cb241-5"><a href="codebook.html#cb241-5" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">colour =</span> threshold_OE)) <span class="sc">+</span></span>
<span id="cb241-6"><a href="codebook.html#cb241-6" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Mov10 overexpression&quot;</span>) <span class="sc">+</span></span>
<span id="cb241-7"><a href="codebook.html#cb241-7" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;log2 fold change&quot;</span>) <span class="sc">+</span> </span>
<span id="cb241-8"><a href="codebook.html#cb241-8" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;-log10 adjusted p-value&quot;</span>) <span class="sc">+</span></span>
<span id="cb241-9"><a href="codebook.html#cb241-9" tabindex="-1"></a>        <span class="co">#scale_y_continuous(limits = c(0,50)) +</span></span>
<span id="cb241-10"><a href="codebook.html#cb241-10" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb241-11"><a href="codebook.html#cb241-11" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.5</span>), <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb241-12"><a href="codebook.html#cb241-12" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.25</span>)))</span></code></pre></div>
<pre><code>## Warning: Removed 7791 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-83-1.png" width="672" /></p>
<p>What if we also wanted to know where the top 10 genes (lowest padj) in our DE list are located on this plot? We could label those dots with the gene name on the Volcano plot using <code>geom_text_repel()</code>.</p>
<p>First, we need to order the res_tableOE tibble by padj, and add an additional column to it, to include on those gene names we want to use to label the plot.</p>
<div class="sourceCode" id="cb243"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb243-1"><a href="codebook.html#cb243-1" tabindex="-1"></a><span class="do">## Create a column to indicate which genes to label</span></span>
<span id="cb243-2"><a href="codebook.html#cb243-2" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">genelabels =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb243-3"><a href="codebook.html#cb243-3" tabindex="-1"></a></span>
<span id="cb243-4"><a href="codebook.html#cb243-4" tabindex="-1"></a>res_tableOE_tb<span class="sc">$</span>genelabels[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>] <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb243-5"><a href="codebook.html#cb243-5" tabindex="-1"></a></span>
<span id="cb243-6"><a href="codebook.html#cb243-6" tabindex="-1"></a><span class="fu">head</span>(res_tableOE_tb)</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="11%" />
<col width="12%" />
<col width="18%" />
<col width="12%" />
<col width="8%" />
<col width="6%" />
<col width="16%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">threshold_OE</th>
<th align="left">genelabels</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MOV10</td>
<td align="right">21681.800</td>
<td align="right">5.0796303</td>
<td align="right">0.1101145</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">MOV10</td>
</tr>
<tr class="even">
<td align="left">H1F0</td>
<td align="right">7881.081</td>
<td align="right">1.5517867</td>
<td align="right">0.0565777</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">H1F0</td>
</tr>
<tr class="odd">
<td align="left">HIST1H1C</td>
<td align="right">1741.383</td>
<td align="right">1.5233152</td>
<td align="right">0.0706090</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">HIST1H1C</td>
</tr>
<tr class="even">
<td align="left">TXNIP</td>
<td align="right">5133.749</td>
<td align="right">1.4193814</td>
<td align="right">0.0696576</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">TXNIP</td>
</tr>
<tr class="odd">
<td align="left">NEAT1</td>
<td align="right">21973.706</td>
<td align="right">0.9162823</td>
<td align="right">0.0466942</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">NEAT1</td>
</tr>
<tr class="even">
<td align="left">KLF10</td>
<td align="right">1694.211</td>
<td align="right">1.2332114</td>
<td align="right">0.0651896</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">KLF10</td>
</tr>
</tbody>
</table>
</div>
<p>Next, we plot it as before with an additional layer for <code>geom_text_repel()</code> wherein we can specify the column of gene labels we just created.</p>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="codebook.html#cb244-1" tabindex="-1"></a><span class="fu">ggplot</span>(res_tableOE_tb, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, </span>
<span id="cb244-2"><a href="codebook.html#cb244-2" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb244-3"><a href="codebook.html#cb244-3" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> threshold_OE)) <span class="sc">+</span></span>
<span id="cb244-4"><a href="codebook.html#cb244-4" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> genelabels)) <span class="sc">+</span></span>
<span id="cb244-5"><a href="codebook.html#cb244-5" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Mov10 overexpression&quot;</span>) <span class="sc">+</span></span>
<span id="cb244-6"><a href="codebook.html#cb244-6" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;log2 fold change&quot;</span>) <span class="sc">+</span> </span>
<span id="cb244-7"><a href="codebook.html#cb244-7" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;-log10 adjusted p-value&quot;</span>) <span class="sc">+</span></span>
<span id="cb244-8"><a href="codebook.html#cb244-8" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb244-9"><a href="codebook.html#cb244-9" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.5</span>), <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb244-10"><a href="codebook.html#cb244-10" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.25</span>))) </span></code></pre></div>
<pre><code>## Warning: Removed 7791 rows containing missing values or values outside the scale range
## (`geom_point()`).</code></pre>
<pre><code>## Warning: Removed 7791 rows containing missing values or values outside the scale range
## (`geom_text_repel()`).</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-85-1.png" width="672" /></p>
</div>
</div>
</div>
<div id="summary-of-differential-expression-analysis-workflow" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Summary of differential expression analysis workflow<a href="codebook.html#summary-of-differential-expression-analysis-workflow" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 07-DGE_summarizing_workflow.md</strong></p>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="codebook.html#cb247-1" tabindex="-1"></a><span class="do">## Setup</span></span>
<span id="cb247-2"><a href="codebook.html#cb247-2" tabindex="-1"></a><span class="do">### Bioconductor and CRAN libraries used</span></span>
<span id="cb247-3"><a href="codebook.html#cb247-3" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<p>We have detailed the various steps in a differential expression analysis workflow, providing theory with example code. To provide a more succinct reference for the code needed to run a DGE analysis, we have summarized the steps in an analysis below:</p>
<div id="import-data-into-dds-object" class="section level3 hasAnchor" number="9.7.1">
<h3><span class="header-section-number">9.7.1</span> 1. Import data into dds object:<a href="codebook.html#import-data-into-dds-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="codebook.html#cb248-1" tabindex="-1"></a><span class="co"># Check that the row names of the metadata equal the column names of the **raw counts** data</span></span>
<span id="cb248-2"><a href="codebook.html#cb248-2" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(raw_counts) <span class="sc">==</span> <span class="fu">rownames</span>(metadata))</span>
<span id="cb248-3"><a href="codebook.html#cb248-3" tabindex="-1"></a></span>
<span id="cb248-4"><a href="codebook.html#cb248-4" tabindex="-1"></a><span class="co"># Create DESeq2Dataset object</span></span>
<span id="cb248-5"><a href="codebook.html#cb248-5" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> raw_counts, <span class="at">colData =</span> metadata, <span class="at">design =</span> <span class="sc">~</span> condition)</span></code></pre></div>
</div>
<div id="exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data" class="section level3 hasAnchor" number="9.7.2">
<h3><span class="header-section-number">9.7.2</span> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:<a href="codebook.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="codebook.html#cb249-1" tabindex="-1"></a><span class="co"># Transform counts for data visualization</span></span>
<span id="cb249-2"><a href="codebook.html#cb249-2" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>)</span>
<span id="cb249-3"><a href="codebook.html#cb249-3" tabindex="-1"></a></span>
<span id="cb249-4"><a href="codebook.html#cb249-4" tabindex="-1"></a><span class="co"># Plot PCA</span></span>
<span id="cb249-5"><a href="codebook.html#cb249-5" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>)</span>
<span id="cb249-6"><a href="codebook.html#cb249-6" tabindex="-1"></a></span>
<span id="cb249-7"><a href="codebook.html#cb249-7" tabindex="-1"></a><span class="co"># Extract the rlog matrix from the object</span></span>
<span id="cb249-8"><a href="codebook.html#cb249-8" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb249-9"><a href="codebook.html#cb249-9" tabindex="-1"></a></span>
<span id="cb249-10"><a href="codebook.html#cb249-10" tabindex="-1"></a><span class="co"># Compute pairwise correlation values</span></span>
<span id="cb249-11"><a href="codebook.html#cb249-11" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb249-12"><a href="codebook.html#cb249-12" tabindex="-1"></a></span>
<span id="cb249-13"><a href="codebook.html#cb249-13" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb249-14"><a href="codebook.html#cb249-14" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
</div>
<div id="run-deseq2" class="section level3 hasAnchor" number="9.7.3">
<h3><span class="header-section-number">9.7.3</span> 3. Run DESeq2:<a href="codebook.html#run-deseq2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb250"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb250-1"><a href="codebook.html#cb250-1" tabindex="-1"></a><span class="co"># **Optional step** - Re-create DESeq2 dataset if the design formula has changed after QC analysis in include other sources of variation</span></span>
<span id="cb250-2"><a href="codebook.html#cb250-2" tabindex="-1"></a></span>
<span id="cb250-3"><a href="codebook.html#cb250-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> raw_counts, <span class="at">colData =</span> metadata, <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb250-4"><a href="codebook.html#cb250-4" tabindex="-1"></a></span>
<span id="cb250-5"><a href="codebook.html#cb250-5" tabindex="-1"></a><span class="co"># Run DESeq2 differential expression analysis</span></span>
<span id="cb250-6"><a href="codebook.html#cb250-6" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb250-7"><a href="codebook.html#cb250-7" tabindex="-1"></a></span>
<span id="cb250-8"><a href="codebook.html#cb250-8" tabindex="-1"></a><span class="co">#  **Optional step** - Output normalized counts to save as a file to access outside RStudio</span></span>
<span id="cb250-9"><a href="codebook.html#cb250-9" tabindex="-1"></a>    normalized_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>)</span>
<span id="cb250-10"><a href="codebook.html#cb250-10" tabindex="-1"></a>    </span>
<span id="cb250-11"><a href="codebook.html#cb250-11" tabindex="-1"></a><span class="fu">write.table</span>(normalized_counts, <span class="at">file=</span><span class="st">&quot;data/normalized_counts.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote=</span>F, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
<div id="check-the-fit-of-the-dispersion-estimates" class="section level3 hasAnchor" number="9.7.4">
<h3><span class="header-section-number">9.7.4</span> 4. Check the fit of the dispersion estimates:<a href="codebook.html#check-the-fit-of-the-dispersion-estimates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="codebook.html#cb251-1" tabindex="-1"></a><span class="co"># Plot dispersion estimates</span></span>
<span id="cb251-2"><a href="codebook.html#cb251-2" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span></code></pre></div>
</div>
<div id="create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions" class="section level3 hasAnchor" number="9.7.5">
<h3><span class="header-section-number">9.7.5</span> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:<a href="codebook.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb252"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb252-1"><a href="codebook.html#cb252-1" tabindex="-1"></a><span class="co"># Output results of Wald test for contrast</span></span>
<span id="cb252-2"><a href="codebook.html#cb252-2" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>, <span class="st">&quot;level_to_compare&quot;</span>, <span class="st">&quot;base_level&quot;</span>)</span>
<span id="cb252-3"><a href="codebook.html#cb252-3" tabindex="-1"></a></span>
<span id="cb252-4"><a href="codebook.html#cb252-4" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> contrast)</span>
<span id="cb252-5"><a href="codebook.html#cb252-5" tabindex="-1"></a></span>
<span id="cb252-6"><a href="codebook.html#cb252-6" tabindex="-1"></a>coef <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)</span>
<span id="cb252-7"><a href="codebook.html#cb252-7" tabindex="-1"></a></span>
<span id="cb252-8"><a href="codebook.html#cb252-8" tabindex="-1"></a>res_tableOE <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef =</span> coef[?], <span class="at">res=</span>res, <span class="at">type =</span> <span class="st">&quot;apeglm&quot;</span>)</span></code></pre></div>
</div>
<div id="output-significant-results" class="section level3 hasAnchor" number="9.7.6">
<h3><span class="header-section-number">9.7.6</span> 6. Output significant results:<a href="codebook.html#output-significant-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="codebook.html#cb253-1" tabindex="-1"></a><span class="do">### Set thresholds</span></span>
<span id="cb253-2"><a href="codebook.html#cb253-2" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb253-3"><a href="codebook.html#cb253-3" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fl">0.58</span> <span class="do">## change in expression of 1.5</span></span>
<span id="cb253-4"><a href="codebook.html#cb253-4" tabindex="-1"></a></span>
<span id="cb253-5"><a href="codebook.html#cb253-5" tabindex="-1"></a><span class="co"># Turn the results object into a data frame</span></span>
<span id="cb253-6"><a href="codebook.html#cb253-6" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb253-7"><a href="codebook.html#cb253-7" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb253-8"><a href="codebook.html#cb253-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) </span>
<span id="cb253-9"><a href="codebook.html#cb253-9" tabindex="-1"></a></span>
<span id="cb253-10"><a href="codebook.html#cb253-10" tabindex="-1"></a><span class="co"># Subset the significant results</span></span>
<span id="cb253-11"><a href="codebook.html#cb253-11" tabindex="-1"></a>sig_res <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res_df, padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span> <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span></code></pre></div>
</div>
<div id="visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc." class="section level3 hasAnchor" number="9.7.7">
<h3><span class="header-section-number">9.7.7</span> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.<a href="codebook.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc." class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis" class="section level3 hasAnchor" number="9.7.8">
<h3><span class="header-section-number">9.7.8</span> 8. Make sure to output the versions of all tools used in the DE analysis:<a href="codebook.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="codebook.html#cb254-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.4.1 (2024-06-14)
## Platform: x86_64-apple-darwin20
## Running under: macOS Sonoma 14.6.1
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: America/Vancouver
## tzcode source: internal
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] ggrepel_0.9.5               pheatmap_1.0.12            
##  [3] RColorBrewer_1.1-3          GOenrichment_0.5.0         
##  [5] igraph_2.0.3                DESeq2_1.44.0              
##  [7] SummarizedExperiment_1.34.0 MatrixGenerics_1.16.0      
##  [9] matrixStats_1.3.0           GenomicRanges_1.56.1       
## [11] GenomeInfoDb_1.40.1         fgsea_1.30.0               
## [13] enrichplot_1.24.2           lubridate_1.9.3            
## [15] forcats_1.0.0               stringr_1.5.1              
## [17] dplyr_1.1.4                 purrr_1.0.2                
## [19] readr_2.1.5                 tidyr_1.3.1                
## [21] tibble_3.2.1                ggplot2_3.5.1              
## [23] tidyverse_2.0.0             clusterProfiler_4.12.3     
## [25] org.Hs.eg.db_3.19.1         AnnotationDbi_1.66.0       
## [27] IRanges_2.38.1              S4Vectors_0.42.1           
## [29] Biobase_2.64.0              BiocGenerics_0.50.0        
## [31] bookdown_0.40              
## 
## loaded via a namespace (and not attached):
##   [1] rstudioapi_0.16.0       jsonlite_1.8.8          magrittr_2.0.3         
##   [4] farver_2.1.2            rmarkdown_2.28          fs_1.6.4               
##   [7] zlibbioc_1.50.0         vctrs_0.6.5             memoise_2.0.1          
##  [10] ggtree_3.12.0           S4Arrays_1.4.1          htmltools_0.5.8.1      
##  [13] SparseArray_1.4.8       gridGraphics_0.5-1      sass_0.4.9             
##  [16] bslib_0.8.0             htmlwidgets_1.6.4       plyr_1.8.9             
##  [19] httr2_1.0.2             cachem_1.1.0            lifecycle_1.0.4        
##  [22] pkgconfig_2.0.3         Matrix_1.7-0            R6_2.5.1               
##  [25] fastmap_1.2.0           gson_0.1.0              GenomeInfoDbData_1.2.12
##  [28] numDeriv_2016.8-1.1     digest_0.6.37           aplot_0.2.3            
##  [31] ggnewscale_0.5.0        colorspace_2.1-1        patchwork_1.2.0        
##  [34] RSQLite_2.3.7           labeling_0.4.3          fansi_1.0.6            
##  [37] timechange_0.3.0        abind_1.4-5             httr_1.4.7             
##  [40] polyclip_1.10-7         compiler_4.4.1          bit64_4.0.5            
##  [43] withr_3.0.1             BiocParallel_1.38.0     viridis_0.6.5          
##  [46] DBI_1.2.3               highr_0.11              ggforce_0.4.2          
##  [49] MASS_7.3-61             DelayedArray_0.30.1     rappdirs_0.3.3         
##  [52] HDO.db_0.99.1           tools_4.4.1             ape_5.8                
##  [55] scatterpie_0.2.3        glue_1.7.0              nlme_3.1-166           
##  [58] GOSemSim_2.30.1         rsconnect_1.3.1         grid_4.4.1             
##  [61] shadowtext_0.1.4        reshape2_1.4.4          generics_0.1.3         
##  [64] gtable_0.3.5            tzdb_0.4.0              data.table_1.15.4      
##  [67] hms_1.1.3               tidygraph_1.3.1         utf8_1.2.4             
##  [70] XVector_0.44.0          pillar_1.9.0            emdbook_1.3.13         
##  [73] yulab.utils_0.1.6       splines_4.4.1           tweenr_2.0.3           
##  [76] treeio_1.28.0           lattice_0.22-6          bit_4.0.5              
##  [79] tidyselect_1.2.1        locfit_1.5-9.10         GO.db_3.19.1           
##  [82] Biostrings_2.72.1       knitr_1.48              gridExtra_2.3          
##  [85] xfun_0.47               graphlayouts_1.1.1      visNetwork_2.1.2       
##  [88] stringi_1.8.4           UCSC.utils_1.0.0        lazyeval_0.2.2         
##  [91] ggfun_0.1.5             yaml_2.3.10             evaluate_0.24.0        
##  [94] codetools_0.2-20        bbmle_1.0.25.1          ggraph_2.2.1           
##  [97] qvalue_2.36.0           ggplotify_0.1.2         cli_3.6.3              
## [100] munsell_0.5.1           jquerylib_0.1.4         Rcpp_1.0.13            
## [103] coda_0.19-4.1           png_0.1-8               bdsmatrix_1.3-7        
## [106] parallel_4.4.1          blob_1.2.4              DOSE_3.30.2            
## [109] mvtnorm_1.2-6           apeglm_1.26.1           viridisLite_0.4.2      
## [112] tidytree_0.4.6          scales_1.3.0            crayon_1.5.3           
## [115] rlang_1.1.4             cowplot_1.1.3           fastmatch_1.1-4        
## [118] KEGGREST_1.44.1</code></pre>
<p><strong>Exercise/Homework: modify this file to analyze the MOV dataset. Include a heatmap and a volcano plot</strong></p>
</div>
</div>
<div id="functional-analysis-of-rna-seq-data" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> Functional analysis of RNA-seq data<a href="codebook.html#functional-analysis-of-rna-seq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.md file = 08-GO_enrichment_analysis.md</strong></p>
<p>Functional analysis tools are useful in interpreting resulting DGE gene lists from RNAseq, and fall into three main types:</p>
<ol style="list-style-type: decimal">
<li>Over-representation analysis</li>
<li>Functional class scoring</li>
<li>Pathway topology</li>
</ol>
<div id="over-representation-analysis-1" class="section level3 hasAnchor" number="9.8.1">
<h3><span class="header-section-number">9.8.1</span> Over-representation analysis<a href="codebook.html#over-representation-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To determine whether any GO terms are over-represented in a significant gene list, the probability of the observed proportion of genes associated with a specific term is compared to the probability of the background set of genes belonging to the same term. This statistical test is known as the âhypergeometric testâ.</p>
<p>We will be using <a href="http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a></p>
</div>
<div id="clusterprofiler-1" class="section level3 hasAnchor" number="9.8.2">
<h3><span class="header-section-number">9.8.2</span> clusterProfiler<a href="codebook.html#clusterprofiler-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb256"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb256-1"><a href="codebook.html#cb256-1" tabindex="-1"></a><span class="co"># you may have to install some of these libraries; use </span></span>
<span id="cb256-2"><a href="codebook.html#cb256-2" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;org.Hs.eg.db&quot;,&quot;clusterProfiler&quot;,&quot;enrichplot&quot;,&quot;fgsea&quot;))</span></span>
<span id="cb256-3"><a href="codebook.html#cb256-3" tabindex="-1"></a></span>
<span id="cb256-4"><a href="codebook.html#cb256-4" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb256-5"><a href="codebook.html#cb256-5" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb256-6"><a href="codebook.html#cb256-6" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb256-7"><a href="codebook.html#cb256-7" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb256-8"><a href="codebook.html#cb256-8" tabindex="-1"></a><span class="fu">library</span>(fgsea) </span></code></pre></div>
<div id="running-clusterprofiler-1" class="section level4 hasAnchor" number="9.8.2.1">
<h4><span class="header-section-number">9.8.2.1</span> Running clusterProfiler<a href="codebook.html#running-clusterprofiler-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="codebook.html#cb257-1" tabindex="-1"></a>res_tableOE <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/res_tableOE.rds&quot;</span>)</span>
<span id="cb257-2"><a href="codebook.html#cb257-2" tabindex="-1"></a></span>
<span id="cb257-3"><a href="codebook.html#cb257-3" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span></span>
<span id="cb257-4"><a href="codebook.html#cb257-4" tabindex="-1"></a><span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb257-5"><a href="codebook.html#cb257-5" tabindex="-1"></a><span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb257-6"><a href="codebook.html#cb257-6" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange))  <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>To perform the over-representation analysis, we need a list of background genes and a list of significant genes. For our background dataset we will use all genes tested for differential expression (all genes in our results table). For our significant gene list we will use genes with p-adjusted values less than 0.05 (we could include a fold change threshold too if we have many DE genes).</p>
<div class="sourceCode" id="cb258"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb258-1"><a href="codebook.html#cb258-1" tabindex="-1"></a><span class="do">## background set of ensgenes</span></span>
<span id="cb258-2"><a href="codebook.html#cb258-2" tabindex="-1"></a>allOE_genes <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span>
<span id="cb258-3"><a href="codebook.html#cb258-3" tabindex="-1"></a></span>
<span id="cb258-4"><a href="codebook.html#cb258-4" tabindex="-1"></a>sigOE <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res_tableOE_tb, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb258-5"><a href="codebook.html#cb258-5" tabindex="-1"></a></span>
<span id="cb258-6"><a href="codebook.html#cb258-6" tabindex="-1"></a>sigOE_genes <span class="ot">=</span> sigOE<span class="sc">$</span>gene</span></code></pre></div>
<p>Now we can perform the GO enrichment analysis and save the results:</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="codebook.html#cb259-1" tabindex="-1"></a><span class="do">## Run GO enrichment analysis</span></span>
<span id="cb259-2"><a href="codebook.html#cb259-2" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> sigOE_genes,</span>
<span id="cb259-3"><a href="codebook.html#cb259-3" tabindex="-1"></a>                <span class="at">universe =</span> allOE_genes,</span>
<span id="cb259-4"><a href="codebook.html#cb259-4" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb259-5"><a href="codebook.html#cb259-5" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb259-6"><a href="codebook.html#cb259-6" tabindex="-1"></a>                <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb259-7"><a href="codebook.html#cb259-7" tabindex="-1"></a>                <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb259-8"><a href="codebook.html#cb259-8" tabindex="-1"></a>                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb259-9"><a href="codebook.html#cb259-9" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb259-10"><a href="codebook.html#cb259-10" tabindex="-1"></a>                <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb259-11"><a href="codebook.html#cb259-11" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb259-12"><a href="codebook.html#cb259-12" tabindex="-1"></a></span>
<span id="cb259-13"><a href="codebook.html#cb259-13" tabindex="-1"></a><span class="do">## Output results from GO analysis to a table</span></span>
<span id="cb259-14"><a href="codebook.html#cb259-14" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ego)</span>
<span id="cb259-15"><a href="codebook.html#cb259-15" tabindex="-1"></a></span>
<span id="cb259-16"><a href="codebook.html#cb259-16" tabindex="-1"></a><span class="do">## make sure you have a results directory</span></span>
<span id="cb259-17"><a href="codebook.html#cb259-17" tabindex="-1"></a><span class="fu">write.csv</span>(cluster_summary, <span class="st">&quot;results/clusterProfiler_Mov10oe.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="visualizing-clusterprofiler-results-1" class="section level4 hasAnchor" number="9.8.2.2">
<h4><span class="header-section-number">9.8.2.2</span> Visualizing clusterProfiler results<a href="codebook.html#visualizing-clusterprofiler-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>dotplot</strong></p>
<p>The dotplot shows the number of genes associated with the first 50 terms (size) and the p-adjusted values for these terms (color). This plot displays the top 50 genes by gene ratio (# genes related to GO term / total number of sig genes), not p-adjusted value.</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="codebook.html#cb260-1" tabindex="-1"></a><span class="do">## Dotplot</span></span>
<span id="cb260-2"><a href="codebook.html#cb260-2" tabindex="-1"></a><span class="do">## </span></span>
<span id="cb260-3"><a href="codebook.html#cb260-3" tabindex="-1"></a><span class="fu">dotplot</span>(ego, <span class="at">showCategory=</span><span class="dv">50</span>)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-98-1.png" width="672" />
To save the figure, click on the Export button in the RStudio Plots tab and Save as PDFâ¦.set PDF size to 8 x 14 to give a figure of appropriate size for the text labels</p>
<p><strong>enrichment GO plot</strong></p>
<p>The next plot is the enrichment GO plot, which shows the relationship between the top 50 most significantly enriched GO terms (padj.), by grouping similar terms together. The color represents the p-values relative to the other displayed terms (brighter red is more significant) and the size of the terms represents the number of genes that are significant from our list.</p>
<p>This plot is useful because it serves to collapse the GO terms into functional categories by showing the overlap between GO terms.</p>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="codebook.html#cb261-1" tabindex="-1"></a><span class="do">## Enrichmap clusters the 50 most significant (by padj) GO terms to visualize relationships between terms</span></span>
<span id="cb261-2"><a href="codebook.html#cb261-2" tabindex="-1"></a></span>
<span id="cb261-3"><a href="codebook.html#cb261-3" tabindex="-1"></a>pwt <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(</span>
<span id="cb261-4"><a href="codebook.html#cb261-4" tabindex="-1"></a>ego,</span>
<span id="cb261-5"><a href="codebook.html#cb261-5" tabindex="-1"></a><span class="at">method =</span> <span class="st">&quot;JC&quot;</span>,</span>
<span id="cb261-6"><a href="codebook.html#cb261-6" tabindex="-1"></a><span class="at">semData =</span> <span class="cn">NULL</span>,</span>
<span id="cb261-7"><a href="codebook.html#cb261-7" tabindex="-1"></a><span class="at">showCategory =</span> <span class="dv">50</span></span>
<span id="cb261-8"><a href="codebook.html#cb261-8" tabindex="-1"></a>)</span>
<span id="cb261-9"><a href="codebook.html#cb261-9" tabindex="-1"></a></span>
<span id="cb261-10"><a href="codebook.html#cb261-10" tabindex="-1"></a><span class="fu">emapplot</span>(pwt, <span class="at">showCategory =</span> <span class="dv">50</span>)</span></code></pre></div>
<pre><code>## Warning: ggrepel: 10 unlabeled data points (too many overlaps). Consider increasing
## max.overlaps</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-99-1.png" width="672" /></p>
<p>To save the figure, click on the Export button in the RStudio Plots tab and Save as PDFâ¦. In the pop-up window, change the PDF size to 24 x 32 to give a figure of appropriate size for the text labels.</p>
<p>Finally, the category netplot shows the relationships between the genes associated with the top five most significant GO terms and the fold changes of the significant genes associated with these terms (color). The size of the GO terms reflects the pvalues of the terms, with the more significant terms being larger. This plot is particularly useful for hypothesis generation in identifying genes that may be important to several of the most affected processes.</p>
<p><strong>netplot</strong></p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="codebook.html#cb263-1" tabindex="-1"></a><span class="do">## To color genes by log2 fold changes, we need to extract the log2 fold changes from our results table creating a named vector</span></span>
<span id="cb263-2"><a href="codebook.html#cb263-2" tabindex="-1"></a>OE_foldchanges <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>log2FoldChange</span>
<span id="cb263-3"><a href="codebook.html#cb263-3" tabindex="-1"></a></span>
<span id="cb263-4"><a href="codebook.html#cb263-4" tabindex="-1"></a><span class="fu">names</span>(OE_foldchanges) <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>gene</span>
<span id="cb263-5"><a href="codebook.html#cb263-5" tabindex="-1"></a></span>
<span id="cb263-6"><a href="codebook.html#cb263-6" tabindex="-1"></a></span>
<span id="cb263-7"><a href="codebook.html#cb263-7" tabindex="-1"></a><span class="do">## Cnetplot details the genes associated with one or more terms - by default gives the top 5 significant terms (by padj)</span></span>
<span id="cb263-8"><a href="codebook.html#cb263-8" tabindex="-1"></a><span class="fu">cnetplot</span>(ego,</span>
<span id="cb263-9"><a href="codebook.html#cb263-9" tabindex="-1"></a>         <span class="at">categorySize=</span><span class="st">&quot;pvalue&quot;</span>,</span>
<span id="cb263-10"><a href="codebook.html#cb263-10" tabindex="-1"></a>         <span class="at">showCategory =</span> <span class="dv">5</span>,</span>
<span id="cb263-11"><a href="codebook.html#cb263-11" tabindex="-1"></a>         <span class="at">foldChange =</span> OE_foldchanges,</span>
<span id="cb263-12"><a href="codebook.html#cb263-12" tabindex="-1"></a>         <span class="at">vertex.label.font=</span><span class="dv">6</span>)</span></code></pre></div>
<pre><code>## Warning in cnetplot.enrichResult(x, ...): Use &#39;color.params = list(foldChange = your_value)&#39; instead of &#39;foldChange&#39;.
##  The foldChange parameter will be removed in the next version.</code></pre>
<pre><code>## Warning: ggrepel: 436 unlabeled data points (too many overlaps). Consider
## increasing max.overlaps</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-100-1.png" width="672" /></p>
<p><strong>Again, to save the figure</strong>, click on the Export button in the RStudio Plots tab and Save as PDFâ¦. Change the PDF size to 24 x 32 to give a figure of appropriate size for the text labels.</p>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea-1" class="section level3 hasAnchor" number="9.8.3">
<h3><span class="header-section-number">9.8.3</span> Gene set enrichment analysis (GSEA)<a href="codebook.html#gene-set-enrichment-analysis-gsea-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="gsea-using-clusterprofiler-1" class="section level4 hasAnchor" number="9.8.3.1">
<h4><span class="header-section-number">9.8.3.1</span> GSEA using clusterProfiler<a href="codebook.html#gsea-using-clusterprofiler-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>GSEA uses the entire list of log2 fold changes from all genes. It is based on looking for enrichment of genesets among the large positive or negative fold changes. Thus, rather than setting an arbitrary threshold to identify âsignificant genesâ, all genes are considered in the analysis. The gene-level statistics from the dataset are aggregated to generate a single pathway-level statistic and statistical significance of each pathway is reported.</p>
<p>Extract and name the fold changes:</p>
<div class="sourceCode" id="cb266"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb266-1"><a href="codebook.html#cb266-1" tabindex="-1"></a><span class="do">## Extract the foldchanges</span></span>
<span id="cb266-2"><a href="codebook.html#cb266-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>log2FoldChange</span>
<span id="cb266-3"><a href="codebook.html#cb266-3" tabindex="-1"></a></span>
<span id="cb266-4"><a href="codebook.html#cb266-4" tabindex="-1"></a><span class="do">## Name each fold change with the corresponding Entrez ID</span></span>
<span id="cb266-5"><a href="codebook.html#cb266-5" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span></code></pre></div>
<p>Next we need to order the fold changes in decreasing order. To do this weâll use the <code>sort()</code> function, which takes a vector as input. This is in contrast to Tidyverseâs <code>arrange()</code>, which requires a data frame.</p>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="codebook.html#cb267-1" tabindex="-1"></a><span class="do">## Sort fold changes in decreasing order</span></span>
<span id="cb267-2"><a href="codebook.html#cb267-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> <span class="fu">sort</span>(foldchanges, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb267-3"><a href="codebook.html#cb267-3" tabindex="-1"></a></span>
<span id="cb267-4"><a href="codebook.html#cb267-4" tabindex="-1"></a><span class="fu">head</span>(foldchanges)</span></code></pre></div>
<pre><code>##    HSPA6    MOV10    ASCL1    HSPA7    SCRT1 SIGLEC14 
## 6.246267 5.079630 4.441203 3.637040 2.925440 2.615129</code></pre>
<p>We can explore the enrichment of BP Gene Ontology terms using gene set enrichment analysis:</p>
<div class="sourceCode" id="cb269"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb269-1"><a href="codebook.html#cb269-1" tabindex="-1"></a><span class="co"># GSEA using gene sets associated with BP Gene Ontology terms</span></span>
<span id="cb269-2"><a href="codebook.html#cb269-2" tabindex="-1"></a></span>
<span id="cb269-3"><a href="codebook.html#cb269-3" tabindex="-1"></a>gseaGO <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">gseGO</span>(</span>
<span id="cb269-4"><a href="codebook.html#cb269-4" tabindex="-1"></a>  <span class="at">geneList =</span> foldchanges,</span>
<span id="cb269-5"><a href="codebook.html#cb269-5" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb269-6"><a href="codebook.html#cb269-6" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb269-7"><a href="codebook.html#cb269-7" tabindex="-1"></a>  <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb269-8"><a href="codebook.html#cb269-8" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb269-9"><a href="codebook.html#cb269-9" tabindex="-1"></a>  <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb269-10"><a href="codebook.html#cb269-10" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb269-11"><a href="codebook.html#cb269-11" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb269-12"><a href="codebook.html#cb269-12" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb269-13"><a href="codebook.html#cb269-13" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb269-14"><a href="codebook.html#cb269-14" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">&quot;fgsea&quot;</span></span>
<span id="cb269-15"><a href="codebook.html#cb269-15" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## using &#39;fgsea&#39; for GSEA analysis, please cite Korotkevich et al (2019).</code></pre>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (8.07% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="codebook.html#cb276-1" tabindex="-1"></a>gseaGO_results <span class="ot">&lt;-</span> gseaGO<span class="sc">@</span>result</span>
<span id="cb276-2"><a href="codebook.html#cb276-2" tabindex="-1"></a></span>
<span id="cb276-3"><a href="codebook.html#cb276-3" tabindex="-1"></a><span class="fu">goplot</span>(gseaGO) <span class="co"># default is 10 categories</span></span></code></pre></div>
<pre><code>## Warning: ggrepel: 84 unlabeled data points (too many overlaps). Consider increasing
## max.overlaps</code></pre>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-103-1.png" width="672" /></p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="codebook.html#cb278-1" tabindex="-1"></a><span class="fu">gseaplot2</span>(gseaGO, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="RNA-seq-analysis_files/figure-html/unnamed-chunk-103-2.png" width="672" /></p>
<p>We can also use our homemade GO enrichment analysis. To do this we need to load the library GOenrichment:</p>
<div class="sourceCode" id="cb279"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb279-1"><a href="codebook.html#cb279-1" tabindex="-1"></a><span class="co"># Uncomment the following if you haven&#39;t yet installed GOenrichment.</span></span>
<span id="cb279-2"><a href="codebook.html#cb279-2" tabindex="-1"></a></span>
<span id="cb279-3"><a href="codebook.html#cb279-3" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;gurinina/GOenrichment&quot;)</span></span>
<span id="cb279-4"><a href="codebook.html#cb279-4" tabindex="-1"></a></span>
<span id="cb279-5"><a href="codebook.html#cb279-5" tabindex="-1"></a><span class="fu">library</span>(GOenrichment)</span>
<span id="cb279-6"><a href="codebook.html#cb279-6" tabindex="-1"></a></span>
<span id="cb279-7"><a href="codebook.html#cb279-7" tabindex="-1"></a><span class="fu">ls</span>(<span class="st">&quot;package:GOenrichment&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;compSCORE&quot;     &quot;dfGOBP&quot;        &quot;hGOBP.gmt&quot;     &quot;hyperG&quot;        &quot;runGORESP&quot;    
## [6] &quot;runNetwork&quot;    &quot;sampleFitdata&quot; &quot;visSetup&quot;      &quot;yGOBP.gmt&quot;</code></pre>
<p>One of the problems with GO enrichment analysis is that the GO annotations are in constant flux.</p>
<p>Here we can use the GO annotations in <code>hGOBP.gmt</code> (downloaded recently) to run GSEA using the <code>fgsea</code> package to run GSEA:</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="codebook.html#cb281-1" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span>  fgsea<span class="sc">::</span><span class="fu">fgseaSimple</span>(<span class="at">pathways =</span> hGOBP.gmt,<span class="at">stats=</span>foldchanges,<span class="at">nperm=</span><span class="dv">1000</span>,<span class="at">maxSize =</span> <span class="dv">300</span>,<span class="at">minSize =</span> <span class="dv">20</span>)</span></code></pre></div>
<pre><code>## Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize, gseaParam, : There are ties in the preranked stats (8.07% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="codebook.html#cb283-1" tabindex="-1"></a>fgsea <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb283-2"><a href="codebook.html#cb283-2" tabindex="-1"></a></span>
<span id="cb283-3"><a href="codebook.html#cb283-3" tabindex="-1"></a>w <span class="ot">=</span> <span class="fu">which</span>(fgsea<span class="sc">$</span>ES <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb283-4"><a href="codebook.html#cb283-4" tabindex="-1"></a></span>
<span id="cb283-5"><a href="codebook.html#cb283-5" tabindex="-1"></a>fposgsea <span class="ot">&lt;-</span> fgsea[w,]</span>
<span id="cb283-6"><a href="codebook.html#cb283-6" tabindex="-1"></a></span>
<span id="cb283-7"><a href="codebook.html#cb283-7" tabindex="-1"></a>fposgsea <span class="ot">&lt;-</span> fposgsea <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(padj)</span></code></pre></div>
<p>We are going to compare these results to runing the GO enrichment function <code>runGORESP</code>. <code>runGORESP</code> uses over-representation analysis to identify enriched GO terms, so we need to define a significance cutoff for the <code>querySet</code>.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="codebook.html#cb284-1" tabindex="-1"></a><span class="fu">args</span>(runGORESP)</span></code></pre></div>
<pre><code>## function (mat, coln, curr_exp = colnames(mat)[coln], sig = 1, 
##     fdrThresh = 0.2, bp_path = NULL, bp_input = NULL, go_path = NULL, 
##     go_input = NULL, minSetSize = 5, maxSetSize = 300) 
## NULL</code></pre>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="codebook.html#cb286-1" tabindex="-1"></a>?runGORESP</span>
<span id="cb286-2"><a href="codebook.html#cb286-2" tabindex="-1"></a></span>
<span id="cb286-3"><a href="codebook.html#cb286-3" tabindex="-1"></a><span class="co"># we&#39;ll define our significance cutoff as 0.58, corresponding to 1.5x change.</span></span>
<span id="cb286-4"><a href="codebook.html#cb286-4" tabindex="-1"></a></span>
<span id="cb286-5"><a href="codebook.html#cb286-5" tabindex="-1"></a><span class="co"># `runGORESP` requires a matrix, so we can turn foldchanges into a matrix using `cbind`:</span></span>
<span id="cb286-6"><a href="codebook.html#cb286-6" tabindex="-1"></a>matx <span class="ot">&lt;-</span> <span class="fu">cbind</span>(foldchanges,foldchanges)</span>
<span id="cb286-7"><a href="codebook.html#cb286-7" tabindex="-1"></a></span>
<span id="cb286-8"><a href="codebook.html#cb286-8" tabindex="-1"></a>hresp <span class="ot">=</span> GOenrichment<span class="sc">::</span><span class="fu">runGORESP</span>(<span class="at">fdrThresh =</span> <span class="fl">0.2</span>,<span class="at">mat=</span>matx,</span>
<span id="cb286-9"><a href="codebook.html#cb286-9" tabindex="-1"></a><span class="at">coln=</span><span class="dv">1</span>,<span class="at">curr_exp =</span> <span class="fu">colnames</span>(matx)[<span class="dv">1</span>], <span class="at">sig =</span> <span class="fl">0.58</span>,</span>
<span id="cb286-10"><a href="codebook.html#cb286-10" tabindex="-1"></a><span class="at">bp_input =</span> hGOBP.gmt,<span class="at">go_input =</span> <span class="cn">NULL</span>,<span class="at">minSetSize =</span> <span class="dv">20</span>,</span>
<span id="cb286-11"><a href="codebook.html#cb286-11" tabindex="-1"></a><span class="at">maxSetSize =</span> <span class="dv">300</span>)</span>
<span id="cb286-12"><a href="codebook.html#cb286-12" tabindex="-1"></a></span>
<span id="cb286-13"><a href="codebook.html#cb286-13" tabindex="-1"></a><span class="fu">names</span>(hresp<span class="sc">$</span>edgeMat)</span></code></pre></div>
<pre><code>## [1] &quot;source&quot;       &quot;target&quot;       &quot;overlapCoeff&quot; &quot;width&quot;        &quot;label&quot;</code></pre>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="codebook.html#cb288-1" tabindex="-1"></a><span class="fu">names</span>(hresp<span class="sc">$</span>enrichInfo)</span></code></pre></div>
<pre><code>##  [1] &quot;filename&quot;            &quot;term&quot;                &quot;nGenes&quot;             
##  [4] &quot;nQuery&quot;              &quot;nOverlap&quot;            &quot;querySetFraction&quot;   
##  [7] &quot;geneSetFraction&quot;     &quot;foldEnrichment&quot;      &quot;P&quot;                  
## [10] &quot;FDR&quot;                 &quot;overlapGenes&quot;        &quot;maxOverlapGeneScore&quot;
## [13] &quot;cluster&quot;             &quot;id&quot;                  &quot;size&quot;               
## [16] &quot;formattedLabel&quot;</code></pre>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="codebook.html#cb290-1" tabindex="-1"></a><span class="co"># View(hresp$enrichInfo[,c(2,3,4,5,10)])</span></span></code></pre></div>
<p>Letâs check the overlap between the enriched terms found using <code>runGORESP</code> and those found using <code>fgseaSimple</code> as they used the same GO term libraries:</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="codebook.html#cb291-1" tabindex="-1"></a>w <span class="ot">=</span> <span class="fu">which</span>(fposgsea<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.2</span>)</span>
<span id="cb291-2"><a href="codebook.html#cb291-2" tabindex="-1"></a></span>
<span id="cb291-3"><a href="codebook.html#cb291-3" tabindex="-1"></a>lens <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(fposgsea<span class="sc">$</span>pathway[w],hresp<span class="sc">$</span>enrichInfo<span class="sc">$</span>term))</span>
<span id="cb291-4"><a href="codebook.html#cb291-4" tabindex="-1"></a></span>
<span id="cb291-5"><a href="codebook.html#cb291-5" tabindex="-1"></a><span class="fu">length</span>(w)</span></code></pre></div>
<pre><code>## [1] 622</code></pre>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="codebook.html#cb293-1" tabindex="-1"></a><span class="fu">dim</span>(hresp<span class="sc">$</span>enrichInfo)</span></code></pre></div>
<pre><code>## [1] 238  16</code></pre>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="codebook.html#cb295-1" tabindex="-1"></a>percent_overlap <span class="ot">&lt;-</span> lens<span class="sc">/</span><span class="fu">nrow</span>(hresp<span class="sc">$</span>enrichInfo)<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb295-2"><a href="codebook.html#cb295-2" tabindex="-1"></a></span>
<span id="cb295-3"><a href="codebook.html#cb295-3" tabindex="-1"></a>percent_overlap</span></code></pre></div>
<pre><code>## [1] 83.19328</code></pre>
<p>80%, thatâs very good, especially because we are using two different GO enrichment methods, over-representation analysis and GSEA. The overlap between these enrichment and the ones using the other GO enrichment tools will be very small because of the differences in the GO annotation libraries.</p>
<p>Now to set up the results for viewing in a network, we use the function <code>visSetup</code>, which creates a set of nodes and edges in the network, where nodes are GO terms (node size proportional to FDR score) and edges represent the overlap between GO terms (proportional to edge width). This network analysis is based on <a href="https://cytoscape.org/">Cytoscape</a>, an open source bioinformatics software platform for visualizing molecular interaction networks.</p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="codebook.html#cb297-1" tabindex="-1"></a>vis <span class="ot">=</span> <span class="fu">visSetup</span>(hresp<span class="sc">$</span>enrichInfo,hresp<span class="sc">$</span>edgeMat)</span>
<span id="cb297-2"><a href="codebook.html#cb297-2" tabindex="-1"></a><span class="fu">names</span>(vis)</span></code></pre></div>
<pre><code>## [1] &quot;nodes&quot; &quot;edges&quot;</code></pre>
<p>Now we use runNetwork to view the map:</p>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="codebook.html#cb299-1" tabindex="-1"></a><span class="fu">runNetwork</span>(vis<span class="sc">$</span>nodes,vis<span class="sc">$</span>edges)</span></code></pre></div>
<div id="htmlwidget-31fa5624230bf2202f94" style="width:100%;height:1300px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-31fa5624230bf2202f94">{"x":{"nodes":{"id":["1","3","4","2","7","6","5","8","10","9","11","12","13","14","15","16","17","18","19","20","21","22","23","24","27","26","25","28","29","30","31","32","33","34","35","36","37","38","39","41","40","42","43","44","45","46","47","48","49","50","51","52","54","53","55","56","59","61","57","60","58","62","64","63","65","69","68","70","66","67","71","72","73","75","74","76","77","79","78","81","80","82","83","84","93","87","92","94","88","91","90","85","86","89","95","96","99","97","101","100","98","103","102","104","105","106","107","108","109","110","111","112","114","113","118","115","116","117","119","120","121","122","123","124","127","126","128","129","125","132","130","133","131","136","137","140","134","147","135","144","145","146","141","142","138","143","139","149","150","154","148","151","153","152","155","156","160","159","158","161","157","162","163","164","165","166","167","168","169","170","173","171","172","177","174","176","175","178","179","184","181","183","180","185","186","182","187","189","190","191","195","200","199","196","193","192","188","197","198","194","201","202","212","204","208","203","211","205","209","207","215","206","210","213","214","216","219","218","217","220","224","221","223","222","225","227","230","226","229","228","231","233","235","236","232","238","237","234"],"filename":["foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges","foldchanges"],"term":["NABA_CORE_MATRISOME|MSIGDB_C2","EXTRACELLULAR MATRIX ORGANIZATION|GOBP","EXTRACELLULAR STRUCTURE ORGANIZATION|GOBP","EXTRACELLULAR MATRIX ORGANIZATION|REACTOME","BETA1 INTEGRIN CELL SURFACE INTERACTIONS|PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA","PID_INTEGRIN1_PATHWAY|MSIGDB_C2","NABA_ECM_GLYCOPROTEINS|MSIGDB_C2","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION|MSIGDB_C2","HDACS DEACETYLATE HISTONES|REACTOME DATABASE ID RELEASE 61","INTEGRIN CELL SURFACE INTERACTIONS|REACTOME DATABASE ID RELEASE 61","NABA_BASEMENT_MEMBRANES|MSIGDB_C2","HALLMARK_TNFA_SIGNALING_VIA_NFKB|MSIGDB_C2","RNA POLYMERASE I PROMOTER OPENING|REACTOME DATABASE ID RELEASE 61","AMYLOID FIBER FORMATION|REACTOME","DNA METHYLATION|REACTOME","REGULATION OF IGF ACTIVITY BY IGFBP|REACTOME DATABASE ID RELEASE 61","ACTIVATED PKN1 STIMULATES TRANSCRIPTION OF AR (ANDROGEN RECEPTOR) REGULATED GENES KLK2 AND KLK3|REACTOME DATABASE ID RELEASE 61","SIRT1 NEGATIVELY REGULATES RRNA EXPRESSION|REACTOME","O-LINKED GLYCOSYLATION|REACTOME","MULTICELLULAR ORGANISMAL SIGNALING|GOBP","CARDIAC CONDUCTION|GOBP","DNA DAMAGE TELOMERE STRESS INDUCED SENESCENCE|REACTOME","POST-TRANSLATIONAL PROTEIN PHOSPHORYLATION|REACTOME","DEFECTIVE B3GALTL CAUSES PETERS-PLUS SYNDROME (PPS)|REACTOME","PRC2 METHYLATES HISTONES AND DNA|REACTOME DATABASE ID RELEASE 61","DISEASES OF GLYCOSYLATION|REACTOME","CONDENSATION OF PROPHASE CHROMOSOMES|REACTOME DATABASE ID RELEASE 61","O-GLYCOSYLATION OF TSR DOMAIN-CONTAINING PROTEINS|REACTOME DATABASE ID RELEASE 61","NCAM1 INTERACTIONS|REACTOME DATABASE ID RELEASE 61","SIGNAL RELEASE|GOBP","ACTIVATION OF RRNA EXPRESSION BY ERCC6 (CSB) AND EHMT2 (G9A)|REACTOME","RUNX1 REGULATES GENES INVOLVED IN MEGAKARYOCYTE DIFFERENTIATION AND PLATELET FUNCTION|REACTOME DATABASE ID RELEASE 61","FORMATION OF THE BETA-CATENIN:TCF TRANSACTIVATING COMPLEX|REACTOME DATABASE ID RELEASE 61","DISEASES ASSOCIATED WITH O-GLYCOSYLATION OF PROTEINS|REACTOME DATABASE ID RELEASE 61","PROTEIN HETEROOLIGOMERIZATION|GOBP","SENESCENCE-ASSOCIATED SECRETORY PHENOTYPE (SASP)|REACTOME","HATS ACETYLATE HISTONES|REACTOME","MEIOTIC RECOMBINATION|REACTOME DATABASE ID RELEASE 61","CELLULAR SENESCENCE|REACTOME","RNA POLYMERASE I CHAIN ELONGATION|REACTOME DATABASE ID RELEASE 61","B-WICH COMPLEX POSITIVELY REGULATES RRNA EXPRESSION|REACTOME","RMTS METHYLATE HISTONE ARGININES|REACTOME","NABA_ECM_REGULATORS|MSIGDB_C2","DEGRADATION OF THE EXTRACELLULAR MATRIX|REACTOME DATABASE ID RELEASE 61","RHO GTPASES ACTIVATE PKNS|REACTOME DATABASE ID RELEASE 61","REGULATION OF BLOOD CIRCULATION|GOBP","INTRINSIC APOPTOTIC SIGNALING PATHWAY BY P53 CLASS MEDIATOR|GOBP","NUCLEOSOME ASSEMBLY|GOBP","PROTEIN OLIGOMERIZATION|GOBP","REGULATION OF HEART CONTRACTION|GOBP","TRANSCRIPTIONAL REGULATION BY SMALL RNAS|REACTOME","CARDIAC CONDUCTION|REACTOME DATABASE ID RELEASE 61","MEIOSIS|REACTOME DATABASE ID RELEASE 61","DIRECT P53 EFFECTORS|PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA","PROTEIN HETEROTETRAMERIZATION|GOBP","CELL-SUBSTRATE ADHESION|GOBP","OXIDATIVE STRESS INDUCED SENESCENCE|REACTOME","POSITIVE REGULATION OF VASCULATURE DEVELOPMENT|GOBP","HALLMARK_APOPTOSIS|MSIGDB_C2","POSITIVE EPIGENETIC REGULATION OF RRNA EXPRESSION|REACTOME DATABASE ID RELEASE 61","PACKAGING OF TELOMERE ENDS|REACTOME DATABASE ID RELEASE 61","HALLMARK_INFLAMMATORY_RESPONSE|MSIGDB_C2","CHROMATIN SILENCING|GOBP","NORC NEGATIVELY REGULATES RRNA EXPRESSION|REACTOME DATABASE ID RELEASE 61","POSITIVE REGULATION OF ANGIOGENESIS|GOBP","SIGNAL RELEASE FROM SYNAPSE|GOBP","NEUROTRANSMITTER SECRETION|GOBP","CHROMATIN ASSEMBLY|GOBP","RNA POLYMERASE I PROMOTER CLEARANCE|REACTOME DATABASE ID RELEASE 61","NEGATIVE EPIGENETIC REGULATION OF RRNA EXPRESSION|REACTOME","RNA POLYMERASE I TRANSCRIPTION|REACTOME","HALLMARK_P53_PATHWAY|MSIGDB_C2","REGULATION OF NEUROTRANSMITTER LEVELS|GOBP","POSITIVE REGULATION OF NEURON PROJECTION DEVELOPMENT|GOBP","ANGIOGENESIS|GOBP","PRESYNAPTIC PROCESS INVOLVED IN CHEMICAL SYNAPTIC TRANSMISSION|GOBP","ER-NUCLEUS SIGNALING PATHWAY|GOBP","NUCLEOSOME ASSEMBLY|REACTOME","DEPOSITION OF NEW CENPA-CONTAINING NUCLEOSOMES AT THE CENTROMERE|REACTOME","DNA REPLICATION-DEPENDENT NUCLEOSOME ORGANIZATION|GOBP","REGULATION OF ION TRANSPORT|GOBP","DNA REPLICATION-DEPENDENT NUCLEOSOME ASSEMBLY|GOBP","ECM PROTEOGLYCANS|REACTOME","ACTIVATION OF ANTERIOR HOX GENES IN HINDBRAIN DEVELOPMENT DURING EARLY EMBRYOGENESIS|REACTOME DATABASE ID RELEASE 61","CHROMATIN SILENCING AT RDNA|GOBP","HALLMARK_MYOGENESIS|MSIGDB_C2","CALCIUM ION-REGULATED EXOCYTOSIS OF NEUROTRANSMITTER|GOBP","DOPAMINE NEUROTRANSMITTER RELEASE CYCLE|REACTOME","COLLAGEN CHAIN TRIMERIZATION|REACTOME DATABASE ID RELEASE 61","HORMONE SECRETION|GOBP","MUSCLE CONTRACTION|REACTOME DATABASE ID RELEASE 61","ACTIVATION OF HOX GENES DURING DIFFERENTIATION|REACTOME","NABA_COLLAGENS|MSIGDB_C2","MULTICELLULAR ORGANISMAL CATABOLIC PROCESS|GOBP","NUCLEOSOME ORGANIZATION|GOBP","HALLMARK_HYPOXIA|MSIGDB_C2","NEUROTRANSMITTER TRANSPORT|GOBP","MEIOTIC SYNAPSIS|REACTOME","INTEGRIN SIGNALLING PATHWAY|PANTHER PATHWAY","REGULATION OF CALCIUM ION-DEPENDENT EXOCYTOSIS|GOBP","RESPONSE TO UNFOLDED PROTEIN|GOBP","POSITIVE REGULATION OF NERVOUS SYSTEM DEVELOPMENT|GOBP","MULTICELLULAR ORGANISMAL MACROMOLECULE METABOLIC PROCESS|GOBP","PID_P53_DOWNSTREAM_PATHWAY|MSIGDB_C2","POTASSIUM ION TRANSPORT|GOBP","GENE SILENCING BY RNA|REACTOME DATABASE ID RELEASE 61","ENDODERM DEVELOPMENT|GOBP","RESPONSE TO TOPOLOGICALLY INCORRECT PROTEIN|GOBP","CHROMATIN ASSEMBLY OR DISASSEMBLY|GOBP","LAMININ INTERACTIONS|REACTOME","NABA_SECRETED_FACTORS|MSIGDB_C2","NEGATIVE REGULATION OF GENE EXPRESSION, EPIGENETIC|GOBP","POSITIVE REGULATION OF NEURON DIFFERENTIATION|GOBP","RUNX1 REGULATES TRANSCRIPTION OF GENES INVOLVED IN DIFFERENTIATION OF HSCS|REACTOME","HORMONE TRANSPORT|GOBP","COLLAGEN BIOSYNTHESIS AND MODIFYING ENZYMES|REACTOME","PROTEIN TETRAMERIZATION|GOBP","PHASE 2 - PLATEAU PHASE|REACTOME","CALCIUM ION REGULATED EXOCYTOSIS|GOBP","GLYCOSAMINOGLYCAN METABOLIC PROCESS|GOBP","SYNAPSE ORGANIZATION|GOBP","MITOTIC PROPHASE|REACTOME","COLLAGEN CATABOLIC PROCESS|GOBP","PID_INTEGRIN_CS_PATHWAY|MSIGDB_C2","BLOOD VESSEL MORPHOGENESIS|GOBP","RESPONSE TO VIRUS|GOBP","SIGNAL TRANSDUCTION BY P53 CLASS MEDIATOR|GOBP","REGULATION OF NEURON PROJECTION DEVELOPMENT|GOBP","INTEGRIN FAMILY CELL SURFACE INTERACTIONS|PATHWAY INTERACTION DATABASE NCI-NATURE CURATED DATA","DNA PACKAGING|GOBP","GLYCOSAMINOGLYCAN BIOSYNTHETIC PROCESS|GOBP","REGULATION OF CATION TRANSMEMBRANE TRANSPORT|GOBP","POSITIVE REGULATION OF MRNA METABOLIC PROCESS|GOBP","PEPTIDE HORMONE SECRETION|GOBP","POSITIVE REGULATION OF DEVELOPMENTAL GROWTH|GOBP","REGULATION OF NEURON DIFFERENTIATION|GOBP","MULTICELLULAR ORGANISM METABOLIC PROCESS|GOBP","BLOOD VESSEL DEVELOPMENT|GOBP","AMINOGLYCAN BIOSYNTHETIC PROCESS|GOBP","POLYSACCHARIDE METABOLIC PROCESS|GOBP","POSITIVE REGULATION OF NEUROGENESIS|GOBP","SUBSTRATE ADHESION-DEPENDENT CELL SPREADING|GOBP","SENSORY PERCEPTION OF SOUND|GOBP","NEGATIVE REGULATION OF NERVOUS SYSTEM DEVELOPMENT|GOBP","AMINOGLYCAN METABOLIC PROCESS|GOBP","MUCOPOLYSACCHARIDE METABOLIC PROCESS|GOBP","REGULATION OF ION TRANSMEMBRANE TRANSPORT|GOBP","COLLAGEN METABOLIC PROCESS|GOBP","MEMBRANE DEPOLARIZATION|GOBP","CELLULAR RESPONSE TO CAMP|GOBP","POSITIVE REGULATION OF CELL PROJECTION ORGANIZATION|GOBP","PEPTIDE SECRETION|GOBP","ANION HOMEOSTASIS|GOBP","VOLTAGE GATED POTASSIUM CHANNELS|REACTOME","CELL-MATRIX ADHESION|GOBP","EXTRACELLULAR MATRIX DISASSEMBLY|GOBP","GLYCOSAMINOGLYCAN METABOLISM|REACTOME","CELL COMMUNICATION INVOLVED IN CARDIAC CONDUCTION|GOBP","ACID SECRETION|GOBP","NEPHRON DEVELOPMENT|GOBP","NON-INTEGRIN MEMBRANE-ECM INTERACTIONS|REACTOME","REGULATION OF HORMONE LEVELS|GOBP","REGULATION OF TRANSMEMBRANE TRANSPORT|GOBP","SENSORY PERCEPTION OF MECHANICAL STIMULUS|GOBP","CARDIAC MUSCLE CELL ACTION POTENTIAL|GOBP","EMBRYONIC MORPHOGENESIS|GOBP","EPIGENETIC REGULATION OF GENE EXPRESSION|REACTOME DATABASE ID RELEASE 61","SYNAPTIC VESICLE EXOCYTOSIS|GOBP","VASCULATURE DEVELOPMENT|GOBP","KIDNEY DEVELOPMENT|GOBP","HDMS DEMETHYLATE HISTONES|REACTOME","CARDIAC MUSCLE CELL ACTION POTENTIAL INVOLVED IN CONTRACTION|GOBP","CARDIOVASCULAR SYSTEM DEVELOPMENT|GOBP","WOUND HEALING, SPREADING OF CELLS|GOBP","DEFENSE RESPONSE TO VIRUS|GOBP","GENE SILENCING|GOBP","EPIBOLY INVOLVED IN WOUND HEALING|GOBP","CALCIUM-DEPENDENT CELL-CELL ADHESION VIA PLASMA MEMBRANE CELL ADHESION MOLECULES|GOBP","MONOVALENT INORGANIC CATION TRANSPORT|GOBP","CARDIAC MUSCLE CELL CONTRACTION|GOBP","REGULATION OF POTASSIUM ION TRANSMEMBRANE TRANSPORT|GOBP","REGULATION OF EPIDERMIS DEVELOPMENT|GOBP","NABA_PROTEOGLYCANS|MSIGDB_C2","SINGLE ORGANISM CELL ADHESION|GOBP","INTRINSIC APOPTOTIC SIGNALING PATHWAY|GOBP","ENDODERMAL CELL DIFFERENTIATION|GOBP","REGULATION OF INTRINSIC APOPTOTIC SIGNALING PATHWAY|GOBP","HALLMARK_INTERFERON_ALPHA_RESPONSE|MSIGDB_C2","REGULATION OF ANGIOGENESIS|GOBP","PROTEIN-DNA COMPLEX ASSEMBLY|GOBP","EPIBOLY|GOBP","POSITIVE REGULATION OF BMP SIGNALING PATHWAY|GOBP","CELLULAR POTASSIUM ION TRANSPORT|GOBP","REGULATION OF CELL ADHESION MEDIATED BY INTEGRIN|GOBP","PID_BETA_CATENIN_NUC_PATHWAY|MSIGDB_C2","HALLMARK_GLYCOLYSIS|MSIGDB_C2","TELOMERE MAINTENANCE|REACTOME","AMEBOIDAL-TYPE CELL MIGRATION|GOBP","POTASSIUM ION TRANSMEMBRANE TRANSPORT|GOBP","RESPONSE TO GLUCOCORTICOID|GOBP","REGULATION OF VASCULATURE DEVELOPMENT|GOBP","APOPTOTIC SIGNALING PATHWAY|GOBP","SYNAPTIC SIGNALING|GOBP","TRANSMISSION OF NERVE IMPULSE|GOBP","COLLAGEN FIBRIL ORGANIZATION|GOBP","REGULATION OF HEART RATE BY CARDIAC CONDUCTION|GOBP","TRANS-SYNAPTIC SIGNALING|GOBP","POTASSIUM CHANNELS|REACTOME DATABASE ID RELEASE 61","ANTEROGRADE TRANS-SYNAPTIC SIGNALING|GOBP","RENAL SYSTEM DEVELOPMENT|GOBP","MUSCLE CONTRACTION|GOBP","HSF1-DEPENDENT TRANSACTIVATION|REACTOME DATABASE ID RELEASE 61","NOTCH SIGNALING PATHWAY|GOBP","CHEMICAL SYNAPTIC TRANSMISSION|GOBP","HALLMARK_KRAS_SIGNALING_UP|MSIGDB_C2","COLLAGEN FORMATION|REACTOME","POSITIVE REGULATION OF OXIDOREDUCTASE ACTIVITY|GOBP","GLOMERULUS DEVELOPMENT|GOBP","HALLMARK_HEDGEHOG_SIGNALING|MSIGDB_C2","CELL-CELL ADHESION|GOBP","POSITIVE REGULATION OF MRNA CATABOLIC PROCESS|GOBP","RESPONSE TO CORTICOSTEROID|GOBP","REGULATION OF CHONDROCYTE DIFFERENTIATION|GOBP","REGULATION OF METAL ION TRANSPORT|GOBP","NONHOMOLOGOUS END-JOINING (NHEJ)|REACTOME","ACTION POTENTIAL|GOBP","CELLULAR RESPONSE TO STARVATION|GOBP","REGULATION OF LIPID KINASE ACTIVITY|GOBP","OSSIFICATION|GOBP","REGULATION OF EPITHELIAL CELL PROLIFERATION|GOBP","ENDOPLASMIC RETICULUM UNFOLDED PROTEIN RESPONSE|GOBP","PROTEIN-PROTEIN INTERACTIONS AT SYNAPSES|REACTOME","CELL-CELL SIGNALING INVOLVED IN CARDIAC CONDUCTION|GOBP","POSITIVE REGULATION OF PEPTIDYL-TYROSINE PHOSPHORYLATION|GOBP","INTERLEUKIN-6 FAMILY SIGNALING|REACTOME DATABASE ID RELEASE 61","REGULATION OF CELL GROWTH|GOBP","REGULATION OF OXIDOREDUCTASE ACTIVITY|GOBP","POSITIVE REGULATION OF EPITHELIAL CELL PROLIFERATION|GOBP"],"nGenes":[259,219,220,271,65,65,185,194,86,65,40,198,54,86,55,112,58,59,94,107,83,73,97,34,64,75,64,35,36,115,67,80,81,50,61,99,128,77,178,81,81,70,199,97,85,204,39,101,292,177,90,133,106,135,33,108,110,96,155,96,45,189,70,97,84,86,86,115,100,100,102,198,134,119,134,90,28,64,64,29,292,29,53,110,31,192,31,21,42,31,193,110,42,54,128,195,98,69,162,56,129,198,57,131,100,116,45,149,133,23,275,88,152,119,35,60,104,24,61,121,91,123,49,26,176,159,109,212,26,143,94,178,38,27,66,272,65,197,95,52,180,27,66,111,128,80,198,53,53,28,182,82,28,40,83,68,116,41,41,69,41,280,205,70,42,226,137,57,209,120,44,31,211,20,73,122,20,20,194,32,45,32,32,177,107,32,107,91,160,160,21,21,76,33,76,197,75,76,76,21,180,219,280,34,22,34,280,94,280,128,184,22,48,280,184,79,35,35,35,243,23,23,23,205,65,81,98,36,115,169,98,82,24,99,24,248,66,99],"nQuery":[724,727,725,736,727,727,730,736,727,736,733,726,749,713,749,740,750,749,716,730,738,722,716,751,752,721,752,750,750,727,750,720,720,714,732,735,727,751,733,749,749,733,744,722,749,740,727,724,739,722,748,716,724,714,700,720,721,750,726,750,728,718,715,752,734,734,734,722,750,750,752,721,733,721,733,732,749,752,752,750,721,750,728,749,752,743,752,714,701,752,738,749,701,727,725,738,732,748,717,728,724,740,725,720,733,747,702,737,724,713,721,717,736,751,748,725,735,713,726,749,715,753,701,713,712,736,734,737,713,723,712,712,750,714,726,745,727,725,712,702,711,714,726,733,752,747,729,700,700,716,711,752,716,750,747,729,735,748,748,728,748,736,727,725,751,737,754,701,731,736,748,713,729,800,730,732,800,800,711,713,748,713,713,736,711,713,711,751,720,720,798,798,725,717,725,713,730,725,725,798,739,727,732,714,801,714,732,752,732,734,736,801,750,732,736,725,715,715,715,740,800,800,800,713,702,729,751,715,715,723,751,731,802,751,802,744,700,751],"nOverlap":[39,32,32,36,16,16,27,25,16,14,11,24,12,15,12,17,12,12,15,16,14,13,15,9,12,13,12,9,9,16,12,13,13,10,11,14,16,12,19,12,12,11,20,13,12,20,8,13,25,18,12,15,13,15,7,13,13,12,16,12,8,18,10,12,11,11,11,13,12,12,12,18,14,13,14,11,6,9,9,6,23,6,8,12,6,17,6,5,7,6,17,12,7,8,13,17,11,9,15,8,13,17,8,13,11,12,7,14,13,5,21,10,14,12,6,8,11,5,8,12,10,12,7,5,15,14,11,17,5,13,10,15,6,5,8,20,8,16,10,7,15,5,8,11,12,9,16,7,7,5,15,9,5,6,9,8,11,6,6,8,6,20,16,8,6,17,12,7,16,11,6,5,16,4,8,11,4,4,15,5,6,5,5,14,10,5,10,9,13,13,4,4,8,5,8,15,8,8,8,4,14,16,19,5,4,5,19,9,19,11,14,4,6,19,14,8,5,5,5,17,4,4,4,15,7,8,9,5,10,13,9,8,4,9,4,17,7,9],"querySetFraction":[5.4,4.4,4.4,4.9,2.2,2.2,3.7,3.4,2.2,1.9,1.5,3.3,1.6,2.1,1.6,2.3,1.6,1.6,2.1,2.2,1.9,1.8,2.1,1.2,1.6,1.8,1.6,1.2,1.2,2.2,1.6,1.8,1.8,1.4,1.5,1.9,2.2,1.6,2.6,1.6,1.6,1.5,2.7,1.8,1.6,2.7,1.1,1.8,3.4,2.5,1.6,2.1,1.8,2.1,1,1.8,1.8,1.6,2.2,1.6,1.1,2.5,1.4,1.6,1.5,1.5,1.5,1.8,1.6,1.6,1.6,2.5,1.9,1.8,1.9,1.5,0.8,1.2,1.2,0.8,3.2,0.8,1.1,1.6,0.8,2.3,0.8,0.7,1,0.8,2.3,1.6,1,1.1,1.8,2.3,1.5,1.2,2.1,1.1,1.8,2.3,1.1,1.8,1.5,1.6,1,1.9,1.8,0.7,2.9,1.4,1.9,1.6,0.8,1.1,1.5,0.7,1.1,1.6,1.4,1.6,1,0.7,2.1,1.9,1.5,2.3,0.7,1.8,1.4,2.1,0.8,0.7,1.1,2.7,1.1,2.2,1.4,1,2.1,0.7,1.1,1.5,1.6,1.2,2.2,1,1,0.7,2.1,1.2,0.7,0.8,1.2,1.1,1.5,0.8,0.8,1.1,0.8,2.7,2.2,1.1,0.8,2.3,1.6,1,2.2,1.5,0.8,0.7,2.2,0.5,1.1,1.5,0.5,0.5,2.1,0.7,0.8,0.7,0.7,1.9,1.4,0.7,1.4,1.2,1.8,1.8,0.5,0.5,1.1,0.7,1.1,2.1,1.1,1.1,1.1,0.5,1.9,2.2,2.6,0.7,0.5,0.7,2.6,1.2,2.6,1.5,1.9,0.5,0.8,2.6,1.9,1.1,0.7,0.7,0.7,2.3,0.5,0.5,0.5,2.1,1,1.1,1.2,0.7,1.4,1.8,1.2,1.1,0.5,1.2,0.5,2.3,1,1.2],"geneSetFraction":[15.1,14.6,14.5,13.3,24.6,24.6,14.6,12.9,18.6,21.5,27.5,12.1,22.2,17.4,21.8,15.2,20.7,20.3,16,15,16.9,17.8,15.5,26.5,18.8,17.3,18.8,25.7,25,13.9,17.9,16.2,16,20,18,14.1,12.5,15.6,10.7,14.8,14.8,15.7,10.1,13.4,14.1,9.800000000000001,20.5,12.9,8.6,10.2,13.3,11.3,12.3,11.1,21.2,12,11.8,12.5,10.3,12.5,17.8,9.5,14.3,12.4,13.1,12.8,12.8,11.3,12,12,11.8,9.1,10.4,10.9,10.4,12.2,21.4,14.1,14.1,20.7,7.9,20.7,15.1,10.9,19.4,8.9,19.4,23.8,16.7,19.4,8.800000000000001,10.9,16.7,14.8,10.2,8.699999999999999,11.2,13,9.300000000000001,14.3,10.1,8.6,14,9.9,11,10.3,15.6,9.4,9.800000000000001,21.7,7.6,11.4,9.199999999999999,10.1,17.1,13.3,10.6,20.8,13.1,9.9,11,9.800000000000001,14.3,19.2,8.5,8.800000000000001,10.1,8,19.2,9.1,10.6,8.4,15.8,18.5,12.1,7.4,12.3,8.1,10.5,13.5,8.300000000000001,18.5,12.1,9.9,9.4,11.2,8.1,13.2,13.2,17.9,8.199999999999999,11,17.9,15,10.8,11.8,9.5,14.6,14.6,11.6,14.6,7.1,7.8,11.4,14.3,7.5,8.800000000000001,12.3,7.7,9.199999999999999,13.6,16.1,7.6,20,11,9,20,20,7.7,15.6,13.3,15.6,15.6,7.9,9.300000000000001,15.6,9.300000000000001,9.9,8.1,8.1,19,19,10.5,15.2,10.5,7.6,10.7,10.5,10.5,19,7.8,7.3,6.8,14.7,18.2,14.7,6.8,9.6,6.8,8.6,7.6,18.2,12.5,6.8,7.6,10.1,14.3,14.3,14.3,7,17.4,17.4,17.4,7.3,10.8,9.9,9.199999999999999,13.9,8.699999999999999,7.7,9.199999999999999,9.800000000000001,16.7,9.1,16.7,6.9,10.6,9.1],"foldEnrichment":[4.08,3.96,3.95,3.6,6.68,6.68,3.96,3.5,5.05,5.84,7.46,3.29,6.03,4.73,5.92,4.12,5.61,5.52,4.33,4.06,4.58,4.83,4.19,7.18,5.09,4.7,5.09,6.98,6.78,3.77,4.86,4.41,4.35,5.43,4.89,3.84,3.39,4.23,2.9,4.02,4.02,4.26,2.73,3.64,3.83,2.66,5.56,3.49,2.32,2.76,3.62,3.06,3.33,3.01,5.75,3.27,3.21,3.39,2.8,3.39,4.82,2.58,3.88,3.36,3.55,3.47,3.47,3.07,3.26,3.26,3.19,2.47,2.83,2.96,2.83,3.32,5.81,3.81,3.81,5.61,2.14,5.61,4.09,2.96,5.25,2.4,5.25,6.46,4.52,5.25,2.39,2.96,4.52,4.02,2.76,2.36,3.04,3.54,2.51,3.88,2.73,2.33,3.81,2.69,2.98,2.81,4.22,2.55,2.65,5.9,2.07,3.08,2.5,2.74,4.65,3.62,2.87,5.65,3.56,2.69,2.98,2.65,3.88,5.22,2.31,2.39,2.74,2.18,5.22,2.47,2.89,2.29,4.28,5.02,3.29,1.99,3.34,2.2,2.86,3.65,2.26,5.02,3.29,2.69,2.54,3.05,2.19,3.58,3.58,4.84,2.24,2.98,4.84,4.07,2.94,3.19,2.57,3.97,3.97,3.15,3.97,1.94,2.12,3.1,3.88,2.04,2.38,3.33,2.08,2.49,3.7,4.38,2.06,5.43,2.97,2.45,5.43,5.43,2.1,4.24,3.62,4.24,4.24,2.15,2.54,4.24,2.54,2.68,2.2,2.2,5.17,5.17,2.86,4.11,2.86,2.07,2.89,2.86,2.86,5.17,2.11,1.98,1.84,3.99,4.93,3.99,1.84,2.6,1.84,2.33,2.06,4.93,3.39,1.84,2.06,2.75,3.88,3.88,3.88,1.9,4.72,4.72,4.72,1.98,2.92,2.68,2.49,3.77,2.36,2.09,2.49,2.65,4.52,2.47,4.52,1.86,2.88,2.47],"P":[6.94e-14,2.82e-11,3.18e-11,2.72e-11,1.19e-09,1.19e-09,1.03e-09,5.45e-08,8.6e-08,8.07e-08,1.38e-07,3.19e-07,4.76e-07,5.22e-07,5.88e-07,7.25e-07,1.08e-06,1.31e-06,1.7e-06,1.88e-06,1.89e-06,2.34e-06,2.55e-06,2.74e-06,3.25e-06,3.21e-06,3.25e-06,3.57e-06,4.61e-06,4.93e-06,5.36e-06,6.74e-06,7.76e-06,1.18e-05,1.23e-05,1.58e-05,1.97e-05,2.35e-05,3.31e-05,3.95e-05,3.95e-05,4.72e-05,4.93e-05,5.61e-05,6.42e-05,6.99e-05,7.32e-05,8.58e-05,8.76e-05,9.94e-05,0.000113,0.000117,0.000141,0.000139,0.000166,0.000171,0.000206,0.000211,0.0002,0.000211,0.000211,0.000228,0.000234,0.000233,0.000254,0.000313,0.000313,0.000321,0.00031,0.00031,0.000373,0.000403,0.000435,0.000448,0.000435,0.000466,0.000462,0.000537,0.000537,0.000565,0.000546,0.000565,0.00067,0.000744,0.000822,0.000778,0.000822,0.000835,0.000787,0.000822,0.000825,0.000744,0.000787,0.000762,0.000899,0.000925,0.000963,0.000938,0.0009859999999999999,0.000975,0.000967,0.0011,0.0011,0.00112,0.00114,0.00119,0.0012,0.00125,0.00128,0.0013,0.00138,0.00147,0.00151,0.00149,0.0016,0.00154,0.00157,0.00159,0.00172,0.00172,0.0019,0.00197,0.00201,0.00232,0.00225,0.00231,0.00229,0.00228,0.00232,0.00246,0.00242,0.00251,0.00248,0.00276,0.00286,0.00275,0.00259,0.00267,0.00262,0.00284,0.0028,0.00276,0.00286,0.00265,0.00276,0.00268,0.00281,0.00317,0.00317,0.00326,0.00311,0.00318,0.00326,0.00324,0.00345,0.00345,0.00373,0.00368,0.00368,0.00378,0.00368,0.00382,0.00395,0.00413,0.00416,0.0044,0.0048,0.00479,0.00476,0.00483,0.00526,0.00515,0.00521,0.00554,0.00534,0.00546,0.00554,0.00554,0.00563,0.00592,0.00588,0.00592,0.00592,0.00603,0.00616,0.00592,0.00616,0.00635,0.0064,0.0064,0.00664,0.00664,0.00681,0.00677,0.00681,0.00646,0.00629,0.00681,0.00681,0.00664,0.00697,0.00739,0.00808,0.0077,0.007889999999999999,0.0077,0.00808,0.00783,0.00808,0.00778,0.00839,0.007889999999999999,0.00808,0.00808,0.00839,0.00856,0.00872,0.00872,0.00872,0.0089,0.00928,0.00928,0.00928,0.009180000000000001,0.009809999999999999,0.009900000000000001,0.0102,0.009820000000000001,0.0101,0.009950000000000001,0.0102,0.0106,0.0108,0.0108,0.0108,0.0108,0.0106,0.0108],"overlapGenes":["COL23A1(1.73)|SSPO(1.55)|IGFBP7(1.45)|FNDC1(1.43)|SLIT3(1.41)|ACAN(1.36)|FBN3(1.32)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|HSPG2(1.02)|SRPX(0.96)|COL3A1(0.94)|NTNG2(0.93)|LAMC1(0.93)|EGFLAM(0.91)|HAPLN2(0.89)|SPOCK1(0.87)|SMOC2(0.86)|SNED1(0.86)|MATN3(0.84)|COL2A1(0.83)|AGRN(0.82)|COL21A1(0.81)|THSD4(0.81)|GLDN(0.79)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|ABI3BP(0.75)|EFEMP1(0.74)|NID2(0.73)|LAMB1(0.71)|LTBP4(0.7)|LUM(0.7)|IGFBP5(0.7)|NPNT(0.67)|THBS3(0.65)|CYR61(0.61)","ITGA11(1.35)|SCUBE1(1.2)|HTRA1(1.16)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|HSPG2(1.02)|CREB3L1(1.01)|SCUBE3(0.99)|JAM2(0.98)|COL3A1(0.94)|CD44(0.94)|LAMC1(0.93)|CSGALNACT1(0.87)|MATN3(0.84)|COL2A1(0.83)|SULF1(0.82)|AGRN(0.82)|MMP2(0.82)|ITGA8(0.81)|ICAM5(0.81)|ICAM4(0.79)|ITGA5(0.78)|LAMA1(0.76)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)|LUM(0.7)|NPNT(0.67)|ITGA9(0.64)|ITGA4(0.64)|ADAMTS4(0.62)","ITGA11(1.35)|SCUBE1(1.2)|HTRA1(1.16)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|HSPG2(1.02)|CREB3L1(1.01)|SCUBE3(0.99)|JAM2(0.98)|COL3A1(0.94)|CD44(0.94)|LAMC1(0.93)|CSGALNACT1(0.87)|MATN3(0.84)|COL2A1(0.83)|SULF1(0.82)|AGRN(0.82)|MMP2(0.82)|ITGA8(0.81)|ICAM5(0.81)|ICAM4(0.79)|ITGA5(0.78)|LAMA1(0.76)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)|LUM(0.7)|NPNT(0.67)|ITGA9(0.64)|ITGA4(0.64)|ADAMTS4(0.62)","COL23A1(1.73)|ITGA11(1.35)|FBN3(1.32)|CAPN12(1.29)|SCUBE1(1.2)|HTRA1(1.16)|COL5A1(1.16)|ADAMTS2(1.13)|COL6A2(1.05)|COL13A1(1.05)|ADAMTS1(1.03)|HSPG2(1.02)|SCUBE3(0.99)|JAM2(0.98)|COL3A1(0.94)|CD44(0.94)|LAMC1(0.93)|MATN3(0.84)|COL2A1(0.83)|AGRN(0.82)|MMP2(0.82)|ITGA8(0.81)|COL21A1(0.81)|ICAM5(0.81)|ICAM4(0.79)|ITGA5(0.78)|NTN4(0.77)|LAMA1(0.76)|EFEMP1(0.74)|NID2(0.73)|LAMB1(0.71)|LTBP4(0.7)|LUM(0.7)|ITGA9(0.64)|ITGA4(0.64)|ADAMTS4(0.62)","ITGA11(1.35)|MDK(1.32)|COL5A1(1.16)|PLAU(1.1)|COL6A2(1.05)|JAM2(0.98)|COL3A1(0.94)|LAMC1(0.93)|COL2A1(0.83)|ITGA8(0.81)|ITGA5(0.78)|LAMA1(0.76)|LAMB1(0.71)|NPNT(0.67)|ITGA9(0.64)|ITGA4(0.64)","ITGA11(1.35)|MDK(1.32)|COL5A1(1.16)|PLAU(1.1)|COL6A2(1.05)|JAM2(0.98)|COL3A1(0.94)|LAMC1(0.93)|COL2A1(0.83)|ITGA8(0.81)|ITGA5(0.78)|LAMA1(0.76)|LAMB1(0.71)|NPNT(0.67)|ITGA9(0.64)|ITGA4(0.64)","SSPO(1.55)|IGFBP7(1.45)|FNDC1(1.43)|SLIT3(1.41)|FBN3(1.32)|SRPX(0.96)|NTNG2(0.93)|LAMC1(0.93)|EGFLAM(0.91)|SMOC2(0.86)|SNED1(0.86)|MATN3(0.84)|AGRN(0.82)|THSD4(0.81)|GLDN(0.79)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|ABI3BP(0.75)|EFEMP1(0.74)|NID2(0.73)|LAMB1(0.71)|LTBP4(0.7)|IGFBP5(0.7)|NPNT(0.67)|THBS3(0.65)|CYR61(0.61)","CRLF1(1.95)|SFRP4(1.66)|SLIT3(1.41)|SCG2(1.38)|HTRA1(1.16)|COL5A1(1.16)|COL6A2(1.05)|COL3A1(0.94)|CD44(0.94)|LAMC1(0.93)|CXCL12(0.88)|SPOCK1(0.87)|MATN3(0.84)|IL32(0.84)|SNAI2(0.84)|SAT1(0.82)|MMP2(0.82)|ITGA5(0.78)|LAMA1(0.76)|ABI3BP(0.75)|NID2(0.73)|NT5E(0.7)|LUM(0.7)|CYR61(0.61)|INHBA(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2AK(0.97)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","COL23A1(1.73)|ITGA11(1.35)|COL13A1(1.05)|HSPG2(1.02)|JAM2(0.98)|CD44(0.94)|AGRN(0.82)|ITGA8(0.81)|ICAM5(0.81)|ICAM4(0.79)|ITGA5(0.78)|LUM(0.7)|ITGA9(0.64)|ITGA4(0.64)","COL6A2(1.05)|HSPG2(1.02)|NTNG2(0.93)|LAMC1(0.93)|AGRN(0.82)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|NID2(0.73)|LAMB1(0.71)|NPNT(0.67)","CCL20(2.01)|G0S2(1.55)|KLF10(1.23)|PLAU(1.1)|PLK2(1.04)|TNFRSF9(0.96)|CD44(0.94)|ABCA1(0.91)|PHLDA2(0.85)|IFIT2(0.84)|BTG2(0.82)|SAT1(0.82)|CDKN1A(0.81)|FOSL1(0.79)|ZC3H12A(0.72)|ATF3(0.71)|EFNA1(0.71)|ZFP36(0.68)|IER3(0.67)|REL(0.66)|CCND1(0.61)|CYR61(0.61)|INHBA(0.59)|DRAM1(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HSPG2(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|UBE2L6(0.76)|BACE1(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","IGFBP7(1.45)|SCG2(1.38)|PAPPA(1.22)|MXRA8(1.1)|FAM20A(1.07)|LAMC1(0.93)|MFI2(0.85)|MATN3(0.84)|MMP2(0.82)|CHGB(0.81)|FGF23(0.81)|NOTUM(0.72)|LAMB1(0.71)|IGFBP5(0.7)|FAM20C(0.65)|CYR61(0.61)|APLP2(0.61)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ADAMTS7(1.28)|ADAMTS2(1.13)|SEMA5B(1.08)|ADAMTS1(1.03)|ADAMTSL3(1)|ADAMTS10(0.98)|THSD4(0.81)|GALNT5(0.77)|GALNT12(0.76)|GYLTL1B(0.76)|GALNT3(0.73)|ADAMTSL4(0.68)|ST3GAL1(0.62)|ADAMTS4(0.62)|C1GALT1C1(0.59)","KCNQ1(1.28)|KCNJ4(1.23)|CACNG6(0.94)|CACNA1G(0.9)|ATP1B2(0.79)|KCNK6(0.74)|SCN1B(0.67)|CNTNAP1(0.64)|CACNA1H(0.61)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","KCNQ1(1.28)|KCNJ4(1.23)|CACNG6(0.94)|CACNA1G(0.9)|ATP1B2(0.79)|KCNK6(0.74)|SCN1B(0.67)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H1D(1.89)|HIST1H4E(1.85)|H1F0(1.55)|HIST1H2BD(1.52)|HIST1H1C(1.52)|HIST1H4D(1.4)|HIST1H1E(1.25)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|CDKN1A(0.81)|HIST1H2AC(0.66)","IGFBP7(1.45)|SCG2(1.38)|MXRA8(1.1)|FAM20A(1.07)|LAMC1(0.93)|MFI2(0.85)|MATN3(0.84)|CHGB(0.81)|FGF23(0.81)|NOTUM(0.72)|LAMB1(0.71)|IGFBP5(0.7)|FAM20C(0.65)|CYR61(0.61)|APLP2(0.61)","ADAMTS7(1.28)|ADAMTS2(1.13)|SEMA5B(1.08)|ADAMTS1(1.03)|ADAMTSL3(1)|ADAMTS10(0.98)|THSD4(0.81)|ADAMTSL4(0.68)|ADAMTS4(0.62)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ADAMTS7(1.28)|ADAMTS2(1.13)|SEMA5B(1.08)|ADAMTS1(1.03)|HSPG2(1.02)|ADAMTSL3(1)|ADAMTS10(0.98)|AGRN(0.82)|THSD4(0.81)|NOTCH3(0.78)|LUM(0.7)|ADAMTSL4(0.68)|ADAMTS4(0.62)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ADAMTS7(1.28)|ADAMTS2(1.13)|SEMA5B(1.08)|ADAMTS1(1.03)|ADAMTSL3(1)|ADAMTS10(0.98)|THSD4(0.81)|ADAMTSL4(0.68)|ADAMTS4(0.62)","GDNF(1.33)|NRTN(1.06)|COL6A2(1.05)|CACNA1G(0.9)|AGRN(0.82)|ST8SIA2(0.7)|CACNA1H(0.61)|CACNA1C(0.6)|CACNB4(0.6)","HRH3(2.52)|SNAP25(1.99)|CRHR1(1.67)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|LTBP4(0.7)|CPLX1(0.69)|SYT16(0.69)|PDX1(0.68)|LIN7A(0.66)|EIF2AK3(0.66)|INHBA(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","MOV10(5.08)|HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|TLE2(1.22)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ADAMTS7(1.28)|ADAMTS2(1.13)|SEMA5B(1.08)|ADAMTS1(1.03)|ADAMTSL3(1)|ADAMTS10(0.98)|THSD4(0.81)|NOTCH3(0.78)|ADAMTSL4(0.68)|ADAMTS4(0.62)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|COL6A2(1.05)|SCUBE3(0.99)|HIST1H3D(0.89)|HSD17B8(0.87)|HIST1H3H(0.72)|MAGI2(0.65)|HIST1H3J(0.64)|ITPR3(0.62)","HIST1H4E(1.85)|HIST1H2BD(1.52)|IGFBP7(1.45)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|CDKN1A(0.81)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2AK(0.97)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","MOV10(5.08)|HIST1H1D(1.89)|HIST1H4E(1.85)|H1F0(1.55)|HIST1H2BD(1.52)|HIST1H1C(1.52)|IGFBP7(1.45)|HIST1H4D(1.4)|HIST1H1E(1.25)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|CDKN1A(0.81)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST1H2AK(0.97)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)|CCND1(0.61)","ADAMTS7(1.28)|MMP28(1.25)|PAPPA(1.22)|ITIH5(1.2)|HTRA1(1.16)|ADAMTS2(1.13)|PLAU(1.1)|FAM20A(1.07)|ADAMTS1(1.03)|ADAMTSL3(1)|ADAMTS10(0.98)|PRSS12(0.9)|SULF1(0.82)|MMP2(0.82)|ADAM21(0.76)|ADAMTSL4(0.68)|FAM20C(0.65)|KAZALD1(0.63)|ADAMTS4(0.62)|PAMR1(0.59)","COL23A1(1.73)|CAPN12(1.29)|SCUBE1(1.2)|HTRA1(1.16)|COL13A1(1.05)|ADAMTS1(1.03)|HSPG2(1.02)|SCUBE3(0.99)|CD44(0.94)|LAMC1(0.93)|MMP2(0.82)|LAMB1(0.71)|ADAMTS4(0.62)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","KCNQ1(1.28)|KCNJ4(1.23)|BDKRB2(1.02)|CACNG6(0.94)|ACE(0.93)|CACNA1G(0.9)|ATP1B2(0.79)|KCNK6(0.74)|ZC3H12A(0.72)|SCN1B(0.67)|FXYD6(0.63)|ITPR3(0.62)|CACNA1H(0.61)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","POU4F2(1.83)|KRCC1(0.73)|FHIT(0.72)|DDIT4(0.72)|RPS27L(0.69)|DYRK2(0.66)|ZMAT3(0.64)|PMAIP1(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","HIST1H4E(1.85)|CRYAB(1.65)|P2RX2(1.53)|NOD2(1.4)|HIST1H4D(1.4)|HIST1H3B(1.23)|SCUBE1(1.2)|SCARA5(1.2)|COL6A2(1.05)|SCUBE3(0.99)|CTH(0.9)|AHNAK(0.89)|HIST1H3D(0.89)|HSD17B8(0.87)|HIST1H3H(0.72)|ZC3H12A(0.72)|SHMT2(0.71)|EIF2AK3(0.66)|RNF213(0.65)|MAGI2(0.65)|ALDH1A3(0.64)|HIST1H3J(0.64)|QPRT(0.62)|ITPR3(0.62)|KCNJ2(0.59)","KCNQ1(1.28)|KCNJ4(1.23)|CACNG6(0.94)|CACNA1G(0.9)|ATP1B2(0.79)|KCNK6(0.74)|ZC3H12A(0.72)|SCN1B(0.67)|FXYD6(0.63)|ITPR3(0.62)|CACNA1H(0.61)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","KCNQ1(1.28)|KCNJ4(1.23)|CACNG6(0.94)|ATP1B2(0.79)|KCNK6(0.74)|SCN1B(0.67)|FXYD6(0.63)|ITPR3(0.62)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|SYCP2(0.77)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","POU4F2(1.83)|EGFR(1.09)|PPM1J(1.08)|GDF15(1.05)|BDKRB2(1.02)|HSPA1A(0.89)|SNAI2(0.84)|HSPA1B(0.82)|BTG2(0.82)|MMP2(0.82)|CDKN1A(0.81)|DDIT4(0.72)|ATF3(0.71)|RPS27L(0.69)|PMAIP1(0.59)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST1H3D(0.89)|HSD17B8(0.87)|HIST1H3H(0.72)|HIST1H3J(0.64)","ITGA11(1.35)|COL13A1(1.05)|COL3A1(0.94)|CD44(0.94)|LAMC1(0.93)|ITGA8(0.81)|BCAM(0.78)|ITGA5(0.78)|LAMB1(0.71)|EFNA1(0.71)|NPNT(0.67)|THBS3(0.65)|ITGA4(0.64)","MOV10(5.08)|HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","RAPGEF3(0.9)|CTH(0.9)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","H1F0(1.55)|TXNIP(1.42)|CD44(0.94)|CTH(0.9)|HSPB1(0.85)|BTG2(0.82)|SAT1(0.82)|MMP2(0.82)|CDKN1A(0.81)|ATF3(0.71)|LUM(0.7)|IER3(0.67)|ERBB3(0.66)|CCND1(0.61)|PLCB2(0.6)|PMAIP1(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H2AC(0.66)","CCL20(2.01)|NOD2(1.4)|BST2(1.17)|TNFRSF9(0.96)|ABCA1(0.91)|BTG2(0.82)|CDKN1A(0.81)|ICAM4(0.79)|ITGA5(0.78)|RGS16(0.75)|IL1R1(0.74)|CD70(0.71)|SCN1B(0.67)|BEST1(0.65)|OSMR(0.64)|KCNA3(0.59)|INHBA(0.59)|KCNJ2(0.59)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST1H2AK(0.97)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","RAPGEF3(0.9)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","HRH3(2.52)|SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|LIN7A(0.66)","HRH3(2.52)|SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|LIN7A(0.66)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H1C(1.52)|TXNIP(1.42)|PLK2(1.04)|ZFP36L1(0.93)|NUPR1(0.92)|BTG2(0.82)|SAT1(0.82)|CDKN1A(0.81)|NHLH2(0.78)|RGS16(0.75)|DDIT4(0.72)|ATF3(0.71)|RPS27L(0.69)|IER3(0.67)|TSPYL2(0.65)|ZMAT3(0.64)|TOB1(0.6)|DRAM1(0.59)","HRH3(2.52)|SNAP25(1.99)|ACHE(1.9)|GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|MOXD1(0.84)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|LIN7A(0.66)","POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SYT4(0.83)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|NDRG4(0.59)","SCG2(1.38)|PLXND1(0.91)|PGF(0.84)|ADM2(0.82)|SAT1(0.82)|ITGA5(0.78)|DLL1(0.74)|EFNA1(0.71)|RAMP2(0.7)|EIF2AK3(0.66)|MCAM(0.65)|RNF213(0.65)|XBP1(0.61)|PDGFRA(0.58)","HRH3(2.52)|SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|LIN7A(0.66)","INSIG1(1.25)|TMCO1(0.95)|HERPUD1(0.73)|ATF3(0.71)|EIF2AK3(0.66)|XBP1(0.61)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H2AC(0.66)|CENPQ(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H2AC(0.66)|CENPQ(0.58)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)","HRH3(2.52)|CRHR1(1.67)|SFRP4(1.66)|P2RX2(1.53)|SHANK1(1.46)|GDNF(1.33)|KCNQ1(1.28)|KCNAB2(1.06)|AHNAK(0.89)|CXCL12(0.88)|FGF23(0.81)|ATP1B2(0.79)|SCN1B(0.67)|KCNG1(0.66)|ABCB4(0.64)|FXYD6(0.63)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)","LAMC1(0.93)|MATN3(0.84)|AGRN(0.82)|ITGA8(0.81)|LAMA1(0.76)|LAMB1(0.71)|LUM(0.7)|ITGA9(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)","ACHE(1.9)|CRYAB(1.65)|IGFBP7(1.45)|PPP1R3C(1.25)|COL6A2(1.05)|BDKRB2(1.02)|COL3A1(0.94)|PTP4A3(0.89)|HSPB8(0.84)|AGRN(0.82)|CDKN1A(0.81)|ENO3(0.77)|PPFIA4(0.71)|SH3BGR(0.67)|ERBB3(0.66)|CACNA1H(0.61)|FKBP1B(0.59)","SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|SYT16(0.69)","SNAP25(1.99)|SYN3(0.74)|PPFIA4(0.71)|CPLX1(0.69)|LIN7A(0.66)","COL23A1(1.73)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|COL21A1(0.81)","CRHR1(1.67)|RIMS2(0.93)|LTBP4(0.7)|PDX1(0.68)|EIF2AK3(0.66)|INHBA(0.59)","KCNQ1(1.28)|KCNJ4(1.23)|LMOD1(0.95)|CACNG6(0.94)|DYSF(0.81)|ATP1B2(0.79)|KCNK6(0.74)|SCN1B(0.67)|FXYD6(0.63)|ITPR3(0.62)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","COL23A1(1.73)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|COL21A1(0.81)","MRC2(1.59)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|ACE(0.93)|COL2A1(0.83)|MMP2(0.82)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","PCK1(1.32)|PPP1R3C(1.25)|COL5A1(1.16)|EGFR(1.09)|SRPX(0.96)|CA12(0.87)|PGF(0.84)|CDKN1A(0.81)|ENO3(0.77)|KLHL24(0.73)|DDIT4(0.72)|ATF3(0.71)|EFNA1(0.71)|PPFIA4(0.71)|ZFP36(0.68)|IER3(0.67)|CYR61(0.61)","HRH3(2.52)|SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|LIN7A(0.66)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|SYCP2(0.77)|HIST1H2AC(0.66)","ITGA11(1.35)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|LAMC1(0.93)|COL2A1(0.83)|ITGA8(0.81)|ITGA5(0.78)|RRAS(0.77)|NTN4(0.77)|LAMA1(0.76)|LAMB1(0.71)|ITGA9(0.64)|ITGA4(0.64)","SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|CACNA1G(0.9)|SYT4(0.83)|SYT6(0.74)|SYT16(0.69)|CACNA1H(0.61)","HSPA6(6.25)|CREB3L1(1.01)|CTH(0.9)|WIPI1(0.88)|HSPB1(0.85)|DNAJB1(0.8)|HERPUD1(0.73)|HSPA1L(0.72)|ATF3(0.71)|ACADVL(0.68)|EIF2AK3(0.66)|TSPYL2(0.65)|XBP1(0.61)","ASCL1(4.44)|POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|LRRC4B(0.86)|SYT4(0.83)|AGRN(0.82)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|PRKCH(0.6)|NDRG4(0.59)","MRC2(1.59)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|ACE(0.93)|COL2A1(0.83)|MMP2(0.82)","POU4F2(1.83)|EGFR(1.09)|PPM1J(1.08)|GDF15(1.05)|BDKRB2(1.02)|SNAI2(0.84)|BTG2(0.82)|MMP2(0.82)|CDKN1A(0.81)|DDIT4(0.72)|ATF3(0.71)|RPS27L(0.69)|PMAIP1(0.59)","KCNQ1(1.28)|KCNJ4(1.23)|KCNH3(1.15)|ATP1B2(0.79)|KCNK6(0.74)|AQP1(0.68)|KCNG1(0.66)|KCNA3(0.59)|KCNJ2(0.59)|KCNIP2(0.58)|KCNA6(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ZFP36L1(0.93)|LAMC1(0.93)|MMP2(0.82)|ITGA5(0.78)|LAMB1(0.71)|ITGA4(0.64)|INHBA(0.59)","HSPA6(6.25)|CREB3L1(1.01)|CTH(0.9)|WIPI1(0.88)|HSPB1(0.85)|DNAJB1(0.8)|HERPUD1(0.73)|HSPA1L(0.72)|ATF3(0.71)|ACADVL(0.68)|EIF2AK3(0.66)|TSPYL2(0.65)|KLHL15(0.64)|XBP1(0.61)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","HSPG2(1.02)|LAMC1(0.93)|LAMA1(0.76)|NID2(0.73)|LAMB1(0.71)","CCL20(2.01)|CRLF1(1.95)|INHBE(1.77)|SFRP4(1.66)|GDNF(1.33)|MDK(1.32)|IL11(1.25)|SCUBE1(1.2)|NRTN(1.06)|GDF15(1.05)|SCUBE3(0.99)|CXCL12(0.88)|ISM2(0.86)|PGF(0.84)|CHRD(0.82)|MEGF6(0.82)|FGF23(0.81)|PIK3IP1(0.77)|EGFL7(0.61)|INHBA(0.59)|IL12A(0.58)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST1H2AK(0.97)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","ASCL1(4.44)|POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SYT4(0.83)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|NDRG4(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","CRHR1(1.67)|RIMS2(0.93)|LTBP4(0.7)|PDX1(0.68)|EIF2AK3(0.66)|INHBA(0.59)","COL23A1(1.73)|COL5A1(1.16)|ADAMTS2(1.13)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|COL21A1(0.81)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|CTH(0.9)|HIST1H3D(0.89)|HSD17B8(0.87)|HIST1H3H(0.72)|SHMT2(0.71)|ALDH1A3(0.64)|HIST1H3J(0.64)|KCNJ2(0.59)","KCNQ1(1.28)|CACNG6(0.94)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)","SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|SYT16(0.69)|SCIN(0.61)","B3GAT1(1.07)|HSPG2(1.02)|CD44(0.94)|CSGALNACT1(0.87)|AGRN(0.82)|GALNT5(0.77)|CHST1(0.71)|LUM(0.7)|GUSB(0.67)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","ACHE(1.9)|SHANK1(1.46)|PCDHB10(1.12)|SYNPO(0.97)|PLXND1(0.91)|AGRN(0.82)|SHANK2(0.75)|ELFN1(0.69)|PCDHB11(0.63)|CACNB4(0.6)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","MRC2(1.59)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|MMP2(0.82)","ITGA11(1.35)|ITGA8(0.81)|ITGA5(0.78)|ITGA9(0.64)|ITGA4(0.64)","SCG2(1.38)|PLXND1(0.91)|PGF(0.84)|ADM2(0.82)|SAT1(0.82)|ITGA5(0.78)|DLL1(0.74)|EFNA1(0.71)|RAMP2(0.7)|EIF2AK3(0.66)|MCAM(0.65)|RNF213(0.65)|EGFL7(0.61)|XBP1(0.61)|PDGFRA(0.58)","BST2(1.17)|CXCL12(0.88)|APOBEC3G(0.87)|HSPB1(0.85)|IFIT2(0.84)|IRF2(0.82)|FOSL1(0.79)|DDIT4(0.72)|ZC3H12A(0.72)|DDX60(0.7)|C19orf66(0.69)|TRIM5(0.62)|PMAIP1(0.59)|IL12A(0.58)","POU4F2(1.83)|PLK2(1.04)|BTG2(0.82)|CDKN1A(0.81)|KRCC1(0.73)|FHIT(0.72)|DDIT4(0.72)|RPS27L(0.69)|DYRK2(0.66)|ZMAT3(0.64)|PMAIP1(0.59)","POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SPOCK1(0.87)|SYT4(0.83)|EFNA1(0.71)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|CRMP1(0.63)|NGEF(0.59)|NDRG4(0.59)","ITGA11(1.35)|ITGA8(0.81)|ITGA5(0.78)|ITGA9(0.64)|ITGA4(0.64)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","B3GAT1(1.07)|HSPG2(1.02)|CSGALNACT1(0.87)|AGRN(0.82)|GALNT5(0.77)|CHST1(0.71)|LUM(0.7)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","CRHR1(1.67)|SHANK1(1.46)|KCNQ1(1.28)|KCNAB2(1.06)|AHNAK(0.89)|ATP1B2(0.79)|SCN1B(0.67)|KCNG1(0.66)|FXYD6(0.63)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","CELF3(1.4)|ZFP36L1(0.93)|BTG2(0.82)|ZC3H12A(0.72)|TOB1(0.6)|CELF4(0.58)","CRHR1(1.67)|RIMS2(0.93)|LTBP4(0.7)|PDX1(0.68)|EIF2AK3(0.66)","POU4F2(1.83)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|SYT4(0.83)|AGRN(0.82)|DLL1(0.74)","ASCL1(4.44)|POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SPOCK1(0.87)|SYT4(0.83)|DLL1(0.74)|EFNA1(0.71)|HES5(0.7)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|CRMP1(0.63)|NGEF(0.59)|NDRG4(0.59)","MRC2(1.59)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|ACE(0.93)|COL2A1(0.83)|MMP2(0.82)","SCG2(1.38)|PLXND1(0.91)|PGF(0.84)|ADM2(0.82)|SAT1(0.82)|ITGA5(0.78)|DLL1(0.74)|EFNA1(0.71)|RAMP2(0.7)|EIF2AK3(0.66)|MCAM(0.65)|RNF213(0.65)|EGFL7(0.61)|XBP1(0.61)|S1PR1(0.6)|PDGFRA(0.58)","B3GAT1(1.07)|HSPG2(1.02)|CSGALNACT1(0.87)|AGRN(0.82)|GALNT5(0.77)|CHST1(0.71)|LUM(0.7)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","NR1D1(1.59)|PPP1R3C(1.25)|PGM5(1.23)|CSGALNACT1(0.87)|CHST1(0.71)|CHST7(0.63)|PYGL(0.59)","ASCL1(4.44)|POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SYT4(0.83)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|PRKCH(0.6)|NDRG4(0.59)","LAMC1(0.93)|ITGA8(0.81)|LAMB1(0.71)|EFNA1(0.71)|ITGA4(0.64)","USH1C(1.56)|P2RX2(1.53)|CDH23(1.39)|KCNQ1(1.28)|SNAI2(0.84)|COL2A1(0.83)|ATP6V1B1(0.81)|EML2(0.68)","ASCL1(4.44)|POU4F2(1.83)|COL3A1(0.94)|SPOCK1(0.87)|CHRD(0.82)|DLL1(0.74)|EFNA1(0.71)|HES5(0.7)|EIF2AK3(0.66)|CRMP1(0.63)|NGEF(0.59)","B3GAT1(1.07)|HSPG2(1.02)|CD44(0.94)|CSGALNACT1(0.87)|AGRN(0.82)|GALNT5(0.77)|CHST1(0.71)|LUM(0.7)|GUSB(0.67)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","B3GAT1(1.07)|CD44(0.94)|CSGALNACT1(0.87)|CHST1(0.71)|LUM(0.7)|GUSB(0.67)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","CRHR1(1.67)|SHANK1(1.46)|KCNQ1(1.28)|KCNAB2(1.06)|AHNAK(0.89)|ATP1B2(0.79)|SCN1B(0.67)|KCNG1(0.66)|FXYD6(0.63)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","MRC2(1.59)|COL5A1(1.16)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|MMP2(0.82)","CACNA1G(0.9)|SCN1B(0.67)|CACNA1H(0.61)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|KCNJ2(0.59)","KCNQ1(1.28)|ZFP36L1(0.93)|RAPGEF3(0.9)|IGFBP5(0.7)|AQP1(0.68)","POU4F2(1.83)|SHANK1(1.46)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|PLXND1(0.91)|SYT4(0.83)|AGRN(0.82)|ABL2(0.69)|SCN1B(0.67)|FAM150A(0.66)|MAGI2(0.65)|CDC42EP3(0.6)|NDRG4(0.59)","CRHR1(1.67)|NOD2(1.4)|SCG2(1.38)|OLFM2(1.06)|RIMS2(0.93)|ABCA1(0.91)|LTBP4(0.7)|PDX1(0.68)|EIF2AK3(0.66)","SFRP4(1.66)|ABCA1(0.91)|CA12(0.87)|FGF23(0.81)|XBP1(0.61)","KCNQ1(1.28)|KCNH3(1.15)|KCNAB2(1.06)|KCNG1(0.66)|KCNA3(0.59)|KCNA6(0.58)","ITGA11(1.35)|COL13A1(1.05)|COL3A1(0.94)|CD44(0.94)|ITGA8(0.81)|BCAM(0.78)|NPNT(0.67)|THBS3(0.65)|ITGA4(0.64)","SCUBE1(1.2)|HTRA1(1.16)|HSPG2(1.02)|SCUBE3(0.99)|CD44(0.94)|LAMC1(0.93)|MMP2(0.82)|ADAMTS4(0.62)","B3GAT1(1.07)|HSPG2(1.02)|CD44(0.94)|CSGALNACT1(0.87)|AGRN(0.82)|CHST1(0.71)|LUM(0.7)|GUSB(0.67)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","KCNQ1(1.28)|CACNA1G(0.9)|ATP1B2(0.79)|SCN1B(0.67)|CACNA1C(0.6)|FKBP1B(0.59)","SNAP25(1.99)|BDKRB2(1.02)|ACE(0.93)|PPFIA4(0.71)|CPLX1(0.69)|ABCB4(0.64)","FOXJ1(2.31)|GDNF(1.33)|SULF1(0.82)|DLL1(0.74)|NPNT(0.67)|MAGI2(0.65)|PLCE1(0.64)|PDGFRA(0.58)","HSPG2(1.02)|LAMC1(0.93)|AGRN(0.82)|NTN4(0.77)|LAMA1(0.76)|LAMB1(0.71)","SNAP25(1.99)|CRHR1(1.67)|NR1D1(1.59)|GIPR(1.25)|IL11(1.25)|DPP4(1.07)|RIMS2(0.93)|ACE(0.93)|RAPGEF3(0.9)|HSD17B8(0.87)|CYP27C1(0.82)|FGF23(0.81)|LTBP4(0.7)|PDX1(0.68)|EIF2AK3(0.66)|ALDH1A3(0.64)|ITPR3(0.62)|CACNA1H(0.61)|CACNA1C(0.6)|INHBA(0.59)","CRHR1(1.67)|SHANK1(1.46)|KCNQ1(1.28)|KCNAB2(1.06)|AHNAK(0.89)|ATP1B2(0.79)|SCN1B(0.67)|KCNG1(0.66)|FXYD6(0.63)|CACNG4(0.6)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","USH1C(1.56)|P2RX2(1.53)|CDH23(1.39)|KCNQ1(1.28)|SNAI2(0.84)|COL2A1(0.83)|ATP6V1B1(0.81)|EML2(0.68)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)|KCNJ2(0.59)|KCNIP2(0.58)","USH1C(1.56)|COL2A1(0.83)|MMP2(0.82)|ITGA8(0.81)|ATP6V1B1(0.81)|ITGA5(0.78)|NPHP3-ACAD11(0.78)|DLL1(0.74)|EFEMP1(0.74)|MBNL1(0.73)|LAMB1(0.71)|ALDH1A3(0.64)|ITGA4(0.64)|CACNA1C(0.6)|INHBA(0.59)|NDRG4(0.59)|PDGFRA(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|SYT16(0.69)","SCG2(1.38)|PLXND1(0.91)|PGF(0.84)|ADM2(0.82)|SAT1(0.82)|ITGA5(0.78)|DLL1(0.74)|EFNA1(0.71)|RAMP2(0.7)|EIF2AK3(0.66)|MCAM(0.65)|RNF213(0.65)|EGFL7(0.61)|XBP1(0.61)|S1PR1(0.6)|PDGFRA(0.58)","FOXJ1(2.31)|GDNF(1.33)|ACE(0.93)|SULF1(0.82)|ITGA8(0.81)|NPHP3-ACAD11(0.78)|DLL1(0.74)|NPNT(0.67)|MAGI2(0.65)|PLCE1(0.64)|PDGFRA(0.58)","HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)|KCNJ2(0.59)","SCG2(1.38)|PLXND1(0.91)|PGF(0.84)|ADM2(0.82)|SAT1(0.82)|ITGA5(0.78)|DLL1(0.74)|EFNA1(0.71)|RAMP2(0.7)|EIF2AK3(0.66)|MCAM(0.65)|RNF213(0.65)|EGFL7(0.61)|XBP1(0.61)|S1PR1(0.6)|PDGFRA(0.58)","COL5A1(1.16)|CD44(0.94)|ITGA5(0.78)|CYR61(0.61)","BST2(1.17)|APOBEC3G(0.87)|IFIT2(0.84)|IRF2(0.82)|DDIT4(0.72)|C19orf66(0.69)|TRIM5(0.62)|PMAIP1(0.59)","MOV10(5.08)|HIST1H4E(1.85)|HIST1H4D(1.4)|HIST1H2AI(1.33)|HIST1H3B(1.23)|HIST1H2AK(0.97)|HIST1H3D(0.89)|HIST1H2AM(0.74)|HIST1H3H(0.72)|HIST1H2AC(0.66)|HIST1H3J(0.64)","COL5A1(1.16)|CD44(0.94)|ITGA5(0.78)|CYR61(0.61)","CDH23(1.39)|DCHS1(1.37)|PCDHB10(1.12)|PCDHB11(0.63)","KCNQ1(1.28)|KCNJ4(1.23)|KCNH3(1.15)|HVCN1(0.93)|ATP6V1B1(0.81)|ATP1B2(0.79)|KCNK6(0.74)|AQP1(0.68)|SCN1B(0.67)|KCNG1(0.66)|SLC25A27(0.63)|KCNA3(0.59)|KCNJ2(0.59)|KCNIP2(0.58)|KCNA6(0.58)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)|KCNJ2(0.59)","KCNQ1(1.28)|KCNAB2(1.06)|ATP1B2(0.79)|KCNG1(0.66)|KCNIP3(0.59)|KCNIP2(0.58)","SFRP4(1.66)|ZFP36L1(0.93)|DLL1(0.74)|CDH3(0.72)|PRKCH(0.6)","ACAN(1.36)|HSPG2(1.02)|HAPLN2(0.89)|SPOCK1(0.87)|LUM(0.7)","EGFR(1.09)|COL13A1(1.05)|JAM2(0.98)|CD44(0.94)|LAMC1(0.93)|HSPB1(0.85)|ITGA8(0.81)|ITGA5(0.78)|LAMB1(0.71)|EFNA1(0.71)|NT5E(0.7)|NPNT(0.67)|ITGA4(0.64)|PDGFRA(0.58)","POU4F2(1.83)|CHAC1(0.92)|CDKN1A(0.81)|KRCC1(0.73)|FHIT(0.72)|DDIT4(0.72)|RPS27L(0.69)|DYRK2(0.66)|ZMAT3(0.64)|PMAIP1(0.59)","MMP2(0.82)|ITGA5(0.78)|LAMB1(0.71)|ITGA4(0.64)|INHBA(0.59)","CREB3L1(1.01)|CD44(0.94)|CXCL12(0.88)|HSPB1(0.85)|SNAI2(0.84)|PDIA2(0.83)|HERPUD1(0.73)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","MOV10(5.08)|TXNIP(1.42)|BST2(1.17)|IFIT2(0.84)|IRF2(0.82)|UBE2L6(0.76)|DDX60(0.7)|TRIM5(0.62)|LPAR6(0.6)","PLXND1(0.91)|RAPGEF3(0.9)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|SULF1(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST1H3B(1.23)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST2H2BF(0.94)|HIST1H2BN(0.94)|HIST1H3D(0.89)|HIST1H3H(0.72)|HIST1H3J(0.64)|CENPQ(0.58)","COL5A1(1.16)|CD44(0.94)|ITGA5(0.78)|CYR61(0.61)","RNF165(1.1)|SULF1(0.82)|HES5(0.7)|CYR61(0.61)","KCNQ1(1.28)|KCNJ4(1.23)|ATP1B2(0.79)|KCNG1(0.66)|KCNA3(0.59)|KCNJ2(0.59)|KCNIP2(0.58)|KCNA6(0.58)","PLAU(1.1)|DPP4(1.07)|SNAI2(0.84)|EFNA1(0.71)|EPHA8(0.68)","SSPO(1.55)|TLE2(1.22)|CACNA1G(0.9)|MDFIC(0.89)|SNAI2(0.84)|MMP2(0.82)|CCND1(0.61)|CYR61(0.61)","COL5A1(1.16)|EGFR(1.09)|B3GAT1(1.07)|CD44(0.94)|CTH(0.9)|AGRN(0.82)|DDIT4(0.72)|CHST1(0.71)|PPFIA4(0.71)|NT5E(0.7)|IER3(0.67)|GUSB(0.67)|CACNA1H(0.61)|CHST6(0.61)|PYGL(0.59)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)|HIST1H2AC(0.66)","SCG2(1.38)|GDNF(1.33)|DPP4(1.07)|NRTN(1.06)|PLXND1(0.91)|PTP4A3(0.89)|ITGA4(0.64)|ZEB2(0.62)","KCNQ1(1.28)|KCNJ4(1.23)|ATP1B2(0.79)|KCNG1(0.66)|KCNA3(0.59)|KCNJ2(0.59)|KCNIP2(0.58)|KCNA6(0.58)","SLIT3(1.41)|ZFP36L1(0.93)|SSTR2(0.68)|AQP1(0.68)","PLXND1(0.91)|RAPGEF3(0.9)|CTH(0.9)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|SULF1(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","POU4F2(1.83)|G0S2(1.55)|TNFRSF9(0.96)|CHAC1(0.92)|CDKN1A(0.81)|KRCC1(0.73)|FHIT(0.72)|DDIT4(0.72)|CD70(0.71)|RPS27L(0.69)|DYRK2(0.66)|ERBB3(0.66)|ZMAT3(0.64)|PMAIP1(0.59)|INHBA(0.59)|IL12A(0.58)","HRH3(2.52)|SNAP25(1.99)|ACHE(1.9)|PCDHB10(1.12)|GPR1(1.09)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|SCN1B(0.67)|NPTX1(0.67)|LIN7A(0.66)|PCDHB11(0.63)|CACNB4(0.6)|KCNIP2(0.58)","CACNA1G(0.9)|SCN1B(0.67)|CNTNAP1(0.64)|CACNA1H(0.61)|CACNG4(0.6)","COL5A1(1.16)|COL3A1(0.94)|COL2A1(0.83)|LUM(0.7)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)|KCNJ2(0.59)","HRH3(2.52)|SNAP25(1.99)|ACHE(1.9)|PCDHB10(1.12)|GPR1(1.09)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|SCN1B(0.67)|NPTX1(0.67)|LIN7A(0.66)|PCDHB11(0.63)|CACNB4(0.6)|KCNIP2(0.58)","KCNQ1(1.28)|KCNJ4(1.23)|KCNH3(1.15)|KCNAB2(1.06)|KCNK6(0.74)|KCNG1(0.66)|KCNA3(0.59)|KCNJ2(0.59)|KCNA6(0.58)","HRH3(2.52)|SNAP25(1.99)|ACHE(1.9)|PCDHB10(1.12)|GPR1(1.09)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|SCN1B(0.67)|NPTX1(0.67)|LIN7A(0.66)|PCDHB11(0.63)|CACNB4(0.6)|KCNIP2(0.58)","FOXJ1(2.31)|GDNF(1.33)|ACE(0.93)|SULF1(0.82)|ITGA8(0.81)|NPHP3-ACAD11(0.78)|DLL1(0.74)|NPNT(0.67)|MAGI2(0.65)|PLCE1(0.64)|PDGFRA(0.58)","CRYAB(1.65)|GDNF(1.33)|KCNQ1(1.28)|BDKRB2(1.02)|LMOD1(0.95)|CACNA1G(0.9)|SULF1(0.82)|DYSF(0.81)|SCN1B(0.67)|HTR1D(0.65)|CACNA1H(0.61)|CACNA1C(0.6)|KCNJ2(0.59)|KCNIP2(0.58)","CRYAB(1.65)|HSPB8(0.84)|DNAJB1(0.8)|HSPA1L(0.72)","ASCL1(4.44)|MDK(1.32)|SNAI2(0.84)|NOTCH3(0.78)|DLL1(0.74)|HES5(0.7)","HRH3(2.52)|SNAP25(1.99)|ACHE(1.9)|PCDHB10(1.12)|GPR1(1.09)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|SYT6(0.74)|PPFIA4(0.71)|CPLX1(0.69)|SYT16(0.69)|SCN1B(0.67)|NPTX1(0.67)|LIN7A(0.66)|PCDHB11(0.63)|CACNB4(0.6)|KCNIP2(0.58)","CCL20(2.01)|SNAP25(1.99)|USH1C(1.56)|G0S2(1.55)|PLAU(1.1)|ACE(0.93)|GYPC(0.87)|ETV1(0.76)|RGS16(0.75)|GALNT3(0.73)|TMEM158(0.69)|SCN1B(0.67)|ALDH1A3(0.64)|INHBA(0.59)","COL23A1(1.73)|COL5A1(1.16)|ADAMTS2(1.13)|COL6A2(1.05)|COL13A1(1.05)|COL3A1(0.94)|COL2A1(0.83)|COL21A1(0.81)","NOD2(1.4)|GDNF(1.33)|FGF23(0.81)|CDH3(0.72)|ABL2(0.69)","FOXJ1(2.31)|SULF1(0.82)|MAGI2(0.65)|PLCE1(0.64)|PDGFRA(0.58)","ACHE(1.9)|SCG2(1.38)|CRMP1(0.63)|GLI1(0.61)|CNTFR(0.59)","CDH23(1.39)|DCHS1(1.37)|PCDHB10(1.12)|EGFR(1.09)|COL13A1(1.05)|JAM2(0.98)|CD44(0.94)|GYPC(0.87)|HSPB1(0.85)|ITGA8(0.81)|ITGA5(0.78)|LAMB1(0.71)|NT5E(0.7)|NPNT(0.67)|ITGA4(0.64)|PCDHB11(0.63)|PDGFRA(0.58)","ZFP36L1(0.93)|BTG2(0.82)|ZC3H12A(0.72)|TOB1(0.6)","SLIT3(1.41)|ZFP36L1(0.93)|SSTR2(0.68)|AQP1(0.68)","ADAMTS7(1.28)|SNAI2(0.84)|EFEMP1(0.74)|SCIN(0.61)","CRHR1(1.67)|P2RX2(1.53)|KCNQ1(1.28)|KCNAB2(1.06)|AHNAK(0.89)|CXCL12(0.88)|ATP1B2(0.79)|SCN1B(0.67)|KCNG1(0.66)|FXYD6(0.63)|CACNA1C(0.6)|CACNB4(0.6)|FKBP1B(0.59)|KCNIP3(0.59)|KCNIP2(0.58)","HIST1H4E(1.85)|HIST1H2BD(1.52)|HIST1H4D(1.4)|HIST3H2BB(1.14)|HIST1H2BF(1.06)|HIST2H2BE(1.02)|HIST1H2BN(0.94)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CNTNAP1(0.64)|CACNA1H(0.61)|CACNA1C(0.6)|KCNJ2(0.59)|KCNIP2(0.58)","KLF10(1.23)|KIAA1324(1.15)|CDKN1A(0.81)|ZC3H12A(0.72)|PDK4(0.71)|EIF2AK3(0.66)|XBP1(0.61)|LAMP2(0.6)|PMAIP1(0.59)","NOD2(1.4)|PIK3IP1(0.77)|EEF1A2(0.77)|EPHA8(0.68)|PDGFRA(0.58)","ACHE(1.9)|MRC2(1.59)|ITGA11(1.35)|EGFR(1.09)|COL13A1(1.05)|CREB3L1(1.01)|SNAI2(0.84)|ATP6V1B1(0.81)|EIF2AK3(0.66)|GLI1(0.61)","NR1D1(1.59)|NOD2(1.4)|SCG2(1.38)|EGFR(1.09)|ZFP36L1(0.93)|LAMC1(0.93)|SNAI2(0.84)|PGF(0.84)|SULF1(0.82)|CDH3(0.72)|LAMB1(0.71)|EGFL7(0.61)|XBP1(0.61)","CREB3L1(1.01)|CTH(0.9)|WIPI1(0.88)|HERPUD1(0.73)|ATF3(0.71)|ACADVL(0.68)|EIF2AK3(0.66)|TSPYL2(0.65)|XBP1(0.61)","SHANK1(1.46)|LRRTM2(1.25)|DLGAP3(1.14)|SYT2(0.94)|LRRC4B(0.86)|SHANK2(0.75)|PPFIA4(0.71)|LIN7A(0.66)","KCNQ1(1.28)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)","CRLF1(1.95)|IL11(1.25)|CD44(0.94)|ACE(0.93)|ITGA5(0.78)|EFNA1(0.71)|HES5(0.7)|ERBB3(0.66)|IL12A(0.58)","CRLF1(1.95)|IL11(1.25)|OSMR(0.64)|CNTFR(0.59)","POU4F2(1.83)|SLIT3(1.41)|BST2(1.17)|RASGRP2(1.16)|EGFR(1.09)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|SPOCK1(0.87)|MFI2(0.85)|SYT4(0.83)|CDKN1A(0.81)|LTBP4(0.7)|TSPYL2(0.65)|PLCE1(0.64)|INHBA(0.59)","NOD2(1.4)|GDNF(1.33)|EGFR(1.09)|FGF23(0.81)|CDH3(0.72)|ABL2(0.69)|NOSTRIN(0.61)","NOD2(1.4)|SCG2(1.38)|EGFR(1.09)|LAMC1(0.93)|PGF(0.84)|CDH3(0.72)|LAMB1(0.71)|EGFL7(0.61)|XBP1(0.61)"],"maxOverlapGeneScore":[1.73,1.35,1.35,1.73,1.35,1.35,1.55,1.95,1.85,1.73,1.05,2.01,1.85,1.85,1.85,1.45,1.85,1.85,1.28,1.28,1.28,1.89,1.45,1.28,1.85,1.28,1.85,1.28,1.33,2.52,1.85,5.08,1.85,1.28,1.85,1.85,1.85,1.85,5.08,1.85,1.85,1.85,1.28,1.73,1.85,1.28,1.83,1.85,1.85,1.28,1.85,1.28,1.85,1.83,1.85,1.35,5.08,0.9,1.55,1.85,1.85,2.01,1.85,1.85,0.9,2.52,2.52,1.85,1.85,1.85,1.85,1.52,2.52,1.83,1.38,2.52,1.25,1.85,1.85,1.85,2.52,1.85,0.93,1.85,1.85,1.9,0.9399999999999999,1.99,1.73,1.67,1.28,1.85,1.73,1.59,1.85,1.32,2.52,1.85,1.35,0.9399999999999999,6.25,4.44,1.59,1.83,1.28,1.85,0.93,6.25,1.85,1.02,2.01,1.85,4.44,1.85,1.67,1.73,1.85,1.28,1.99,1.07,1.9,1.85,1.59,1.35,1.38,1.17,1.83,1.83,1.35,1.85,1.07,1.67,1.4,1.67,1.83,4.44,1.59,1.38,1.07,1.59,4.44,0.93,1.56,4.44,1.07,1.07,1.67,1.59,0.9,1.28,1.83,1.67,1.66,1.28,1.35,1.2,1.07,1.28,1.99,2.31,1.02,1.99,1.67,1.56,1.28,1.56,1.85,1.99,1.38,2.31,1.85,1.28,1.38,1.16,1.17,5.08,1.16,1.39,1.28,1.28,1.28,1.66,1.36,1.09,1.83,0.82,1.01,5.08,0.91,1.85,1.16,1.1,1.28,1.1,1.55,1.16,1.85,1.38,1.28,1.41,0.91,1.83,2.52,0.9,1.16,1.28,2.52,1.28,2.52,2.31,1.65,1.65,4.44,2.52,2.01,1.73,1.4,2.31,1.9,1.39,0.93,1.41,1.28,1.67,1.85,1.28,1.23,1.4,1.9,1.59,1.01,1.46,1.28,1.95,1.95,1.83,1.4,1.4],"size":[40,32.99,32.99,32.99,28.47,28.47,28.47,23.01,22.67,22.67,22.11,21,20.53,20.5,20.43,20.22,19.72,19.51,19.21,19.2,19.2,18.96,18.9,18.85,18.77,18.77,18.77,18.69,18.36,18.31,18.24,17.95,17.78,17.21,17.19,16.86,16.57,16.36,15.89,15.7,15.7,15.47,15.44,15.28,15.12,15.02,14.99,14.79,14.79,14.63,14.47,14.44,14.22,14.22,14.01,13.99,13.81,13.81,13.81,13.81,13.81,13.73,13.73,13.73,13.63,13.41,13.41,13.39,13.41,13.41,13.19,13.1,13.03,13,13.03,12.98,12.98,12.81,12.81,12.79,12.81,12.79,12.56,12.44,12.42,12.42,12.42,12.42,12.42,12.42,12.42,12.44,12.42,12.42,12.32,12.3,12.28,12.29,12.28,12.28,12.28,12.15,12.15,12.14,12.12,12.07,12.07,12.03,12,11.99,11.92,11.84,11.83,11.84,11.79,11.81,11.79,11.79,11.71,11.71,11.57,11.53,11.52,11.37,11.37,11.37,11.37,11.37,11.37,11.31,11.32,11.3,11.31,11.26,11.26,11.26,11.27,11.26,11.26,11.26,11.26,11.26,11.26,11.26,11.26,11.26,11.26,11.14,11.14,11.13,11.14,11.14,11.13,11.13,11.07,11.07,10.99,11,11,10.98,11,10.97,10.93,10.88,10.88,10.8,10.7,10.7,10.7,10.7,10.6,10.61,10.6,10.56,10.58,10.56,10.56,10.56,10.55,10.51,10.51,10.51,10.51,10.49,10.48,10.51,10.48,10.45,10.45,10.45,10.43,10.43,10.43,10.43,10.43,10.45,10.46,10.43,10.43,10.43,10.4,10.33,10.27,10.28,10.27,10.28,10.27,10.27,10.27,10.27,10.23,10.27,10.27,10.27,10.23,10.21,10.2,10.2,10.2,10.18,10.14,10.14,10.14,10.14,10.07,10.06,10.05,10.07,10.05,10.06,10.05,10,10,10,10,10,10,10],"formattedLabel":["NABA_CORE_MATRISOME|MSIGDB_C2","EXTRACELLULAR\nMATRIX\nORGANIZATION|GOBP","EXTRACELLULAR\nSTRUCTURE\nORGANIZATION|GOBP","EXTRACELLULAR\nMATRIX\nORGANIZATION|REACTOME","BETA1 INTEGRIN CELL\nSURFACE\nINTERACTIONS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","PID_INTEGRIN1_PATHWAY|MSIGDB_C2","NABA_ECM_GLYCOPROTEINS|MSIGDB_C2","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION|MSIGDB_C2","HDACS DEACETYLATE\nHISTONES|REACTOME\nDATABASE ID RELEASE\n61","INTEGRIN CELL\nSURFACE\nINTERACTIONS|REACTOME\nDATABASE ID RELEASE\n61","NABA_BASEMENT_MEMBRANES|MSIGDB_C2","HALLMARK_TNFA_SIGNALING_VIA_NFKB|MSIGDB_C2","RNA POLYMERASE I\nPROMOTER\nOPENING|REACTOME\nDATABASE ID RELEASE\n61","AMYLOID FIBER\nFORMATION|REACTOME","DNA\nMETHYLATION|REACTOME","REGULATION OF IGF\nACTIVITY BY\nIGFBP|REACTOME\nDATABASE ID RELEASE\n61","ACTIVATED PKN1\nSTIMULATES\nTRANSCRIPTION OF AR\n(ANDROGEN RECEPTOR)\nREGULATED GENES\nKLK2 AND\nKLK3|REACTOME\nDATABASE ID RELEASE\n61","SIRT1 NEGATIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME","O-LINKED\nGLYCOSYLATION|REACTOME","MULTICELLULAR\nORGANISMAL\nSIGNALING|GOBP","CARDIAC\nCONDUCTION|GOBP","DNA DAMAGE TELOMERE\nSTRESS INDUCED\nSENESCENCE|REACTOME","POST-TRANSLATIONAL\nPROTEIN\nPHOSPHORYLATION|REACTOME","DEFECTIVE B3GALTL\nCAUSES PETERS-PLUS\nSYNDROME\n(PPS)|REACTOME","PRC2 METHYLATES\nHISTONES AND\nDNA|REACTOME\nDATABASE ID RELEASE\n61","DISEASES OF\nGLYCOSYLATION|REACTOME","CONDENSATION OF\nPROPHASE\nCHROMOSOMES|REACTOME\nDATABASE ID RELEASE\n61","O-GLYCOSYLATION OF\nTSR\nDOMAIN-CONTAINING\nPROTEINS|REACTOME\nDATABASE ID RELEASE\n61","NCAM1\nINTERACTIONS|REACTOME\nDATABASE ID RELEASE\n61","SIGNAL RELEASE|GOBP","ACTIVATION OF RRNA\nEXPRESSION BY ERCC6\n(CSB) AND EHMT2\n(G9A)|REACTOME","RUNX1 REGULATES\nGENES INVOLVED IN\nMEGAKARYOCYTE\nDIFFERENTIATION AND\nPLATELET\nFUNCTION|REACTOME\nDATABASE ID RELEASE\n61","FORMATION OF THE\nBETA-CATENIN:TCF\nTRANSACTIVATING\nCOMPLEX|REACTOME\nDATABASE ID RELEASE\n61","DISEASES ASSOCIATED\nWITH\nO-GLYCOSYLATION OF\nPROTEINS|REACTOME\nDATABASE ID RELEASE\n61","PROTEIN\nHETEROOLIGOMERIZATION|GOBP","SENESCENCE-ASSOCIATED\nSECRETORY PHENOTYPE\n(SASP)|REACTOME","HATS ACETYLATE\nHISTONES|REACTOME","MEIOTIC\nRECOMBINATION|REACTOME\nDATABASE ID RELEASE\n61","CELLULAR\nSENESCENCE|REACTOME","RNA POLYMERASE I\nCHAIN\nELONGATION|REACTOME\nDATABASE ID RELEASE\n61","B-WICH COMPLEX\nPOSITIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME","RMTS METHYLATE\nHISTONE\nARGININES|REACTOME","NABA_ECM_REGULATORS|MSIGDB_C2","DEGRADATION OF THE\nEXTRACELLULAR\nMATRIX|REACTOME\nDATABASE ID RELEASE\n61","RHO GTPASES\nACTIVATE\nPKNS|REACTOME\nDATABASE ID RELEASE\n61","REGULATION OF BLOOD\nCIRCULATION|GOBP","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nBY P53 CLASS\nMEDIATOR|GOBP","NUCLEOSOME\nASSEMBLY|GOBP","PROTEIN\nOLIGOMERIZATION|GOBP","REGULATION OF HEART\nCONTRACTION|GOBP","TRANSCRIPTIONAL\nREGULATION BY SMALL\nRNAS|REACTOME","CARDIAC\nCONDUCTION|REACTOME\nDATABASE ID RELEASE\n61","MEIOSIS|REACTOME\nDATABASE ID RELEASE\n61","DIRECT P53\nEFFECTORS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","PROTEIN\nHETEROTETRAMERIZATION|GOBP","CELL-SUBSTRATE\nADHESION|GOBP","OXIDATIVE STRESS\nINDUCED\nSENESCENCE|REACTOME","POSITIVE REGULATION\nOF VASCULATURE\nDEVELOPMENT|GOBP","HALLMARK_APOPTOSIS|MSIGDB_C2","POSITIVE EPIGENETIC\nREGULATION OF RRNA\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","PACKAGING OF\nTELOMERE\nENDS|REACTOME\nDATABASE ID RELEASE\n61","HALLMARK_INFLAMMATORY_RESPONSE|MSIGDB_C2","CHROMATIN\nSILENCING|GOBP","NORC NEGATIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","POSITIVE REGULATION\nOF\nANGIOGENESIS|GOBP","SIGNAL RELEASE FROM\nSYNAPSE|GOBP","NEUROTRANSMITTER\nSECRETION|GOBP","CHROMATIN\nASSEMBLY|GOBP","RNA POLYMERASE I\nPROMOTER\nCLEARANCE|REACTOME\nDATABASE ID RELEASE\n61","NEGATIVE EPIGENETIC\nREGULATION OF RRNA\nEXPRESSION|REACTOME","RNA POLYMERASE I\nTRANSCRIPTION|REACTOME","HALLMARK_P53_PATHWAY|MSIGDB_C2","REGULATION OF\nNEUROTRANSMITTER\nLEVELS|GOBP","POSITIVE REGULATION\nOF NEURON\nPROJECTION\nDEVELOPMENT|GOBP","ANGIOGENESIS|GOBP","PRESYNAPTIC PROCESS\nINVOLVED IN\nCHEMICAL SYNAPTIC\nTRANSMISSION|GOBP","ER-NUCLEUS\nSIGNALING\nPATHWAY|GOBP","NUCLEOSOME\nASSEMBLY|REACTOME","DEPOSITION OF NEW\nCENPA-CONTAINING\nNUCLEOSOMES AT THE\nCENTROMERE|REACTOME","DNA\nREPLICATION-DEPENDENT\nNUCLEOSOME\nORGANIZATION|GOBP","REGULATION OF ION\nTRANSPORT|GOBP","DNA\nREPLICATION-DEPENDENT\nNUCLEOSOME\nASSEMBLY|GOBP","ECM\nPROTEOGLYCANS|REACTOME","ACTIVATION OF\nANTERIOR HOX GENES\nIN HINDBRAIN\nDEVELOPMENT DURING\nEARLY\nEMBRYOGENESIS|REACTOME\nDATABASE ID RELEASE\n61","CHROMATIN SILENCING\nAT RDNA|GOBP","HALLMARK_MYOGENESIS|MSIGDB_C2","CALCIUM\nION-REGULATED\nEXOCYTOSIS OF\nNEUROTRANSMITTER|GOBP","DOPAMINE\nNEUROTRANSMITTER\nRELEASE\nCYCLE|REACTOME","COLLAGEN CHAIN\nTRIMERIZATION|REACTOME\nDATABASE ID RELEASE\n61","HORMONE\nSECRETION|GOBP","MUSCLE\nCONTRACTION|REACTOME\nDATABASE ID RELEASE\n61","ACTIVATION OF HOX\nGENES DURING\nDIFFERENTIATION|REACTOME","NABA_COLLAGENS|MSIGDB_C2","MULTICELLULAR\nORGANISMAL\nCATABOLIC\nPROCESS|GOBP","NUCLEOSOME\nORGANIZATION|GOBP","HALLMARK_HYPOXIA|MSIGDB_C2","NEUROTRANSMITTER\nTRANSPORT|GOBP","MEIOTIC\nSYNAPSIS|REACTOME","INTEGRIN SIGNALLING\nPATHWAY|PANTHER\nPATHWAY","REGULATION OF\nCALCIUM\nION-DEPENDENT\nEXOCYTOSIS|GOBP","RESPONSE TO\nUNFOLDED\nPROTEIN|GOBP","POSITIVE REGULATION\nOF NERVOUS SYSTEM\nDEVELOPMENT|GOBP","MULTICELLULAR\nORGANISMAL\nMACROMOLECULE\nMETABOLIC\nPROCESS|GOBP","PID_P53_DOWNSTREAM_PATHWAY|MSIGDB_C2","POTASSIUM ION\nTRANSPORT|GOBP","GENE SILENCING BY\nRNA|REACTOME\nDATABASE ID RELEASE\n61","ENDODERM\nDEVELOPMENT|GOBP","RESPONSE TO\nTOPOLOGICALLY\nINCORRECT\nPROTEIN|GOBP","CHROMATIN ASSEMBLY\nOR DISASSEMBLY|GOBP","LAMININ\nINTERACTIONS|REACTOME","NABA_SECRETED_FACTORS|MSIGDB_C2","NEGATIVE REGULATION\nOF GENE EXPRESSION,\nEPIGENETIC|GOBP","POSITIVE REGULATION\nOF NEURON\nDIFFERENTIATION|GOBP","RUNX1 REGULATES\nTRANSCRIPTION OF\nGENES INVOLVED IN\nDIFFERENTIATION OF\nHSCS|REACTOME","HORMONE\nTRANSPORT|GOBP","COLLAGEN\nBIOSYNTHESIS AND\nMODIFYING\nENZYMES|REACTOME","PROTEIN\nTETRAMERIZATION|GOBP","PHASE 2 - PLATEAU\nPHASE|REACTOME","CALCIUM ION\nREGULATED\nEXOCYTOSIS|GOBP","GLYCOSAMINOGLYCAN\nMETABOLIC\nPROCESS|GOBP","SYNAPSE\nORGANIZATION|GOBP","MITOTIC\nPROPHASE|REACTOME","COLLAGEN CATABOLIC\nPROCESS|GOBP","PID_INTEGRIN_CS_PATHWAY|MSIGDB_C2","BLOOD VESSEL\nMORPHOGENESIS|GOBP","RESPONSE TO\nVIRUS|GOBP","SIGNAL TRANSDUCTION\nBY P53 CLASS\nMEDIATOR|GOBP","REGULATION OF\nNEURON PROJECTION\nDEVELOPMENT|GOBP","INTEGRIN FAMILY\nCELL SURFACE\nINTERACTIONS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","DNA PACKAGING|GOBP","GLYCOSAMINOGLYCAN\nBIOSYNTHETIC\nPROCESS|GOBP","REGULATION OF\nCATION\nTRANSMEMBRANE\nTRANSPORT|GOBP","POSITIVE REGULATION\nOF MRNA METABOLIC\nPROCESS|GOBP","PEPTIDE HORMONE\nSECRETION|GOBP","POSITIVE REGULATION\nOF DEVELOPMENTAL\nGROWTH|GOBP","REGULATION OF\nNEURON\nDIFFERENTIATION|GOBP","MULTICELLULAR\nORGANISM METABOLIC\nPROCESS|GOBP","BLOOD VESSEL\nDEVELOPMENT|GOBP","AMINOGLYCAN\nBIOSYNTHETIC\nPROCESS|GOBP","POLYSACCHARIDE\nMETABOLIC\nPROCESS|GOBP","POSITIVE REGULATION\nOF\nNEUROGENESIS|GOBP","SUBSTRATE\nADHESION-DEPENDENT\nCELL SPREADING|GOBP","SENSORY PERCEPTION\nOF SOUND|GOBP","NEGATIVE REGULATION\nOF NERVOUS SYSTEM\nDEVELOPMENT|GOBP","AMINOGLYCAN\nMETABOLIC\nPROCESS|GOBP","MUCOPOLYSACCHARIDE\nMETABOLIC\nPROCESS|GOBP","REGULATION OF ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","COLLAGEN METABOLIC\nPROCESS|GOBP","MEMBRANE\nDEPOLARIZATION|GOBP","CELLULAR RESPONSE\nTO CAMP|GOBP","POSITIVE REGULATION\nOF CELL PROJECTION\nORGANIZATION|GOBP","PEPTIDE\nSECRETION|GOBP","ANION\nHOMEOSTASIS|GOBP","VOLTAGE GATED\nPOTASSIUM\nCHANNELS|REACTOME","CELL-MATRIX\nADHESION|GOBP","EXTRACELLULAR\nMATRIX\nDISASSEMBLY|GOBP","GLYCOSAMINOGLYCAN\nMETABOLISM|REACTOME","CELL COMMUNICATION\nINVOLVED IN CARDIAC\nCONDUCTION|GOBP","ACID SECRETION|GOBP","NEPHRON\nDEVELOPMENT|GOBP","NON-INTEGRIN\nMEMBRANE-ECM\nINTERACTIONS|REACTOME","REGULATION OF\nHORMONE LEVELS|GOBP","REGULATION OF\nTRANSMEMBRANE\nTRANSPORT|GOBP","SENSORY PERCEPTION\nOF MECHANICAL\nSTIMULUS|GOBP","CARDIAC MUSCLE CELL\nACTION\nPOTENTIAL|GOBP","EMBRYONIC\nMORPHOGENESIS|GOBP","EPIGENETIC\nREGULATION OF GENE\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","SYNAPTIC VESICLE\nEXOCYTOSIS|GOBP","VASCULATURE\nDEVELOPMENT|GOBP","KIDNEY\nDEVELOPMENT|GOBP","HDMS DEMETHYLATE\nHISTONES|REACTOME","CARDIAC MUSCLE CELL\nACTION POTENTIAL\nINVOLVED IN\nCONTRACTION|GOBP","CARDIOVASCULAR\nSYSTEM\nDEVELOPMENT|GOBP","WOUND HEALING,\nSPREADING OF\nCELLS|GOBP","DEFENSE RESPONSE TO\nVIRUS|GOBP","GENE SILENCING|GOBP","EPIBOLY INVOLVED IN\nWOUND HEALING|GOBP","CALCIUM-DEPENDENT\nCELL-CELL ADHESION\nVIA PLASMA MEMBRANE\nCELL ADHESION\nMOLECULES|GOBP","MONOVALENT\nINORGANIC CATION\nTRANSPORT|GOBP","CARDIAC MUSCLE CELL\nCONTRACTION|GOBP","REGULATION OF\nPOTASSIUM ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","REGULATION OF\nEPIDERMIS\nDEVELOPMENT|GOBP","NABA_PROTEOGLYCANS|MSIGDB_C2","SINGLE ORGANISM\nCELL ADHESION|GOBP","INTRINSIC APOPTOTIC\nSIGNALING\nPATHWAY|GOBP","ENDODERMAL CELL\nDIFFERENTIATION|GOBP","REGULATION OF\nINTRINSIC APOPTOTIC\nSIGNALING\nPATHWAY|GOBP","HALLMARK_INTERFERON_ALPHA_RESPONSE|MSIGDB_C2","REGULATION OF\nANGIOGENESIS|GOBP","PROTEIN-DNA COMPLEX\nASSEMBLY|GOBP","EPIBOLY|GOBP","POSITIVE REGULATION\nOF BMP SIGNALING\nPATHWAY|GOBP","CELLULAR POTASSIUM\nION TRANSPORT|GOBP","REGULATION OF CELL\nADHESION MEDIATED\nBY INTEGRIN|GOBP","PID_BETA_CATENIN_NUC_PATHWAY|MSIGDB_C2","HALLMARK_GLYCOLYSIS|MSIGDB_C2","TELOMERE\nMAINTENANCE|REACTOME","AMEBOIDAL-TYPE CELL\nMIGRATION|GOBP","POTASSIUM ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","RESPONSE TO\nGLUCOCORTICOID|GOBP","REGULATION OF\nVASCULATURE\nDEVELOPMENT|GOBP","APOPTOTIC SIGNALING\nPATHWAY|GOBP","SYNAPTIC\nSIGNALING|GOBP","TRANSMISSION OF\nNERVE IMPULSE|GOBP","COLLAGEN FIBRIL\nORGANIZATION|GOBP","REGULATION OF HEART\nRATE BY CARDIAC\nCONDUCTION|GOBP","TRANS-SYNAPTIC\nSIGNALING|GOBP","POTASSIUM\nCHANNELS|REACTOME\nDATABASE ID RELEASE\n61","ANTEROGRADE\nTRANS-SYNAPTIC\nSIGNALING|GOBP","RENAL SYSTEM\nDEVELOPMENT|GOBP","MUSCLE\nCONTRACTION|GOBP","HSF1-DEPENDENT\nTRANSACTIVATION|REACTOME\nDATABASE ID RELEASE\n61","NOTCH SIGNALING\nPATHWAY|GOBP","CHEMICAL SYNAPTIC\nTRANSMISSION|GOBP","HALLMARK_KRAS_SIGNALING_UP|MSIGDB_C2","COLLAGEN\nFORMATION|REACTOME","POSITIVE REGULATION\nOF OXIDOREDUCTASE\nACTIVITY|GOBP","GLOMERULUS\nDEVELOPMENT|GOBP","HALLMARK_HEDGEHOG_SIGNALING|MSIGDB_C2","CELL-CELL\nADHESION|GOBP","POSITIVE REGULATION\nOF MRNA CATABOLIC\nPROCESS|GOBP","RESPONSE TO\nCORTICOSTEROID|GOBP","REGULATION OF\nCHONDROCYTE\nDIFFERENTIATION|GOBP","REGULATION OF METAL\nION TRANSPORT|GOBP","NONHOMOLOGOUS\nEND-JOINING\n(NHEJ)|REACTOME","ACTION\nPOTENTIAL|GOBP","CELLULAR RESPONSE\nTO STARVATION|GOBP","REGULATION OF LIPID\nKINASE\nACTIVITY|GOBP","OSSIFICATION|GOBP","REGULATION OF\nEPITHELIAL CELL\nPROLIFERATION|GOBP","ENDOPLASMIC\nRETICULUM UNFOLDED\nPROTEIN\nRESPONSE|GOBP","PROTEIN-PROTEIN\nINTERACTIONS AT\nSYNAPSES|REACTOME","CELL-CELL SIGNALING\nINVOLVED IN CARDIAC\nCONDUCTION|GOBP","POSITIVE REGULATION\nOF\nPEPTIDYL-TYROSINE\nPHOSPHORYLATION|GOBP","INTERLEUKIN-6\nFAMILY\nSIGNALING|REACTOME\nDATABASE ID RELEASE\n61","REGULATION OF CELL\nGROWTH|GOBP","REGULATION OF\nOXIDOREDUCTASE\nACTIVITY|GOBP","POSITIVE REGULATION\nOF EPITHELIAL CELL\nPROLIFERATION|GOBP"],"label":["NABA_CORE_MATRISOME|MSIGDB_C2","EXTRACELLULAR\nMATRIX\nORGANIZATION|GOBP","EXTRACELLULAR\nSTRUCTURE\nORGANIZATION|GOBP","EXTRACELLULAR\nMATRIX\nORGANIZATION|REACTOME","BETA1 INTEGRIN CELL\nSURFACE\nINTERACTIONS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","PID_INTEGRIN1_PATHWAY|MSIGDB_C2","NABA_ECM_GLYCOPROTEINS|MSIGDB_C2","HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION|MSIGDB_C2","HDACS DEACETYLATE\nHISTONES|REACTOME\nDATABASE ID RELEASE\n61","INTEGRIN CELL\nSURFACE\nINTERACTIONS|REACTOME\nDATABASE ID RELEASE\n61","NABA_BASEMENT_MEMBRANES|MSIGDB_C2","HALLMARK_TNFA_SIGNALING_VIA_NFKB|MSIGDB_C2","RNA POLYMERASE I\nPROMOTER\nOPENING|REACTOME\nDATABASE ID RELEASE\n61","AMYLOID FIBER\nFORMATION|REACTOME","DNA\nMETHYLATION|REACTOME","REGULATION OF IGF\nACTIVITY BY\nIGFBP|REACTOME\nDATABASE ID RELEASE\n61","ACTIVATED PKN1\nSTIMULATES\nTRANSCRIPTION OF AR\n(ANDROGEN RECEPTOR)\nREGULATED GENES\nKLK2 AND\nKLK3|REACTOME\nDATABASE ID RELEASE\n61","SIRT1 NEGATIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME","O-LINKED\nGLYCOSYLATION|REACTOME","MULTICELLULAR\nORGANISMAL\nSIGNALING|GOBP","CARDIAC\nCONDUCTION|GOBP","DNA DAMAGE TELOMERE\nSTRESS INDUCED\nSENESCENCE|REACTOME","POST-TRANSLATIONAL\nPROTEIN\nPHOSPHORYLATION|REACTOME","DEFECTIVE B3GALTL\nCAUSES PETERS-PLUS\nSYNDROME\n(PPS)|REACTOME","PRC2 METHYLATES\nHISTONES AND\nDNA|REACTOME\nDATABASE ID RELEASE\n61","DISEASES OF\nGLYCOSYLATION|REACTOME","CONDENSATION OF\nPROPHASE\nCHROMOSOMES|REACTOME\nDATABASE ID RELEASE\n61","O-GLYCOSYLATION OF\nTSR\nDOMAIN-CONTAINING\nPROTEINS|REACTOME\nDATABASE ID RELEASE\n61","NCAM1\nINTERACTIONS|REACTOME\nDATABASE ID RELEASE\n61","SIGNAL RELEASE|GOBP","ACTIVATION OF RRNA\nEXPRESSION BY ERCC6\n(CSB) AND EHMT2\n(G9A)|REACTOME","RUNX1 REGULATES\nGENES INVOLVED IN\nMEGAKARYOCYTE\nDIFFERENTIATION AND\nPLATELET\nFUNCTION|REACTOME\nDATABASE ID RELEASE\n61","FORMATION OF THE\nBETA-CATENIN:TCF\nTRANSACTIVATING\nCOMPLEX|REACTOME\nDATABASE ID RELEASE\n61","DISEASES ASSOCIATED\nWITH\nO-GLYCOSYLATION OF\nPROTEINS|REACTOME\nDATABASE ID RELEASE\n61","PROTEIN\nHETEROOLIGOMERIZATION|GOBP","SENESCENCE-ASSOCIATED\nSECRETORY PHENOTYPE\n(SASP)|REACTOME","HATS ACETYLATE\nHISTONES|REACTOME","MEIOTIC\nRECOMBINATION|REACTOME\nDATABASE ID RELEASE\n61","CELLULAR\nSENESCENCE|REACTOME","RNA POLYMERASE I\nCHAIN\nELONGATION|REACTOME\nDATABASE ID RELEASE\n61","B-WICH COMPLEX\nPOSITIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME","RMTS METHYLATE\nHISTONE\nARGININES|REACTOME","NABA_ECM_REGULATORS|MSIGDB_C2","DEGRADATION OF THE\nEXTRACELLULAR\nMATRIX|REACTOME\nDATABASE ID RELEASE\n61","RHO GTPASES\nACTIVATE\nPKNS|REACTOME\nDATABASE ID RELEASE\n61","REGULATION OF BLOOD\nCIRCULATION|GOBP","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nBY P53 CLASS\nMEDIATOR|GOBP","NUCLEOSOME\nASSEMBLY|GOBP","PROTEIN\nOLIGOMERIZATION|GOBP","REGULATION OF HEART\nCONTRACTION|GOBP","TRANSCRIPTIONAL\nREGULATION BY SMALL\nRNAS|REACTOME","CARDIAC\nCONDUCTION|REACTOME\nDATABASE ID RELEASE\n61","MEIOSIS|REACTOME\nDATABASE ID RELEASE\n61","DIRECT P53\nEFFECTORS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","PROTEIN\nHETEROTETRAMERIZATION|GOBP","CELL-SUBSTRATE\nADHESION|GOBP","OXIDATIVE STRESS\nINDUCED\nSENESCENCE|REACTOME","POSITIVE REGULATION\nOF VASCULATURE\nDEVELOPMENT|GOBP","HALLMARK_APOPTOSIS|MSIGDB_C2","POSITIVE EPIGENETIC\nREGULATION OF RRNA\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","PACKAGING OF\nTELOMERE\nENDS|REACTOME\nDATABASE ID RELEASE\n61","HALLMARK_INFLAMMATORY_RESPONSE|MSIGDB_C2","CHROMATIN\nSILENCING|GOBP","NORC NEGATIVELY\nREGULATES RRNA\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","POSITIVE REGULATION\nOF\nANGIOGENESIS|GOBP","SIGNAL RELEASE FROM\nSYNAPSE|GOBP","NEUROTRANSMITTER\nSECRETION|GOBP","CHROMATIN\nASSEMBLY|GOBP","RNA POLYMERASE I\nPROMOTER\nCLEARANCE|REACTOME\nDATABASE ID RELEASE\n61","NEGATIVE EPIGENETIC\nREGULATION OF RRNA\nEXPRESSION|REACTOME","RNA POLYMERASE I\nTRANSCRIPTION|REACTOME","HALLMARK_P53_PATHWAY|MSIGDB_C2","REGULATION OF\nNEUROTRANSMITTER\nLEVELS|GOBP","POSITIVE REGULATION\nOF NEURON\nPROJECTION\nDEVELOPMENT|GOBP","ANGIOGENESIS|GOBP","PRESYNAPTIC PROCESS\nINVOLVED IN\nCHEMICAL SYNAPTIC\nTRANSMISSION|GOBP","ER-NUCLEUS\nSIGNALING\nPATHWAY|GOBP","NUCLEOSOME\nASSEMBLY|REACTOME","DEPOSITION OF NEW\nCENPA-CONTAINING\nNUCLEOSOMES AT THE\nCENTROMERE|REACTOME","DNA\nREPLICATION-DEPENDENT\nNUCLEOSOME\nORGANIZATION|GOBP","REGULATION OF ION\nTRANSPORT|GOBP","DNA\nREPLICATION-DEPENDENT\nNUCLEOSOME\nASSEMBLY|GOBP","ECM\nPROTEOGLYCANS|REACTOME","ACTIVATION OF\nANTERIOR HOX GENES\nIN HINDBRAIN\nDEVELOPMENT DURING\nEARLY\nEMBRYOGENESIS|REACTOME\nDATABASE ID RELEASE\n61","CHROMATIN SILENCING\nAT RDNA|GOBP","HALLMARK_MYOGENESIS|MSIGDB_C2","CALCIUM\nION-REGULATED\nEXOCYTOSIS OF\nNEUROTRANSMITTER|GOBP","DOPAMINE\nNEUROTRANSMITTER\nRELEASE\nCYCLE|REACTOME","COLLAGEN CHAIN\nTRIMERIZATION|REACTOME\nDATABASE ID RELEASE\n61","HORMONE\nSECRETION|GOBP","MUSCLE\nCONTRACTION|REACTOME\nDATABASE ID RELEASE\n61","ACTIVATION OF HOX\nGENES DURING\nDIFFERENTIATION|REACTOME","NABA_COLLAGENS|MSIGDB_C2","MULTICELLULAR\nORGANISMAL\nCATABOLIC\nPROCESS|GOBP","NUCLEOSOME\nORGANIZATION|GOBP","HALLMARK_HYPOXIA|MSIGDB_C2","NEUROTRANSMITTER\nTRANSPORT|GOBP","MEIOTIC\nSYNAPSIS|REACTOME","INTEGRIN SIGNALLING\nPATHWAY|PANTHER\nPATHWAY","REGULATION OF\nCALCIUM\nION-DEPENDENT\nEXOCYTOSIS|GOBP","RESPONSE TO\nUNFOLDED\nPROTEIN|GOBP","POSITIVE REGULATION\nOF NERVOUS SYSTEM\nDEVELOPMENT|GOBP","MULTICELLULAR\nORGANISMAL\nMACROMOLECULE\nMETABOLIC\nPROCESS|GOBP","PID_P53_DOWNSTREAM_PATHWAY|MSIGDB_C2","POTASSIUM ION\nTRANSPORT|GOBP","GENE SILENCING BY\nRNA|REACTOME\nDATABASE ID RELEASE\n61","ENDODERM\nDEVELOPMENT|GOBP","RESPONSE TO\nTOPOLOGICALLY\nINCORRECT\nPROTEIN|GOBP","CHROMATIN ASSEMBLY\nOR DISASSEMBLY|GOBP","LAMININ\nINTERACTIONS|REACTOME","NABA_SECRETED_FACTORS|MSIGDB_C2","NEGATIVE REGULATION\nOF GENE EXPRESSION,\nEPIGENETIC|GOBP","POSITIVE REGULATION\nOF NEURON\nDIFFERENTIATION|GOBP","RUNX1 REGULATES\nTRANSCRIPTION OF\nGENES INVOLVED IN\nDIFFERENTIATION OF\nHSCS|REACTOME","HORMONE\nTRANSPORT|GOBP","COLLAGEN\nBIOSYNTHESIS AND\nMODIFYING\nENZYMES|REACTOME","PROTEIN\nTETRAMERIZATION|GOBP","PHASE 2 - PLATEAU\nPHASE|REACTOME","CALCIUM ION\nREGULATED\nEXOCYTOSIS|GOBP","GLYCOSAMINOGLYCAN\nMETABOLIC\nPROCESS|GOBP","SYNAPSE\nORGANIZATION|GOBP","MITOTIC\nPROPHASE|REACTOME","COLLAGEN CATABOLIC\nPROCESS|GOBP","PID_INTEGRIN_CS_PATHWAY|MSIGDB_C2","BLOOD VESSEL\nMORPHOGENESIS|GOBP","RESPONSE TO\nVIRUS|GOBP","SIGNAL TRANSDUCTION\nBY P53 CLASS\nMEDIATOR|GOBP","REGULATION OF\nNEURON PROJECTION\nDEVELOPMENT|GOBP","INTEGRIN FAMILY\nCELL SURFACE\nINTERACTIONS|PATHWAY\nINTERACTION\nDATABASE NCI-NATURE\nCURATED DATA","DNA PACKAGING|GOBP","GLYCOSAMINOGLYCAN\nBIOSYNTHETIC\nPROCESS|GOBP","REGULATION OF\nCATION\nTRANSMEMBRANE\nTRANSPORT|GOBP","POSITIVE REGULATION\nOF MRNA METABOLIC\nPROCESS|GOBP","PEPTIDE HORMONE\nSECRETION|GOBP","POSITIVE REGULATION\nOF DEVELOPMENTAL\nGROWTH|GOBP","REGULATION OF\nNEURON\nDIFFERENTIATION|GOBP","MULTICELLULAR\nORGANISM METABOLIC\nPROCESS|GOBP","BLOOD VESSEL\nDEVELOPMENT|GOBP","AMINOGLYCAN\nBIOSYNTHETIC\nPROCESS|GOBP","POLYSACCHARIDE\nMETABOLIC\nPROCESS|GOBP","POSITIVE REGULATION\nOF\nNEUROGENESIS|GOBP","SUBSTRATE\nADHESION-DEPENDENT\nCELL SPREADING|GOBP","SENSORY PERCEPTION\nOF SOUND|GOBP","NEGATIVE REGULATION\nOF NERVOUS SYSTEM\nDEVELOPMENT|GOBP","AMINOGLYCAN\nMETABOLIC\nPROCESS|GOBP","MUCOPOLYSACCHARIDE\nMETABOLIC\nPROCESS|GOBP","REGULATION OF ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","COLLAGEN METABOLIC\nPROCESS|GOBP","MEMBRANE\nDEPOLARIZATION|GOBP","CELLULAR RESPONSE\nTO CAMP|GOBP","POSITIVE REGULATION\nOF CELL PROJECTION\nORGANIZATION|GOBP","PEPTIDE\nSECRETION|GOBP","ANION\nHOMEOSTASIS|GOBP","VOLTAGE GATED\nPOTASSIUM\nCHANNELS|REACTOME","CELL-MATRIX\nADHESION|GOBP","EXTRACELLULAR\nMATRIX\nDISASSEMBLY|GOBP","GLYCOSAMINOGLYCAN\nMETABOLISM|REACTOME","CELL COMMUNICATION\nINVOLVED IN CARDIAC\nCONDUCTION|GOBP","ACID SECRETION|GOBP","NEPHRON\nDEVELOPMENT|GOBP","NON-INTEGRIN\nMEMBRANE-ECM\nINTERACTIONS|REACTOME","REGULATION OF\nHORMONE LEVELS|GOBP","REGULATION OF\nTRANSMEMBRANE\nTRANSPORT|GOBP","SENSORY PERCEPTION\nOF MECHANICAL\nSTIMULUS|GOBP","CARDIAC MUSCLE CELL\nACTION\nPOTENTIAL|GOBP","EMBRYONIC\nMORPHOGENESIS|GOBP","EPIGENETIC\nREGULATION OF GENE\nEXPRESSION|REACTOME\nDATABASE ID RELEASE\n61","SYNAPTIC VESICLE\nEXOCYTOSIS|GOBP","VASCULATURE\nDEVELOPMENT|GOBP","KIDNEY\nDEVELOPMENT|GOBP","HDMS DEMETHYLATE\nHISTONES|REACTOME","CARDIAC MUSCLE CELL\nACTION POTENTIAL\nINVOLVED IN\nCONTRACTION|GOBP","CARDIOVASCULAR\nSYSTEM\nDEVELOPMENT|GOBP","WOUND HEALING,\nSPREADING OF\nCELLS|GOBP","DEFENSE RESPONSE TO\nVIRUS|GOBP","GENE SILENCING|GOBP","EPIBOLY INVOLVED IN\nWOUND HEALING|GOBP","CALCIUM-DEPENDENT\nCELL-CELL ADHESION\nVIA PLASMA MEMBRANE\nCELL ADHESION\nMOLECULES|GOBP","MONOVALENT\nINORGANIC CATION\nTRANSPORT|GOBP","CARDIAC MUSCLE CELL\nCONTRACTION|GOBP","REGULATION OF\nPOTASSIUM ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","REGULATION OF\nEPIDERMIS\nDEVELOPMENT|GOBP","NABA_PROTEOGLYCANS|MSIGDB_C2","SINGLE ORGANISM\nCELL ADHESION|GOBP","INTRINSIC APOPTOTIC\nSIGNALING\nPATHWAY|GOBP","ENDODERMAL CELL\nDIFFERENTIATION|GOBP","REGULATION OF\nINTRINSIC APOPTOTIC\nSIGNALING\nPATHWAY|GOBP","HALLMARK_INTERFERON_ALPHA_RESPONSE|MSIGDB_C2","REGULATION OF\nANGIOGENESIS|GOBP","PROTEIN-DNA COMPLEX\nASSEMBLY|GOBP","EPIBOLY|GOBP","POSITIVE REGULATION\nOF BMP SIGNALING\nPATHWAY|GOBP","CELLULAR POTASSIUM\nION TRANSPORT|GOBP","REGULATION OF CELL\nADHESION MEDIATED\nBY INTEGRIN|GOBP","PID_BETA_CATENIN_NUC_PATHWAY|MSIGDB_C2","HALLMARK_GLYCOLYSIS|MSIGDB_C2","TELOMERE\nMAINTENANCE|REACTOME","AMEBOIDAL-TYPE CELL\nMIGRATION|GOBP","POTASSIUM ION\nTRANSMEMBRANE\nTRANSPORT|GOBP","RESPONSE TO\nGLUCOCORTICOID|GOBP","REGULATION OF\nVASCULATURE\nDEVELOPMENT|GOBP","APOPTOTIC SIGNALING\nPATHWAY|GOBP","SYNAPTIC\nSIGNALING|GOBP","TRANSMISSION OF\nNERVE IMPULSE|GOBP","COLLAGEN FIBRIL\nORGANIZATION|GOBP","REGULATION OF HEART\nRATE BY CARDIAC\nCONDUCTION|GOBP","TRANS-SYNAPTIC\nSIGNALING|GOBP","POTASSIUM\nCHANNELS|REACTOME\nDATABASE ID RELEASE\n61","ANTEROGRADE\nTRANS-SYNAPTIC\nSIGNALING|GOBP","RENAL SYSTEM\nDEVELOPMENT|GOBP","MUSCLE\nCONTRACTION|GOBP","HSF1-DEPENDENT\nTRANSACTIVATION|REACTOME\nDATABASE ID RELEASE\n61","NOTCH SIGNALING\nPATHWAY|GOBP","CHEMICAL SYNAPTIC\nTRANSMISSION|GOBP","HALLMARK_KRAS_SIGNALING_UP|MSIGDB_C2","COLLAGEN\nFORMATION|REACTOME","POSITIVE REGULATION\nOF OXIDOREDUCTASE\nACTIVITY|GOBP","GLOMERULUS\nDEVELOPMENT|GOBP","HALLMARK_HEDGEHOG_SIGNALING|MSIGDB_C2","CELL-CELL\nADHESION|GOBP","POSITIVE REGULATION\nOF MRNA CATABOLIC\nPROCESS|GOBP","RESPONSE TO\nCORTICOSTEROID|GOBP","REGULATION OF\nCHONDROCYTE\nDIFFERENTIATION|GOBP","REGULATION OF METAL\nION TRANSPORT|GOBP","NONHOMOLOGOUS\nEND-JOINING\n(NHEJ)|REACTOME","ACTION\nPOTENTIAL|GOBP","CELLULAR RESPONSE\nTO STARVATION|GOBP","REGULATION OF LIPID\nKINASE\nACTIVITY|GOBP","OSSIFICATION|GOBP","REGULATION OF\nEPITHELIAL CELL\nPROLIFERATION|GOBP","ENDOPLASMIC\nRETICULUM UNFOLDED\nPROTEIN\nRESPONSE|GOBP","PROTEIN-PROTEIN\nINTERACTIONS AT\nSYNAPSES|REACTOME","CELL-CELL SIGNALING\nINVOLVED IN CARDIAC\nCONDUCTION|GOBP","POSITIVE REGULATION\nOF\nPEPTIDYL-TYROSINE\nPHOSPHORYLATION|GOBP","INTERLEUKIN-6\nFAMILY\nSIGNALING|REACTOME\nDATABASE ID RELEASE\n61","REGULATION OF CELL\nGROWTH|GOBP","REGULATION OF\nOXIDOREDUCTASE\nACTIVITY|GOBP","POSITIVE REGULATION\nOF EPITHELIAL CELL\nPROLIFERATION|GOBP"],"color.background":["#FF00CC","#FF00CC","#FF00CC","#FF00CC","#FF00CC","#FF00CC","#FF00CC","#FF00CC","#33CCFF","#FF00CC","#FF00CC","#33CC00","#33CCFF","#33CCFF","#33CCFF","#9900FF","#33CCFF","#33CCFF","#FF9900","#FFFF00","#FFFF00","#33CCFF","#9900FF","#FF9900","#33CCFF","#FF9900","#33CCFF","#FF9900","#FFCCFF","#FF00CC","#33CCFF","#33CCFF","#33CCFF","#FF9900","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#FF9900","#FF00CC","#33CCFF","#FFFF00","#FF0000","#33CCFF","#33CCFF","#FFFF00","#33CCFF","#FFFF00","#33CCFF","#006600","#33CCFF","#FF00CC","#33CCFF","#CCCC00","#009999","#33CCFF","#33CCFF","#993300","#33CCFF","#33CCFF","#CCCC00","#FF00CC","#FF00CC","#33CCFF","#33CCFF","#33CCFF","#33CCFF","#CC99CC","#FF00CC","#CCCCFF","#6699CC","#FF00CC","#FFCC99","#33CCFF","#33CCFF","#33CCFF","#FFFF00","#33CCFF","#FF00CC","#33CCFF","#33CCFF","#9966FF","#FF00CC","#FF00CC","#FF00CC","#FF00CC","#FFFF00","#33CCFF","#FF00CC","#FF00CC","#33CCFF","#CC6600","#FF00CC","#33CCFF","#FF00CC","#FF00CC","#FFCC99","#CCCCFF","#FF00CC","#006600","#FFFF00","#33CCFF","#FF00CC","#FFCC99","#33CCFF","#FF00CC","#CCFFFF","#33CCFF","#CCCCFF","#33CCFF","#FF00CC","#FF00CC","#33CCFF","#FFFF00","#FF00CC","#99CC00","#FF00CC","#33CCFF","#FF00CC","#FF00CC","#6699CC","#FF99FF","#FF0000","#CCCCFF","#FF00CC","#33CCFF","#99CC00","#FFFF00","#0066FF","#FF00CC","#CCCCFF","#CCCCFF","#FF00CC","#6699CC","#99CC00","#99CCFF","#CCCCFF","#FF00CC","#66FFCC","#CCCCFF","#99CC00","#99CC00","#FFFF00","#FF00CC","#FFFF00","#CC9900","#CCCCFF","#FF00CC","#9999CC","#FFFF00","#FF00CC","#FF00CC","#99CC00","#FFFF00","#FF00CC","#CC33FF","#FF00CC","#FF00CC","#FFFF00","#66FFCC","#FFFF00","#FF00CC","#33CCFF","#FF00CC","#6699CC","#CC33FF","#33CCFF","#FFFF00","#6699CC","#006699","#FF99FF","#33CCFF","#006699","#FF00CC","#FFFF00","#FFFF00","#FFFF00","#F5DF16","#FF00CC","#FF00CC","#FF0000","#FF00CC","#B5185E","#99FF00","#CCCC00","#33CCFF","#006699","#996600","#FFFF00","#33CCCC","#990000","#00FFFF","#33CCFF","#CC6666","#FFFF00","#CC0000","#CCCC00","#FF0000","#FF00CC","#FFFF00","#FF00CC","#FFFF00","#FF00CC","#FFFF00","#FF00CC","#CC33FF","#FFFF00","#9999FF","#3366FF","#FF00CC",null,"#FF00CC",null,"#CC33FF",null,"#FF00CC","#0066FF","#CC0000",null,"#FFFF00","#33CCFF","#FFFF00",null,null,null,null,"#FFCC99","#FF00CC","#FFFF00",null,null,"#CCCCFF",null,null],"FDR":[3.02e-10,3.40513e-08,3.48708e-08,3.50738e-08,7.29524e-07,7.43915e-07,7.46461e-07,2.96e-05,3.70995e-05,3.79679e-05,5.45e-05,0.000116,0.000159,0.000162,0.000171,0.000197,0.000277,0.000317,0.000388,0.000396749,0.000399609,0.000462,0.000482,0.000497,0.000515279,0.0005247,0.000531177,0.000555,0.000691,0.000715,0.000753,0.000916,0.00102,0.00151,0.00153,0.00191,0.00232,0.00269,0.00369,0.00410838,0.00411944,0.00489,0.00499,0.00555,0.00621,0.00661,0.00678,0.00763174,0.0076832,0.00865,0.009639999999999999,0.009820000000000001,0.0115469,0.0116113,0.0131,0.0133,0.0148718,0.0149936,0.0150544,0.0151244,0.0153546,0.0156211,0.0156688,0.0160572,0.017,0.0194107,0.0194462,0.0199,0.0199006,0.0200272,0.0229,0.0243,0.0257959,0.026,0.0260098,0.0264263,0.0267205,0.0290894,0.0293792,0.0295181,0.0297,0.0301039,0.0351,0.0375123,0.0379152,0.0381888,0.0382094,0.0387,0.0387023,0.038766,0.038804,0.0388249,0.0391748,0.0391783,0.0412,0.0419,0.0419029,0.0421,0.0425,0.0427035,0.0431333,0.0459548,0.0460257,0.0467,0.0472,0.0490228,0.0497792,0.0502,0.0512,0.0514,0.054,0.0571637,0.0576,0.0577409,0.0581793,0.0584,0.0589,0.0598207,0.0613537,0.0627485,0.0683,0.0704,0.0709,0.0768151,0.0768773,0.076906,0.0773297,0.0789641,0.0789728,0.08078249999999999,0.081,0.0822,0.08278820000000001,0.0831522,0.0832239,0.0832855,0.08409999999999999,0.0842706,0.0844,0.08449810000000001,0.0847823,0.0852805,0.08535719999999999,0.0856441,0.0857464,0.0857777,0.08605980000000001,0.0910169,0.0910957,0.09135989999999999,0.0914,0.0916331,0.091701,0.093112,0.0975719,0.09774819999999999,0.09935919999999999,0.101066,0.101862,0.102,0.102709,0.103,0.105,0.108583,0.110088,0.115,0.120888,0.123066,0.123761,0.124,0.130577,0.131,0.132472,0.133207,0.134,0.136269,0.136592,0.136754,0.137,0.138292,0.138741,0.139175,0.141134,0.142,0.142125,0.142462,0.142591,0.143189,0.143638,0.145302,0.146208,0.146577,0.146766,0.14773,0.148301,0.148302,0.148415,0.148596,0.150792,0.150901,0.151,0.159,0.162399,0.162955,0.163093,0.163097,0.164469,0.16553,0.166727,0.167627,0.167908,0.168104,0.168105,0.1682,0.168407,0.172,0.172414,0.173201,0.173469,0.176,0.179661,0.179762,0.18012,0.181647,0.187052,0.187659,0.189506,0.18997,0.191,0.191341,0.191923,0.195924,0.196959,0.197007,0.197898,0.198043,0.198614,0.201845],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"borderWidthSelected":[4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4],"color.highlight.border":["#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066"],"color.highlight.background":["#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff"],"color.hover.background":["#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066"],"color.hover.border":["#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff"],"font.face":["Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial"],"shape":["dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot"],"font.size":[26,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22],"font.bold":[true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[4,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],"x":[-0.3372711693372009,-0.2322485364352678,-0.2116089201246598,-0.2246434973155781,-0.2780147066627254,-0.2501405992019327,-0.4277036245468082,-0.5061712445492477,-0.5668069389458485,-0.1480812202882785,-0.3288043973589406,-0.8452269337470091,-0.5847105158794762,-0.5897506921774527,-0.5789796552814773,0.2126001278294583,-0.5635010765448389,-0.5791375020240059,0.4287989294247638,0.6607093371350059,0.6281691297743857,-0.6516379642757635,0.2519026375147275,0.4118295823067468,-0.5897141526006506,0.3495805821059432,-0.5698090153761155,0.3620314370042288,0.9971404159663866,0.1642066512149514,-0.5627721459443267,-0.5530857913389525,-0.602975151931885,0.3862095099203329,-0.4414823178413027,-0.6447969042518671,-0.5133751998581635,-0.6000244086494664,-0.6337471712506506,-0.5864803273130315,-0.600639183588391,-0.4648290476287855,0.3811413623822406,-0.1009736714191023,-0.6534977147189325,0.6072325021703486,-0.9206234571519842,-0.4971899754469572,-0.4773834699697691,0.6246888961100385,-0.6430615285006409,0.5827436365678393,-0.6485277995154743,0.6166038047858946,-0.5445942276017937,-0.2356625215550726,-0.6542783203531848,0.3577024876891572,0.03262499516585371,-0.6051941859773571,-0.6117405663488316,-0.7709713307359128,-0.5333797766451098,-0.6399192844215427,0.3163824052414443,0.1970311524762423,0.2248845438494849,-0.5196569653849401,-0.6245716529788923,-0.623377863921424,-0.6111923624131225,-0.7429223038366395,0.2404162452697878,0.7393100605318372,-0.1322712821071539,0.2666149705745415,0.8804567546648723,-0.6217116151351307,-0.6258373093535068,-0.5580146718435846,0.6884941249559697,-0.546258278005836,-0.3506478712481694,-0.6220866517548834,-0.5323929428108694,-0.8784175481404335,0.2700619681049246,0.1746139613517168,-0.2865052365882637,0.02516153117469533,0.6058541203738919,-0.634220444657384,-0.2463622783223048,-0.2187130593153488,-0.5224734957129173,0.1861276142775297,0.2860668405312086,-0.6818856816316723,-0.2502598118060375,0.2770253888760572,0.8401616960684928,0.74346136704713,-0.2276028651427149,0.5589377398354398,0.7261335720339108,-0.6644344245996886,0.00625212191156943,0.7953292988553946,-0.5239570701836546,-0.3079356651273506,0.3531633896394979,-0.4877497114193112,0.694422070707704,-0.6529890470490476,0.0008116346113491435,-0.2797531482224189,-0.4143241017423712,0.5845939420085724,0.2980152503446285,0.5257055953850807,0.1996620559047271,-0.6624670560365682,-0.2547153472331866,-0.1904502927305445,-0.1468235152010264,-0.4662733824315234,-0.9561157598035633,0.6984590248909472,-0.2101423691476239,-0.5548158181561198,0.571216143634788,0.6270761643132594,-0.7412318603604506,0.03040349077355509,0.6484380572447379,0.6663915523742139,-0.2010832802867922,-0.0802765442255795,0.5392091983065097,0.03331613874727579,0.6904583753107312,-0.2412798186211473,0.785572786030333,0.6489541945243729,0.4843617207437725,0.4995368228323329,0.6648576418468,-0.2743210091122825,0.6016264286250363,0.5801782449499133,0.7352950360056261,-0.04894814033923378,-1,0.8124204836063231,-0.2794451406670487,-0.1099479320691068,0.5709233407378014,0.7235123331933468,0.2542265945834932,0.1095846794098705,-0.1843639904453681,-0.06976092529132882,0.6584237022567372,0.7219600023486754,0.6831799295011178,0.04157975365887023,-0.5982984775267201,0.2247139997235168,-0.06903787285693241,0.08553272520475885,-0.5575988749138834,0.7064252493136189,-0.1102854763642669,0.2184544006770073,-0.4117731900563363,-0.464813058752003,0.226890858753076,0.1445368928501232,0.7374803847800002,0.6875512096894578,0.6054344392077884,0.9699978159446661,-0.5111589716596296,-0.1686121797703858,-0.8618336757171954,-0.08876613409764411,-0.7638105509578801,0.0194545235495216,0.3518342171625486,-0.551088608980986,0.1656318452774221,0.05932515891362633,0.8141001225851732,-0.9056922456919909,1,0.4674693032647235,-0.6844156505683412,-0.7125667883997575,0.7647458022819185,0.2542328781893262,0.4003441794833826,-0.9137164000341611,0.2269478103224725,0.805741268781605,-0.3606301739238493,0.6779178202823708,0.2311893531002918,0.767056659692777,0.2543738140258902,0.06045164310319673,0.7748249835777701,0.1101748174465731,-0.564755443066793,0.2045471059528305,0.2458050431768495,-0.2990865098693701,0.2074419133986181,0.1390622463822551,-0.07192813490084982,-0.06331766749439149,-0.7473918569313958,0.306032628502005,0.8941755856961311,0.7106571209437749,-0.6501890188543484,0.710494257673298,0.3448532054865829,-0.2897606444424969,0.9414437707886478,0.07395234409411278,0.8358938331919408,0.05560972156529709,0.665493242703884,-0.6302862917009663,-0.6300517026310997,0.5490775312979357,0.2117282512682557,0.08707114856140508],"y":[0.3976155036981759,0.4052847861422735,0.4058990548695267,0.4203576264420856,0.5156644113828952,0.5071813446849753,0.509248921358695,0.2853974728558106,-0.409146995392679,0.4962223964234678,0.4730870777065561,0.5436897542055752,-0.3937390083833823,-0.451415699793472,-0.3740502954096763,-0.1795110968526622,-0.3843902166119272,-0.4250974895599154,-0.877118924113946,-0.3414981943368124,-0.4373291323519651,-0.3422747731708794,-0.2366235217290459,-0.8238055184030612,-0.4134320450854991,-0.8766369214454091,-0.3968611326765213,-0.8264165524786429,0.1408722337406501,0.8769726567455556,-0.3722553891113001,-0.4064260616014296,-0.3751722314944216,-0.9111455832444383,-0.5662702300969791,-0.4599952983834428,-0.4010611165652127,-0.3577098628007828,-0.3634835296739484,-0.3531610861314333,-0.3881892215837147,-0.4079866941301872,-0.749523302471234,0.4177796143098256,-0.429318125600515,-0.4104222927225322,0.3506201617801223,-0.3579276779860999,-0.5862323215399055,-0.3898379389147646,-0.4451870597753776,-0.3695732913160118,-0.3812106537266304,-0.80490600232534,-0.4541723075604769,0.7026293504936809,-0.400107556568299,0.3766174075791722,-0.5489560126709051,-0.4555001812236733,-0.340586795678349,0.6134878761381892,-0.4871626902483261,-0.4182467926568303,0.3319147069472537,0.8495599837110779,0.846494183675311,-0.358258218768886,-0.4611331147419869,-0.4202241811586859,-0.4313066112256527,0.7087513722757739,0.8746796292564971,0.5360217329772958,-0.965206362334678,0.8275327394027014,0.1439435448226867,-0.3737513663093253,-0.342048378551546,-0.4537390837517297,-0.5025368861039543,-0.4406316841052428,0.4388707470214581,-0.448537612215926,-0.4339094954752891,0.2015942219815392,0.8606239999105092,0.778416772677553,0.3509184789016793,0.9457684942001596,-0.3446602304506963,-0.4018127277337028,0.3554374861387208,0.3376814055091524,-0.3221748226544937,-0.02855000151749909,0.808024098155375,-0.3773532245232035,0.4732658483936787,0.8876640665600481,0.1850167017332656,0.5720050547555717,0.3096404095328651,-0.7848689100190587,-0.1245556630550446,-0.417732297301196,0.1850123496236491,0.1447413069948831,-0.3397002403920427,0.4925559100653298,-0.1266189527108079,-0.4757749423173863,0.5790883075062807,-0.3652978063568521,0.9178767059001658,0.3720663887412115,-0.5344794196050737,-0.454403787460227,0.8509674028248717,0.1826928986436918,0.5717851894871315,-0.4450016921710256,0.2914282400229933,0.5624903326070698,-0.917821632435053,0.8808254355955341,0.2837487261198961,0.5189196742753379,0.5730011021952022,-0.3123235385338279,0.1862031422824135,-0.4868469457938494,0.3878394819866291,0.9816265305383358,0.5876314909947156,0.5345223255615366,0.2964034284575014,-0.963625085261403,0.1040892787613845,-0.3896270049743888,0.6182169087909914,0.7971437914752353,0.3199722309277386,0.4433774161947823,0.1683980010486914,0.1208716037251265,-0.4771346477726646,0.307081966224519,-0.2773573718090178,0.7802631607517581,0.6103033830675304,1,0.1090427544279065,-0.1211406580425227,0.6689085022786336,0.3716301262446515,0.1351327089263572,-0.4348386554905471,0.7047592317706735,-0.7006294707921392,0.4888054342595463,0.9487394993127078,-0.5048843186540959,0.2939780244460637,-0.3812179494502789,0.2348870090799235,-0.4763432562706599,0.8942857294324016,-0.913969199777697,-0.7804755070504262,-0.4254144012971506,-0.3956088841105992,-0.8838983198469103,-0.4918719343914728,0.9200160745762926,-0.4440768748913774,-0.4230760707932407,0.6950060672195968,-0.2498850111770503,-0.409130375328933,-0.5347242930780178,0.01371475884723528,0.4151658049545124,0.6931695555408037,0.3906183469405813,0.3040250168353071,0.1902495894581506,-0.1818943410793448,0.2919285181264297,-0.3392772485809253,-0.4541371570937049,-0.06088727938070926,-0.1641835807601647,0.1007988823557102,-0.1001303923925592,0.4685428441679955,-0.3988387681108548,0.5252373263291836,-0.1535814524129946,-0.6482982730471185,0.3311905496162921,0.4259343061717438,0.7818173719905284,-0.2960406551839153,0.3454385708930148,-0.4354388164805345,0.8065918980538769,-0.1019041069262592,0.785695190738201,-0.726936356916557,-0.395272041850896,-0.2805610865320447,0.8338214093890419,0.8047215266052863,0.09243058121637593,0.3181968727180389,-0.9249299200151406,-0.7512363510728023,-0.7095857727708368,0.6478059733358656,0.3216969931816569,-0.6055678217171515,0.3980199451312219,-0.4882702529589089,-0.2502293822437531,-0.3334265990772697,-0.009081082827475107,-0.9518740090207526,0.2938801788372492,-1,0.1053776198459015,0.7430287322913456,-0.3930980514368246,0.7523459960914545,0.6879313886261913,0.614717766178501,-0.9930230730250028,-0.9337596477422497]},"edges":{"from":["1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","4","5","5","6","6","6","6","6","6","7","7","7","7","7","8","9","9","9","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","10","11","11","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","13","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","14","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","15","16","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","17","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","18","19","19","19","19","20","20","20","20","20","20","20","20","20","20","20","20","20","20","20","21","21","21","21","21","21","21","21","21","21","21","21","21","21","21","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","22","24","24","24","24","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","25","26","26","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","27","28","28","30","30","30","30","30","30","30","30","30","30","30","30","30","30","30","30","30","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","31","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","32","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","33","35","35","35","35","35","35","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","36","37","37","37","37","37","37","37","37","37","37","37","37","37","37","37","37","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","38","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","39","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","40","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","41","42","42","42","42","42","42","42","42","42","42","44","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","45","46","46","46","46","46","46","46","46","46","46","46","46","47","47","47","48","48","48","48","48","48","48","48","48","48","48","48","48","49","49","49","49","49","50","50","50","50","50","50","50","50","50","50","50","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","51","52","52","52","52","52","52","52","53","54","54","54","54","54","54","54","54","54","54","54","54","54","54","54","54","54","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","55","56","56","56","56","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","58","59","59","59","59","59","59","59","59","59","59","59","59","59","59","59","59","59","60","60","60","60","60","60","60","60","60","60","60","60","60","60","60","60","60","60","60","61","61","61","63","63","63","63","63","63","63","63","63","63","63","63","63","63","63","63","63","63","63","64","64","64","64","64","64","64","64","64","64","65","65","66","66","66","66","66","66","66","66","66","66","66","66","66","66","66","66","66","67","67","67","67","67","67","67","67","67","67","67","67","67","67","67","67","68","68","68","68","68","68","68","68","68","68","68","68","68","69","69","69","69","69","69","69","69","69","69","69","69","70","70","70","70","70","70","70","70","70","70","71","71","71","71","71","71","71","71","71","71","71","71","71","71","71","73","73","73","73","73","73","73","73","73","73","73","74","74","74","74","75","75","75","75","75","75","75","76","76","76","76","76","76","76","76","76","76","77","77","77","78","78","78","78","78","78","78","78","78","78","78","78","78","79","79","79","79","79","79","79","79","79","79","79","79","80","80","80","80","80","80","80","80","80","80","80","80","81","81","81","81","81","81","81","81","81","81","81","81","81","81","81","81","82","82","82","82","82","82","82","82","82","82","82","82","82","82","82","83","84","84","84","84","84","84","84","85","85","85","85","85","85","86","86","86","86","86","86","86","86","86","88","88","88","88","88","88","88","88","89","89","89","89","89","89","90","90","90","90","90","90","91","91","91","91","92","92","92","92","92","92","92","92","93","93","93","93","93","93","93","93","93","93","93","93","94","94","94","94","94","94","94","95","95","95","95","97","97","97","97","97","98","98","99","99","99","99","99","99","99","100","100","100","100","100","100","101","101","101","101","102","102","102","102","102","103","103","103","103","103","103","105","105","105","105","105","106","106","106","107","107","108","109","109","109","110","112","112","112","113","113","114","114","114","114","114","115","115","115","115","115","117","117","117","117","118","118","118","119","119","119","119","119","120","120","120","120","120","121","122","122","123","123","123","124","124","124","124","125","125","125","126","127","127","127","129","129","129","129","129","130","130","130","130","131","132","132","133","133","133","133","133","133","133","133","133","134","134","135","135","135","136","136","137","137","137","137","138","138","139","139","139","139","139","139","139","139","139","140","140","140","141","143","145","146","147","147","149","150","150","152","152","152","152","158","158","158","158","159","159","159","159","159","159","159","159","159","161","161","161","163","163","163","163","163","163","163","165","165","165","165","165","165","165","165","166","167","167","168","168","168","168","169","170","170","171","171","171","171","171","171","171","173","173","175","175","177","178","178","178","178","178","179","179","179","179","179","179","181","184","184","184","184","184","185","186","190","194","198","198","199","203","203","203","203","204","207","208","209","209","209","211","211","212","215","215","219","222","227","228","232"],"to":["5","6","7","11","83","86","88","89","102","110","115","123","149","180","208","216","3","4","6","7","9","11","44","83","86","88","89","102","110","115","123","124","125","134","149","156","157","182","208","216","4","6","7","9","11","44","83","86","88","89","102","110","115","123","124","125","134","149","156","157","182","208","216","6","7","9","11","44","83","86","88","89","102","110","115","123","124","125","134","149","156","157","182","208","216","11","110","7","11","101","110","124","125","11","101","110","124","125","208","101","124","125","13","14","15","17","18","25","27","31","32","33","36","37","38","39","40","41","42","45","51","54","55","58","59","60","63","64","66","67","71","78","79","81","82","84","85","93","95","106","109","113","122","167","173","101","110","14","15","17","18","22","25","27","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","225","15","17","18","25","27","31","32","33","36","37","38","39","40","41","45","51","54","55","58","59","60","63","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","191","17","18","22","25","27","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","225","23","18","22","25","27","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","225","22","25","27","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","24","26","28","34","21","46","50","52","90","117","150","159","165","171","184","203","204","227","235","46","50","52","80","90","117","133","139","159","163","165","171","184","203","235","25","27","31","32","33","36","38","39","40","41","45","51","54","58","59","60","63","66","67","70","71","78","79","84","85","97","106","109","113","122","132","167","188","225","26","28","34","43","27","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","28","34","31","32","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","34","43","68","69","73","76","91","92","94","99","100","118","119","136","168","209","211","212","213","32","33","36","37","38","39","40","41","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","112","113","122","132","167","173","176","188","191","33","36","37","38","39","40","41","42","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","113","122","132","167","173","176","188","191","36","37","38","39","40","41","45","48","51","54","55","58","59","60","63","64","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","113","122","132","167","173","188","191","49","55","81","82","93","116","38","39","40","41","45","51","54","55","58","59","60","63","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","38","40","41","42","45","51","55","58","60","64","78","79","81","82","93","173","39","40","41","45","48","51","54","55","58","59","60","63","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","113","122","132","167","173","188","191","225","40","41","45","51","54","55","58","59","60","63","66","67","71","78","79","81","82","84","85","93","97","173","188","225","41","45","48","51","54","55","58","59","60","63","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","113","122","132","167","173","188","191","45","48","51","54","55","58","59","60","63","66","67","70","71","78","79","81","82","84","85","93","95","97","106","109","113","122","132","167","173","188","191","55","64","81","82","93","95","109","112","173","176","156","51","54","55","58","59","60","63","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","50","52","90","117","159","165","171","181","184","203","227","235","128","186","202","55","58","70","78","79","81","82","93","95","109","132","173","191","55","81","82","93","116","52","90","117","159","165","171","181","184","203","227","235","54","55","58","59","60","63","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","90","117","150","165","171","184","203","104","55","58","60","63","66","67","71","78","79","81","82","93","97","122","173","188","225","59","60","63","64","66","67","70","71","81","82","84","85","93","95","106","109","112","113","116","122","132","167","173","176","191","124","125","146","155","59","60","63","66","67","70","71","78","79","84","85","95","97","106","109","113","122","132","167","188","191","225","60","63","66","67","71","78","79","81","82","84","85","93","97","113","167","173","188","63","64","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","65","190","201","64","66","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","191","66","67","71","81","82","93","112","167","173","176","190","201","67","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","191","71","78","79","81","82","84","85","93","97","106","113","122","167","173","188","191","69","73","76","92","94","99","100","119","168","209","211","212","213","73","76","92","94","99","100","119","168","209","211","212","213","78","79","81","82","93","95","109","132","173","191","78","79","81","82","84","85","93","97","106","113","122","167","173","188","191","76","92","94","99","100","119","168","209","211","212","213","127","147","169","172","103","114","129","137","140","145","148","92","94","99","100","119","168","209","211","212","213","98","108","231","79","84","85","95","97","106","109","113","122","132","167","188","191","84","85","95","97","106","109","113","122","132","167","188","191","117","133","139","159","163","165","171","181","184","203","222","235","82","84","85","93","95","106","109","112","113","116","122","132","167","173","176","191","84","85","93","95","106","109","112","113","116","122","132","167","173","176","191","110","85","93","97","106","167","173","188","93","97","106","167","173","188","88","89","101","102","115","123","134","149","216","89","101","102","115","123","134","149","216","102","115","123","134","149","216","117","150","165","171","184","203","118","136","151","162","99","100","119","168","209","211","212","213","95","106","109","112","113","116","122","132","167","173","176","191","99","158","209","211","212","213","233","109","132","173","191","106","113","122","167","188","108","231","100","119","168","209","211","212","213","119","168","209","211","212","213","110","115","124","125","115","123","134","149","216","114","129","137","140","145","148","152","179","198","199","205","122","173","188","166","182","231","132","173","191","157","167","173","176","173","188","129","137","140","145","148","123","134","149","208","216","133","139","163","222","136","151","162","168","209","211","212","213","130","135","138","143","160","178","173","188","134","149","216","125","155","185","220","155","185","220","174","147","169","172","137","140","142","145","148","135","138","143","160","224","173","191","139","163","165","171","181","184","203","222","235","149","216","138","143","160","151","162","140","145","148","238","143","160","159","163","165","171","181","184","203","222","235","142","145","148","164","160","148","185","169","172","216","227","235","179","198","199","205","209","211","212","213","163","165","171","184","203","215","222","227","235","170","207","218","165","171","181","184","203","222","235","171","179","184","203","215","222","227","235","182","173","188","209","211","212","213","172","207","218","179","184","203","215","222","227","235","176","191","177","195","195","209","211","212","213","220","184","198","199","205","227","235","222","203","215","222","227","235","220","202","201","221","199","205","205","215","222","227","235","227","218","216","211","212","213","212","213","213","227","235","237","235","235","234","236"],"overlapCoeff":[1,0.62,0.62,1,0.64,1,1,0.54,0.53,0.65,0.73,0.59,0.57,1,0.5,0.61,0.8100000000000001,0.8100000000000001,0.8,0.8,1,0.72,1,1,1,1,0.87,0.84,1,1,0.96,1,1,0.74,0.91,0.85,1,0.62,0.73,1,1,0.83,0.83,0.98,0.65,0.67,0.87,0.71,0.71,0.76,0.74,1,0.58,0.84,1,1,0.65,0.79,1,0.73,0.62,1,0.63,0.83,0.83,0.98,0.65,0.67,0.87,0.71,0.71,0.76,0.74,1,0.58,0.84,1,1,0.65,0.79,1,0.73,0.62,1,0.63,0.65,0.57,1,0.52,0.58,0.83,0.5,0.5,0.52,0.58,0.83,0.5,0.5,0.5,0.52,0.96,0.96,0.83,0.52,0.82,0.79,0.76,0.7,0.73,0.85,0.58,0.57,0.52,0.65,0.58,0.55,0.5600000000000001,0.5600000000000001,0.54,0.54,0.52,0.52,0.64,0.76,0.55,0.66,0.6,0.57,0.66,0.6,0.66,0.5600000000000001,0.5600000000000001,0.76,0.76,0.57,0.57,0.71,0.5,0.52,0.52,0.52,0.52,0.72,0.52,0.55,0.65,0.91,0.9399999999999999,0.9399999999999999,0.9399999999999999,0.7,0.9399999999999999,0.9399999999999999,0.9399999999999999,0.9399999999999999,0.9399999999999999,0.96,0.83,0.9399999999999999,0.96,0.9399999999999999,0.96,0.59,0.9399999999999999,0.78,0.9399999999999999,0.9399999999999999,0.64,0.84,0.96,0.9399999999999999,0.98,0.65,1,0.98,0.78,1,0.7,0.7,0.72,0.72,0.9399999999999999,0.9399999999999999,0.8100000000000001,0.78,0.7,0.9399999999999999,0.78,0.65,0.9399999999999999,0.9399999999999999,0.78,0.98,0.5,0.65,0.7,0.78,0.54,0.89,0.84,0.83,0.77,0.77,0.73,0.61,0.6,0.62,0.52,0.64,0.62,0.6,0.6,0.58,0.57,0.57,0.64,0.8,0.62,0.57,0.57,0.57,0.57,0.57,0.5600000000000001,0.5600000000000001,0.72,0.72,0.57,0.57,0.77,0.52,0.57,0.62,0.57,0.57,0.5,0.53,0.93,0.93,0.6899999999999999,0.93,0.98,0.93,0.93,0.93,0.93,0.82,0.93,0.93,0.93,0.93,0.6,0.93,0.76,0.93,0.93,0.64,0.84,0.93,0.93,0.96,0.62,0.93,0.96,0.76,0.93,0.6899999999999999,0.6899999999999999,0.72,0.72,0.93,0.93,0.77,0.76,0.6899999999999999,0.93,0.76,0.67,0.93,0.93,0.76,1,0.5,0.62,0.6899999999999999,0.76,0.53,1,0.88,0.66,0.88,0.88,0.88,0.88,0.88,0.88,0.79,0.88,0.88,0.88,0.88,0.55,1,0.72,0.88,0.88,0.64,0.84,0.88,0.88,0.88,0.59,0.88,0.88,0.72,0.88,0.66,0.66,0.72,0.72,0.88,0.88,0.77,0.72,0.66,0.88,0.72,0.59,0.88,0.88,0.72,0.88,0.55,0.59,0.66,0.72,0.5,0.64,0.86,0.86,0.86,0.86,0.86,0.86,0.76,0.86,0.86,0.95,0.95,0.54,0.86,0.71,0.86,0.86,0.64,0.84,0.86,0.95,0.95,0.63,0.95,1,0.71,0.95,0.64,0.64,0.72,0.72,0.86,0.86,0.87,0.71,0.64,0.86,0.71,0.63,0.86,0.86,0.71,1,0.5,0.63,0.64,0.75,1,0.61,1,0.92,1,0.8,0.79,0.59,0.59,0.96,0.66,1,0.83,0.84,0.84,1,1,0.7,1,1,1,0.66,0.58,0.66,0.96,0.51,0.54,1,0.54,0.8100000000000001,0.8100000000000001,0.8100000000000001,1,1,0.62,0.59,0.57,0.52,0.55,0.59,0.6,1,0.52,0.52,0.52,0.52,0.7,1,0.53,0.52,0.52,0.52,0.52,0.51,0.52,0.59,0.59,0.52,0.52,0.67,0.52,0.51,0.52,0.5600000000000001,0.51,0.52,0.62,0.54,1,1,1,0.68,0.8,0.8,0.8,0.8100000000000001,0.8,0.7,0.8100000000000001,0.83,0.8,0.8,0.5,0.8,0.67,0.8,0.8100000000000001,0.67,0.87,0.8,0.8,0.8,0.53,0.8,0.8,0.67,0.8,0.59,0.59,0.72,0.72,0.8,0.8,0.77,0.6899999999999999,0.61,0.8,0.6899999999999999,0.53,0.8,1,0.72,0.8,0.5,0.53,0.61,0.67,0.97,1,0.83,0.8,0.8,0.8,0.72,0.8,0.88,0.8,0.8,0.53,0.8,0.6899999999999999,0.8,0.8,0.64,0.84,0.88,0.83,0.83,0.53,0.83,0.83,0.6899999999999999,0.83,0.62,0.62,0.76,0.76,0.88,0.88,0.77,0.6899999999999999,0.59,0.8,0.6899999999999999,0.73,0.8,0.8,0.6899999999999999,1,0.5,0.53,0.59,0.6899999999999999,0.97,0.66,1,1,0.75,0.96,1,1,0.9,0.88,0.68,0.89,0.93,1,1,0.75,0.75,0.75,0.75,0.78,0.78,0.78,0.6899999999999999,0.76,0.8100000000000001,0.78,0.78,0.76,0.66,0.76,0.76,0.64,0.84,0.79,1,0.8100000000000001,0.55,0.99,0.8100000000000001,0.66,1,0.62,0.62,0.76,0.76,0.79,0.79,0.77,0.66,0.57,0.76,0.67,0.58,0.76,0.76,0.66,1,0.5,0.55,0.57,0.66,0.6899999999999999,0.64,0.6,0.66,0.6899999999999999,0.66,0.64,0.5,0.65,0.52,0.65,0.64,0.64,0.84,0.6899999999999999,0.68,0.68,0.51,0.66,0.68,0.52,0.66,0.59,0.59,0.72,0.72,0.6899999999999999,0.6899999999999999,0.77,0.54,0.55,0.68,0.54,0.68,0.64,0.52,0.7,0.5,0.5,0.51,0.52,0.63,0.62,0.68,0.65,0.64,0.63,0.63,0.54,0.63,0.64,0.67,0.87,0.63,0.65,0.64,0.5,0.64,0.64,0.54,0.64,0.61,0.61,0.72,0.72,0.68,0.68,0.77,0.5600000000000001,0.57,0.63,0.5600000000000001,0.63,0.64,0.54,0.65,0.5,0.55,0.54,1,1,0.72,0.72,0.68,0.54,0.6899999999999999,1,0.63,0.63,0.6,0.57,0.54,0.64,0.84,0.66,0.54,0.53,0.54,0.52,0.54,0.59,0.59,0.72,0.72,0.53,0.53,0.77,0.55,0.52,0.5600000000000001,0.53,0.53,0.5,0.51,0.58,0.6,0.5600000000000001,0.57,0.54,0.5,0.64,0.76,0.52,0.57,0.5600000000000001,0.5600000000000001,0.76,0.76,0.71,0.5,0.75,0.66,0.66,0.66,0.5600000000000001,0.66,1,0.67,0.87,0.68,0.66,0.66,0.66,0.66,0.5600000000000001,0.66,0.59,0.59,0.72,0.72,0.66,0.66,0.77,0.5600000000000001,0.58,0.66,0.5600000000000001,0.66,0.68,0.5600000000000001,0.66,0.5,0.5600000000000001,0.64,0.54,0.63,0.63,0.6,0.58,0.61,0.67,1,1,0.5600000000000001,0.53,0.55,0.51,0.54,0.62,0.62,0.79,0.79,0.52,0.52,0.77,0.67,0.52,0.6,0.54,0.85,0.63,0.53,0.6899999999999999,0.63,0.64,0.84,0.63,1,0.86,0.89,0.86,0.53,0.89,0.61,0.61,0.72,0.72,0.7,0.7,0.77,0.53,0.55,0.6899999999999999,0.53,0.63,0.63,0.53,1,0.5,0.51,0.5600000000000001,0.63,0.52,0.6899999999999999,0.63,0.64,0.84,0.63,0.86,0.99,0.99,0.99,0.52,1,0.59,0.59,0.72,0.72,0.6899999999999999,0.6899999999999999,0.77,0.52,0.55,0.6899999999999999,0.52,0.67,0.63,0.52,1,0.5,0.51,0.67,0.64,0.59,0.72,0.72,0.71,0.5,0.5,0.61,0.5,0.59,0.79,0.6,0.6,0.64,0.84,0.6,0.6,0.6,0.6,0.6,0.6,0.59,0.59,0.72,0.72,0.6,0.6,0.77,0.55,0.6,0.6,0.6,0.6,0.55,0.51,1,0.72,0.53,0.96,1,0.88,0.87,0.53,0.88,1,0.54,1,1,1,1,0.67,0.67,1,0.86,0.86,1,1,0.77,1,1,1,0.5,1,1,0.72,0.72,0.68,1,0.72,0.58,0.96,1,0.88,0.87,0.53,0.88,1,0.52,1,0.57,0.64,0.84,0.58,0.62,0.62,0.62,0.62,0.62,0.59,0.59,0.72,0.72,0.7,0.7,0.77,0.55,1,0.57,0.83,0.62,0.5,0.51,1,1,0.51,0.57,0.58,0.59,0.53,1,0.67,1,0.53,0.53,0.51,0.51,0.5,0.59,0.59,0.72,0.72,0.77,1,0.5,0.5,0.64,0.54,0.64,0.64,0.64,0.64,0.64,0.64,0.67,0.64,0.72,0.72,0.64,0.64,0.68,0.67,0.64,0.67,0.64,0.64,1,0.67,0.67,0.64,0.64,0.64,0.67,0.73,0.73,1,1,0.84,0.84,0.84,0.84,0.84,0.67,0.84,0.84,0.84,0.84,0.84,0.67,1,0.84,0.67,0.84,0.87,0.67,0.84,1,0.67,0.67,0.55,0.53,0.54,0.51,0.53,0.62,0.62,0.76,0.76,0.52,0.52,0.77,0.55,0.5,0.51,0.52,0.51,0.76,0.53,0.9,0.76,0.91,0.64,0.64,0.76,0.76,0.61,0.61,0.77,0.55,0.58,0.53,0.53,1,0.5,0.51,1,0.89,1,0.54,0.87,1,0.87,0.61,0.61,0.72,0.72,0.58,0.58,0.84,0.55,0.58,0.5600000000000001,0.53,1,0.5,0.51,0.57,0.54,0.59,0.54,0.72,0.72,1,1,0.6,0.5,1,1,1,0.84,1,0.62,0.62,0.76,0.76,0.58,0.58,0.8100000000000001,0.55,0.5600000000000001,0.54,0.51,0.98,0.5,0.51,0.5600000000000001,0.84,0.61,0.61,0.72,0.72,0.5600000000000001,0.5600000000000001,0.9399999999999999,0.55,0.5600000000000001,0.54,0.51,1,0.5,0.51,0.55,1,1,1,1,0.9,1,0.68,0.93,1,1,1,1,1,1,1,1,0.9,1,0.68,0.93,1,1,1,1,1,0.86,0.86,1,1,0.77,0.91,1,1,0.5,0.9,0.62,0.62,0.76,0.76,0.57,0.57,0.8100000000000001,0.55,0.55,0.53,0.5,0.97,0.5,0.51,0.55,1,1,0.95,0.96,0.79,0.93,1,0.76,0.76,0.76,0.76,1,1,1,1,1,1,1,0.53,1,1,1,1,0.9,1,0.68,0.93,1,1,1,1,1,0.68,0.68,0.68,1,0.62,0.62,0.86,0.59,0.59,0.86,0.59,0.59,0.86,0.64,0.61,0.86,0.62,0.62,0.86,0.59,0.59,0.86,0.59,0.59,0.86,0.64,0.61,0.86,0.71,1,1,0.59,0.98,0.74,0.77,1,0.78,0.65,1,0.71,1,0.76,0.76,0.72,1,0.72,1,0.76,0.72,0.72,0.72,1,0.76,0.72,0.72,1,0.76,0.76,0.72,1,0.72,1,0.76,0.72,0.72,0.72,1,0.76,0.72,0.72,1,0.52,1,0.77,0.55,0.57,0.5600000000000001,0.52,0.51,0.77,0.55,0.57,0.5600000000000001,0.52,0.51,1,0.6899999999999999,0.74,0.6899999999999999,1,0.6899999999999999,0.6899999999999999,0.6899999999999999,1,0.6899999999999999,0.74,0.6899999999999999,1,0.6899999999999999,0.6899999999999999,0.6899999999999999,1,0.96,0.5600000000000001,1,1,0.92,0.67,1,0.51,0.6,0.58,0.59,0.53,1,1,0.87,1,1,1,1,1,1,1,1,1,0.77,0.77,0.77,1,0.77,0.68,0.77,0.77,0.9399999999999999,0.71,1,0.77,0.9,0.62,0.95,0.95,0.95,0.95,0.52,0.93,0.84,0.5,0.79,0.57,0.55,0.58,0.55,0.65,1,1,0.68,0.93,1,0.93,0.93,0.93,0.93,0.66,0.66,0.73,0.73,0.73,0.73,0.78,0.52,0.96,0.96,0.53,1,1,1,0.63,1,0.62,0.61,0.79,1,0.66,0.82,1,1,1,0.76,0.65,0.5,0.51,0.93,1,1,0.86,0.5,0.77,0.74,0.62,0.5,0.8,0.5,0.51,0.8,0.5600000000000001,1,1,0.78,0.61,0.5,0.57,0.55,1,0.71,0.71,0.71,0.5,1,0.77,1,1,0.93,0.93,0.93,0.93,1,0.99,1,1,0.91,0.6,0.5,0.52,1,1,0.73,1,0.65,0.5,0.5,0.65,0.5,0.5,1,1,1,1,0.55,1,0.58,0.68,0.66,1,1,0.78,0.9399999999999999,1,0.5,0.72,1,1,0.62,0.65,1,0.66,0.5600000000000001,0.85,0.58,1,0.55,1,0.78,0.93,1,1,0.58,0.58,0.55,0.61,1,0.91,0.54,1,0.6899999999999999,0.71,1,0.72,0.62,0.78,0.67,0.78,0.86,0.66,1,0.9,0.66,1,1,1,0.68,0.8100000000000001,0.54,0.82,0.55,0.55,1,0.51,0.51,0.51,0.51,0.54,0.63,0.65,0.62,0.65,0.61,0.5600000000000001,0.63,1,1,1,1,0.6899999999999999,0.71,1,0.72,0.62,0.76,0.67,1,0.64,0.97,0.71,0.88,0.6899999999999999,1,0.92,1,0.5,0.51,1,1,1,1,1,1,1,0.58,1,0.65,1,0.74,1,0.79,0.5,0.5,1,1,1,0.6,0.6,0.6,0.6,1,0.5600000000000001,1,1,0.76,0.52,0.5,1,0.66,1,0.75,0.97,0.79,0.92,1,1,1,1,0.64,0.64,0.65,0.62,0.71,0.71,0.71,1,0.59,1,1,1,1,1,1,0.53,0.88,1,0.67,0.92,1,0.5],"width":[5,1.92,1.92,5,2.13,5,5,1.3,1.21,2.22,2.87,1.73,1.53,5,1,1.86,3.5,3.47,3.4,3.4,5,2.8,5,5,5,5,3.96,3.74,5,5,4.67,5,5,2.91,4.25,3.82,5,2,2.82,5,5,3.65,3.65,4.88,2.2,2.36,3.94,2.71,2.71,3.07,2.89,5,1.67,3.69,5,5,2.17,3.34,5,2.85,2,5,2.06,3.65,3.65,4.88,2.2,2.36,3.94,2.71,2.71,3.07,2.89,5,1.67,3.69,5,5,2.17,3.34,5,2.85,2,5,2.06,2.2,1.52,5,1.2,1.68,3.61,1,1,1.2,1.68,3.61,1,1,1,1.18,4.69,4.69,3.67,1.19,3.55,3.34,3.1,2.62,2.88,3.81,1.6,1.54,1.19,2.21,1.68,1.37,1.44,1.44,1.34,1.33,1.19,1.19,2.09,3.04,1.37,2.3,1.84,1.57,2.3,1.84,2.3,1.5,1.5,3.07,3.07,1.56,1.56,2.68,1,1.19,1.19,1.19,1.19,2.77,1.18,1.4,2.22,4.26,4.56,4.56,4.56,2.63,4.56,4.56,4.56,4.56,4.56,4.7,3.67,4.56,4.7,4.56,4.7,1.74,4.56,3.22,4.56,4.56,2.09,3.76,4.7,4.56,4.85,2.19,5,4.85,3.22,5,2.63,2.63,2.79,2.79,4.56,4.56,3.45,3.22,2.63,4.56,3.22,2.19,4.56,4.56,3.22,4.85,1,2.19,2.63,3.22,1.3,4.13,3.76,3.64,3.12,3.12,2.85,1.9,1.84,1.93,1.19,2.09,1.93,1.84,1.84,1.61,1.56,1.56,2.09,3.4,1.93,1.56,1.56,1.56,1.56,1.56,1.5,1.5,2.79,2.79,1.56,1.56,3.19,1.17,1.56,1.93,1.56,1.56,1,1.28,4.42,4.42,2.53,4.42,4.85,4.42,4.42,4.42,4.42,3.55,4.42,4.42,4.42,4.42,1.8,4.42,3.11,4.42,4.42,2.09,3.76,4.42,4.42,4.71,1.95,4.42,4.71,3.11,4.42,2.53,2.53,2.79,2.79,4.42,4.42,3.19,3.11,2.53,4.42,3.11,2.38,4.42,4.42,3.11,5,1,1.95,2.53,3.11,1.22,5,4.03,2.24,4.03,4.03,4.03,4.03,4.03,4.03,3.34,4.03,4.03,4.03,4.03,1.41,5,2.79,4.03,4.03,2.09,3.76,4.03,4.03,4.03,1.69,4.03,4.03,2.79,4.03,2.24,2.24,2.79,2.79,4.03,4.03,3.19,2.79,2.24,4.03,2.79,1.69,4.03,4.03,2.79,4.03,1.36,1.69,2.24,2.79,1,2.15,3.92,3.92,3.92,3.92,3.92,3.92,3.1,3.92,3.92,4.59,4.59,1.34,3.92,2.69,3.92,3.92,2.09,3.76,3.92,4.59,4.59,2.02,4.59,5,2.69,4.59,2.15,2.15,2.79,2.79,3.92,3.92,3.97,2.69,2.15,3.92,2.69,2.02,3.92,3.92,2.69,5,1,2.02,2.15,2.97,5,1.91,5,4.36,5,3.43,3.28,1.71,1.71,4.67,2.28,5,3.67,3.71,3.75,5,5,2.63,5,5,5,2.3,1.63,2.3,4.67,1.05,1.34,5,1.34,3.48,3.45,3.5,5,5,2,1.75,1.54,1.16,1.38,1.71,1.82,5,1.16,1.16,1.16,1.16,2.59,5,1.27,1.16,1.16,1.16,1.16,1.05,1.16,1.75,1.75,1.16,1.16,2.33,1.16,1.05,1.16,1.49,1.05,1.16,1.93,1.31,5,5,5,2.41,3.38,3.38,3.38,3.5,3.38,2.62,3.5,3.62,3.38,3.38,1,3.38,2.38,3.38,3.5,2.33,3.93,3.38,3.38,3.38,1.25,3.38,3.38,2.38,3.38,1.75,1.75,2.79,2.79,3.38,3.38,3.19,2.5,1.88,3.38,2.5,1.25,3.38,5,2.75,3.38,1,1.25,1.88,2.38,4.77,5,3.62,3.38,3.38,3.38,2.75,3.38,4,3.38,3.38,1.25,3.38,2.5,3.38,3.38,2.09,3.76,4,3.62,3.62,1.25,3.62,3.62,2.5,3.62,2,2,3.07,3.07,4,4,3.19,2.5,1.75,3.38,2.5,2.88,3.38,3.38,2.5,5,1,1.25,1.75,2.5,4.77,2.26,5,5,2.98,4.64,5,5,4.24,4.02,2.43,4.09,4.48,5,5,2.98,2.98,2.98,2.98,3.21,3.21,3.21,2.49,3.09,3.45,3.21,3.21,3.09,2.25,3.09,3.09,2.09,3.76,3.33,5,3.45,1.42,4.88,3.45,2.25,5,2,2,3.07,3.07,3.33,3.33,3.19,2.25,1.54,3.09,2.37,1.66,3.09,3.09,2.25,5,1,1.42,1.54,2.25,2.5,2.1,1.8,2.3,2.5,2.3,2.1,1,2.2,1.2,2.2,2.1,2.09,3.76,2.5,2.4,2.4,1.11,2.3,2.4,1.2,2.3,1.75,1.75,2.79,2.79,2.5,2.5,3.19,1.3,1.41,2.4,1.3,2.4,2.1,1.2,2.6,1,1,1.05,1.2,2.04,1.94,2.4,2.23,2.14,2.04,2.04,1.35,2.04,2.14,2.33,3.93,2.04,2.23,2.14,1,2.14,2.14,1.35,2.14,1.88,1.88,2.79,2.79,2.43,2.43,3.19,1.44,1.52,2.04,1.44,2.04,2.14,1.35,2.23,1,1.37,1.35,5,5,2.79,2.79,2.42,1.33,2.51,5,2.04,2.04,1.8,1.53,1.28,2.09,3.76,2.25,1.33,1.21,1.28,1.12,1.28,1.75,1.75,2.79,2.79,1.2,1.2,3.19,1.41,1.12,1.44,1.2,1.2,1,1.05,1.68,1.84,1.44,1.57,1.33,1,2.09,3.04,1.17,1.57,1.5,1.5,3.07,3.07,2.68,1,3.03,2.3,2.3,2.3,1.47,2.3,5,2.33,3.93,2.4,2.3,2.3,2.3,2.3,1.47,2.3,1.75,1.75,2.79,2.79,2.3,2.3,3.19,1.47,1.64,2.3,1.47,2.3,2.4,1.47,2.3,1,1.48,2.09,1.31,2.04,2.04,1.8,1.62,1.91,2.33,5,5,1.5,1.21,1.4,1.08,1.31,2,2,3.34,3.34,1.15,1.15,3.19,2.33,1.18,1.8,1.31,3.81,2.04,1.25,2.53,2.04,2.09,3.76,2.04,5,3.91,4.11,3.91,1.25,4.11,1.88,1.88,2.79,2.79,2.63,2.63,3.19,1.25,1.41,2.53,1.25,2.04,2.04,1.25,5,1,1.05,1.44,2.04,1.15,2.53,2.04,2.09,3.76,2.04,3.91,4.9,4.9,4.9,1.15,5,1.75,1.75,2.79,2.79,2.53,2.53,3.19,1.15,1.41,2.53,1.15,2.33,2.04,1.15,5,1,1.05,2.33,2.09,1.69,2.79,2.79,2.68,1,1,1.91,1,1.69,3.35,1.8,1.8,2.09,3.76,1.8,1.8,1.8,1.8,1.8,1.8,1.75,1.75,2.79,2.79,1.8,1.8,3.19,1.41,1.8,1.8,1.8,1.8,1.36,1.05,5,2.77,1.23,4.67,5,4.05,3.97,1.27,4,5,1.35,5,5,5,5,2.33,2.33,5,3.88,3.88,5,5,3.19,5,5,5,1,5,5,2.79,2.79,2.42,5,2.77,1.61,4.67,5,4.05,3.97,1.27,4,5,1.15,5,1.53,2.09,3.76,1.62,1.98,1.98,1.98,1.98,1.98,1.75,1.75,2.79,2.79,2.6,2.6,3.19,1.41,5,1.53,3.67,1.98,1,1.05,5,5,1.08,1.57,1.65,1.75,1.24,5,2.33,5,1.25,1.21,1.08,1.08,1,1.75,1.75,2.79,2.79,3.19,5,1,1,2.12,1.31,2.09,2.09,2.09,2.09,2.09,2.09,2.33,2.09,2.79,2.79,2.09,2.09,2.42,2.33,2.09,2.33,2.09,2.09,5,2.33,2.33,2.09,2.09,2.09,2.33,2.85,2.85,5,5,3.76,3.76,3.76,3.76,3.76,2.33,3.76,3.76,3.76,3.76,3.76,2.33,5,3.76,2.33,3.76,3.93,2.33,3.76,5,2.33,2.33,1.42,1.21,1.32,1.08,1.24,2,2,3.07,3.07,1.15,1.15,3.19,1.41,1,1.07,1.18,1.05,3.08,1.23,4.17,3.08,4.25,2.12,2.12,3.07,3.07,1.92,1.92,3.19,1.41,1.67,1.25,1.25,5,1,1.05,5,4.08,5,1.34,3.93,5,3.93,1.88,1.88,2.79,2.79,1.62,1.62,3.71,1.41,1.62,1.45,1.21,5,1,1.05,1.54,1.34,1.69,1.34,2.79,2.79,5,5,1.8,1,5,5,5,3.72,5,2,2,3.07,3.07,1.64,1.64,3.45,1.41,1.48,1.32,1.08,4.84,1,1.05,1.48,3.72,1.88,1.88,2.79,2.79,1.48,1.48,4.48,1.41,1.48,1.32,1.08,5,1,1.05,1.4,5,5,5,5,4.24,5,2.43,4.48,5,5,5,5,5,5,5,5,4.24,5,2.43,4.48,5,5,5,5,5,3.88,3.88,5,5,3.19,4.3,5,5,1,4.17,2,2,3.07,3.07,1.55,1.55,3.45,1.41,1.39,1.24,1,4.76,1,1.05,1.39,5,5,4.62,4.67,3.29,4.48,5,3.09,3.09,3.09,3.09,5,5,5,5,5,5,5,1.24,5,5,5,5,4.24,5,2.43,4.48,5,5,5,5,5,2.43,2.43,2.43,5,2,2,3.88,1.75,1.75,3.88,1.75,1.75,3.88,2.12,1.88,3.88,2,2,3.88,1.75,1.75,3.88,1.75,1.75,3.88,2.12,1.88,3.88,2.67,5,5,1.68,4.8,2.9,3.19,5,3.25,2.18,5,2.67,5,3.07,3.07,2.79,5,2.79,5,3.07,2.79,2.79,2.79,5,3.07,2.79,2.79,5,3.07,3.07,2.79,5,2.79,5,3.07,2.79,2.79,2.79,5,3.07,2.79,2.79,5,1.17,5,3.19,1.41,1.58,1.51,1.18,1.05,3.19,1.41,1.58,1.51,1.18,1.05,5,2.52,2.9,2.52,5,2.52,2.52,2.52,5,2.52,2.9,2.52,5,2.52,2.52,2.52,5,4.7,1.44,5,5,4.4,2.33,5,1.08,1.76,1.65,1.75,1.24,5,5,3.97,5,5,5,5,5,5,5,5,5,3.19,3.19,3.19,5,3.19,2.42,3.19,3.19,4.48,2.68,5,3.19,4.24,1.95,4.62,4.62,4.62,4.62,1.19,4.44,3.75,1,3.31,1.52,1.41,1.64,1.41,2.22,5,5,2.43,4.48,5,4.43,4.43,4.43,4.43,2.29,2.29,2.86,2.86,2.86,2.86,3.26,1.13,4.69,4.69,1.21,5,5,5,2.05,5,1.97,1.85,3.3,5,2.27,3.6,5,5,5,3.04,2.17,1,1.05,4.47,5,5,3.92,1,3.2,2.91,2,1,3.36,1,1.05,3.37,1.48,5,5,3.26,1.9,1,1.53,1.36,5,2.67,2.67,2.67,1,5,3.17,5,5,4.48,4.48,4.48,4.48,5,4.92,5,5,4.24,1.8,1,1.16,5,5,2.88,5,2.23,1,1,2.23,1,1,5,5,5,5,1.36,5,1.61,2.42,2.27,5,5,3.2,4.49,5,1,2.76,5,5,1.95,2.16,5,2.25,1.47,3.79,1.67,5,1.43,5,3.2,4.41,5,5,1.61,1.61,1.36,1.85,5,4.24,1.29,5,2.52,2.68,5,2.75,1.94,3.26,2.33,3.27,3.89,2.27,5,4.2,2.29,5,5,5,2.43,3.49,1.33,3.6,1.4,1.4,5,1.1,1.1,1.1,1.1,1.29,2.07,2.16,2,2.18,1.88,1.49,2.07,5,5,5,5,2.52,2.68,5,2.75,1.94,3.09,2.33,5,2.14,4.75,2.65,4.05,2.52,5,4.33,5,1,1.05,5,5,5,5,5,5,5,1.65,5,2.16,5,2.94,5,3.33,1,1,5,5,5,1.8,1.8,1.8,1.8,5,1.5,5,5,3.04,1.15,1,5,2.25,5,3,4.75,3.33,4.32,5,5,5,5,2.16,2.16,2.18,1.94,2.65,2.67,2.65,5,1.73,5,5,5,5,5,5,1.25,4,5,2.33,4.33,5,1],"color":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","shadow":{"enabled":true,"size":25},"physics":false},"manipulation":{"enabled":false},"interaction":{"hover":true,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":null,"width":"100%","height":1300,"idselection":{"enabled":true,"style":"width: 500px; height = 31px; font-size: 18px; color: #000066;border: 3px solid #4d88ff;","useLabels":true,"main":"Select by id","values":["158","227","17","84","85","31","197","135","138","14","74","153","209","202","40","7","147","127","119","92","178","21","52","171","165","184","172","159","220","235","155","56","199","154","230","39","213","109","70","93","64","115","123","88","208","216","149","25","24","174","44","78","53","34","26","22","15","132","82","81","94","83","166","107","182","231","175","195","167","77","156","3","2","4","33","106","176","218","130","120","160","57","8","192","217","96","62","189","214","87","72","12","37","10","173","91","118","206","9","125","101","232","47","186","170","110","54","38","97","150","122","179","143","134","89","102","20","215","90","11","86","1","5","43","180","111","29","67","112","142","161","68","99","157","225","63","210","48","79","95","28","19","229","59","58","136","151","117","193","6","124","104","144","60","65","200","148","137","234","224","131","103","145","114","75","219","236","61","23","205","198","105","27","76","35","55","49","116","191","233","190","46","100","133","196","238","223","183","228","50","203","162","16","187","139","80","226","222","140","129","73","237","181","163","201","207","221","194","108","98","126","45","42","41","66","13","71","32","113","36","164","141","69","30","128","185","18","146","121","212","168","188","211","51","204","169","152","177"]},"byselection":{"enabled":true,"style":"width: 200px; height = 31px; font-size: 18px; color: #000066;border: 3px solid #4d88ff;","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false,"variable":"FDR","main":"Select by FDR","values":[3.02e-10,3.40513e-08,3.48708e-08,3.50738e-08,7.29524e-07,7.43915e-07,7.46461e-07,2.96e-05,3.70995e-05,3.79679e-05,5.45e-05,0.000116,0.000159,0.000162,0.000171,0.000197,0.000277,0.000317,0.000388,0.000396749,0.000399609,0.000462,0.000482,0.000497,0.000515279,0.0005247,0.000531177,0.000555,0.000691,0.000715,0.000753,0.000916,0.00102,0.00151,0.00153,0.00191,0.00232,0.00269,0.00369,0.00410838,0.00411944,0.00489,0.00499,0.00555,0.00621,0.00661,0.00678,0.00763174,0.0076832,0.00865,0.009639999999999999,0.009820000000000001,0.0115469,0.0116113,0.0131,0.0133,0.0148718,0.0149936,0.0150544,0.0151244,0.0153546,0.0156211,0.0156688,0.0160572,0.017,0.0194107,0.0194462,0.0199,0.0199006,0.0200272,0.0229,0.0243,0.0257959,0.026,0.0260098,0.0264263,0.0267205,0.0290894,0.0293792,0.0295181,0.0297,0.0301039,0.0351,0.0375123,0.0379152,0.0381888,0.0382094,0.0387,0.0387023,0.038766,0.038804,0.0388249,0.0391748,0.0391783,0.0412,0.0419,0.0419029,0.0421,0.0425,0.0427035,0.0431333,0.0459548,0.0460257,0.0467,0.0472,0.0490228,0.0497792,0.0502,0.0512,0.0514,0.054,0.0571637,0.0576,0.0577409,0.0581793,0.0584,0.0589,0.0598207,0.0613537,0.0627485,0.0683,0.0704,0.0709,0.0768151,0.0768773,0.076906,0.0773297,0.0789641,0.0789728,0.08078249999999999,0.081,0.0822,0.08278820000000001,0.0831522,0.0832239,0.0832855,0.08409999999999999,0.0842706,0.0844,0.08449810000000001,0.0847823,0.0852805,0.08535719999999999,0.0856441,0.0857464,0.0857777,0.08605980000000001,0.0910169,0.0910957,0.09135989999999999,0.0914,0.0916331,0.091701,0.093112,0.0975719,0.09774819999999999,0.09935919999999999,0.101066,0.101862,0.102,0.102709,0.103,0.105,0.108583,0.110088,0.115,0.120888,0.123066,0.123761,0.124,0.130577,0.131,0.132472,0.133207,0.134,0.136269,0.136592,0.136754,0.137,0.138292,0.138741,0.139175,0.141134,0.142,0.142125,0.142462,0.142591,0.143189,0.143638,0.145302,0.146208,0.146577,0.146766,0.14773,0.148301,0.148302,0.148415,0.148596,0.150792,0.150901,0.151,0.159,0.162399,0.162955,0.163093,0.163097,0.164469,0.16553,0.166727,0.167627,0.167908,0.168104,0.168105,0.1682,0.168407,0.172,0.172414,0.173201,0.173469,0.176,0.179661,0.179762,0.18012,0.181647,0.187052,0.187659,0.189506,0.18997,0.191,0.191341,0.191923,0.195924,0.196959,0.197007,0.197898,0.198043,0.198614,0.201845]},"main":{"text":"foldchanges","style":"font-size:40px;text-align:center;"},"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":300,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","highlight":{"enabled":true,"hoverNearest":true,"degree":5,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"},"igraphlayout":{"type":"full"},"events":{"select":"function(nodes) {\nShiny.onInputChange('current_node_id', nodes.nodes);\n;}"}},"evals":["events.select"],"jsHooks":[]}</script>
<p>This is one of the best visualizations available out of all the GO packages.</p>
<p>There are other gene sets available for GSEA analysis in clusterProfiler (Disease Ontology, Reactome pathways, etc.). In addition, it is possible to supply your own gene set GMT file, such as a GMT for MSigDB called c2.</p>
<p>** The C2 subcollection CGP: Chemical and genetic perturbations. Gene sets that represent expression signatures of genetic and chemical perturbations.**</p>
</div>
</div>
<div id="other-tools-and-resources-1" class="section level3 hasAnchor" number="9.8.4">
<h3><span class="header-section-number">9.8.4</span> Other tools and resources<a href="codebook.html#other-tools-and-resources-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p><a href="http://genemania.org/">GeneMANIA</a>. GeneMANIA finds other genes that are related to a set of input genes, using a very large set of functional association data curated from the literature. Association data include protein and genetic interactions, pathways, co-expression, co-localization and protein domain similarity.</p></li>
<li><p><a href="http://revigo.irb.hr/">ReviGO</a>. Revigo is an online GO enrichment tool that allows you to copy-paste your significant gene list and your background gene list. The output is a visualization of enriched GO terms in a hierarchical tree.</p></li>
<li><p><a href="http://amigo.geneontology.org/amigo">AmiGO</a>. AmiGO is the current official web-based set of tools for searching and browsing the Gene Ontology database.</p></li>
<li><p><a href="http://david.abcc.ncifcrf.gov/tools.jsp">DAVID</a>. The fold enrichment is defined as the ratio of the two proportions; one is the proportion of genes in your list belong to certain pathway, and the other is the proportion of genes in the background information (i.e., universe genes) that belong to that pathway.</p></li>
<li><p>etc.</p></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="functional-analysis-of-rnaseq-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RNA-seq-analysis.pdf", "RNA-seq-analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
