<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Codebook | 2024_RNA-seq-analysis</title>
  <meta name="description" content="Chapter 9 Codebook | 2024_RNA-seq-analysis" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Codebook | 2024_RNA-seq-analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bookdown.org/ggiaever/2024_RNA-seq-analysis/img/football.png" />
  
  <meta name="github-repo" content="gurinina/2024_RNA-seq-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Codebook | 2024_RNA-seq-analysis" />
  
  
  <meta name="twitter:image" content="https://bookdown.org/ggiaever/2024_RNA-seq-analysis/img/football.png" />




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="functional-analysis-of-rnaseq-data.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet" />
<script src="libs/vis-9.1.0/vis-network.min.js"></script>
<script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="css/git.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>RNA-seq analysis workshop</a></li>
<li class="chapter" data-level="1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html"><i class="fa fa-check"></i><b>1</b> Differential gene expression (DGE) analysis overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#review-of-the-dataset"><i class="fa fa-check"></i><b>1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="1.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#setting-up"><i class="fa fa-check"></i><b>1.2</b> Setting up</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-libraries"><i class="fa fa-check"></i><b>1.2.1</b> Loading libraries</a></li>
<li class="chapter" data-level="1.2.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-data"><i class="fa fa-check"></i><b>1.2.2</b> Loading data</a></li>
<li class="chapter" data-level="1.2.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#viewing-data"><i class="fa fa-check"></i><b>1.2.3</b> Viewing data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-analysis-overview"><i class="fa fa-check"></i><b>1.3</b> Differential gene expression analysis overview</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#rna-seq-count-distribution"><i class="fa fa-check"></i><b>1.3.1</b> RNA-seq count distribution</a></li>
<li class="chapter" data-level="1.3.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#modeling-count-data"><i class="fa fa-check"></i><b>1.3.2</b> Modeling count data</a></li>
<li class="chapter" data-level="1.3.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates"><i class="fa fa-check"></i><b>1.3.3</b> Improving mean estimates (i.e. reducing variance) with biological replicates</a></li>
<li class="chapter" data-level="1.3.4" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>1.3.4</b> Differential expression analysis workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="count-normalization.html"><a href="count-normalization.html"><i class="fa fa-check"></i><b>2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="2.1" data-path="count-normalization.html"><a href="count-normalization.html#normalization"><i class="fa fa-check"></i><b>2.1</b> Normalization</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="count-normalization.html"><a href="count-normalization.html#common-normalization-methods"><i class="fa fa-check"></i><b>2.1.1</b> Common normalization methods</a></li>
<li class="chapter" data-level="2.1.2" data-path="count-normalization.html"><a href="count-normalization.html#tpm-not-recommended"><i class="fa fa-check"></i><b>2.1.2</b> TPM (Not Recommended)</a></li>
<li class="chapter" data-level="2.1.3" data-path="count-normalization.html"><a href="count-normalization.html#rpkmfpkm-not-recommended"><i class="fa fa-check"></i><b>2.1.3</b> RPKM/FPKM (Not Recommended)</a></li>
<li class="chapter" data-level="2.1.4" data-path="count-normalization.html"><a href="count-normalization.html#deseq2-normalization-median-of-ratios-method"><i class="fa fa-check"></i><b>2.1.4</b> DESeq2 Normalization: Median of Ratios Method</a></li>
<li class="chapter" data-level="2.1.5" data-path="count-normalization.html"><a href="count-normalization.html#summary-of-key-reasons-for-using-deseq2"><i class="fa fa-check"></i><b>2.1.5</b> Summary of Key Reasons for Using DESeq2</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="count-normalization.html"><a href="count-normalization.html#count-normalization-of-mov10"><i class="fa fa-check"></i><b>2.2</b> Count normalization of Mov10</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="count-normalization.html"><a href="count-normalization.html#match-the-metadata-and-counts-data"><i class="fa fa-check"></i><b>2.2.1</b> 1. Match the metadata and counts data</a></li>
<li class="chapter" data-level="2.2.2" data-path="count-normalization.html"><a href="count-normalization.html#create-deseq2-object"><i class="fa fa-check"></i><b>2.2.2</b> 2. Create DESEq2 object</a></li>
<li class="chapter" data-level="2.2.3" data-path="count-normalization.html"><a href="count-normalization.html#generate-the-mov10-normalized-counts"><i class="fa fa-check"></i><b>2.2.3</b> 3. Generate the Mov10 normalized counts</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#sample-level-qc"><i class="fa fa-check"></i><b>3.1</b> Sample-level QC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quality-control.html"><a href="quality-control.html#recommended-resources"><i class="fa fa-check"></i><b>3.1.1</b> Recommended Resources</a></li>
<li class="chapter" data-level="3.1.2" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>3.1.2</b> Hierarchical Clustering Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#gene-level-qc"><i class="fa fa-check"></i><b>3.2</b> Gene-level QC</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#mov10-quality-assessment-and-exploratory-analysis-using-deseq2"><i class="fa fa-check"></i><b>3.3</b> Mov10 quality assessment and exploratory analysis using DESeq2</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="quality-control.html"><a href="quality-control.html#transform-normalized-counts-using-the-rlog-transformation"><i class="fa fa-check"></i><b>3.3.1</b> Transform normalized counts using the rlog transformation</a></li>
<li class="chapter" data-level="3.3.2" data-path="quality-control.html"><a href="quality-control.html#principal-components-analysis-pca"><i class="fa fa-check"></i><b>3.3.2</b> Principal components analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.3" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.3</b> Hierarchical Clustering</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html"><i class="fa fa-check"></i><b>4</b> DGE analysis workflow</a>
<ul>
<li class="chapter" data-level="4.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#running-deseq2"><i class="fa fa-check"></i><b>4.1</b> Running DESeq2</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#set-the-design-formula"><i class="fa fa-check"></i><b>4.1.1</b> Set the Design Formula</a></li>
<li class="chapter" data-level="4.1.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-differential-expression-analysis"><i class="fa fa-check"></i><b>4.1.2</b> MOV10 Differential Expression Analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#deseq2-differential-gene-expression-analysis-workflow"><i class="fa fa-check"></i><b>4.2</b> DESeq2 differential gene expression analysis workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-1-estimate-size-factors"><i class="fa fa-check"></i><b>4.2.1</b> Step 1: Estimate size factors</a></li>
<li class="chapter" data-level="4.2.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-2-estimate-gene-wise-dispersion"><i class="fa fa-check"></i><b>4.2.2</b> Step 2: Estimate Gene-wise Dispersion</a></li>
<li class="chapter" data-level="4.2.3" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-3-fit-curve-to-gene-wise-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.3</b> Step 3: Fit Curve to Gene-wise Dispersion Estimates</a></li>
<li class="chapter" data-level="4.2.4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-4-assessing-model-fit-with-dispersion-plots"><i class="fa fa-check"></i><b>4.2.4</b> Step 4: Assessing Model Fit with Dispersion Plots</a></li>
<li class="chapter" data-level="4.2.5" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#assessing-and-addressing-issues"><i class="fa fa-check"></i><b>4.2.5</b> Assessing and Addressing Issues</a></li>
<li class="chapter" data-level="4.2.6" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-differential-expression-analysis-exploring-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.6</b> MOV10 Differential Expression Analysis: Exploring Dispersion Estimates</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html"><i class="fa fa-check"></i><b>5</b> Negative Binomial model fitting</a>
<ul>
<li class="chapter" data-level="5.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#generalized-linear-model-fit-for-each-gene"><i class="fa fa-check"></i><b>5.1</b> Generalized Linear Model fit for each gene</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#negative-binomial-model-formula"><i class="fa fa-check"></i><b>5.1.1</b> Negative Binomial Model Formula</a></li>
<li class="chapter" data-level="5.1.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#estimating-beta-coefficients-using-the-design-matrix"><i class="fa fa-check"></i><b>5.1.2</b> Estimating Beta Coefficients Using the Design Matrix</a></li>
<li class="chapter" data-level="5.1.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#step-by-step-process"><i class="fa fa-check"></i><b>5.1.3</b> Step-by-Step Process:</a></li>
<li class="chapter" data-level="5.1.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#example"><i class="fa fa-check"></i><b>5.1.4</b> Example:</a></li>
<li class="chapter" data-level="5.1.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#calculation"><i class="fa fa-check"></i><b>5.1.5</b> Calculation:</a></li>
<li class="chapter" data-level="5.1.6" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#matrix-multiplication-form"><i class="fa fa-check"></i><b>5.1.6</b> Matrix Multiplication Form:</a></li>
<li class="chapter" data-level="5.1.7" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#log2-fold-change-and-adjustments"><i class="fa fa-check"></i><b>5.1.7</b> <strong>Log2 Fold Change and Adjustments</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#shrunken-log2-foldchanges-lfc"><i class="fa fa-check"></i><b>5.2</b> Shrunken log2 foldchanges (LFC)</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#hypothesis-testing"><i class="fa fa-check"></i><b>5.2.1</b> Hypothesis Testing</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#statistical-test-for-lfc-estimates-wald-test"><i class="fa fa-check"></i><b>5.3</b> Statistical test for LFC estimates: Wald test</a></li>
<li class="chapter" data-level="5.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-differential-expression-analysis-control-versus-overexpression"><i class="fa fa-check"></i><b>5.4</b> MOV10 Differential Expression Analysis: Control versus Overexpression</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#creating-contrasts-for-hypothesis-testing"><i class="fa fa-check"></i><b>5.4.1</b> Creating Contrasts for Hypothesis Testing</a></li>
<li class="chapter" data-level="5.4.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#ma-plot"><i class="fa fa-check"></i><b>5.4.2</b> MA Plot</a></li>
<li class="chapter" data-level="5.4.3" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#multiple-testing-correction"><i class="fa fa-check"></i><b>5.4.3</b> Multiple Testing Correction</a></li>
<li class="chapter" data-level="5.4.4" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#benjamini-hochberg-adjustment"><i class="fa fa-check"></i><b>5.4.4</b> Benjamini-Hochberg Adjustment</a></li>
<li class="chapter" data-level="5.4.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-de-analysis-control-vs.-knockdown"><i class="fa fa-check"></i><b>5.4.5</b> MOV10 DE Analysis: <strong>Control vs. Knockdown</strong></a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#summarizing-results"><i class="fa fa-check"></i><b>5.5</b> Summarizing Results</a>
<ul>
<li class="chapter" data-level="5.5.1" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#extracting-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.5.1</b> Extracting Significant Differentially Expressed Genes</a></li>
<li class="chapter" data-level="5.5.2" data-path="negative-binomial-model-fitting.html"><a href="negative-binomial-model-fitting.html#mov10-knockdown-analysis-control-vs.-knockdown"><i class="fa fa-check"></i><b>5.5.2</b> MOV10 Knockdown Analysis: Control vs. Knockdown</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html"><i class="fa fa-check"></i><b>6</b> Visualizing RNA-seq results</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#plotting-signicant-de-genes"><i class="fa fa-check"></i><b>6.0.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="6.0.2" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#heatmap"><i class="fa fa-check"></i><b>6.0.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.0.3" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#volcano-plots"><i class="fa fa-check"></i><b>6.0.3</b> Volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html"><i class="fa fa-check"></i><b>7</b> Summary of DGE workflow</a>
<ul>
<li class="chapter" data-level="7.0.1" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#import-data-into-dds-object"><i class="fa fa-check"></i><b>7.0.1</b> 1. Import data into dds object:</a></li>
<li class="chapter" data-level="7.0.2" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data"><i class="fa fa-check"></i><b>7.0.2</b> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:</a></li>
<li class="chapter" data-level="7.0.3" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#run-deseq2"><i class="fa fa-check"></i><b>7.0.3</b> 3. Run DESeq2:</a></li>
<li class="chapter" data-level="7.0.4" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#check-the-fit-of-the-dispersion-estimates"><i class="fa fa-check"></i><b>7.0.4</b> 4. Check the fit of the dispersion estimates:</a></li>
<li class="chapter" data-level="7.0.5" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions"><i class="fa fa-check"></i><b>7.0.5</b> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:</a></li>
<li class="chapter" data-level="7.0.6" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#output-significant-results"><i class="fa fa-check"></i><b>7.0.6</b> 6. Output significant results:</a></li>
<li class="chapter" data-level="7.0.7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc."><i class="fa fa-check"></i><b>7.0.7</b> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.</a></li>
<li class="chapter" data-level="7.0.8" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis"><i class="fa fa-check"></i><b>7.0.8</b> 8. Make sure to output the versions of all tools used in the DE analysis:</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html"><i class="fa fa-check"></i><b>8</b> Functional analysis of RNAseq data</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#ontology-and-go-terms"><i class="fa fa-check"></i><b>8.0.1</b> Ontology and GO Terms</a></li>
<li class="chapter" data-level="8.0.2" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#go-terms-gene-ontology-terms"><i class="fa fa-check"></i><b>8.0.2</b> GO Terms (Gene Ontology Terms)</a></li>
<li class="chapter" data-level="8.0.3" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#go-enrichment-methods"><i class="fa fa-check"></i><b>8.0.3</b> GO Enrichment Methods</a></li>
<li class="chapter" data-level="8.0.4" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#example-1"><i class="fa fa-check"></i><b>8.0.4</b> Example</a></li>
<li class="chapter" data-level="8.0.5" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#over-representation-analysis-ora-using-clusterprofiler"><i class="fa fa-check"></i><b>8.0.5</b> Over-representation Analysis (ORA) using <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="8.0.6" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#visualizing-enrichment-results"><i class="fa fa-check"></i><b>8.0.6</b> Visualizing Enrichment Results</a></li>
<li class="chapter" data-level="8.0.7" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gsea-using-clusterprofiler-and-fgsea"><i class="fa fa-check"></i><b>8.0.7</b> GSEA Using clusterProfiler and fgsea</a></li>
<li class="chapter" data-level="8.0.8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#visualizing-gsea-results"><i class="fa fa-check"></i><b>8.0.8</b> Visualizing GSEA Results</a></li>
<li class="chapter" data-level="8.0.9" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gsea-using-fgsea"><i class="fa fa-check"></i><b>8.0.9</b> GSEA using fgsea</a></li>
<li class="chapter" data-level="8.0.10" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#goenrichment-package-analysis"><i class="fa fa-check"></i><b>8.0.10</b> <code>GOenrichment</code> Package Analysis</a></li>
<li class="chapter" data-level="8.0.11" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#other-tools-and-resources"><i class="fa fa-check"></i><b>8.0.11</b> Other tools and resources</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="codebook.html"><a href="codebook.html"><i class="fa fa-check"></i><b>9</b> Codebook</a>
<ul>
<li class="chapter" data-level="9.1" data-path="codebook.html"><a href="codebook.html#dge-analysis-overview"><i class="fa fa-check"></i><b>9.1</b> DGE analysis overview</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="codebook.html"><a href="codebook.html#setting-up-1"><i class="fa fa-check"></i><b>9.1.1</b> Setting up</a></li>
<li class="chapter" data-level="9.1.2" data-path="codebook.html"><a href="codebook.html#dge-analysis-workflow-1"><i class="fa fa-check"></i><b>9.1.2</b> DGE analysis workflow</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="codebook.html"><a href="codebook.html#count-normalization-1"><i class="fa fa-check"></i><b>9.2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="codebook.html"><a href="codebook.html#match-the-metadata-and-counts-data-1"><i class="fa fa-check"></i><b>9.2.1</b> 1. Match the metadata and counts data</a></li>
<li class="chapter" data-level="9.2.2" data-path="codebook.html"><a href="codebook.html#create-deseq2-object-1"><i class="fa fa-check"></i><b>9.2.2</b> 2. Create DESEq2 object</a></li>
<li class="chapter" data-level="9.2.3" data-path="codebook.html"><a href="codebook.html#generate-the-mov10-normalized-counts-1"><i class="fa fa-check"></i><b>9.2.3</b> 3. Generate the Mov10 normalized counts</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="codebook.html"><a href="codebook.html#quality-control-1"><i class="fa fa-check"></i><b>9.3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="codebook.html"><a href="codebook.html#principal-components-analysis-pca-1"><i class="fa fa-check"></i><b>9.3.1</b> Principal components analysis (PCA)</a></li>
<li class="chapter" data-level="9.3.2" data-path="codebook.html"><a href="codebook.html#hierarchical-clustering-1"><i class="fa fa-check"></i><b>9.3.2</b> Hierarchical Clustering</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="codebook.html"><a href="codebook.html#dge-analysis-workflow-2"><i class="fa fa-check"></i><b>9.4</b> DGE analysis workflow</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="codebook.html"><a href="codebook.html#set-the-design-formula-1"><i class="fa fa-check"></i><b>9.4.1</b> Set the Design Formula</a></li>
<li class="chapter" data-level="9.4.2" data-path="codebook.html"><a href="codebook.html#mov10-differential-expression-analysis-1"><i class="fa fa-check"></i><b>9.4.2</b> MOV10 Differential Expression Analysis</a></li>
<li class="chapter" data-level="9.4.3" data-path="codebook.html"><a href="codebook.html#deseq2-differential-gene-expression-analysis-workflow-1"><i class="fa fa-check"></i><b>9.4.3</b> DESeq2 differential gene expression analysis workflow</a></li>
<li class="chapter" data-level="9.4.4" data-path="codebook.html"><a href="codebook.html#mov10-differential-expression-analysis-exploring-dispersion-estimates-1"><i class="fa fa-check"></i><b>9.4.4</b> MOV10 Differential Expression Analysis: Exploring Dispersion Estimates</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="codebook.html"><a href="codebook.html#negative-binomial-model-fitting-1"><i class="fa fa-check"></i><b>9.5</b> Negative Binomial model fitting</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="codebook.html"><a href="codebook.html#example-2"><i class="fa fa-check"></i><b>9.5.1</b> Example:</a></li>
<li class="chapter" data-level="9.5.2" data-path="codebook.html"><a href="codebook.html#calculation-1"><i class="fa fa-check"></i><b>9.5.2</b> Calculation:</a></li>
<li class="chapter" data-level="9.5.3" data-path="codebook.html"><a href="codebook.html#matrix-multiplication-form-1"><i class="fa fa-check"></i><b>9.5.3</b> Matrix Multiplication Form:</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="codebook.html"><a href="codebook.html#mov10-differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.6</b> MOV10 Differential Expression Analysis:</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="codebook.html"><a href="codebook.html#mov10-de-analysis-control-vs.-knockdown-1"><i class="fa fa-check"></i><b>9.6.1</b> MOV10 DE Analysis: <strong>Control vs. Knockdown</strong></a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="codebook.html"><a href="codebook.html#summarizing-results-1"><i class="fa fa-check"></i><b>9.7</b> Summarizing Results</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="codebook.html"><a href="codebook.html#extracting-significant-differentially-expressed-genes-1"><i class="fa fa-check"></i><b>9.7.1</b> Extracting Significant Differentially Expressed Genes</a></li>
<li class="chapter" data-level="9.7.2" data-path="codebook.html"><a href="codebook.html#mov10-knockdown-analysis-control-vs.-knockdown-1"><i class="fa fa-check"></i><b>9.7.2</b> MOV10 Knockdown Analysis: Control vs. Knockdown</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="codebook.html"><a href="codebook.html#visualizing-rna-seq-results-1"><i class="fa fa-check"></i><b>9.8</b> Visualizing RNA-seq results</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="codebook.html"><a href="codebook.html#heatmap-1"><i class="fa fa-check"></i><b>9.8.1</b> Heatmap</a></li>
<li class="chapter" data-level="9.8.2" data-path="codebook.html"><a href="codebook.html#volcano-plots-1"><i class="fa fa-check"></i><b>9.8.2</b> Volcano plots</a></li>
</ul></li>
<li class="chapter" data-level="9.9" data-path="codebook.html"><a href="codebook.html#summary-of-dge-workflow-1"><i class="fa fa-check"></i><b>9.9</b> Summary of DGE workflow</a>
<ul>
<li class="chapter" data-level="9.9.1" data-path="codebook.html"><a href="codebook.html#import-data-into-dds-object-1"><i class="fa fa-check"></i><b>9.9.1</b> 1. Import data into dds object:</a></li>
<li class="chapter" data-level="9.9.2" data-path="codebook.html"><a href="codebook.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data-1"><i class="fa fa-check"></i><b>9.9.2</b> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:</a></li>
<li class="chapter" data-level="9.9.3" data-path="codebook.html"><a href="codebook.html#run-deseq2-1"><i class="fa fa-check"></i><b>9.9.3</b> 3. Run DESeq2:</a></li>
<li class="chapter" data-level="9.9.4" data-path="codebook.html"><a href="codebook.html#check-the-fit-of-the-dispersion-estimates-1"><i class="fa fa-check"></i><b>9.9.4</b> 4. Check the fit of the dispersion estimates:</a></li>
<li class="chapter" data-level="9.9.5" data-path="codebook.html"><a href="codebook.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions-1"><i class="fa fa-check"></i><b>9.9.5</b> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:</a></li>
<li class="chapter" data-level="9.9.6" data-path="codebook.html"><a href="codebook.html#output-significant-results-1"><i class="fa fa-check"></i><b>9.9.6</b> 6. Output significant results:</a></li>
<li class="chapter" data-level="9.9.7" data-path="codebook.html"><a href="codebook.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc.-1"><i class="fa fa-check"></i><b>9.9.7</b> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.</a></li>
<li class="chapter" data-level="9.9.8" data-path="codebook.html"><a href="codebook.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis-1"><i class="fa fa-check"></i><b>9.9.8</b> 8. Make sure to output the versions of all tools used in the DE analysis:</a></li>
</ul></li>
<li class="chapter" data-level="9.10" data-path="codebook.html"><a href="codebook.html#functional-analysis-of-rna-seq-data"><i class="fa fa-check"></i><b>9.10</b> Functional analysis of RNA-seq data</a>
<ul>
<li class="chapter" data-level="9.10.1" data-path="codebook.html"><a href="codebook.html#over-representation-analysis-ora-using-clusterprofiler-1"><i class="fa fa-check"></i><b>9.10.1</b> Over-representation Analysis (ORA) using <code>clusterProfiler</code></a></li>
<li class="chapter" data-level="9.10.2" data-path="codebook.html"><a href="codebook.html#visualizing-enrichment-results-1"><i class="fa fa-check"></i><b>9.10.2</b> Visualizing Enrichment Results</a></li>
<li class="chapter" data-level="9.10.3" data-path="codebook.html"><a href="codebook.html#gsea-using-clusterprofiler-and-fgsea-1"><i class="fa fa-check"></i><b>9.10.3</b> GSEA Using clusterProfiler and fgsea</a></li>
<li class="chapter" data-level="9.10.4" data-path="codebook.html"><a href="codebook.html#visualizing-gsea-results-1"><i class="fa fa-check"></i><b>9.10.4</b> Visualizing GSEA Results</a></li>
<li class="chapter" data-level="9.10.5" data-path="codebook.html"><a href="codebook.html#gsea-using-fgsea-1"><i class="fa fa-check"></i><b>9.10.5</b> GSEA using fgsea</a></li>
<li class="chapter" data-level="9.10.6" data-path="codebook.html"><a href="codebook.html#goenrichment-package-analysis-1"><i class="fa fa-check"></i><b>9.10.6</b> <code>GOenrichment</code> Package Analysis</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2024_RNA-seq-analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="codebook" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Codebook<a href="codebook.html#codebook" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>28 total points</p>
<div id="dge-analysis-overview" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> DGE analysis overview<a href="codebook.html#dge-analysis-overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 01-DGE_setup_and_overview.Rmd</strong></p>
<div id="setting-up-1" class="section level3 hasAnchor" number="9.1.1">
<h3><span class="header-section-number">9.1.1</span> Setting up<a href="codebook.html#setting-up-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="loading-libraries-technically-we-dont-need-to-load-all-of-these-libraries-because-they-are-in-the-file-setup.r-which-is-sourced-in-the-first-.rmd-of-the-module.-however-its-good-practice-to-load-them-all-at-the-beginning-of-your-script." class="section level4 hasAnchor" number="9.1.1.1">
<h4><span class="header-section-number">9.1.1.1</span> Loading libraries; technically we don’t need to load all of these libraries because they are in the file setup.R which is sourced in the first .Rmd of the module. However, it’s good practice to load them all at the beginning of your script.<a href="codebook.html#loading-libraries-technically-we-dont-need-to-load-all-of-these-libraries-because-they-are-in-the-file-setup.r-which-is-sourced-in-the-first-.rmd-of-the-module.-however-its-good-practice-to-load-them-all-at-the-beginning-of-your-script." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="codebook.html#cb182-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb182-2"><a href="codebook.html#cb182-2" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb182-3"><a href="codebook.html#cb182-3" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb182-4"><a href="codebook.html#cb182-4" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb182-5"><a href="codebook.html#cb182-5" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb182-6"><a href="codebook.html#cb182-6" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span></code></pre></div>
</div>
<div id="loading-data-1" class="section level4 hasAnchor" number="9.1.1.2">
<h4><span class="header-section-number">9.1.1.2</span> Loading data<a href="codebook.html#loading-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="codebook.html#cb183-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/Mov10_full_counts.txt&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb183-2"><a href="codebook.html#cb183-2" tabindex="-1"></a></span>
<span id="cb183-3"><a href="codebook.html#cb183-3" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/Mov10_full_meta.txt&quot;</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Use class() to inspect our data and make sure we are working with data frames:</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="codebook.html#cb184-1" tabindex="-1"></a><span class="do">### Check classes of the data we just brought in</span></span>
<span id="cb184-2"><a href="codebook.html#cb184-2" tabindex="-1"></a><span class="fu">class</span>(meta)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="codebook.html#cb186-1" tabindex="-1"></a><span class="fu">class</span>(data)</span></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
</div>
<div id="viewing-data-1" class="section level4 hasAnchor" number="9.1.1.3">
<h4><span class="header-section-number">9.1.1.3</span> Viewing data<a href="codebook.html#viewing-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb188"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb188-1"><a href="codebook.html#cb188-1" tabindex="-1"></a><span class="fu">rownames</span>(meta)</span></code></pre></div>
<pre><code>## [1] &quot;Irrel_kd_1&quot; &quot;Irrel_kd_2&quot; &quot;Irrel_kd_3&quot; &quot;Mov10_kd_2&quot; &quot;Mov10_kd_3&quot;
## [6] &quot;Mov10_oe_1&quot; &quot;Mov10_oe_2&quot; &quot;Mov10_oe_3&quot;</code></pre>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="codebook.html#cb190-1" tabindex="-1"></a><span class="fu">names</span>(data)</span></code></pre></div>
<pre><code>## [1] &quot;Mov10_kd_2&quot; &quot;Mov10_kd_3&quot; &quot;Mov10_oe_1&quot; &quot;Mov10_oe_2&quot; &quot;Mov10_oe_3&quot;
## [6] &quot;Irrel_kd_1&quot; &quot;Irrel_kd_2&quot; &quot;Irrel_kd_3&quot;</code></pre>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="codebook.html#cb192-1" tabindex="-1"></a>data[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>]</span></code></pre></div>
<div class="kable-table">
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">Mov10_kd_2</th>
<th align="right">Mov10_kd_3</th>
<th align="right">Mov10_oe_1</th>
<th align="right">Mov10_oe_2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1/2-SBSRNA4</td>
<td align="right">57</td>
<td align="right">41</td>
<td align="right">64</td>
<td align="right">55</td>
</tr>
<tr class="even">
<td align="left">A1BG</td>
<td align="right">71</td>
<td align="right">40</td>
<td align="right">100</td>
<td align="right">81</td>
</tr>
<tr class="odd">
<td align="left">A1BG-AS1</td>
<td align="right">256</td>
<td align="right">177</td>
<td align="right">220</td>
<td align="right">189</td>
</tr>
<tr class="even">
<td align="left">A1CF</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<p>You’ll notice that the colnames of the data and the rownames of metadata are the same except that they are in a different order. This is important later as we will be merging the metadata with the data based on these sample names.</p>
</div>
</div>
<div id="dge-analysis-workflow-1" class="section level3 hasAnchor" number="9.1.2">
<h3><span class="header-section-number">9.1.2</span> DGE analysis workflow<a href="codebook.html#dge-analysis-workflow-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="rna-seq-count-distribution-1" class="section level4 hasAnchor" number="9.1.2.1">
<h4><span class="header-section-number">9.1.2.1</span> RNA-seq count distribution<a href="codebook.html#rna-seq-count-distribution-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To determine the appropriate statistical model, we need information about the distribution of counts. To get an idea about how RNA-seq counts are distributed, let’s plot the counts for a single sample, ‘Mov10_oe_1’:</p>
<div class="sourceCode" id="cb193"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb193-1"><a href="codebook.html#cb193-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb193-2"><a href="codebook.html#cb193-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb193-3"><a href="codebook.html#cb193-3" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb193-4"><a href="codebook.html#cb193-4" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-110-1.png" width="672" /></p>
<p>If we zoom in close to zero, we can see a large number of genes with counts of zero:</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="codebook.html#cb194-1" tabindex="-1"></a><span class="fu">ggplot</span>(data) <span class="sc">+</span></span>
<span id="cb194-2"><a href="codebook.html#cb194-2" tabindex="-1"></a><span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">x =</span> Mov10_oe_1), <span class="at">stat =</span> <span class="st">&quot;bin&quot;</span>, <span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb194-3"><a href="codebook.html#cb194-3" tabindex="-1"></a>   <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">500</span>)  <span class="sc">+</span></span>
<span id="cb194-4"><a href="codebook.html#cb194-4" tabindex="-1"></a>   <span class="fu">xlab</span>(<span class="st">&quot;Raw expression counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb194-5"><a href="codebook.html#cb194-5" tabindex="-1"></a>   <span class="fu">ylab</span>(<span class="st">&quot;Number of genes&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-111-1.png" width="672" /></p>
<p>These images illustrate some common features of RNA-seq count data, including a low counts associated with a large proportion of genes, and a long right tail due to the lack of any upper limit for expression.</p>
</div>
<div id="modeling-count-data-1" class="section level4 hasAnchor" number="9.1.2.2">
<h4><span class="header-section-number">9.1.2.2</span> Modeling count data<a href="codebook.html#modeling-count-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>By plotting the <em>mean versus the variance</em> of our data we should be able to see that the variance &gt; mean and therefore it does not fit the Poisson distribution and is better suited to the Negative Binomial (NB) model.</p>
<p>To calculate the mean and variance of our data, we will use the <code>apply</code> function. The <code>apply</code> function allows you to apply a function to the margins (rows or columns) of a matrix. The syntax for <code>apply</code> is as follows: <code>apply(X, MARGIN, FUN)</code>, where <code>X</code> is a matrix, <code>MARGIN</code> is the margin of the matrix to apply the function to (1 = rows, 2 = columns), and <code>FUN</code> is the function to apply.</p>
<p>We will use <code>MARGIN = 1</code> to apply the mean and variance of the counts for each row (gene) across the ‘Mov10 overexpression’ replicates. We will then create a data frame with the mean and variance of the counts for each gene.</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="codebook.html#cb195-1" tabindex="-1"></a>mean_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, mean)</span>
<span id="cb195-2"><a href="codebook.html#cb195-2" tabindex="-1"></a></span>
<span id="cb195-3"><a href="codebook.html#cb195-3" tabindex="-1"></a>variance_counts <span class="ot">&lt;-</span> <span class="fu">apply</span>(data[, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>], <span class="dv">1</span>, var)</span>
<span id="cb195-4"><a href="codebook.html#cb195-4" tabindex="-1"></a></span>
<span id="cb195-5"><a href="codebook.html#cb195-5" tabindex="-1"></a><span class="co"># for ggplot we need the data to be in a data.frame</span></span>
<span id="cb195-6"><a href="codebook.html#cb195-6" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mean_counts, variance_counts)</span></code></pre></div>
<p>Run the following code to plot the <em>mean versus variance</em> for the ‘Mov10 overexpression’ replicates:</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="codebook.html#cb196-1" tabindex="-1"></a><span class="fu">ggplot</span>(df) <span class="sc">+</span></span>
<span id="cb196-2"><a href="codebook.html#cb196-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_counts, <span class="at">y =</span> variance_counts)) <span class="sc">+</span></span>
<span id="cb196-3"><a href="codebook.html#cb196-3" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_counts, <span class="at">y =</span> mean_counts, <span class="at">color =</span> <span class="st">&quot;red&quot;</span>)) <span class="sc">+</span></span>
<span id="cb196-4"><a href="codebook.html#cb196-4" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb196-5"><a href="codebook.html#cb196-5" tabindex="-1"></a>  <span class="fu">scale_x_log10</span>()</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-113-1.png" width="672" /></p>
<p>By plotting the mean versus the variance of our data, we can see that the variance is greater than the mean, indicating that the data does not fit the Poisson distribution and is better suited to the Negative Binomial (NB) model.</p>
</div>
</div>
</div>
<div id="count-normalization-1" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> Count normalization<a href="codebook.html#count-normalization-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 02-DGE_count_normalization.Rmd</strong></p>
<p><strong>Exercise</strong> points +1</p>
<p>Determine the normalized counts for your gene of interest, PD1, given the raw counts and size factors below.</p>
<p>NOTE: You will need to run the code below to generate the raw counts dataframe (PD1) and the size factor vector (size_factors), then use these objects to determine the normalized counts values:</p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="codebook.html#cb197-1" tabindex="-1"></a><span class="co"># Raw counts for PD1</span></span>
<span id="cb197-2"><a href="codebook.html#cb197-2" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">21</span>, <span class="dv">58</span>, <span class="dv">17</span>, <span class="dv">97</span>, <span class="dv">83</span>, <span class="dv">10</span>)</span>
<span id="cb197-3"><a href="codebook.html#cb197-3" tabindex="-1"></a><span class="fu">names</span>(PD1) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;Sample&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)</span>
<span id="cb197-4"><a href="codebook.html#cb197-4" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(PD1)</span>
<span id="cb197-5"><a href="codebook.html#cb197-5" tabindex="-1"></a>PD1 <span class="ot">&lt;-</span> <span class="fu">t</span>(PD1)</span>
<span id="cb197-6"><a href="codebook.html#cb197-6" tabindex="-1"></a></span>
<span id="cb197-7"><a href="codebook.html#cb197-7" tabindex="-1"></a><span class="co"># Size factors for each sample</span></span>
<span id="cb197-8"><a href="codebook.html#cb197-8" tabindex="-1"></a>size_factors <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">1.32</span>, <span class="fl">0.70</span>, <span class="fl">1.04</span>, <span class="fl">1.27</span>, <span class="fl">1.11</span>, <span class="fl">0.85</span>)</span></code></pre></div>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="codebook.html#cb198-1" tabindex="-1"></a><span class="co"># Your code here</span></span></code></pre></div>
<p>Now that we know the theory of count normalization, we will normalize the counts for the Mov10 dataset using DESeq2. This requires a few steps:</p>
<ol style="list-style-type: decimal">
<li>Ensure the row names of the metadata dataframe are present and in the same order as the column names of the counts dataframe.</li>
<li>Create a <code>DESeqDataSet</code> object</li>
<li>Generate the normalized counts</li>
</ol>
<div id="match-the-metadata-and-counts-data-1" class="section level3 hasAnchor" number="9.2.1">
<h3><span class="header-section-number">9.2.1</span> 1. Match the metadata and counts data<a href="codebook.html#match-the-metadata-and-counts-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We should always make sure that we have sample names that match between the two files, and that the samples are in the right order. DESeq2 will output an error if this is not the case.</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="codebook.html#cb199-1" tabindex="-1"></a><span class="do">### Check that sample names match in both files</span></span>
<span id="cb199-2"><a href="codebook.html#cb199-2" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">%in%</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb201"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb201-1"><a href="codebook.html#cb201-1" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">==</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>The colnames of our data don’t match the rownames of our metadata so we need to reorder them. We can use the <code>match</code> function:</p>
<div class="sourceCode" id="cb203"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb203-1"><a href="codebook.html#cb203-1" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">rownames</span>(meta),<span class="fu">colnames</span>(data))</span>
<span id="cb203-2"><a href="codebook.html#cb203-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> data[,idx]</span>
<span id="cb203-3"><a href="codebook.html#cb203-3" tabindex="-1"></a></span>
<span id="cb203-4"><a href="codebook.html#cb203-4" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(data) <span class="sc">==</span> <span class="fu">rownames</span>(meta))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p><strong>Exercise</strong> points = +2</p>
<p>Suppose we had sample names matching in the counts matrix and metadata file, but they were out of order. Write the line(s) of code required to create a new matrix with columns ordered such that they were identical to the row names of the metadata.</p>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="codebook.html#cb205-1" tabindex="-1"></a><span class="co"># write the code to reorder the columns of the &#39;counts&#39; matrix to match the rownames of the metadata</span></span>
<span id="cb205-2"><a href="codebook.html#cb205-2" tabindex="-1"></a></span>
<span id="cb205-3"><a href="codebook.html#cb205-3" tabindex="-1"></a></span>
<span id="cb205-4"><a href="codebook.html#cb205-4" tabindex="-1"></a><span class="co"># write the code to reorder the columns of the &#39;counts&#39; matrix to match the rownames of the metadata</span></span></code></pre></div>
</div>
<div id="create-deseq2-object-1" class="section level3 hasAnchor" number="9.2.2">
<h3><span class="header-section-number">9.2.2</span> 2. Create DESEq2 object<a href="codebook.html#create-deseq2-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Bioconductor software packages often define and use a custom class within R for storing data (input data, intermediate data and also result data). These custom data structures are similar to <code>lists</code> in that they can contain multiple different data types/structures within them. But, unlike lists they have pre-specified <code>data slots</code>, which hold specific types/classes of data. The data stored in these pre-specified slots can be accessed by using specific package-defined functions.</p>
<p>Let’s start by creating the <code>DESeqDataSet</code> object and then we can talk a bit more about what is stored inside it. To create the object we will need the <strong>count matrix</strong> and the <strong>metadata</strong> table as input. We will also need to specify a <strong>design formula</strong>. The design formula specifies the column(s) in the metadata table and how they should be used in the analysis. For our dataset we only have one column we are interested in, that is <code>~sampletype</code>. This column has three factor levels, which tells DESeq2 that for each gene we want to evaluate gene expression change with respect to these different levels.</p>
<div class="sourceCode" id="cb206"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb206-1"><a href="codebook.html#cb206-1" tabindex="-1"></a><span class="do">## Create DESeq2Dataset object</span></span>
<span id="cb206-2"><a href="codebook.html#cb206-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data, <span class="at">colData =</span> meta, <span class="at">design =</span> <span class="sc">~</span> sampletype)</span></code></pre></div>
<p>You can use DESeq2-specific functions to access the different slots and retrieve information, if you wish. For example, suppose we wanted the original count matrix we would use <code>counts()</code>:</p>
<div class="sourceCode" id="cb207"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb207-1"><a href="codebook.html#cb207-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">counts</span>(dds[,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]))</span></code></pre></div>
<pre><code>##             Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3
## 1/2-SBSRNA4         45         31         39         57         41
## A1BG                77         58         40         71         40
## A1BG-AS1           213        172        126        256        177
## A1CF                 0          0          0          0          1
## A2LD1               91         80         50        146         81
## A2M                  9          8          4         10          9</code></pre>
<p>As we go through the workflow we will use the relevant functions to check what information gets stored inside our object. We can also run:</p>
<div class="sourceCode" id="cb209"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb209-1"><a href="codebook.html#cb209-1" tabindex="-1"></a><span class="fu">slotNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;design&quot;             &quot;dispersionFunction&quot; &quot;rowRanges&quot;         
## [4] &quot;colData&quot;            &quot;assays&quot;             &quot;NAMES&quot;             
## [7] &quot;elementMetadata&quot;    &quot;metadata&quot;</code></pre>
</div>
<div id="generate-the-mov10-normalized-counts-1" class="section level3 hasAnchor" number="9.2.3">
<h3><span class="header-section-number">9.2.3</span> 3. Generate the Mov10 normalized counts<a href="codebook.html#generate-the-mov10-normalized-counts-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to normalize the count data in order to be able to make fair gene comparisons between samples.</p>
<p><img src="img/deseq_workflow_normalization_2018.png" width="400" alt="Alt text" /></p>
<p>To perform the <strong>median of ratios method</strong> of normalization, DESeq2 has a single <code>estimateSizeFactors()</code> function that will generate size factors for us. We will use the function in the example below, but <strong>in a typical RNA-seq analysis this step is automatically performed by the <code>DESeq()</code> function</strong>, which we will see later.</p>
<div class="sourceCode" id="cb211"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb211-1"><a href="codebook.html#cb211-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span></code></pre></div>
<p>By assigning the results back to the <code>dds</code> object we are filling in the slots of the <code>DESeqDataSet</code> object with the appropriate information. We can take a look at the normalization factor applied to each sample using:</p>
<div class="sourceCode" id="cb212"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb212-1"><a href="codebook.html#cb212-1" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 
##  1.1224020  0.9625632  0.7477715  1.5646728  0.9351760  1.2016082 
## Mov10_oe_2 Mov10_oe_3 
##  1.1205912  0.6534987</code></pre>
<p>Now, to retrieve the normalized counts matrix from <code>dds</code>, we use the <code>counts()</code> function and add the argument <code>normalized=TRUE</code>.</p>
<div class="sourceCode" id="cb214"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb214-1"><a href="codebook.html#cb214-1" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can save this normalized data matrix to file for later use:</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="codebook.html#cb215-1" tabindex="-1"></a><span class="fu">write.table</span>(normalized_counts, <span class="at">file=</span><span class="st">&quot;data/normalized_counts.txt&quot;</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote=</span>F, <span class="at">col.names=</span><span class="cn">NA</span>)</span></code></pre></div>
</div>
</div>
<div id="quality-control-1" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Quality Control<a href="codebook.html#quality-control-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 03-DGE_QC_analysis.Rmd</strong></p>
<p><strong>Exercise</strong> points = +5</p>
<p>The figure below was generated from a time course experiment with sample groups ‘Ctrl’ and ‘Sci’ and the following timepoints: 0h, 2h, 8h, and 16h.</p>
<p><img src="img/PCA_example3.png" width="600" alt="Alt text" /></p>
<p><strong>Determine the sources explaining the variation represented by PC1 and PC2.</strong></p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Do the sample groups separate well?</strong></p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Do the replicates cluster together for each sample group?</strong></p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Are there any outliers in the data?</strong></p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Should we have any other concerns regarding the samples in the dataset?</strong></p>
<ul>
<li>Ans:</li>
</ul>
<p><strong>Example of Variance Stabilization</strong>
Raw RNA-Seq count data has different levels of variability:</p>
<p>Without stabilization: Lowly expressed genes show high variability between samples (often due to noise), and highly expressed genes show less variability.
With stabilization (e.g., rlog or vst): Variance across genes is more evenly distributed, which makes the data easier to work with in analyses like clustering or PCA.</p>
<blockquote>
<p>The rlog transformation of the normalized counts is only necessary for these visualization methods during this quality assessment.</p>
</blockquote>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="codebook.html#cb216-1" tabindex="-1"></a><span class="do">### Transform counts for data visualization</span></span>
<span id="cb216-2"><a href="codebook.html#cb216-2" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>The <code>blind=TRUE</code> argument results in a transformation unbiased to sample condition information. When performing quality assessment, it is important to include this option. The <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#blind-dispersion-estimation">DESeq2 vignette</a> has more details.</p>
<p>The <code>rlog</code> function returns a <code>DESeqTransform</code> object, another type of DESeq-specific object. The reason you don’t just get a matrix of transformed values is because all of the parameters (i.e. size factors) that went into computing the rlog transform are stored in that object. We use this object to plot the PCA and heirarchical clustering figures for quality assessment.</p>
<div id="principal-components-analysis-pca-1" class="section level3 hasAnchor" number="9.3.1">
<h3><span class="header-section-number">9.3.1</span> Principal components analysis (PCA)<a href="codebook.html#principal-components-analysis-pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>DESeq2 has a built-in function for plotting PCA plots, that uses <code>ggplot2</code> under the hood. This is great because it saves us having to type out lines of code and having to fiddle with the different ggplot2 layers. In addition, it takes the <code>rlog</code> object as an input directly, hence saving us the trouble of extracting the relevant information from it.</p>
<p>The function <code>plotPCA()</code> requires two arguments as input: an <code>rlog</code> object and the <code>intgroup</code> (the column in our metadata that we are interested in).</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="codebook.html#cb217-1" tabindex="-1"></a><span class="do">### Plot PCA </span></span>
<span id="cb217-2"><a href="codebook.html#cb217-2" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-127-1.png" width="576" style="display: block; margin: auto;" /></p>
<p><strong>What does this plot tell you about the similarity of samples? Does it fit the expectation from the experimental design?</strong> By default the function uses the <em>top 500 most variable genes</em>. You can change this by adding the <code>ntop</code> argument and specifying how many genes you want to use to draw the plot.</p>
</div>
<div id="hierarchical-clustering-1" class="section level3 hasAnchor" number="9.3.2">
<h3><span class="header-section-number">9.3.2</span> Hierarchical Clustering<a href="codebook.html#hierarchical-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Since there is no built-in function for heatmaps in DESeq2 we will be using the <code>pheatmap()</code> function from the <code>pheatmap</code> package. This function requires a matrix/dataframe of numeric values as input, and so the first thing we need to is retrieve that information from the <code>rld</code> object:</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="codebook.html#cb218-1" tabindex="-1"></a><span class="do">### Extract the rlog matrix from the object</span></span>
<span id="cb218-2"><a href="codebook.html#cb218-2" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)    <span class="do">## assay() is function from the &quot;SummarizedExperiment&quot; package that was loaded when you loaded DESeq2</span></span></code></pre></div>
<p>Then we need to compute the pairwise correlation values for samples. We can do this using the <code>cor()</code> function:</p>
<div class="sourceCode" id="cb219"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb219-1"><a href="codebook.html#cb219-1" tabindex="-1"></a><span class="do">### Compute pairwise correlation values</span></span>
<span id="cb219-2"><a href="codebook.html#cb219-2" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)    <span class="do">## cor() is a base R function</span></span>
<span id="cb219-3"><a href="codebook.html#cb219-3" tabindex="-1"></a></span>
<span id="cb219-4"><a href="codebook.html#cb219-4" tabindex="-1"></a><span class="fu">head</span>(rld_cor)   <span class="do">## check the output of cor(), make note of the rownames and colnames</span></span></code></pre></div>
<pre><code>##            Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3
## Irrel_kd_1  1.0000000  0.9999614  0.9999532  0.9997202  0.9997748
## Irrel_kd_2  0.9999614  1.0000000  0.9999544  0.9996918  0.9997568
## Irrel_kd_3  0.9999532  0.9999544  1.0000000  0.9996816  0.9997574
## Mov10_kd_2  0.9997202  0.9996918  0.9996816  1.0000000  0.9999492
## Mov10_kd_3  0.9997748  0.9997568  0.9997574  0.9999492  1.0000000
## Mov10_oe_1  0.9996700  0.9996984  0.9997067  0.9994868  0.9996154
##            Mov10_oe_1 Mov10_oe_2 Mov10_oe_3
## Irrel_kd_1  0.9996700  0.9996599  0.9995804
## Irrel_kd_2  0.9996984  0.9996825  0.9996227
## Irrel_kd_3  0.9997067  0.9997090  0.9996026
## Mov10_kd_2  0.9994868  0.9994565  0.9993869
## Mov10_kd_3  0.9996154  0.9995905  0.9995235
## Mov10_oe_1  1.0000000  0.9999505  0.9999196</code></pre>
<p>And now to plot the correlation values as a heatmap:</p>
<div class="sourceCode" id="cb221"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb221-1"><a href="codebook.html#cb221-1" tabindex="-1"></a><span class="do">### Plot heatmap</span></span>
<span id="cb221-2"><a href="codebook.html#cb221-2" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
<p>Overall, we observe pretty high correlations across the board ( &gt; 0.999) suggesting no outlying sample(s). Also, similar to the PCA plot you see the samples clustering together by sample group. Together, these plots suggest to us that the data are of good quality and we have the green light to proceed to differential expression analysis.</p>
<p><strong>Exercise</strong> points = +3</p>
<p>The pheatmap function has a number of different arguments that we can alter from default values to enhance the aesthetics of the plot. Try adding the arguments <code>color</code>, <code>border_color</code>, <code>fontsize_row</code>, <code>fontsize_col</code>, <code>show_rownames</code> and <code>show_colnames</code> to your pheatmap. How does your plot change. Take a look through the help pages (?pheatmap) and identify what each of the added arguments is contributing to the plot.</p>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-131-1.png" width="768" /></p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="codebook.html#cb222-1" tabindex="-1"></a><span class="co"># The color palette &quot;Blues&quot; is a good choice for this heatmap</span></span>
<span id="cb222-2"><a href="codebook.html#cb222-2" tabindex="-1"></a>heat.colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">&quot;Blues&quot;</span>)</span>
<span id="cb222-3"><a href="codebook.html#cb222-3" tabindex="-1"></a></span>
<span id="cb222-4"><a href="codebook.html#cb222-4" tabindex="-1"></a><span class="co"># Your code here</span></span>
<span id="cb222-5"><a href="codebook.html#cb222-5" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor, <span class="at">border_color =</span> <span class="st">&quot;grey&quot;</span>, <span class="at">fontsize_row =</span> <span class="dv">12</span>, <span class="at">fontsize_col =</span> <span class="dv">12</span>, <span class="at">show_rownames =</span> <span class="cn">TRUE</span>, <span class="at">show_colnames =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> heat.colors)</span></code></pre></div>
</div>
</div>
<div id="dge-analysis-workflow-2" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> DGE analysis workflow<a href="codebook.html#dge-analysis-workflow-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 04_DGE_DESeq2_analysis.Rmd</strong></p>
<div id="set-the-design-formula-1" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Set the Design Formula<a href="codebook.html#set-the-design-formula-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The main purpose of the <strong>design formula</strong> in DESeq2 is to specify the factors that are influencing gene expression so that their effects can be <strong>accounted for</strong> or <strong>controlled</strong> during the analysis. This allows DESeq2 to isolate the effect of the variable you’re primarily interested in while adjusting for other known sources of variation. The design formula should have all of the factors in your metadata that account for major sources of variation in your data. The last factor entered in the formula should be the condition of interest.</p>
<p>For example:</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="codebook.html#cb223-1" tabindex="-1"></a><span class="fu">design</span>(dds)</span></code></pre></div>
<pre><code>## ~sampletype</code></pre>
<p>This shows that <strong>sampletype</strong> is the condition of interest and as such is the only covariate. DESeq2 will test for differential expression between the two sample types, adjusting for any other variation.</p>
<p><strong>Exercise</strong> points = +3</p>
<ol style="list-style-type: decimal">
<li>Suppose you wanted to study the expression differences between the two age groups in the metadata shown above, and major sources of variation were <code>sex</code> and <code>treatment</code>, how would the design formula be written?</li>
</ol>
<div class="sourceCode" id="cb225"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb225-1"><a href="codebook.html#cb225-1" tabindex="-1"></a><span class="co"># the default behavior of `results` is to return the comparison of the last group in the design formula, so your condition of interest should come last:</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Based on our Mov10 metadata dataframe, which factors could we include in our design formula?</li>
</ol>
<ul>
<li>Ans:</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>What would you do if you wanted to include a factor in your design formula that is not in your metadata?</li>
</ol>
<ul>
<li>Ans:</li>
</ul>
</div>
<div id="mov10-differential-expression-analysis-1" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> MOV10 Differential Expression Analysis<a href="codebook.html#mov10-differential-expression-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Running Differential Expression in Two Lines of Code</strong></p>
<p>To obtain differential expression results from our raw count data, we only need to run two lines of code!</p>
<p>First, we create a DESeqDataSet, as we did in the <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html#2-create-deseq2-object">‘Count normalization’</a> lesson, specifying the location of our raw counts and metadata, and applying our design formula:</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="codebook.html#cb226-1" tabindex="-1"></a><span class="do">## Create DESeq object</span></span>
<span id="cb226-2"><a href="codebook.html#cb226-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> data, <span class="at">colData =</span> meta, <span class="at">design =</span> <span class="sc">~</span> sampletype)</span></code></pre></div>
<p>Next, we run the actual differential expression analysis with a single call to the <code>DESeq()</code> function. This function handles everything—from <strong>normalization</strong> to <strong>linear</strong> modeling—all in one step. During execution, <code>DESeq2</code> will print messages detailing the steps being performed: estimating size factors, estimating dispersions, gene-wise dispersion estimates, modeling the mean-dispersion relationship, and statistical testing for differential expression.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="codebook.html#cb227-1" tabindex="-1"></a><span class="do">## Run analysis</span></span>
<span id="cb227-2"><a href="codebook.html#cb227-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<p>By re-assigning the result to back to the same variable name (<code>dds</code>), we update our <code>DESeqDataSet</code> object, which will now contain the results of each step in the analysis, effectively filling in the <code>slots</code> of our <code>DESeqDataSet</code> object.</p>
</div>
<div id="deseq2-differential-gene-expression-analysis-workflow-1" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> DESeq2 differential gene expression analysis workflow<a href="codebook.html#deseq2-differential-gene-expression-analysis-workflow-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Everything from normalization to linear modeling was carried out by the use of a single function!</p>
<p>With the 2 lines of code above, we just completed the workflow for the differential gene expression analysis with DESeq2. The steps in the analysis are output below:</p>
<p><img src="img/deseq2_workflow_separate.png" width="200" alt="Alt text" /></p>
<div id="step-1-estimate-size-factors-1" class="section level4 hasAnchor" number="9.4.3.1">
<h4><span class="header-section-number">9.4.3.1</span> Step 1: Estimate size factors<a href="codebook.html#step-1-estimate-size-factors-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>MOV10 DE analysis: examining the size factors</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="codebook.html#cb234-1" tabindex="-1"></a><span class="do">## Check the size factors</span></span>
<span id="cb234-2"><a href="codebook.html#cb234-2" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 
##  1.1224020  0.9625632  0.7477715  1.5646728  0.9351760  1.2016082 
## Mov10_oe_2 Mov10_oe_3 
##  1.1205912  0.6534987</code></pre>
<p>Take a look at the total number of reads for each sample:</p>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="codebook.html#cb236-1" tabindex="-1"></a><span class="do">## Total number of raw counts per sample</span></span>
<span id="cb236-2"><a href="codebook.html#cb236-2" tabindex="-1"></a><span class="fu">colSums</span>(<span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">FALSE</span>))</span></code></pre></div>
<pre><code>## Irrel_kd_1 Irrel_kd_2 Irrel_kd_3 Mov10_kd_2 Mov10_kd_3 Mov10_oe_1 
##   22687366   19381680   14962754   32826936   19360003   23447317 
## Mov10_oe_2 Mov10_oe_3 
##   21713289   12737889</code></pre>
<p><strong>How do the numbers correlate with the size factor?</strong></p>
<ul>
<li>Ans:</li>
</ul>
</div>
</div>
<div id="mov10-differential-expression-analysis-exploring-dispersion-estimates-1" class="section level3 hasAnchor" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> MOV10 Differential Expression Analysis: Exploring Dispersion Estimates<a href="codebook.html#mov10-differential-expression-analysis-exploring-dispersion-estimates-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now, let’s explore the dispersion estimates for the MOV10 dataset:</p>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="codebook.html#cb238-1" tabindex="-1"></a><span class="do">## Plot dispersion estimates</span></span>
<span id="cb238-2"><a href="codebook.html#cb238-2" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-139-1.png" width="672" /></p>
<p>**Since we have a small sample size, for many genes we see quite a bit of shrinkage.</p>
<p><strong>Exercise</strong> points = +1</p>
<p><strong>Do you think our data are a good fit for the model?</strong></p>
<ul>
<li>Ans:</li>
</ul>
</div>
</div>
<div id="negative-binomial-model-fitting-1" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Negative Binomial model fitting<a href="codebook.html#negative-binomial-model-fitting-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 05_DGE_DESeq2_analysis2.Rmd</strong></p>
<ol start="3" style="list-style-type: decimal">
<li><strong>Estimate the Beta Coefficients:</strong>
“DESeq2 estimates the <strong>beta coefficients</strong> by fitting a <strong>negative binomial GLM</strong> to the data using <strong>Maximum Likelihood Estimation (MLE)</strong>. DESeq2 substitutes the mean (<span class="math inline">\(\mu_{ij}\)</span>) with <span class="math inline">\(2^{\beta_i X_j}\)</span> in the negative binomial model, using MLE to predict expected counts based on experimental covariates and estimate gene expression changes while controlling for design factors.</li>
</ol>
<p>The model is fit using the <strong><code>DESeq()</code></strong> function in R, which carries out the entire estimation process:</p>
<div class="sourceCode" id="cb239"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb239-1"><a href="codebook.html#cb239-1" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## using pre-existing size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## found already estimated dispersions, replacing these</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<ol start="4" style="list-style-type: decimal">
<li><strong>Extract the Beta Coefficients:</strong>
Once the model is fitted, the estimated beta coefficients (log2 fold changes) for each gene can be extracted using the <strong><code>coef()</code></strong> function:</li>
</ol>
<div class="sourceCode" id="cb247"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb247-1"><a href="codebook.html#cb247-1" tabindex="-1"></a>beta_coefficients <span class="ot">&lt;-</span> <span class="fu">coef</span>(dds)</span></code></pre></div>
<p>This gives a matrix of estimated beta coefficients, where each row corresponds to a gene and each column corresponds to a covariate in the design matrix.</p>
<div id="example-2" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> Example:<a href="codebook.html#example-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For example, if the design matrix includes an intercept (control) and two conditions (MOV10 knockdown and MOV10 overexpression), the beta coefficients <span class="math inline">\(\beta_i\)</span> for gene <span class="math inline">\(i\)</span> would represent:</p>
<ul>
<li><p><span class="math inline">\(\beta_{i1}\)</span>: the log2 fold change for the control group (intercept)</p></li>
<li><p><span class="math inline">\(\beta_{i2}\)</span>: the log2 fold change for the MOV10 knockdown group</p></li>
<li><p><span class="math inline">\(\beta_{i3}\)</span>: the log2 fold change for the MOV10 overexpression group.</p></li>
</ul>
</div>
<div id="calculation-1" class="section level3 hasAnchor" number="9.5.2">
<h3><span class="header-section-number">9.5.2</span> Calculation:<a href="codebook.html#calculation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To calculate the fitted (expected) log2 counts for <strong>gene <span class="math inline">\(i\)</span></strong> in <strong>sample <span class="math inline">\(j\)</span></strong>, you can take the <strong>dot product</strong> of the row of the design matrix <span class="math inline">\(X_j\)</span> for that sample and the beta coefficients <span class="math inline">\(\beta_i\)</span> for that gene:</p>
<p><span class="math display">\[
\log_2(\mu_{ij}) = \beta_i X_j = \beta_{i1} X_{j1} + \beta_{i2} X_{j2}  + \dots + \beta_{iP} X_{jP}
\]</span></p>
<p>Where:</p>
<ul>
<li><span class="math inline">\(P\)</span> is the number of covariates in the design matrix</li>
</ul>
<p>The expected mean <span class="math inline">\(\mu_{ij}\)</span> on the original count scale is obtained by exponentiating the log2-scale fitted values:</p>
<p><span class="math display">\[
\mu_{ij} = 2^{\beta_i X_j}
\]</span></p>
<p><strong>Exercise</strong> points = +1</p>
<p>In the DESeq2 workflow, what is the purpose of the design matrix in the context of fitting the Negative Binomial model for each gene?</p>
<ul>
<li>Ans:</li>
</ul>
</div>
<div id="matrix-multiplication-form-1" class="section level3 hasAnchor" number="9.5.3">
<h3><span class="header-section-number">9.5.3</span> Matrix Multiplication Form:<a href="codebook.html#matrix-multiplication-form-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you want to calculate the fitted values for all genes and samples at once, you can express the model as a <strong>matrix multiplication</strong>:</p>
<p><span class="math display">\[
\log_2(\mu) = \beta X^T
\]</span></p>
<p>This computes the expected log2 counts for all genes across all samples.</p>
<p>In R, you can calculate the fitted log2 counts for all genes and samples using:</p>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="codebook.html#cb248-1" tabindex="-1"></a>log2_fitted <span class="ot">&lt;-</span> beta_coefficients <span class="sc">%*%</span> <span class="fu">t</span>(X)</span></code></pre></div>
<p>This step multiplies the beta coefficients matrix by the transpose of the design matrix to compute the fitted log2 counts.</p>
<p>STOP Tues Oct 22</p>
</div>
</div>
<div id="mov10-differential-expression-analysis-2" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> MOV10 Differential Expression Analysis:<a href="codebook.html#mov10-differential-expression-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Control versus Overexpression</p>
<p><strong>Building the Results Table</strong></p>
<p>To build the results table, we use the <code>results()</code> function. You can specify the <code>contrast</code> to be tested using the contrast argument. In this example, we’ll save the unshrunken and shrunken results of <strong>Control vs. Mov10 overexpression</strong> to different variables. We’ll also set the <code>alpha</code> to 0.05, which is more stringent than the default value of 0.1.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="codebook.html#cb249-1" tabindex="-1"></a><span class="co"># define contrasts</span></span>
<span id="cb249-2"><a href="codebook.html#cb249-2" tabindex="-1"></a>contrast_oe <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sampletype&quot;</span>, </span>
<span id="cb249-3"><a href="codebook.html#cb249-3" tabindex="-1"></a>                 <span class="st">&quot;MOV10_overexpression&quot;</span>,</span>
<span id="cb249-4"><a href="codebook.html#cb249-4" tabindex="-1"></a>                 <span class="st">&quot;control&quot;</span>)</span>
<span id="cb249-5"><a href="codebook.html#cb249-5" tabindex="-1"></a></span>
<span id="cb249-6"><a href="codebook.html#cb249-6" tabindex="-1"></a><span class="co"># extract results table</span></span>
<span id="cb249-7"><a href="codebook.html#cb249-7" tabindex="-1"></a>res_tableOE_unshrunken <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, </span>
<span id="cb249-8"><a href="codebook.html#cb249-8" tabindex="-1"></a>                          <span class="at">contrast =</span> contrast_oe, </span>
<span id="cb249-9"><a href="codebook.html#cb249-9" tabindex="-1"></a>                          <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb249-10"><a href="codebook.html#cb249-10" tabindex="-1"></a></span>
<span id="cb249-11"><a href="codebook.html#cb249-11" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                                 
## [2] &quot;sampletype_MOV10_knockdown_vs_control&quot;     
## [3] &quot;sampletype_MOV10_overexpression_vs_control&quot;</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="codebook.html#cb251-1" tabindex="-1"></a><span class="co"># shrink log2 fold changes</span></span>
<span id="cb251-2"><a href="codebook.html#cb251-2" tabindex="-1"></a>res_tableOE <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(<span class="at">dds =</span> dds, </span>
<span id="cb251-3"><a href="codebook.html#cb251-3" tabindex="-1"></a>  <span class="at">coef =</span>  <span class="st">&quot;sampletype_MOV10_overexpression_vs_control&quot;</span>, </span>
<span id="cb251-4"><a href="codebook.html#cb251-4" tabindex="-1"></a>  <span class="at">res =</span> res_tableOE_unshrunken)</span></code></pre></div>
<pre><code>## using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="codebook.html#cb253-1" tabindex="-1"></a><span class="co"># save the results for future use</span></span>
<span id="cb253-2"><a href="codebook.html#cb253-2" tabindex="-1"></a><span class="fu">saveRDS</span>(res_tableOE, <span class="at">file =</span> <span class="st">&quot;data/res_tableOE.RDS&quot;</span>)</span></code></pre></div>
<p><strong>MA Plot</strong></p>
<p>A <strong>MA plot</strong> visualizes the relationship between the <strong>mean expression</strong> (on the x-axis) and the <strong>log2 fold change</strong> (on the y-axis) for all genes in a dataset. Genes that are significantly differentially expressed are highlighted, while most other genes cluster around zero log2 fold change.</p>
<p>When <strong>shrinkage</strong> is applied, as in DESeq2, large log2 fold change estimates for genes with low counts or high variability are pulled toward more moderate values, reducing the noise in the plot. This helps focus attention on the more reliable fold changes and makes the plot cleaner, especially for low-expression genes that tend to have more variable LFC estimates without shrinkage.</p>
<p>This plot is also a great way to illustrate the effect of LFC shrinkage. DESeq2 provides a simple function to generate an MA plot.</p>
<p><strong>Shrunken &amp; Unshrunken Results:</strong></p>
<div class="sourceCode" id="cb254"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb254-1"><a href="codebook.html#cb254-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb254-2"><a href="codebook.html#cb254-2" tabindex="-1"></a></span>
<span id="cb254-3"><a href="codebook.html#cb254-3" tabindex="-1"></a><span class="fu">plotMA</span>(res_tableOE_unshrunken, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb254-4"><a href="codebook.html#cb254-4" tabindex="-1"></a></span>
<span id="cb254-5"><a href="codebook.html#cb254-5" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="dv">10</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>,<span class="at">lwd =</span> <span class="dv">2</span>)</span>
<span id="cb254-6"><a href="codebook.html#cb254-6" tabindex="-1"></a></span>
<span id="cb254-7"><a href="codebook.html#cb254-7" tabindex="-1"></a><span class="co"># Shrunken results:</span></span>
<span id="cb254-8"><a href="codebook.html#cb254-8" tabindex="-1"></a><span class="fu">plotMA</span>(res_tableOE, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb254-9"><a href="codebook.html#cb254-9" tabindex="-1"></a></span>
<span id="cb254-10"><a href="codebook.html#cb254-10" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="dv">10</span>,<span class="at">col=</span><span class="st">&quot;red&quot;</span>,<span class="at">lwd =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-144-1.png" width="672" /></p>
<p><strong>MOV10 DE Analysis: Exploring the Results</strong></p>
<p>The results table in <code>DESeq2</code> looks similar to a data.frame and can be treated like one for accessing or subsetting data. However, it is stored as a <code>DESeqResults</code> object, which is important to keep in mind when working with visualization tools.</p>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="codebook.html#cb255-1" tabindex="-1"></a><span class="fu">class</span>(res_tableOE)</span></code></pre></div>
<pre><code>## [1] &quot;DESeqResults&quot;
## attr(,&quot;package&quot;)
## [1] &quot;DESeq2&quot;</code></pre>
<p>Let’s go through some of the columns in the results table to get a better idea of what we are looking at. To extract information regarding the meaning of each column we can use <code>mcols()</code>:</p>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="codebook.html#cb257-1" tabindex="-1"></a><span class="fu">mcols</span>(res_tableOE, <span class="at">use.names =</span> T)</span></code></pre></div>
<pre><code>## DataFrame with 5 rows and 2 columns
##                        type            description
##                 &lt;character&gt;            &lt;character&gt;
## baseMean       intermediate mean of normalized c..
## log2FoldChange      results log2 fold change (MA..
## lfcSE               results posterior SD: sample..
## pvalue              results Wald test p-value: s..
## padj                results   BH adjusted p-values</code></pre>
<p>Now let’s take a look at what information is stored in the results:</p>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="codebook.html#cb259-1" tabindex="-1"></a><span class="fu">head</span>(res_tableOE)</span></code></pre></div>
<pre><code>## log2 fold change (MAP): sampletype MOV10_overexpression vs control 
## Wald test p-value: sampletype MOV10 overexpression vs control 
## DataFrame with 6 rows and 5 columns
##               baseMean log2FoldChange     lfcSE    pvalue      padj
##              &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## 1/2-SBSRNA4  45.652040     0.16858121  0.209275 0.1610752 0.2655661
## A1BG         61.093102     0.13623770  0.176713 0.2401909 0.3603094
## A1BG-AS1    175.665807    -0.04016953  0.116916 0.6789312 0.7748342
## A1CF          0.237692     0.00626568  0.210057 0.7945932        NA
## A2LD1        89.617985     0.29007260  0.195380 0.0333343 0.0741149
## A2M           5.860084    -0.09623512  0.233587 0.1057823 0.1900466</code></pre>
<div class="sourceCode" id="cb261"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb261-1"><a href="codebook.html#cb261-1" tabindex="-1"></a><span class="fu">names</span>(res_tableOE)</span></code></pre></div>
<pre><code>## [1] &quot;baseMean&quot;       &quot;log2FoldChange&quot; &quot;lfcSE&quot;          &quot;pvalue&quot;        
## [5] &quot;padj&quot;</code></pre>
<div id="mov10-de-analysis-control-vs.-knockdown-1" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> MOV10 DE Analysis: <strong>Control vs. Knockdown</strong><a href="codebook.html#mov10-de-analysis-control-vs.-knockdown-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After examining the overexpression results, let’s move on to the comparison between <strong>Control vs. Knockdown</strong>. We’ll use contrasts in the <code>results()</code> function to extract the results table and store it in the <code>res_tableKD</code> variable. You can also use <code>coef</code> to specify the contrast directly in the <code>lfcShrink()</code> function.</p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="codebook.html#cb263-1" tabindex="-1"></a><span class="co"># define contrast</span></span>
<span id="cb263-2"><a href="codebook.html#cb263-2" tabindex="-1"></a>contrast_kd <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">&quot;sampletype&quot;</span>, </span>
<span id="cb263-3"><a href="codebook.html#cb263-3" tabindex="-1"></a>                  <span class="st">&quot;MOV10_knockdown&quot;</span>,</span>
<span id="cb263-4"><a href="codebook.html#cb263-4" tabindex="-1"></a>                  <span class="st">&quot;control&quot;</span>)</span>
<span id="cb263-5"><a href="codebook.html#cb263-5" tabindex="-1"></a></span>
<span id="cb263-6"><a href="codebook.html#cb263-6" tabindex="-1"></a><span class="co"># extract results table</span></span>
<span id="cb263-7"><a href="codebook.html#cb263-7" tabindex="-1"></a>res_tableKD_unshrunken <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, </span>
<span id="cb263-8"><a href="codebook.html#cb263-8" tabindex="-1"></a>                                  <span class="at">contrast =</span> contrast_kd, </span>
<span id="cb263-9"><a href="codebook.html#cb263-9" tabindex="-1"></a>                                  <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb263-10"><a href="codebook.html#cb263-10" tabindex="-1"></a></span>
<span id="cb263-11"><a href="codebook.html#cb263-11" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span></code></pre></div>
<pre><code>## [1] &quot;Intercept&quot;                                 
## [2] &quot;sampletype_MOV10_knockdown_vs_control&quot;     
## [3] &quot;sampletype_MOV10_overexpression_vs_control&quot;</code></pre>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="codebook.html#cb265-1" tabindex="-1"></a><span class="co"># use the `coef` argument to specify the contrast directly</span></span>
<span id="cb265-2"><a href="codebook.html#cb265-2" tabindex="-1"></a>res_tableKD <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef =</span> <span class="dv">2</span>, </span>
<span id="cb265-3"><a href="codebook.html#cb265-3" tabindex="-1"></a>                         <span class="at">res =</span> res_tableKD_unshrunken)</span></code></pre></div>
<pre><code>## using &#39;apeglm&#39; for LFC shrinkage. If used in published research, please cite:
##     Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
##     sequence count data: removing the noise and preserving large differences.
##     Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
<div class="sourceCode" id="cb267"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb267-1"><a href="codebook.html#cb267-1" tabindex="-1"></a><span class="do">## Save results for future use</span></span>
<span id="cb267-2"><a href="codebook.html#cb267-2" tabindex="-1"></a><span class="fu">saveRDS</span>(res_tableKD, <span class="at">file =</span> <span class="st">&quot;data/res_tableKD.RDS&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="summarizing-results-1" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> Summarizing Results<a href="codebook.html#summarizing-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To summarize the results, DESeq2 offers the <code>summary()</code> function, which conveniently reports the number of genes that are significantly differentially expressed at a specified threshold (<code>default FDR &lt; 0.05</code>). Note that, even though the output refers to p-values, it actually summarizes the results using <strong>adjusted p-values (padj/FDR)</strong>.</p>
<p>Let’s start by summarizing the results for the OE vs. control comparison:</p>
<div class="sourceCode" id="cb268"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb268-1"><a href="codebook.html#cb268-1" tabindex="-1"></a><span class="do">## Summarize results</span></span>
<span id="cb268-2"><a href="codebook.html#cb268-2" tabindex="-1"></a><span class="fu">summary</span>(res_tableOE)</span></code></pre></div>
<pre><code>## 
## out of 19748 with nonzero total read count
## adjusted p-value &lt; 0.05
## LFC &gt; 0 (up)       : 3103, 16%
## LFC &lt; 0 (down)     : 3408, 17%
## outliers [1]       : 0, 0%
## low counts [2]     : 4171, 21%
## (mean count &lt; 5)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<p>In addition to reporting the number of up- and down-regulated genes at the default significance threshold, this function also provides information on:</p>
<ul>
<li><strong>Number of genes tested</strong> (genes with non-zero total read count)</li>
<li><strong>Number of genes excluded</strong> from multiple test correction due to low mean counts</li>
</ul>
<div id="extracting-significant-differentially-expressed-genes-1" class="section level3 hasAnchor" number="9.7.1">
<h3><span class="header-section-number">9.7.1</span> Extracting Significant Differentially Expressed Genes<a href="codebook.html#extracting-significant-differentially-expressed-genes-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In some cases, using only the FDR threshold doesn’t sufficiently reduce the number of significant genes, making it difficult to extract biologically meaningful results. To increase stringency, we can apply an additional fold change threshold.</p>
<p>Although the summary() function doesn’t include an argument for fold change thresholds, we can define our own criteria.</p>
<p>Let’s start by setting the thresholds for both adjusted p-value (FDR &lt; 0.05) and log2 fold change (|log2FC| &gt; 0.58, corresponding to a 1.5-fold change):</p>
<div class="sourceCode" id="cb270"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb270-1"><a href="codebook.html#cb270-1" tabindex="-1"></a><span class="do">### Set thresholds</span></span>
<span id="cb270-2"><a href="codebook.html#cb270-2" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb270-3"><a href="codebook.html#cb270-3" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fl">0.58</span></span></code></pre></div>
<p>Next, we’ll convert the results table to a tibble for easier subsetting:</p>
<div class="sourceCode" id="cb271"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb271-1"><a href="codebook.html#cb271-1" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span></span>
<span id="cb271-2"><a href="codebook.html#cb271-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb271-3"><a href="codebook.html#cb271-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb271-4"><a href="codebook.html#cb271-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>Now, we can filter the table to retain only the genes that meet the significance and fold change criteria:</p>
<div class="sourceCode" id="cb272"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb272-1"><a href="codebook.html#cb272-1" tabindex="-1"></a>sigOE <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span></span>
<span id="cb272-2"><a href="codebook.html#cb272-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span> </span>
<span id="cb272-3"><a href="codebook.html#cb272-3" tabindex="-1"></a>                <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span>
<span id="cb272-4"><a href="codebook.html#cb272-4" tabindex="-1"></a></span>
<span id="cb272-5"><a href="codebook.html#cb272-5" tabindex="-1"></a><span class="co"># Save the results for future use</span></span>
<span id="cb272-6"><a href="codebook.html#cb272-6" tabindex="-1"></a><span class="fu">saveRDS</span>(sigOE,<span class="st">&quot;data/sigOE.RDS&quot;</span>)</span></code></pre></div>
<p><strong>Exercise</strong> points = +2</p>
<p>How many genes are differentially expressed in the <strong>Overexpression vs. Control</strong> comparison based on the criteria we just defined? Does this reduce the number of significant genes compared to using only the FDR threshold?</p>
<div class="sourceCode" id="cb273"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb273-1"><a href="codebook.html#cb273-1" tabindex="-1"></a><span class="co"># Your code here</span></span></code></pre></div>
<p><strong>Does this reduce our results?</strong></p>
<div class="sourceCode" id="cb274"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb274-1"><a href="codebook.html#cb274-1" tabindex="-1"></a><span class="co"># Your code here</span></span></code></pre></div>
</div>
<div id="mov10-knockdown-analysis-control-vs.-knockdown-1" class="section level3 hasAnchor" number="9.7.2">
<h3><span class="header-section-number">9.7.2</span> MOV10 Knockdown Analysis: Control vs. Knockdown<a href="codebook.html#mov10-knockdown-analysis-control-vs.-knockdown-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Next, let’s perform the same analysis for the <strong>Control vs. Mov10 knockdown</strong> comparison. We’ll use the same thresholds for adjusted p-value (FDR &lt; 0.05) and log2 fold change (|log2FC| &gt; 0.58).</p>
<div class="sourceCode" id="cb275"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb275-1"><a href="codebook.html#cb275-1" tabindex="-1"></a>res_tableKD_tb <span class="ot">&lt;-</span> res_tableKD <span class="sc">%&gt;%</span></span>
<span id="cb275-2"><a href="codebook.html#cb275-2" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb275-3"><a href="codebook.html#cb275-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb275-4"><a href="codebook.html#cb275-4" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb275-5"><a href="codebook.html#cb275-5" tabindex="-1"></a></span>
<span id="cb275-6"><a href="codebook.html#cb275-6" tabindex="-1"></a>sigKD <span class="ot">&lt;-</span> res_tableKD_tb <span class="sc">%&gt;%</span></span>
<span id="cb275-7"><a href="codebook.html#cb275-7" tabindex="-1"></a>         dplyr<span class="sc">::</span><span class="fu">filter</span>(padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span> </span>
<span id="cb275-8"><a href="codebook.html#cb275-8" tabindex="-1"></a>         <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span>
<span id="cb275-9"><a href="codebook.html#cb275-9" tabindex="-1"></a></span>
<span id="cb275-10"><a href="codebook.html#cb275-10" tabindex="-1"></a><span class="co"># We&#39;ll save this object for use in the homework</span></span>
<span id="cb275-11"><a href="codebook.html#cb275-11" tabindex="-1"></a><span class="fu">saveRDS</span>(sigKD,<span class="st">&quot;data/sigKD.RDS&quot;</span>)</span></code></pre></div>
<p><strong>How many genes are differentially expressed in the Knockdown compared to Control?</strong></p>
<div class="sourceCode" id="cb276"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb276-1"><a href="codebook.html#cb276-1" tabindex="-1"></a><span class="co"># Your code here</span></span></code></pre></div>
</div>
</div>
<div id="visualizing-rna-seq-results-1" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> Visualizing RNA-seq results<a href="codebook.html#visualizing-rna-seq-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 06_DGE_visualizing_results.Rmd</strong></p>
<p>Let’s start by loading a few libraries (if not already loaded):</p>
<div class="sourceCode" id="cb277"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb277-1"><a href="codebook.html#cb277-1" tabindex="-1"></a><span class="co"># load libraries</span></span>
<span id="cb277-2"><a href="codebook.html#cb277-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb277-3"><a href="codebook.html#cb277-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb277-4"><a href="codebook.html#cb277-4" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb277-5"><a href="codebook.html#cb277-5" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb277-6"><a href="codebook.html#cb277-6" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb277-7"><a href="codebook.html#cb277-7" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb277-8"><a href="codebook.html#cb277-8" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<p>When we are working with large amounts of data it can be useful to display that information graphically to gain more insight.</p>
<p>Let’s create tibble objects from the meta and normalized_counts data frames before we start plotting. This will enable us to use the tidyverse functionality more easily.</p>
<div class="sourceCode" id="cb278"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb278-1"><a href="codebook.html#cb278-1" tabindex="-1"></a><span class="co"># Read in the metadata</span></span>
<span id="cb278-2"><a href="codebook.html#cb278-2" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/Mov10_full_meta.txt&quot;</span>, </span>
<span id="cb278-3"><a href="codebook.html#cb278-3" tabindex="-1"></a>                   <span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb278-4"><a href="codebook.html#cb278-4" tabindex="-1"></a></span>
<span id="cb278-5"><a href="codebook.html#cb278-5" tabindex="-1"></a><span class="co"># Create a tibble for meta data</span></span>
<span id="cb278-6"><a href="codebook.html#cb278-6" tabindex="-1"></a>mov10_meta <span class="ot">&lt;-</span> meta <span class="sc">%&gt;%</span> </span>
<span id="cb278-7"><a href="codebook.html#cb278-7" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;samplename&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb278-8"><a href="codebook.html#cb278-8" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb278-9"><a href="codebook.html#cb278-9" tabindex="-1"></a>        </span>
<span id="cb278-10"><a href="codebook.html#cb278-10" tabindex="-1"></a><span class="co"># read in the normalized counts</span></span>
<span id="cb278-11"><a href="codebook.html#cb278-11" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">&quot;data/normalized_counts.txt&quot;</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb278-12"><a href="codebook.html#cb278-12" tabindex="-1"></a></span>
<span id="cb278-13"><a href="codebook.html#cb278-13" tabindex="-1"></a><span class="fu">all</span>(mov10_meta<span class="sc">$</span>samplename <span class="sc">==</span> <span class="fu">colnames</span>(normalized_counts))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb280"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb280-1"><a href="codebook.html#cb280-1" tabindex="-1"></a><span class="co"># Create a tibble for normalized_counts</span></span>
<span id="cb280-2"><a href="codebook.html#cb280-2" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> normalized_counts <span class="sc">%&gt;%</span> </span>
<span id="cb280-3"><a href="codebook.html#cb280-3" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb280-4"><a href="codebook.html#cb280-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb280-5"><a href="codebook.html#cb280-5" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>One way to visualize results would be to simply plot the expression data for a handful of genes. We could do that by picking out specific genes of interest or selecting a range of genes.</p>
<div id="using-deseq2-plotcounts-to-plot-expression-of-a-single-gene-1" class="section level4 hasAnchor" number="9.8.0.1">
<h4><span class="header-section-number">9.8.0.1</span> <strong>Using DESeq2 <code>plotCounts()</code> to plot expression of a single gene</strong><a href="codebook.html#using-deseq2-plotcounts-to-plot-expression-of-a-single-gene-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To pick out a specific gene of interest to plot, for example Mov10, we can use the <code>plotCounts()</code> from DESeq2. <code>plotCounts()</code> is a function that allows us to plot the normalized counts for a single gene across all samples.</p>
<div class="sourceCode" id="cb281"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb281-1"><a href="codebook.html#cb281-1" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, </span>
<span id="cb281-2"><a href="codebook.html#cb281-2" tabindex="-1"></a>           <span class="at">gene=</span><span class="st">&quot;MOV10&quot;</span>, </span>
<span id="cb281-3"><a href="codebook.html#cb281-3" tabindex="-1"></a>           <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>) </span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-159-1.png" width="672" /></p>
<div class="sourceCode" id="cb282"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb282-1"><a href="codebook.html#cb282-1" tabindex="-1"></a><span class="co"># We can also color the points by sample type</span></span>
<span id="cb282-2"><a href="codebook.html#cb282-2" tabindex="-1"></a>sampletype <span class="ot">=</span> <span class="fu">as.factor</span>(mov10_meta<span class="sc">$</span>sampletype)</span>
<span id="cb282-3"><a href="codebook.html#cb282-3" tabindex="-1"></a></span>
<span id="cb282-4"><a href="codebook.html#cb282-4" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb282-5"><a href="codebook.html#cb282-5" tabindex="-1"></a></span>
<span id="cb282-6"><a href="codebook.html#cb282-6" tabindex="-1"></a><span class="fu">display.brewer.all</span>()</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-159-2.png" width="672" /></p>
<div class="sourceCode" id="cb283"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb283-1"><a href="codebook.html#cb283-1" tabindex="-1"></a>col <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">8</span>,<span class="st">&quot;Dark2&quot;</span>)</span>
<span id="cb283-2"><a href="codebook.html#cb283-2" tabindex="-1"></a><span class="fu">palette</span>(col)</span>
<span id="cb283-3"><a href="codebook.html#cb283-3" tabindex="-1"></a></span>
<span id="cb283-4"><a href="codebook.html#cb283-4" tabindex="-1"></a><span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">&quot;MOV10&quot;</span>, </span>
<span id="cb283-5"><a href="codebook.html#cb283-5" tabindex="-1"></a>           <span class="at">intgroup=</span><span class="st">&quot;sampletype&quot;</span>,</span>
<span id="cb283-6"><a href="codebook.html#cb283-6" tabindex="-1"></a>           <span class="at">col =</span> <span class="fu">as.numeric</span>(sampletype),</span>
<span id="cb283-7"><a href="codebook.html#cb283-7" tabindex="-1"></a>           <span class="at">pch =</span> <span class="dv">19</span>) </span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-159-3.png" width="672" /></p>
<p><strong>This function only allows for plotting the counts of a single gene at a time.</strong></p>
</div>
<div id="using-ggplot2-to-plot-expression-of-a-single-gene-1" class="section level4 hasAnchor" number="9.8.0.2">
<h4><span class="header-section-number">9.8.0.2</span> <strong>Using ggplot2 to plot expression of a single gene</strong><a href="codebook.html#using-ggplot2-to-plot-expression-of-a-single-gene-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We can also use ggplot2 to plot the <code>MOV10</code> counts. We can save the output of <code>plotCounts()</code> to a variable specifying the <code>returnData=TRUE</code> argument. This will save the normalized counts for the gene <code>MOV10</code> to a data frame object. We can then use ggplot2 to plot the normalized counts for <code>MOV10</code> across all samples.</p>
<div class="sourceCode" id="cb284"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb284-1"><a href="codebook.html#cb284-1" tabindex="-1"></a><span class="co"># Save `plotCounts()` to a data frame object</span></span>
<span id="cb284-2"><a href="codebook.html#cb284-2" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene =</span> <span class="st">&quot;MOV10&quot;</span>, <span class="at">intgroup =</span> <span class="st">&quot;sampletype&quot;</span>, </span>
<span id="cb284-3"><a href="codebook.html#cb284-3" tabindex="-1"></a>                <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb284-4"><a href="codebook.html#cb284-4" tabindex="-1"></a></span>
<span id="cb284-5"><a href="codebook.html#cb284-5" tabindex="-1"></a><span class="co"># Plot using ggplot2</span></span>
<span id="cb284-6"><a href="codebook.html#cb284-6" tabindex="-1"></a><span class="fu">ggplot</span>(d, <span class="fu">aes</span>(<span class="at">x =</span> sampletype, <span class="at">y =</span> count, <span class="at">color =</span> sampletype)) <span class="sc">+</span> </span>
<span id="cb284-7"><a href="codebook.html#cb284-7" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="at">w =</span> <span class="fl">0.1</span>,<span class="at">h =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb284-8"><a href="codebook.html#cb284-8" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">rownames</span>(d))) <span class="sc">+</span> </span>
<span id="cb284-9"><a href="codebook.html#cb284-9" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb284-10"><a href="codebook.html#cb284-10" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;MOV10&quot;</span>) <span class="sc">+</span></span>
<span id="cb284-11"><a href="codebook.html#cb284-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-160-1.png" width="672" /></p>
<blockquote>
<p>Note that in the plot below (code above), we are using <code>geom_text_repel()</code> from the <code>ggrepel</code> package to label our individual points on the plot.</p>
</blockquote>
</div>
<div id="using-ggplot2-to-plot-multiple-genes-e.g.-top-20-1" class="section level4 hasAnchor" number="9.8.0.3">
<h4><span class="header-section-number">9.8.0.3</span> Using <code>ggplot2</code> to plot multiple genes (e.g. top 20)<a href="codebook.html#using-ggplot2-to-plot-multiple-genes-e.g.-top-20-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Often it is helpful to check the expression of multiple genes of interest at the same time. This often first requires some data wrangling.</p>
<p>We are going to plot the normalized count values for the <strong>top 20 differentially expressed genes (by padj values)</strong>.</p>
<p>To do this, we first need to determine the gene names of our top 20 genes by ordering our results and extracting the top 20 genes (by padj values):</p>
<div class="sourceCode" id="cb285"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb285-1"><a href="codebook.html#cb285-1" tabindex="-1"></a>res_tableOE <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/res_tableOE.RDS&quot;</span>)</span>
<span id="cb285-2"><a href="codebook.html#cb285-2" tabindex="-1"></a></span>
<span id="cb285-3"><a href="codebook.html#cb285-3" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span></span>
<span id="cb285-4"><a href="codebook.html#cb285-4" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb285-5"><a href="codebook.html#cb285-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb285-6"><a href="codebook.html#cb285-6" tabindex="-1"></a>  <span class="fu">as_tibble</span>()</span>
<span id="cb285-7"><a href="codebook.html#cb285-7" tabindex="-1"></a></span>
<span id="cb285-8"><a href="codebook.html#cb285-8" tabindex="-1"></a><span class="do">## Order results by padj values</span></span>
<span id="cb285-9"><a href="codebook.html#cb285-9" tabindex="-1"></a>top20_sigOE_genes <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> </span>
<span id="cb285-10"><a href="codebook.html#cb285-10" tabindex="-1"></a>        <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span>   </span>
<span id="cb285-11"><a href="codebook.html#cb285-11" tabindex="-1"></a><span class="co">#Arrange rows by padj values</span></span>
<span id="cb285-12"><a href="codebook.html#cb285-12" tabindex="-1"></a>        <span class="fu">pull</span>(gene) <span class="sc">%&gt;%</span>      </span>
<span id="cb285-13"><a href="codebook.html#cb285-13" tabindex="-1"></a><span class="co">#Extract character vector of ordered genes</span></span>
<span id="cb285-14"><a href="codebook.html#cb285-14" tabindex="-1"></a>        <span class="fu">head</span>(<span class="at">n=</span><span class="dv">20</span>)      </span>
<span id="cb285-15"><a href="codebook.html#cb285-15" tabindex="-1"></a><span class="co">#Extract the first 20 genes</span></span></code></pre></div>
<p>Then, we can extract the normalized count values for these top 20 genes:</p>
<div class="sourceCode" id="cb286"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb286-1"><a href="codebook.html#cb286-1" tabindex="-1"></a><span class="do">## Normalized counts for top 20 significant genes</span></span>
<span id="cb286-2"><a href="codebook.html#cb286-2" tabindex="-1"></a>top20_sigOE_norm <span class="ot">&lt;-</span> normalized_counts <span class="sc">%&gt;%</span></span>
<span id="cb286-3"><a href="codebook.html#cb286-3" tabindex="-1"></a>        <span class="fu">filter</span>(gene <span class="sc">%in%</span> top20_sigOE_genes)</span></code></pre></div>
<p>Now that we have the normalized counts for each of the top 20 genes for all 8 samples, to plot using <code>ggplot()</code>, we need to <code>pivot_longer</code> top20_sigOE_norm from a wide format to a long format so the counts for all samples will be in a single column to allow us to give ggplot the one column with the values we want it to plot.</p>
<p>The <code>pivot_longer()</code> function in the <strong>tidyr</strong> package will perform this operation and will output the normalized counts for all genes for <em>Mov10_oe_1</em> listed in the first 20 rows, followed by the normalized counts for <em>Mov10_oe_2</em> in the next 20 rows, so on and so forth.</p>
<p><img src="img/melt_wide_to_long_format.png" width="800" alt="Alt text" /></p>
<div class="sourceCode" id="cb287"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb287-1"><a href="codebook.html#cb287-1" tabindex="-1"></a><span class="co"># Pivot the data frame</span></span>
<span id="cb287-2"><a href="codebook.html#cb287-2" tabindex="-1"></a>pivoted_top20_sigOE <span class="ot">&lt;-</span> top20_sigOE_norm <span class="sc">%&gt;%</span></span>
<span id="cb287-3"><a href="codebook.html#cb287-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="fu">colnames</span>(top20_sigOE_norm)[<span class="dv">2</span><span class="sc">:</span><span class="dv">9</span>], </span>
<span id="cb287-4"><a href="codebook.html#cb287-4" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;samplename&quot;</span>, </span>
<span id="cb287-5"><a href="codebook.html#cb287-5" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;normalized_counts&quot;</span>)</span>
<span id="cb287-6"><a href="codebook.html#cb287-6" tabindex="-1"></a></span>
<span id="cb287-7"><a href="codebook.html#cb287-7" tabindex="-1"></a><span class="do">## Check the column header in the &quot;pivoted&quot; data frame</span></span>
<span id="cb287-8"><a href="codebook.html#cb287-8" tabindex="-1"></a><span class="fu">head</span>(pivoted_top20_sigOE)</span></code></pre></div>
<div class="kable-table">
<table>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="left">samplename</th>
<th align="right">normalized_counts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_1</td>
<td align="right">6717.735</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_2</td>
<td align="right">6454.641</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Irrel_kd_3</td>
<td align="right">6801.543</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_kd_2</td>
<td align="right">6869.168</td>
</tr>
<tr class="odd">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_kd_3</td>
<td align="right">8730.977</td>
</tr>
<tr class="even">
<td align="left">ADAMTS1</td>
<td align="left">Mov10_oe_1</td>
<td align="right">13252.239</td>
</tr>
</tbody>
</table>
</div>
<p>Now, if we want our counts colored by sample group, then we need to combine the metadata information with the melted normalized counts data into the same data frame for input to <code>ggplot()</code>:</p>
<p><code>inner_join(x,y)</code> will merge 2 data frames by the colname in x that matches a column name in y in this case <code>samplename</code> column.</p>
<div class="sourceCode" id="cb288"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb288-1"><a href="codebook.html#cb288-1" tabindex="-1"></a>pivoted_top20_sigOE <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb288-2"><a href="codebook.html#cb288-2" tabindex="-1"></a>  mov10_meta, </span>
<span id="cb288-3"><a href="codebook.html#cb288-3" tabindex="-1"></a>  pivoted_top20_sigOE)</span></code></pre></div>
<pre><code>## Joining with `by = join_by(samplename)`</code></pre>
<p>Now that we have a data frame in a format that can be utilised by ggplot easily, let’s plot!</p>
<div class="sourceCode" id="cb290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb290-1"><a href="codebook.html#cb290-1" tabindex="-1"></a><span class="do">## plot using ggplot2</span></span>
<span id="cb290-2"><a href="codebook.html#cb290-2" tabindex="-1"></a><span class="fu">ggplot</span>(pivoted_top20_sigOE) <span class="sc">+</span></span>
<span id="cb290-3"><a href="codebook.html#cb290-3" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> gene, <span class="at">y =</span> normalized_counts, </span>
<span id="cb290-4"><a href="codebook.html#cb290-4" tabindex="-1"></a>          <span class="at">color =</span> sampletype)) <span class="sc">+</span></span>
<span id="cb290-5"><a href="codebook.html#cb290-5" tabindex="-1"></a>        <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb290-6"><a href="codebook.html#cb290-6" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb290-7"><a href="codebook.html#cb290-7" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;log10 Normalized Counts&quot;</span>) <span class="sc">+</span></span>
<span id="cb290-8"><a href="codebook.html#cb290-8" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Top 20 Significant DE Genes&quot;</span>) <span class="sc">+</span></span>
<span id="cb290-9"><a href="codebook.html#cb290-9" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb290-10"><a href="codebook.html#cb290-10" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, </span>
<span id="cb290-11"><a href="codebook.html#cb290-11" tabindex="-1"></a>                                         <span class="at">hjust =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb290-12"><a href="codebook.html#cb290-12" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-165-1.png" width="672" /></p>
</div>
<div id="heatmap-1" class="section level3 hasAnchor" number="9.8.1">
<h3><span class="header-section-number">9.8.1</span> Heatmap<a href="codebook.html#heatmap-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will plot a heatmap of the <code>sigOE</code> genes to visualize their expression across all samples using <code>pheatmap()</code>.</p>
<div class="sourceCode" id="cb291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb291-1"><a href="codebook.html#cb291-1" tabindex="-1"></a>sigOE <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/sigOE.RDS&quot;</span>)</span>
<span id="cb291-2"><a href="codebook.html#cb291-2" tabindex="-1"></a></span>
<span id="cb291-3"><a href="codebook.html#cb291-3" tabindex="-1"></a>norm_OEsig <span class="ot">&lt;-</span> normalized_counts[,<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>)] <span class="sc">%&gt;%</span> </span>
<span id="cb291-4"><a href="codebook.html#cb291-4" tabindex="-1"></a>              <span class="fu">filter</span>(gene <span class="sc">%in%</span> sigOE<span class="sc">$</span>gene) <span class="sc">%&gt;%</span> </span>
<span id="cb291-5"><a href="codebook.html#cb291-5" tabindex="-1"></a>              <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb291-6"><a href="codebook.html#cb291-6" tabindex="-1"></a>              <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) </span></code></pre></div>
<p>Now let’s draw the heatmap using pheatmap. We can also add annotations to the heatmap to indicate the sample type. The <code>annotation_col</code> argument in <code>pheatmap()</code>requires a data frame with the same row names as the column names of the matrix used to generate the heatmap.</p>
<div class="sourceCode" id="cb292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb292-1"><a href="codebook.html#cb292-1" tabindex="-1"></a><span class="do">### Annotate our heatmap (optional)</span></span>
<span id="cb292-2"><a href="codebook.html#cb292-2" tabindex="-1"></a></span>
<span id="cb292-3"><a href="codebook.html#cb292-3" tabindex="-1"></a>annotation <span class="ot">&lt;-</span> mov10_meta <span class="sc">%&gt;%</span> </span>
<span id="cb292-4"><a href="codebook.html#cb292-4" tabindex="-1"></a>  <span class="fu">filter</span>(samplename <span class="sc">%in%</span> <span class="fu">colnames</span>(norm_OEsig)) <span class="sc">%&gt;%</span></span>
<span id="cb292-5"><a href="codebook.html#cb292-5" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(samplename, sampletype) <span class="sc">%&gt;%</span> </span>
<span id="cb292-6"><a href="codebook.html#cb292-6" tabindex="-1"></a>    <span class="fu">data.frame</span>(<span class="at">row.names =</span> <span class="st">&quot;samplename&quot;</span>)</span>
<span id="cb292-7"><a href="codebook.html#cb292-7" tabindex="-1"></a></span>
<span id="cb292-8"><a href="codebook.html#cb292-8" tabindex="-1"></a>annotation<span class="sc">$</span>sampletype <span class="ot">=</span> <span class="fu">factor</span>(annotation<span class="sc">$</span>sampletype)</span>
<span id="cb292-9"><a href="codebook.html#cb292-9" tabindex="-1"></a></span>
<span id="cb292-10"><a href="codebook.html#cb292-10" tabindex="-1"></a><span class="do">### Set a color palette</span></span>
<span id="cb292-11"><a href="codebook.html#cb292-11" tabindex="-1"></a>heat_colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">6</span>, <span class="st">&quot;YlOrRd&quot;</span>)</span>
<span id="cb292-12"><a href="codebook.html#cb292-12" tabindex="-1"></a></span>
<span id="cb292-13"><a href="codebook.html#cb292-13" tabindex="-1"></a><span class="do">### Set annotation colors</span></span>
<span id="cb292-14"><a href="codebook.html#cb292-14" tabindex="-1"></a>ann_colors <span class="ot">=</span> <span class="fu">list</span>(</span>
<span id="cb292-15"><a href="codebook.html#cb292-15" tabindex="-1"></a>  <span class="at">sampletype =</span> <span class="fu">c</span>(<span class="at">control=</span><span class="st">&quot;navy&quot;</span>,</span>
<span id="cb292-16"><a href="codebook.html#cb292-16" tabindex="-1"></a>  <span class="at">MOV10_overexpression=</span> <span class="st">&quot;green&quot;</span>))</span>
<span id="cb292-17"><a href="codebook.html#cb292-17" tabindex="-1"></a></span>
<span id="cb292-18"><a href="codebook.html#cb292-18" tabindex="-1"></a><span class="do">### Run pheatmap</span></span>
<span id="cb292-19"><a href="codebook.html#cb292-19" tabindex="-1"></a><span class="fu">pheatmap</span>(norm_OEsig, </span>
<span id="cb292-20"><a href="codebook.html#cb292-20" tabindex="-1"></a>         <span class="at">annotation_colors =</span> ann_colors, </span>
<span id="cb292-21"><a href="codebook.html#cb292-21" tabindex="-1"></a>         <span class="at">color =</span> heat_colors, </span>
<span id="cb292-22"><a href="codebook.html#cb292-22" tabindex="-1"></a>         <span class="at">cluster_rows =</span> T, </span>
<span id="cb292-23"><a href="codebook.html#cb292-23" tabindex="-1"></a>         <span class="at">show_rownames =</span> F,</span>
<span id="cb292-24"><a href="codebook.html#cb292-24" tabindex="-1"></a>         <span class="at">annotation_col =</span> annotation, </span>
<span id="cb292-25"><a href="codebook.html#cb292-25" tabindex="-1"></a>         <span class="at">border_color =</span> <span class="cn">NA</span>, </span>
<span id="cb292-26"><a href="codebook.html#cb292-26" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">12</span>, </span>
<span id="cb292-27"><a href="codebook.html#cb292-27" tabindex="-1"></a>         <span class="at">scale =</span> <span class="st">&quot;row&quot;</span>, </span>
<span id="cb292-28"><a href="codebook.html#cb292-28" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">12</span>)</span></code></pre></div>
<blockquote>
<p><em>NOTE:</em> There are several additional arguments we have included in the function for aesthetics. One important one is <code>scale="row"</code>, in which Z-scores are plotted, rather than the actual normalized count value.</p>
</blockquote>
</div>
<div id="volcano-plots-1" class="section level3 hasAnchor" number="9.8.2">
<h3><span class="header-section-number">9.8.2</span> Volcano plots<a href="codebook.html#volcano-plots-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>A commonly used plot in RNA-seq analysis is the <strong>volcano plot</strong>. in which you have the log transformed adjusted p-values plotted on the y-axis and log2 fold change values on the x-axis.</p>
<p>To generate a volcano plot, we first need to have a column in our results data indicating whether or not the gene is considered differentially expressed based on p-adjusted values.</p>
<div class="sourceCode" id="cb293"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb293-1"><a href="codebook.html#cb293-1" tabindex="-1"></a><span class="do">## create a column for thresholding in the results table</span></span>
<span id="cb293-2"><a href="codebook.html#cb293-2" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> </span>
<span id="cb293-3"><a href="codebook.html#cb293-3" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">threshold_OE =</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> </span>
<span id="cb293-4"><a href="codebook.html#cb293-4" tabindex="-1"></a>                  <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;=</span> <span class="fl">0.58</span>)</span></code></pre></div>
<p>Now we can start plotting. The <code>geom_point</code> object is most applicable, as this is essentially a scatter plot:</p>
<div class="sourceCode" id="cb294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb294-1"><a href="codebook.html#cb294-1" tabindex="-1"></a><span class="fu">ggplot</span>(res_tableOE_tb) <span class="sc">+</span></span>
<span id="cb294-2"><a href="codebook.html#cb294-2" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, </span>
<span id="cb294-3"><a href="codebook.html#cb294-3" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">colour =</span> threshold_OE)) <span class="sc">+</span></span>
<span id="cb294-4"><a href="codebook.html#cb294-4" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Mov10 overexpression&quot;</span>) <span class="sc">+</span></span>
<span id="cb294-5"><a href="codebook.html#cb294-5" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;log2 fold change&quot;</span>) <span class="sc">+</span> </span>
<span id="cb294-6"><a href="codebook.html#cb294-6" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;-log10 adjusted p-value&quot;</span>) <span class="sc">+</span></span>
<span id="cb294-7"><a href="codebook.html#cb294-7" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span>      </span>
<span id="cb294-8"><a href="codebook.html#cb294-8" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb294-9"><a href="codebook.html#cb294-9" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.5</span>), </span>
<span id="cb294-10"><a href="codebook.html#cb294-10" tabindex="-1"></a>                                  <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb294-11"><a href="codebook.html#cb294-11" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.25</span>)))  </span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-169-1.png" width="672" /></p>
<p>What if we also wanted to know where the top 10 genes (lowest padj) in our DE list are located on this plot? We could label those dots with the gene name on the Volcano plot using <code>geom_text_repel()</code>.</p>
<p>First, we need to order the res_tableOE tibble by <code>padj</code>, and add an additional column to it, to include on those gene names we want to use to label the plot.</p>
<div class="sourceCode" id="cb295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb295-1"><a href="codebook.html#cb295-1" tabindex="-1"></a><span class="do">## Create a column to indicate which genes to label</span></span>
<span id="cb295-2"><a href="codebook.html#cb295-2" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE_tb <span class="sc">%&gt;%</span> </span>
<span id="cb295-3"><a href="codebook.html#cb295-3" tabindex="-1"></a>  <span class="fu">arrange</span>(padj) <span class="sc">%&gt;%</span> </span>
<span id="cb295-4"><a href="codebook.html#cb295-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">genelabels =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb295-5"><a href="codebook.html#cb295-5" tabindex="-1"></a></span>
<span id="cb295-6"><a href="codebook.html#cb295-6" tabindex="-1"></a>res_tableOE_tb<span class="sc">$</span>genelabels[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>] <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>]</span>
<span id="cb295-7"><a href="codebook.html#cb295-7" tabindex="-1"></a></span>
<span id="cb295-8"><a href="codebook.html#cb295-8" tabindex="-1"></a><span class="fu">head</span>(res_tableOE_tb)</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="11%" />
<col width="12%" />
<col width="18%" />
<col width="12%" />
<col width="8%" />
<col width="6%" />
<col width="16%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">gene</th>
<th align="right">baseMean</th>
<th align="right">log2FoldChange</th>
<th align="right">lfcSE</th>
<th align="right">pvalue</th>
<th align="right">padj</th>
<th align="left">threshold_OE</th>
<th align="left">genelabels</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MOV10</td>
<td align="right">21681.800</td>
<td align="right">5.0833015</td>
<td align="right">0.1100637</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">MOV10</td>
</tr>
<tr class="even">
<td align="left">H1F0</td>
<td align="right">7881.081</td>
<td align="right">1.5488442</td>
<td align="right">0.0565866</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">H1F0</td>
</tr>
<tr class="odd">
<td align="left">HIST1H1C</td>
<td align="right">1741.383</td>
<td align="right">1.5234178</td>
<td align="right">0.0706100</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">HIST1H1C</td>
</tr>
<tr class="even">
<td align="left">TXNIP</td>
<td align="right">5133.749</td>
<td align="right">1.4237003</td>
<td align="right">0.0697011</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">TXNIP</td>
</tr>
<tr class="odd">
<td align="left">NEAT1</td>
<td align="right">21973.706</td>
<td align="right">0.9164119</td>
<td align="right">0.0466768</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">NEAT1</td>
</tr>
<tr class="even">
<td align="left">KLF10</td>
<td align="right">1694.211</td>
<td align="right">1.2330697</td>
<td align="right">0.0651906</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="left">TRUE</td>
<td align="left">KLF10</td>
</tr>
</tbody>
</table>
</div>
<p>Next, we plot it as before with an additiona layer for <code>geom_text_repel()</code> wherein we can specify the column of gene labels we just created.</p>
<div class="sourceCode" id="cb296"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb296-1"><a href="codebook.html#cb296-1" tabindex="-1"></a><span class="fu">ggplot</span>(res_tableOE_tb, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, </span>
<span id="cb296-2"><a href="codebook.html#cb296-2" tabindex="-1"></a>        <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb296-3"><a href="codebook.html#cb296-3" tabindex="-1"></a>        <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">colour =</span> threshold_OE)) <span class="sc">+</span></span>
<span id="cb296-4"><a href="codebook.html#cb296-4" tabindex="-1"></a>        <span class="fu">geom_text_repel</span>(<span class="fu">aes</span>(<span class="at">label =</span> genelabels)) <span class="sc">+</span></span>
<span id="cb296-5"><a href="codebook.html#cb296-5" tabindex="-1"></a>        <span class="fu">ggtitle</span>(<span class="st">&quot;Mov10 overexpression&quot;</span>) <span class="sc">+</span></span>
<span id="cb296-6"><a href="codebook.html#cb296-6" tabindex="-1"></a>        <span class="fu">xlab</span>(<span class="st">&quot;log2 fold change&quot;</span>) <span class="sc">+</span> </span>
<span id="cb296-7"><a href="codebook.html#cb296-7" tabindex="-1"></a>        <span class="fu">ylab</span>(<span class="st">&quot;-log10 adjusted p-value&quot;</span>) <span class="sc">+</span></span>
<span id="cb296-8"><a href="codebook.html#cb296-8" tabindex="-1"></a>        <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb296-9"><a href="codebook.html#cb296-9" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb296-10"><a href="codebook.html#cb296-10" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.5</span>), </span>
<span id="cb296-11"><a href="codebook.html#cb296-11" tabindex="-1"></a>                                  <span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb296-12"><a href="codebook.html#cb296-12" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.25</span>))) </span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-171-1.png" width="672" /></p>
</div>
</div>
<div id="summary-of-dge-workflow-1" class="section level2 hasAnchor" number="9.9">
<h2><span class="header-section-number">9.9</span> Summary of DGE workflow<a href="codebook.html#summary-of-dge-workflow-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 07-DGE_summarizing_workflow.Rmd</strong></p>
<div class="sourceCode" id="cb297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb297-1"><a href="codebook.html#cb297-1" tabindex="-1"></a><span class="do">## Setup</span></span>
<span id="cb297-2"><a href="codebook.html#cb297-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<p>We have detailed the various steps in a differential expression analysis workflow, providing theory with example code. To provide a more succinct reference for the code needed to run a DGE analysis, we have summarized the steps in an analysis below:</p>
<div id="import-data-into-dds-object-1" class="section level3 hasAnchor" number="9.9.1">
<h3><span class="header-section-number">9.9.1</span> 1. Import data into dds object:<a href="codebook.html#import-data-into-dds-object-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb298"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb298-1"><a href="codebook.html#cb298-1" tabindex="-1"></a><span class="co"># check that the rownames of the count matrix </span></span>
<span id="cb298-2"><a href="codebook.html#cb298-2" tabindex="-1"></a><span class="co"># match the colnames of the metadata</span></span>
<span id="cb298-3"><a href="codebook.html#cb298-3" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(raw_counts) <span class="sc">==</span> <span class="fu">rownames</span>(metadata))</span>
<span id="cb298-4"><a href="codebook.html#cb298-4" tabindex="-1"></a></span>
<span id="cb298-5"><a href="codebook.html#cb298-5" tabindex="-1"></a><span class="co"># Create DESeq2Dataset object</span></span>
<span id="cb298-6"><a href="codebook.html#cb298-6" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> raw_counts, <span class="at">colData =</span> metadata, </span>
<span id="cb298-7"><a href="codebook.html#cb298-7" tabindex="-1"></a>                              <span class="at">design =</span> <span class="sc">~</span> condition)</span></code></pre></div>
</div>
<div id="exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data-1" class="section level3 hasAnchor" number="9.9.2">
<h3><span class="header-section-number">9.9.2</span> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:<a href="codebook.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb299-1"><a href="codebook.html#cb299-1" tabindex="-1"></a><span class="co"># Transform counts for data visualization</span></span>
<span id="cb299-2"><a href="codebook.html#cb299-2" tabindex="-1"></a>rld <span class="ot">&lt;-</span> <span class="fu">rlog</span>(dds, <span class="at">blind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb299-3"><a href="codebook.html#cb299-3" tabindex="-1"></a></span>
<span id="cb299-4"><a href="codebook.html#cb299-4" tabindex="-1"></a><span class="co"># Plot PCA</span></span>
<span id="cb299-5"><a href="codebook.html#cb299-5" tabindex="-1"></a><span class="fu">plotPCA</span>(rld, <span class="at">intgroup =</span> <span class="st">&quot;sampletype&quot;</span>)</span>
<span id="cb299-6"><a href="codebook.html#cb299-6" tabindex="-1"></a></span>
<span id="cb299-7"><a href="codebook.html#cb299-7" tabindex="-1"></a><span class="co"># Extract the rlog matrix from the object</span></span>
<span id="cb299-8"><a href="codebook.html#cb299-8" tabindex="-1"></a>rld_mat <span class="ot">&lt;-</span> <span class="fu">assay</span>(rld)</span>
<span id="cb299-9"><a href="codebook.html#cb299-9" tabindex="-1"></a></span>
<span id="cb299-10"><a href="codebook.html#cb299-10" tabindex="-1"></a><span class="co"># Compute pairwise correlation values</span></span>
<span id="cb299-11"><a href="codebook.html#cb299-11" tabindex="-1"></a>rld_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(rld_mat)</span>
<span id="cb299-12"><a href="codebook.html#cb299-12" tabindex="-1"></a></span>
<span id="cb299-13"><a href="codebook.html#cb299-13" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb299-14"><a href="codebook.html#cb299-14" tabindex="-1"></a><span class="fu">pheatmap</span>(rld_cor)</span></code></pre></div>
</div>
<div id="run-deseq2-1" class="section level3 hasAnchor" number="9.9.3">
<h3><span class="header-section-number">9.9.3</span> 3. Run DESeq2:<a href="codebook.html#run-deseq2-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="codebook.html#cb300-1" tabindex="-1"></a><span class="co"># **Optional step** - Re-create DESeq2 dataset if the design formula has changed after QC analysis in include other sources of variation</span></span>
<span id="cb300-2"><a href="codebook.html#cb300-2" tabindex="-1"></a></span>
<span id="cb300-3"><a href="codebook.html#cb300-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> raw_counts, </span>
<span id="cb300-4"><a href="codebook.html#cb300-4" tabindex="-1"></a>                              <span class="at">colData =</span> metadata, <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb300-5"><a href="codebook.html#cb300-5" tabindex="-1"></a></span>
<span id="cb300-6"><a href="codebook.html#cb300-6" tabindex="-1"></a><span class="co"># Run DESeq2 differential expression analysis</span></span>
<span id="cb300-7"><a href="codebook.html#cb300-7" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span>
<span id="cb300-8"><a href="codebook.html#cb300-8" tabindex="-1"></a></span>
<span id="cb300-9"><a href="codebook.html#cb300-9" tabindex="-1"></a><span class="co"># Output normalized counts to save as a file to access outside RStudio</span></span>
<span id="cb300-10"><a href="codebook.html#cb300-10" tabindex="-1"></a>normalized_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb300-11"><a href="codebook.html#cb300-11" tabindex="-1"></a>    </span>
<span id="cb300-12"><a href="codebook.html#cb300-12" tabindex="-1"></a><span class="fu">write.table</span>(normalized_counts, <span class="at">file =</span> <span class="st">&quot;data/normalized_counts.txt&quot;</span>, </span>
<span id="cb300-13"><a href="codebook.html#cb300-13" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>, <span class="at">quote =</span> F, <span class="at">col.names =</span> <span class="cn">NA</span>)</span></code></pre></div>
</div>
<div id="check-the-fit-of-the-dispersion-estimates-1" class="section level3 hasAnchor" number="9.9.4">
<h3><span class="header-section-number">9.9.4</span> 4. Check the fit of the dispersion estimates:<a href="codebook.html#check-the-fit-of-the-dispersion-estimates-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="codebook.html#cb301-1" tabindex="-1"></a><span class="co"># Plot dispersion estimates</span></span>
<span id="cb301-2"><a href="codebook.html#cb301-2" tabindex="-1"></a><span class="fu">plotDispEsts</span>(dds)</span></code></pre></div>
</div>
<div id="create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions-1" class="section level3 hasAnchor" number="9.9.5">
<h3><span class="header-section-number">9.9.5</span> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:<a href="codebook.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="codebook.html#cb302-1" tabindex="-1"></a><span class="co"># Output results of Wald test for contrast</span></span>
<span id="cb302-2"><a href="codebook.html#cb302-2" tabindex="-1"></a>contrast <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;condition&quot;</span>, </span>
<span id="cb302-3"><a href="codebook.html#cb302-3" tabindex="-1"></a>              <span class="st">&quot;level_to_compare&quot;</span>, </span>
<span id="cb302-4"><a href="codebook.html#cb302-4" tabindex="-1"></a>              <span class="st">&quot;base_level&quot;</span>)</span>
<span id="cb302-5"><a href="codebook.html#cb302-5" tabindex="-1"></a></span>
<span id="cb302-6"><a href="codebook.html#cb302-6" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> contrast)</span>
<span id="cb302-7"><a href="codebook.html#cb302-7" tabindex="-1"></a></span>
<span id="cb302-8"><a href="codebook.html#cb302-8" tabindex="-1"></a>coef <span class="ot">=</span> <span class="fu">resultsNames</span>(dds)</span>
<span id="cb302-9"><a href="codebook.html#cb302-9" tabindex="-1"></a></span>
<span id="cb302-10"><a href="codebook.html#cb302-10" tabindex="-1"></a>res_table <span class="ot">&lt;-</span> <span class="fu">lfcShrink</span>(dds, </span>
<span id="cb302-11"><a href="codebook.html#cb302-11" tabindex="-1"></a>                       <span class="at">coef =</span> coef[<span class="dv">2</span>], </span>
<span id="cb302-12"><a href="codebook.html#cb302-12" tabindex="-1"></a>                       <span class="at">res =</span> res, </span>
<span id="cb302-13"><a href="codebook.html#cb302-13" tabindex="-1"></a>                       <span class="at">type =</span> <span class="st">&quot;apeglm&quot;</span>)</span></code></pre></div>
</div>
<div id="output-significant-results-1" class="section level3 hasAnchor" number="9.9.6">
<h3><span class="header-section-number">9.9.6</span> 6. Output significant results:<a href="codebook.html#output-significant-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="codebook.html#cb303-1" tabindex="-1"></a><span class="do">### Set thresholds</span></span>
<span id="cb303-2"><a href="codebook.html#cb303-2" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb303-3"><a href="codebook.html#cb303-3" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fl">0.58</span> <span class="do">## change in expression of 1.5</span></span>
<span id="cb303-4"><a href="codebook.html#cb303-4" tabindex="-1"></a></span>
<span id="cb303-5"><a href="codebook.html#cb303-5" tabindex="-1"></a><span class="co"># Turn the results object into a data frame</span></span>
<span id="cb303-6"><a href="codebook.html#cb303-6" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> res <span class="sc">%&gt;%</span></span>
<span id="cb303-7"><a href="codebook.html#cb303-7" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb303-8"><a href="codebook.html#cb303-8" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) </span>
<span id="cb303-9"><a href="codebook.html#cb303-9" tabindex="-1"></a></span>
<span id="cb303-10"><a href="codebook.html#cb303-10" tabindex="-1"></a><span class="co"># Subset the significant results</span></span>
<span id="cb303-11"><a href="codebook.html#cb303-11" tabindex="-1"></a>sig_res <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res_df, </span>
<span id="cb303-12"><a href="codebook.html#cb303-12" tabindex="-1"></a>                  padj <span class="sc">&lt;</span> padj.cutoff <span class="sc">&amp;</span>  </span>
<span id="cb303-13"><a href="codebook.html#cb303-13" tabindex="-1"></a>                  <span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> lfc.cutoff)</span></code></pre></div>
</div>
<div id="visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc.-1" class="section level3 hasAnchor" number="9.9.7">
<h3><span class="header-section-number">9.9.7</span> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.<a href="codebook.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc.-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis-1" class="section level3 hasAnchor" number="9.9.8">
<h3><span class="header-section-number">9.9.8</span> 8. Make sure to output the versions of all tools used in the DE analysis:<a href="codebook.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="codebook.html#cb304-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.2.3 (2023-03-15)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS 14.6.1
## 
## Matrix products: default
## LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats4    stats     graphics  grDevices utils     datasets 
## [7] methods   base     
## 
## other attached packages:
##  [1] ggraph_2.2.1                igraph_2.1.1               
##  [3] visNetwork_2.1.2            GOSemSim_2.24.0            
##  [5] rmarkdown_2.29              org.Hs.eg.db_3.16.0        
##  [7] clusterProfiler_4.6.2       enrichplot_1.18.4          
##  [9] fgsea_1.24.0                GO.db_3.16.0               
## [11] AnnotationDbi_1.60.2        GOenrichment_0.5.0         
## [13] ggrepel_0.9.6               gplots_3.2.0               
## [15] lubridate_1.9.3             forcats_1.0.0              
## [17] stringr_1.5.1               dplyr_1.1.4                
## [19] purrr_1.0.2                 readr_2.1.5                
## [21] tidyr_1.3.1                 tibble_3.2.1               
## [23] ggplot2_3.5.1               tidyverse_2.0.0            
## [25] RColorBrewer_1.1-3          DESeq2_1.38.3              
## [27] SummarizedExperiment_1.28.0 Biobase_2.58.0             
## [29] MatrixGenerics_1.10.0       matrixStats_1.4.1          
## [31] GenomicRanges_1.50.2        GenomeInfoDb_1.34.9        
## [33] IRanges_2.32.0              S4Vectors_0.36.2           
## [35] BiocGenerics_0.44.0         bookdown_0.41              
## [37] pheatmap_1.0.12             knitr_1.49                 
## 
## loaded via a namespace (and not attached):
##   [1] utf8_1.2.4             tidyselect_1.2.1      
##   [3] RSQLite_2.3.7          htmlwidgets_1.6.4     
##   [5] grid_4.2.3             BiocParallel_1.32.6   
##   [7] PKI_0.1-14             devtools_2.4.5        
##   [9] scatterpie_0.2.4       munsell_0.5.1         
##  [11] codetools_0.2-20       miniUI_0.1.1.1        
##  [13] withr_3.0.2            colorspace_2.1-1      
##  [15] rstudioapi_0.17.1      DOSE_3.24.2           
##  [17] labeling_0.4.3         bbmle_1.0.25.1        
##  [19] GenomeInfoDbData_1.2.9 polyclip_1.10-7       
##  [21] bit64_4.5.2            farver_2.1.2          
##  [23] downloader_0.4         coda_0.19-4.1         
##  [25] vctrs_0.6.5            treeio_1.22.0         
##  [27] generics_0.1.3         gson_0.1.0            
##  [29] xfun_0.49              timechange_0.3.0      
##  [31] R6_2.5.1               apeglm_1.20.0         
##  [33] graphlayouts_1.2.0     locfit_1.5-9.10       
##  [35] bitops_1.0-9           cachem_1.1.0          
##  [37] gridGraphics_0.5-1     DelayedArray_0.24.0   
##  [39] promises_1.3.0         scales_1.3.0          
##  [41] gtable_0.3.6           processx_3.8.4        
##  [43] tidygraph_1.3.1        rlang_1.1.4           
##  [45] splines_4.2.3          lazyeval_0.2.2        
##  [47] BiocManager_1.30.25    yaml_2.3.10           
##  [49] reshape2_1.4.4         httpuv_1.6.15         
##  [51] rsconnect_1.3.2        qvalue_2.30.0         
##  [53] tools_4.2.3            usethis_3.0.0         
##  [55] ggplotify_0.1.2        ellipsis_0.3.2        
##  [57] jquerylib_0.1.4        sessioninfo_1.2.2     
##  [59] Rcpp_1.0.13-1          plyr_1.8.9            
##  [61] base64enc_0.1-3        zlibbioc_1.44.0       
##  [63] RCurl_1.98-1.16        ps_1.8.1              
##  [65] openssl_2.2.2          viridis_0.6.5         
##  [67] cowplot_1.1.3          urlchecker_1.0.1      
##  [69] fs_1.6.5               magrittr_2.0.3        
##  [71] data.table_1.16.2      mvtnorm_1.3-2         
##  [73] ggnewscale_0.5.0       pkgload_1.4.0         
##  [75] hms_1.1.3              patchwork_1.3.0       
##  [77] mime_0.12              evaluate_1.0.1        
##  [79] xtable_1.8-4           HDO.db_0.99.1         
##  [81] XML_3.99-0.17          emdbook_1.3.13        
##  [83] gridExtra_2.3          compiler_4.2.3        
##  [85] bdsmatrix_1.3-7        KernSmooth_2.23-24    
##  [87] crayon_1.5.3           shadowtext_0.1.4      
##  [89] htmltools_0.5.8.1      ggfun_0.1.7           
##  [91] later_1.3.2            tzdb_0.4.0            
##  [93] geneplotter_1.76.0     aplot_0.2.3           
##  [95] DBI_1.2.3              tweenr_2.0.3          
##  [97] MASS_7.3-58.2          Matrix_1.5-3          
##  [99] cli_3.6.3              parallel_4.2.3        
## [101] pkgconfig_2.0.3        numDeriv_2016.8-1.1   
## [103] ggtree_3.6.2           annotate_1.76.0       
## [105] bslib_0.8.0            XVector_0.38.0        
## [107] yulab.utils_0.1.8      callr_3.7.6           
## [109] digest_0.6.37          Biostrings_2.66.0     
## [111] fastmatch_1.1-4        tidytree_0.4.6        
## [113] curl_6.0.0             shiny_1.9.1           
## [115] gtools_3.9.5           lifecycle_1.0.4       
## [117] nlme_3.1-166           jsonlite_1.8.9        
## [119] askpass_1.2.1          desc_1.4.3            
## [121] viridisLite_0.4.2      fansi_1.0.6           
## [123] pillar_1.9.0           lattice_0.22-6        
## [125] KEGGREST_1.38.0        fastmap_1.2.0         
## [127] httr_1.4.7             pkgbuild_1.4.5        
## [129] glue_1.8.0             remotes_2.5.0         
## [131] png_0.1-8              bit_4.5.0             
## [133] ggforce_0.4.2          stringi_1.8.4         
## [135] sass_0.4.9             profvis_0.4.0         
## [137] blob_1.2.4             caTools_1.18.3        
## [139] memoise_2.0.1          ape_5.8</code></pre>
<p><strong>Exercise/Homework: modify the <code>lessons/07-DGE_summarizing_workflow.Rmd</code> file to analyze the MOV dataset, starting with Mov10_full_counts.txt in your data folder. Compare the “MOV10_knockdown” to the “control”. Include a heatmap and a volcano plot</strong>. Make sure you rename the file first to <code>Lastname_07-DGE_summarizing_workflow.Rmd</code> and save it in your main directory, not in the lessons folder. points = +10</p>
</div>
</div>
<div id="functional-analysis-of-rna-seq-data" class="section level2 hasAnchor" number="9.10">
<h2><span class="header-section-number">9.10</span> Functional analysis of RNA-seq data<a href="codebook.html#functional-analysis-of-rna-seq-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>.Rmd file = 08-GO_enrichment_analysis.Rmd</strong></p>
<div id="over-representation-analysis-ora-using-clusterprofiler-1" class="section level3 hasAnchor" number="9.10.1">
<h3><span class="header-section-number">9.10.1</span> Over-representation Analysis (ORA) using <code>clusterProfiler</code><a href="codebook.html#over-representation-analysis-ora-using-clusterprofiler-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="codebook.html#cb306-1" tabindex="-1"></a><span class="co"># Uncomment the following if you haven&#39;t yet installed the required packages</span></span>
<span id="cb306-2"><a href="codebook.html#cb306-2" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;ggraph&quot;, &quot;igraph&quot;, &quot;visNetwork&quot;, </span></span>
<span id="cb306-3"><a href="codebook.html#cb306-3" tabindex="-1"></a><span class="co"># &quot;GO.db&quot;, &quot;GOSemSim&quot;), force = TRUE)</span></span>
<span id="cb306-4"><a href="codebook.html#cb306-4" tabindex="-1"></a></span>
<span id="cb306-5"><a href="codebook.html#cb306-5" tabindex="-1"></a><span class="co"># IMPORTANT!!! Uncomment the following to update the GOenrichment package</span></span>
<span id="cb306-6"><a href="codebook.html#cb306-6" tabindex="-1"></a><span class="co"># remove.package(&quot;GOenrichment&quot;)</span></span>
<span id="cb306-7"><a href="codebook.html#cb306-7" tabindex="-1"></a><span class="co"># devtools::install_github(&quot;gurinina/GOenrichment&quot;, force = TRUE)</span></span>
<span id="cb306-8"><a href="codebook.html#cb306-8" tabindex="-1"></a></span>
<span id="cb306-9"><a href="codebook.html#cb306-9" tabindex="-1"></a><span class="do">## Load required libraries</span></span>
<span id="cb306-10"><a href="codebook.html#cb306-10" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)<span class="co">#</span></span>
<span id="cb306-11"><a href="codebook.html#cb306-11" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)<span class="co">#</span></span>
<span id="cb306-12"><a href="codebook.html#cb306-12" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb306-13"><a href="codebook.html#cb306-13" tabindex="-1"></a><span class="fu">library</span>(enrichplot)<span class="co">#</span></span>
<span id="cb306-14"><a href="codebook.html#cb306-14" tabindex="-1"></a><span class="fu">library</span>(fgsea) <span class="co">#</span></span>
<span id="cb306-15"><a href="codebook.html#cb306-15" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb306-16"><a href="codebook.html#cb306-16" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb306-17"><a href="codebook.html#cb306-17" tabindex="-1"></a><span class="fu">library</span>(visNetwork)</span>
<span id="cb306-18"><a href="codebook.html#cb306-18" tabindex="-1"></a><span class="fu">library</span>(GO.db)</span>
<span id="cb306-19"><a href="codebook.html#cb306-19" tabindex="-1"></a><span class="fu">library</span>(GOSemSim)</span>
<span id="cb306-20"><a href="codebook.html#cb306-20" tabindex="-1"></a><span class="fu">library</span>(GOenrichment)</span></code></pre></div>
<div id="running-clusterprofiler-we-first-need-to-load-the-results-of-the-differential-expression-analysis.-1" class="section level4 hasAnchor" number="9.10.1.1">
<h4><span class="header-section-number">9.10.1.1</span> Running clusterProfiler: we first need to load the results of the differential expression analysis.<a href="codebook.html#running-clusterprofiler-we-first-need-to-load-the-results-of-the-differential-expression-analysis.-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="codebook.html#cb307-1" tabindex="-1"></a>res_tableOE <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/res_tableOE.RDS&quot;</span>)</span>
<span id="cb307-2"><a href="codebook.html#cb307-2" tabindex="-1"></a></span>
<span id="cb307-3"><a href="codebook.html#cb307-3" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span> </span>
<span id="cb307-4"><a href="codebook.html#cb307-4" tabindex="-1"></a>  <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb307-5"><a href="codebook.html#cb307-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange))  <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>To perform the over-representation analysis, we need a list of background genes and a list of significant genes:</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="codebook.html#cb308-1" tabindex="-1"></a><span class="do">## background set of genes</span></span>
<span id="cb308-2"><a href="codebook.html#cb308-2" tabindex="-1"></a>allOE_genes <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span>
<span id="cb308-3"><a href="codebook.html#cb308-3" tabindex="-1"></a></span>
<span id="cb308-4"><a href="codebook.html#cb308-4" tabindex="-1"></a></span>
<span id="cb308-5"><a href="codebook.html#cb308-5" tabindex="-1"></a>sigOE <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res_tableOE_tb, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb308-6"><a href="codebook.html#cb308-6" tabindex="-1"></a></span>
<span id="cb308-7"><a href="codebook.html#cb308-7" tabindex="-1"></a><span class="do">## significant genes</span></span>
<span id="cb308-8"><a href="codebook.html#cb308-8" tabindex="-1"></a>sigOE_genes <span class="ot">=</span> sigOE<span class="sc">$</span>gene</span></code></pre></div>
<p>The<code>enrichGO()</code> function performs the ORA for the significant genes of interest (<code>sigOE_genes</code>) compared to the background gene list (<code>allOE_genes</code>) and returns the enriched GO terms and their associated p-values.</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="codebook.html#cb309-1" tabindex="-1"></a><span class="do">## Run GO enrichment analysis</span></span>
<span id="cb309-2"><a href="codebook.html#cb309-2" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> sigOE_genes,</span>
<span id="cb309-3"><a href="codebook.html#cb309-3" tabindex="-1"></a> <span class="at">universe =</span> allOE_genes,</span>
<span id="cb309-4"><a href="codebook.html#cb309-4" tabindex="-1"></a> <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb309-5"><a href="codebook.html#cb309-5" tabindex="-1"></a> <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb309-6"><a href="codebook.html#cb309-6" tabindex="-1"></a> <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb309-7"><a href="codebook.html#cb309-7" tabindex="-1"></a> <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb309-8"><a href="codebook.html#cb309-8" tabindex="-1"></a> <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb309-9"><a href="codebook.html#cb309-9" tabindex="-1"></a> <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb309-10"><a href="codebook.html#cb309-10" tabindex="-1"></a> <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb309-11"><a href="codebook.html#cb309-11" tabindex="-1"></a> <span class="at">readable =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can extract the gene lists associated with each enriched GO term from the <code>ego</code> object to create edges between terms based on shared genes.</p>
<div class="sourceCode" id="cb310"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb310-1"><a href="codebook.html#cb310-1" tabindex="-1"></a><span class="co"># Extract the result as a data frame</span></span>
<span id="cb310-2"><a href="codebook.html#cb310-2" tabindex="-1"></a>enrich_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(ego)</span>
<span id="cb310-3"><a href="codebook.html#cb310-3" tabindex="-1"></a></span>
<span id="cb310-4"><a href="codebook.html#cb310-4" tabindex="-1"></a><span class="fu">names</span>(enrich_df)</span></code></pre></div>
<pre><code>## [1] &quot;ID&quot;          &quot;Description&quot; &quot;GeneRatio&quot;   &quot;BgRatio&quot;    
## [5] &quot;pvalue&quot;      &quot;p.adjust&quot;    &quot;qvalue&quot;      &quot;geneID&quot;     
## [9] &quot;Count&quot;</code></pre>
<div class="sourceCode" id="cb312"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb312-1"><a href="codebook.html#cb312-1" tabindex="-1"></a><span class="fu">head</span>(enrich_df)</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="1%" />
<col width="1%" />
<col width="3%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="0%" />
<col width="89%" />
<col width="0%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">GeneRatio</th>
<th align="left">BgRatio</th>
<th align="right">pvalue</th>
<th align="right">p.adjust</th>
<th align="right">qvalue</th>
<th align="left">geneID</th>
<th align="right">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="GO:0006403" class="uri">GO:0006403</a></td>
<td align="left"><a href="GO:0006403" class="uri">GO:0006403</a></td>
<td align="left">RNA localization</td>
<td align="left">110/5343</td>
<td align="left">188/14702</td>
<td align="right">0</td>
<td align="right">6.0e-07</td>
<td align="right">5.0e-07</td>
<td align="left">ALKBH5/ALYREF/ARC/ATR/ATXN2/BICD1/CASC3/CCT2/CCT3/CCT5/CCT6A/CCT7/CETN2/CETN3/CHTOP/DCP2/DDX19A/DDX39A/DHX9/DKC1/EIF4A3/EIF5AL1/ENY2/EXOSC10/FBL/FMR1/FUBP3/FYTTD1/GLE1/HNRNPU/HSF1/IGF2BP1/IGF2BP2/KHDRBS1/KHSRP/KPNB1/MCM3AP/MVP/MYO1C/NCBP1/NHP2/NOL6/NOP10/NSUN2/NUP107/NUP133/NUP188/NUP205/NUP210/NUP214/NUP50/NUP54/NUP85/NUP88/NUP93/NUP98/NUTF2/NXF1/NXF3/NXT1/NXT2/PABPN1/PARN/PEG10/PHAX/PIH1D1/POLDIP3/POLR2D/POM121/POM121C/POM121L2/PRPF6/QKI/RAE1/RAN/RBM15B/RBM8A/RFTN1/RUVBL1/RUVBL2/SARNP/SEC13/SEH1L/SENP2/SETD2/SHQ1/SIDT2/SLBP/SMG5/SMG6/SMG7/SNUPN/SRSF3/SRSF7/STAU1/SUPT6H/TGFBR2/THOC1/THOC2/THOC6/THOC7/TOMM20/UPF1/XPO1/XPO5/YBX1/ZC3H11A/ZFP36/ZFP36L1/ZNF385A</td>
<td align="right">110</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0042254" class="uri">GO:0042254</a></td>
<td align="left"><a href="GO:0042254" class="uri">GO:0042254</a></td>
<td align="left">ribosome biogenesis</td>
<td align="left">153/5343</td>
<td align="left">282/14702</td>
<td align="right">0</td>
<td align="right">6.0e-07</td>
<td align="right">5.0e-07</td>
<td align="left">AATF/ABCE1/ABT1/BMS1/BRIX1/BYSL/C1QBP/CHD7/CUL4A/CUL4B/DDX18/DDX21/DDX27/DDX28/DDX47/DDX49/DDX51/DDX54/DDX56/DHX30/DHX37/DKC1/EBNA1BP2/EIF2A/EIF4A3/EIF6/EMG1/ERAL1/ERCC2/ERI2/ERI3/EXOSC10/EXOSC3/EXOSC5/EXOSC6/EXOSC9/FBL/FBLL1/FCF1/FRG1/FTSJ3/GEMIN4/GLUL/GNL3L/GPATCH4/GRWD1/GTPBP10/HEATR1/HELB/IMP3/IMP4/ISG20L2/KRI1/LSG1/LSM6/LTV1/MAK16/MDN1/MPHOSPH10/MRPL1/MRPL20/MRPL36/MRPL44/MRPS11/MRPS2/MRPS7/MRTO4/MYBBP1A/NAT10/NGDN/NGRN/NHP2/NLE1/NMD3/NOC2L/NOC4L/NOL10/NOL11/NOL6/NOL8/NOL9/NOLC1/NOP10/NOP14/NOP16/NOP2/NSUN5/NUDT16/NUP88/PA2G4/PDCD11/PELP1/PES1/PIH1D1/POP4/PPAN/PWP1/PWP2/RAN/REXO4/RIOK3/RPL10A/RPL23A/RPL24/RPL26/RPL26L1/RPL7L1/RPP30/RPP40/RPS15/RPS27/RPS27L/RPSA/RPUSD1/RPUSD4/RRP1/RRP15/RRP1B/RRP36/RRP7A/RRP9/RRS1/RSL1D1/RSL24D1/SBDS/SDAD1/SHQ1/SIRT7/SRFBP1/SURF6/SUV39H1/TFB1M/TRMT112/TRMT2B/TSR1/TSR2/URB1/URB2/UTP14A/UTP15/UTP18/UTP20/UTP3/WDR18/WDR43/WDR46/WDR55/WDR74/XPO1/XRCC5/XRN2/YTHDF2/ZNF622</td>
<td align="right">153</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:1903311" class="uri">GO:1903311</a></td>
<td align="left"><a href="GO:1903311" class="uri">GO:1903311</a></td>
<td align="left">regulation of mRNA metabolic process</td>
<td align="left">149/5343</td>
<td align="left">274/14702</td>
<td align="right">0</td>
<td align="right">6.0e-07</td>
<td align="right">5.0e-07</td>
<td align="left">AHCYL1/AKT1/ALKBH5/ANGEL2/APEX1/BTG2/C1QBP/CARHSP1/CASC3/CDK9/CELF1/CELF2/CELF3/CELF4/CIRBP/CLNS1A/CNOT10/CNOT7/CPEB1/CPEB3/CPSF7/CSDE1/DAZAP1/DCP1A/DCP2/DDX5/DHX34/DHX9/E2F1/EIF4A3/ELAVL1/EXOSC3/EXOSC5/EXOSC9/FAM172A/FASTK/FASTKD5/FMR1/FTO/FUS/FXR2/GDNF/GTPBP1/HNRNPA0/HNRNPD/HNRNPK/HNRNPL/HNRNPM/HNRNPU/HSF1/HSPA8/IGF2BP1/IGF2BP2/KHDRBS1/KHDRBS3/KHSRP/LARP1/LARP4B/LARP7/MAPKAPK2/MBNL1/MBNL2/MBNL3/METTL3/MLH1/MOV10/NCBP1/NCL/NOVA1/NUP98/PABPC1/PABPC4/PAF1/PAIP1/PAPOLA/PARN/PATL1/PDE12/POLR2D/POLR2G/PRDX6/PRKCA/PRMT5/PRPF19/PRR5L/PTBP1/PUF60/QKI/RAVER1/RAVER2/RBFOX2/RBM10/RBM11/RBM15/RBM15B/RBM23/RBM24/RBM3/RBM38/RBM39/RBM4/RBM42/RBM8A/RBMX/RC3H1/RC3H2/RNPS1/ROCK1/SAFB/SAFB2/SAMD4A/SAMD4B/SECISBP2/SF1/SF3B4/SFSWAP/SLTM/SMU1/SNRNP70/SRPK1/SRPK2/SRSF3/SRSF6/SRSF7/SRSF9/SUPT6H/SYNCRIP/TBRG4/TIA1/TIRAP/TNKS1BP1/TNRC6B/TOB1/TRAF2/TRAF5/TTC5/U2AF2/UPF1/VIM/WDR77/YBX1/YTHDF2/YTHDF3/ZBTB7A/ZC3H12A/ZC3H14/ZC3HAV1/ZFP36/ZFP36L1</td>
<td align="right">149</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0043484" class="uri">GO:0043484</a></td>
<td align="left"><a href="GO:0043484" class="uri">GO:0043484</a></td>
<td align="left">regulation of RNA splicing</td>
<td align="left">102/5343</td>
<td align="left">172/14702</td>
<td align="right">0</td>
<td align="right">6.0e-07</td>
<td align="right">5.0e-07</td>
<td align="left">AHNAK/AHNAK2/ATXN7L3/C1QBP/CCNL1/CCNL2/CDK12/CELF1/CELF2/CELF3/CELF4/CIRBP/CLK1/CLNS1A/DAZAP1/DDX5/ENY2/ERN1/ESRP1/FAM172A/FAM50A/FASTK/FMR1/FUS/FXR2/GRSF1/HNRNPF/HNRNPH2/HNRNPK/HNRNPL/HNRNPU/HOXB-AS3/HSPA1A/HSPA8/KAT2A/KHDRBS1/KHDRBS3/LARP7/MBNL1/MBNL2/MBNL3/NCBP1/NCL/NOVA1/NUP98/PIK3R1/POLR2A/PQBP1/PRDX6/PRMT5/PRPF19/PTBP1/PTBP2/PUF60/QKI/RAVER1/RAVER2/RBFOX2/RBM10/RBM11/RBM12B/RBM15/RBM15B/RBM22/RBM23/RBM24/RBM3/RBM38/RBM39/RBM4/RBM42/RBM8A/RBMX/RNPS1/RPS26/RRP1B/SF1/SF3B3/SF3B4/SFSWAP/SLC38A2/SMU1/SNRNP70/SRPK1/SRPK2/SRSF3/SRSF6/SRSF7/SRSF9/SUPT3H/TADA2B/TADA3/TAF5L/TAF6L/TIA1/TMBIM6/TRRAP/U2AF2/USP22/WDR77/ZBTB7A/ZNF326</td>
<td align="right">102</td>
</tr>
<tr class="odd">
<td align="left"><a href="GO:0016571" class="uri">GO:0016571</a></td>
<td align="left"><a href="GO:0016571" class="uri">GO:0016571</a></td>
<td align="left">histone methylation</td>
<td align="left">86/5343</td>
<td align="left">140/14702</td>
<td align="right">0</td>
<td align="right">1.0e-06</td>
<td align="right">8.0e-07</td>
<td align="left">AASS/ARID4A/ARID4B/ASH1L/ASH2L/ATRX/AUTS2/BCOR/BEND3/BOD1/BRCA1/BRD4/CARM1/CHTOP/CLNS1A/CTCF/CTCFL/CTNNB1/CTR9/DOT1L/EHMT1/EHMT2/EZH1/FBL/FBLL1/GFI1/HCFC1/HCFC2/JARID2/KANSL3/KDM3A/KDM4A/MCRS1/MECOM/MECP2/MEN1/METTL23/MLLT6/MTF2/MTHFR/MYB/NCOA6/NFYA/NFYB/NFYC/NR1H4/NSD1/OGT/PAF1/PAXIP1/PHF19/PIH1D1/PRDM13/PRDM2/PRDM4/PRMT1/PRMT2/PRMT5/PRMT7/PWP1/PYGO2/RNF20/RTF1/SETD1A/SETD1B/SETD2/SETD3/SETD5/SETD6/SIRT7/SMARCA5/SMARCB1/SMYD3/SMYD5/SUPT6H/SUV39H1/SUZ12/TET1/TET2/TRMT112/WDR5/WDR5B/WDR77/WDR82/ZNF274/ZNF335</td>
<td align="right">86</td>
</tr>
<tr class="even">
<td align="left"><a href="GO:0051168" class="uri">GO:0051168</a></td>
<td align="left"><a href="GO:0051168" class="uri">GO:0051168</a></td>
<td align="left">nuclear export</td>
<td align="left">90/5343</td>
<td align="left">150/14702</td>
<td align="right">0</td>
<td align="right">1.9e-06</td>
<td align="right">1.6e-06</td>
<td align="left">ABCE1/AHCYL1/ALKBH5/ALYREF/ANP32B/BAG3/CASC3/CDK5/CHTOP/CSE1L/DDX19A/DDX39A/DHX9/DNAJC27/EIF4A3/EIF6/EMD/ENY2/FYTTD1/GAS6/GLE1/KHDRBS1/LSG1/LTV1/MCM3AP/MDN1/NCBP1/NEMF/NMD3/NOL6/NSUN2/NUP107/NUP133/NUP188/NUP214/NUP85/NUP88/NUP93/NUTF2/NXF1/NXF3/NXT1/PABPN1/PARK7/PHAX/PKD1/POLDIP3/POLR2D/POM121/POM121C/POM121L2/PPM1A/PRKACA/PTPN11/RAE1/RAN/RANBP3/RANBP3L/RANGAP1/RAPGEF3/RBM15B/RBM22/RBM8A/RPS15/RRS1/SARNP/SDAD1/SETD2/SFN/SIRT6/SIRT7/SMG5/SMG6/SMG7/SRSF3/SUPT6H/TGFB1/THOC1/THOC2/THOC6/THOC7/UBE2I/UPF1/XPO1/XPO4/XPO5/XPO6/XPO7/YWHAE/ZC3H11A</td>
<td align="right">90</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb313-1"><a href="codebook.html#cb313-1" tabindex="-1"></a><span class="co"># GO term IDs </span></span>
<span id="cb313-2"><a href="codebook.html#cb313-2" tabindex="-1"></a>term_ids <span class="ot">&lt;-</span> enrich_df<span class="sc">$</span>ID  </span>
<span id="cb313-3"><a href="codebook.html#cb313-3" tabindex="-1"></a></span>
<span id="cb313-4"><a href="codebook.html#cb313-4" tabindex="-1"></a><span class="co"># Split gene lists by &quot;/&quot; and assign term IDs as names</span></span>
<span id="cb313-5"><a href="codebook.html#cb313-5" tabindex="-1"></a>gene_lists <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(enrich_df<span class="sc">$</span>geneID, <span class="st">&quot;/&quot;</span>) </span>
<span id="cb313-6"><a href="codebook.html#cb313-6" tabindex="-1"></a></span>
<span id="cb313-7"><a href="codebook.html#cb313-7" tabindex="-1"></a><span class="co"># Assign term IDs as names</span></span>
<span id="cb313-8"><a href="codebook.html#cb313-8" tabindex="-1"></a><span class="fu">names</span>(gene_lists) <span class="ot">&lt;-</span> term_ids  </span></code></pre></div>
<p>Using the gene lists associated with each GO term, we can create edges between terms based on the shared genes.</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="codebook.html#cb314-1" tabindex="-1"></a><span class="co"># Load GO data for semantic similarity calculation</span></span>
<span id="cb314-2"><a href="codebook.html#cb314-2" tabindex="-1"></a>go_data <span class="ot">&lt;-</span> GOSemSim<span class="sc">::</span><span class="fu">godata</span>(<span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>, </span>
<span id="cb314-3"><a href="codebook.html#cb314-3" tabindex="-1"></a>                            <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb314-4"><a href="codebook.html#cb314-4" tabindex="-1"></a>                            <span class="at">keytype =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb314-5"><a href="codebook.html#cb314-5" tabindex="-1"></a>                            <span class="at">computeIC =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## preparing gene to GO mapping data...</code></pre>
<pre><code>## preparing IC data...</code></pre>
<div class="sourceCode" id="cb317"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb317-1"><a href="codebook.html#cb317-1" tabindex="-1"></a><span class="co"># Compute semantic similarity</span></span>
<span id="cb317-2"><a href="codebook.html#cb317-2" tabindex="-1"></a>similarity_matrix <span class="ot">&lt;-</span> GOSemSim<span class="sc">::</span><span class="fu">mgoSim</span>(term_ids, term_ids, </span>
<span id="cb317-3"><a href="codebook.html#cb317-3" tabindex="-1"></a>                     <span class="at">semData =</span> go_data, </span>
<span id="cb317-4"><a href="codebook.html#cb317-4" tabindex="-1"></a>                     <span class="at">measure =</span> <span class="st">&quot;Wang&quot;</span>, <span class="at">combine =</span> <span class="cn">NULL</span>)</span>
<span id="cb317-5"><a href="codebook.html#cb317-5" tabindex="-1"></a></span>
<span id="cb317-6"><a href="codebook.html#cb317-6" tabindex="-1"></a><span class="co"># Convert similarity matrix to edges</span></span>
<span id="cb317-7"><a href="codebook.html#cb317-7" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">from =</span> <span class="fu">rep</span>(term_ids, </span>
<span id="cb317-8"><a href="codebook.html#cb317-8" tabindex="-1"></a>                    <span class="at">each =</span> <span class="fu">length</span>(term_ids)),</span>
<span id="cb317-9"><a href="codebook.html#cb317-9" tabindex="-1"></a>                    <span class="at">to =</span> <span class="fu">rep</span>(term_ids, </span>
<span id="cb317-10"><a href="codebook.html#cb317-10" tabindex="-1"></a>                    <span class="at">times =</span> <span class="fu">length</span>(term_ids)),</span>
<span id="cb317-11"><a href="codebook.html#cb317-11" tabindex="-1"></a>                    <span class="at">similarity =</span> <span class="fu">as.vector</span>(similarity_matrix))</span>
<span id="cb317-12"><a href="codebook.html#cb317-12" tabindex="-1"></a></span>
<span id="cb317-13"><a href="codebook.html#cb317-13" tabindex="-1"></a><span class="co"># Filter out edges with zero similarity or self-loops</span></span>
<span id="cb317-14"><a href="codebook.html#cb317-14" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">subset</span>(edges, similarity <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> from <span class="sc">!=</span> to)</span>
<span id="cb317-15"><a href="codebook.html#cb317-15" tabindex="-1"></a></span>
<span id="cb317-16"><a href="codebook.html#cb317-16" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges <span class="sc">%&gt;%</span> </span>
<span id="cb317-17"><a href="codebook.html#cb317-17" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(similarity <span class="sc">&gt;=</span> <span class="fl">0.5</span>)</span>
<span id="cb317-18"><a href="codebook.html#cb317-18" tabindex="-1"></a></span>
<span id="cb317-19"><a href="codebook.html#cb317-19" tabindex="-1"></a></span>
<span id="cb317-20"><a href="codebook.html#cb317-20" tabindex="-1"></a><span class="co"># Convert term IDs to names</span></span>
<span id="cb317-21"><a href="codebook.html#cb317-21" tabindex="-1"></a>edges<span class="sc">$</span>to <span class="ot">&lt;-</span> <span class="fu">Term</span>(edges<span class="sc">$</span>to)</span>
<span id="cb317-22"><a href="codebook.html#cb317-22" tabindex="-1"></a>edges<span class="sc">$</span>from <span class="ot">&lt;-</span> <span class="fu">Term</span>(edges<span class="sc">$</span>from)</span>
<span id="cb317-23"><a href="codebook.html#cb317-23" tabindex="-1"></a></span>
<span id="cb317-24"><a href="codebook.html#cb317-24" tabindex="-1"></a><span class="fu">head</span>(edges)</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="39%" />
<col width="48%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">from</th>
<th align="left">to</th>
<th align="right">similarity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">RNA localization</td>
<td align="left">establishment of RNA localization</td>
<td align="right">0.728</td>
</tr>
<tr class="even">
<td align="left">ribosome biogenesis</td>
<td align="left">ribosomal small subunit biogenesis</td>
<td align="right">0.848</td>
</tr>
<tr class="odd">
<td align="left">ribosome biogenesis</td>
<td align="left">ribosome assembly</td>
<td align="right">0.623</td>
</tr>
<tr class="even">
<td align="left">regulation of mRNA metabolic process</td>
<td align="left">regulation of RNA splicing</td>
<td align="right">0.752</td>
</tr>
<tr class="odd">
<td align="left">regulation of mRNA metabolic process</td>
<td align="left">regulation of mRNA processing</td>
<td align="right">0.863</td>
</tr>
<tr class="even">
<td align="left">regulation of mRNA metabolic process</td>
<td align="left">regulation of mRNA splicing, via spliceosome</td>
<td align="right">0.720</td>
</tr>
</tbody>
</table>
</div>
<p>Now we can create a network plot to visualize the relationships between the enriched GO terms based on their semantic similarity.</p>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb318-1"><a href="codebook.html#cb318-1" tabindex="-1"></a><span class="co"># Create an igraph object from the edges data frame</span></span>
<span id="cb318-2"><a href="codebook.html#cb318-2" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(edges, <span class="at">directed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb318-3"><a href="codebook.html#cb318-3" tabindex="-1"></a></span>
<span id="cb318-4"><a href="codebook.html#cb318-4" tabindex="-1"></a><span class="co"># Visualize the graph using ggraph</span></span>
<span id="cb318-5"><a href="codebook.html#cb318-5" tabindex="-1"></a><span class="fu">ggraph</span>(g, <span class="at">layout =</span> <span class="st">&#39;fr&#39;</span>) <span class="sc">+</span>  <span class="co"># Fruchterman-Reingold layout</span></span>
<span id="cb318-6"><a href="codebook.html#cb318-6" tabindex="-1"></a>  <span class="fu">geom_edge_link</span>(<span class="fu">aes</span>(<span class="at">edge_alpha =</span> <span class="fl">0.5</span>)) <span class="sc">+</span>  <span class="co"># Edges with some transparency</span></span>
<span id="cb318-7"><a href="codebook.html#cb318-7" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="at">size =</span> <span class="dv">5</span>, <span class="at">color =</span> <span class="st">&quot;blue&quot;</span>) <span class="sc">+</span>  <span class="co"># Nodes sized uniformly</span></span>
<span id="cb318-8"><a href="codebook.html#cb318-8" tabindex="-1"></a>  <span class="fu">geom_node_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> name), <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Node labels</span></span>
<span id="cb318-9"><a href="codebook.html#cb318-9" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span>  <span class="co"># Clean theme without axes</span></span>
<span id="cb318-10"><a href="codebook.html#cb318-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;GO Terms Semantic Similarity Network&quot;</span>)  <span class="co"># Title for the plot</span></span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-186-1.png" width="672" /></p>
<p>Edges based can also be defined by Jaccard similarity; defined as the size of the intersection each pair of GO terms divided by the size of their union</p>
<div class="sourceCode" id="cb319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb319-1"><a href="codebook.html#cb319-1" tabindex="-1"></a><span class="co"># Define a function to compute Jaccard similarity</span></span>
<span id="cb319-2"><a href="codebook.html#cb319-2" tabindex="-1"></a>jaccard_index <span class="ot">&lt;-</span> <span class="cf">function</span>(genes1, genes2) {</span>
<span id="cb319-3"><a href="codebook.html#cb319-3" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(genes1, genes2)) <span class="sc">/</span> <span class="fu">length</span>(<span class="fu">union</span>(genes1, genes2))</span>
<span id="cb319-4"><a href="codebook.html#cb319-4" tabindex="-1"></a>}</span>
<span id="cb319-5"><a href="codebook.html#cb319-5" tabindex="-1"></a></span>
<span id="cb319-6"><a href="codebook.html#cb319-6" tabindex="-1"></a><span class="co"># Create an empty data frame to store edges</span></span>
<span id="cb319-7"><a href="codebook.html#cb319-7" tabindex="-1"></a>edges <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">from =</span> <span class="fu">character</span>(), <span class="at">to =</span> <span class="fu">character</span>(), </span>
<span id="cb319-8"><a href="codebook.html#cb319-8" tabindex="-1"></a>                    <span class="at">similarity =</span> <span class="fu">numeric</span>())</span>
<span id="cb319-9"><a href="codebook.html#cb319-9" tabindex="-1"></a>  <span class="co"># Compute Jaccard similarity for each pair of terms</span></span>
<span id="cb319-10"><a href="codebook.html#cb319-10" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">length</span>(term_ids) <span class="sc">-</span> <span class="dv">1</span>)) {</span>
<span id="cb319-11"><a href="codebook.html#cb319-11" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> (i <span class="sc">+</span> <span class="dv">1</span>)<span class="sc">:</span><span class="fu">length</span>(term_ids)) {</span>
<span id="cb319-12"><a href="codebook.html#cb319-12" tabindex="-1"></a>  similarity <span class="ot">&lt;-</span> <span class="fu">jaccard_index</span>(gene_lists[[i]], gene_lists[[j]])</span>
<span id="cb319-13"><a href="codebook.html#cb319-13" tabindex="-1"></a>  <span class="cf">if</span> (similarity <span class="sc">&gt;</span> <span class="dv">0</span>) {  <span class="co"># Only keep edges with some overlap</span></span>
<span id="cb319-14"><a href="codebook.html#cb319-14" tabindex="-1"></a>  edges <span class="ot">&lt;-</span> <span class="fu">rbind</span>(edges, <span class="fu">data.frame</span>(<span class="at">from =</span> term_ids[i], </span>
<span id="cb319-15"><a href="codebook.html#cb319-15" tabindex="-1"></a>                                   <span class="at">to =</span> term_ids[j], </span>
<span id="cb319-16"><a href="codebook.html#cb319-16" tabindex="-1"></a>                                   <span class="at">similarity =</span> similarity))</span>
<span id="cb319-17"><a href="codebook.html#cb319-17" tabindex="-1"></a>  }}}</span>
<span id="cb319-18"><a href="codebook.html#cb319-18" tabindex="-1"></a></span>
<span id="cb319-19"><a href="codebook.html#cb319-19" tabindex="-1"></a>edges <span class="ot">&lt;-</span> edges <span class="sc">%&gt;%</span> </span>
<span id="cb319-20"><a href="codebook.html#cb319-20" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(similarity <span class="sc">&gt;=</span> <span class="fl">0.5</span>)</span>
<span id="cb319-21"><a href="codebook.html#cb319-21" tabindex="-1"></a></span>
<span id="cb319-22"><a href="codebook.html#cb319-22" tabindex="-1"></a><span class="co"># Convert term IDs to names</span></span>
<span id="cb319-23"><a href="codebook.html#cb319-23" tabindex="-1"></a>edges<span class="sc">$</span>to <span class="ot">&lt;-</span> <span class="fu">Term</span>(edges<span class="sc">$</span>to)</span>
<span id="cb319-24"><a href="codebook.html#cb319-24" tabindex="-1"></a>edges<span class="sc">$</span>from <span class="ot">&lt;-</span> <span class="fu">Term</span>(edges<span class="sc">$</span>from)</span></code></pre></div>
<p>Now we can create a network plot to visualize the relationships between the enriched GO terms based on the Jaccard similarity of their gene lists.</p>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb320-1"><a href="codebook.html#cb320-1" tabindex="-1"></a><span class="co"># Create a graph from the edges</span></span>
<span id="cb320-2"><a href="codebook.html#cb320-2" tabindex="-1"></a>g <span class="ot">&lt;-</span> <span class="fu">graph_from_data_frame</span>(edges, <span class="at">directed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb320-3"><a href="codebook.html#cb320-3" tabindex="-1"></a></span>
<span id="cb320-4"><a href="codebook.html#cb320-4" tabindex="-1"></a><span class="co"># Visualize the graph using ggraph</span></span>
<span id="cb320-5"><a href="codebook.html#cb320-5" tabindex="-1"></a><span class="fu">ggraph</span>(g, <span class="at">layout =</span> <span class="st">&#39;fr&#39;</span>) <span class="sc">+</span>  <span class="co"># Fruchterman-Reingold layout</span></span>
<span id="cb320-6"><a href="codebook.html#cb320-6" tabindex="-1"></a>  <span class="fu">geom_edge_link</span>(<span class="fu">aes</span>(<span class="at">edge_alpha =</span> <span class="fl">0.5</span>)) <span class="sc">+</span>  <span class="co"># Edges with some transparency</span></span>
<span id="cb320-7"><a href="codebook.html#cb320-7" tabindex="-1"></a>  <span class="fu">geom_node_point</span>(<span class="at">size =</span> <span class="dv">5</span>, </span>
<span id="cb320-8"><a href="codebook.html#cb320-8" tabindex="-1"></a>                  <span class="at">color =</span> <span class="st">&quot;blue&quot;</span>) <span class="sc">+</span> </span>
<span id="cb320-9"><a href="codebook.html#cb320-9" tabindex="-1"></a>  <span class="fu">geom_node_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> name), </span>
<span id="cb320-10"><a href="codebook.html#cb320-10" tabindex="-1"></a>                 <span class="at">repel =</span> <span class="cn">TRUE</span>, </span>
<span id="cb320-11"><a href="codebook.html#cb320-11" tabindex="-1"></a>                 <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span>  <span class="co"># Node labels</span></span>
<span id="cb320-12"><a href="codebook.html#cb320-12" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span>  <span class="co"># Clean theme without axes</span></span>
<span id="cb320-13"><a href="codebook.html#cb320-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">&quot;GO Terms Jaccard Overlap Network&quot;</span>) </span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-188-1.png" width="672" /></p>
<p>The resulting network plots shows the relationships between the enriched GO terms based on either Semantic or Jaccard similarity of their gene lists.</p>
</div>
</div>
<div id="visualizing-enrichment-results-1" class="section level3 hasAnchor" number="9.10.2">
<h3><span class="header-section-number">9.10.2</span> Visualizing Enrichment Results<a href="codebook.html#visualizing-enrichment-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Network Plot</strong>:
The <code>emapplot()</code> function generates a network plot where each node represents an enriched gene set, and edges between nodes indicate the similarity or overlap between those gene sets.</p>
<div class="sourceCode" id="cb321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb321-1"><a href="codebook.html#cb321-1" tabindex="-1"></a>pwt <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(</span>
<span id="cb321-2"><a href="codebook.html#cb321-2" tabindex="-1"></a>  ego,</span>
<span id="cb321-3"><a href="codebook.html#cb321-3" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;JC&quot;</span>,</span>
<span id="cb321-4"><a href="codebook.html#cb321-4" tabindex="-1"></a>  <span class="at">semData =</span> <span class="cn">NULL</span></span>
<span id="cb321-5"><a href="codebook.html#cb321-5" tabindex="-1"></a>)</span>
<span id="cb321-6"><a href="codebook.html#cb321-6" tabindex="-1"></a></span>
<span id="cb321-7"><a href="codebook.html#cb321-7" tabindex="-1"></a><span class="co"># category labels</span></span>
<span id="cb321-8"><a href="codebook.html#cb321-8" tabindex="-1"></a><span class="fu">emapplot</span>(pwt, <span class="at">showCategory =</span> <span class="dv">30</span>, </span>
<span id="cb321-9"><a href="codebook.html#cb321-9" tabindex="-1"></a>         <span class="at">node_label =</span> <span class="st">&quot;category&quot;</span>, </span>
<span id="cb321-10"><a href="codebook.html#cb321-10" tabindex="-1"></a>         <span class="at">cex_label_group =</span> <span class="dv">3</span>, </span>
<span id="cb321-11"><a href="codebook.html#cb321-11" tabindex="-1"></a>         <span class="at">layout.params =</span> <span class="fu">list</span>(<span class="st">&quot;mds&quot;</span>)) <span class="sc">+</span></span>
<span id="cb321-12"><a href="codebook.html#cb321-12" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-189-1.png" width="672" /></p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="codebook.html#cb322-1" tabindex="-1"></a><span class="co"># group labels </span></span>
<span id="cb322-2"><a href="codebook.html#cb322-2" tabindex="-1"></a><span class="fu">emapplot</span>(pwt, <span class="at">showCategory =</span> <span class="dv">150</span>, </span>
<span id="cb322-3"><a href="codebook.html#cb322-3" tabindex="-1"></a>         <span class="at">node_label =</span> <span class="st">&quot;group&quot;</span>) <span class="sc">+</span></span>
<span id="cb322-4"><a href="codebook.html#cb322-4" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-189-2.png" width="672" />
To save any of these figures if they become too cluttered, click on the Export button in the RStudio Plots tab and Save as PDF. In the pop-up window, change the PDF size to 24 x 32 to give a figure of appropriate size for the text labels.</p>
<p><strong>cnetplot</strong>: Visualizes connections between genes and top GO terms, highlighting shared pathways.This plot is particularly useful for hypothesis generation in identifying genes that may be important to several of the most affected processes.</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="codebook.html#cb323-1" tabindex="-1"></a><span class="do">## Extract the foldchanges</span></span>
<span id="cb323-2"><a href="codebook.html#cb323-2" tabindex="-1"></a>OE_foldchanges <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>log2FoldChange</span>
<span id="cb323-3"><a href="codebook.html#cb323-3" tabindex="-1"></a></span>
<span id="cb323-4"><a href="codebook.html#cb323-4" tabindex="-1"></a><span class="fu">names</span>(OE_foldchanges) <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>gene</span>
<span id="cb323-5"><a href="codebook.html#cb323-5" tabindex="-1"></a></span>
<span id="cb323-6"><a href="codebook.html#cb323-6" tabindex="-1"></a><span class="do">## cnetplot for the top 5 categories</span></span>
<span id="cb323-7"><a href="codebook.html#cb323-7" tabindex="-1"></a><span class="fu">cnetplot</span>(ego,</span>
<span id="cb323-8"><a href="codebook.html#cb323-8" tabindex="-1"></a>  <span class="at">categorySize=</span><span class="st">&quot;padj&quot;</span>,</span>
<span id="cb323-9"><a href="codebook.html#cb323-9" tabindex="-1"></a>  <span class="at">node_label =</span> <span class="st">&quot;all&quot;</span>,</span>
<span id="cb323-10"><a href="codebook.html#cb323-10" tabindex="-1"></a>  <span class="at">showCategory =</span> <span class="dv">5</span>,</span>
<span id="cb323-11"><a href="codebook.html#cb323-11" tabindex="-1"></a>  <span class="at">foldChange =</span> OE_foldchanges,</span>
<span id="cb323-12"><a href="codebook.html#cb323-12" tabindex="-1"></a>  <span class="at">vertex.label.font =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-190-1.png" width="672" /></p>
</div>
<div id="gsea-using-clusterprofiler-and-fgsea-1" class="section level3 hasAnchor" number="9.10.3">
<h3><span class="header-section-number">9.10.3</span> GSEA Using clusterProfiler and fgsea<a href="codebook.html#gsea-using-clusterprofiler-and-fgsea-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prepare fold changes, sort by expression, and run GSEA with <code>gseGO()</code> for clusterProfiler or <code>fgseaSimple()</code> for <code>fgsea</code>.</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="codebook.html#cb324-1" tabindex="-1"></a><span class="do">## Extract the foldchanges</span></span>
<span id="cb324-2"><a href="codebook.html#cb324-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>log2FoldChange</span>
<span id="cb324-3"><a href="codebook.html#cb324-3" tabindex="-1"></a></span>
<span id="cb324-4"><a href="codebook.html#cb324-4" tabindex="-1"></a><span class="do">## Name each fold change with the gene name</span></span>
<span id="cb324-5"><a href="codebook.html#cb324-5" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span>
<span id="cb324-6"><a href="codebook.html#cb324-6" tabindex="-1"></a></span>
<span id="cb324-7"><a href="codebook.html#cb324-7" tabindex="-1"></a><span class="do">## Sort fold changes in decreasing order</span></span>
<span id="cb324-8"><a href="codebook.html#cb324-8" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> <span class="fu">sort</span>(foldchanges, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>We can explore the enrichment of BP Gene Ontology terms using gene set enrichment analysis (GSEA) using gene sets associated with BP Gene Ontology terms</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="codebook.html#cb325-1" tabindex="-1"></a>gseaGO <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">gseGO</span>(</span>
<span id="cb325-2"><a href="codebook.html#cb325-2" tabindex="-1"></a>  <span class="at">geneList =</span> foldchanges,</span>
<span id="cb325-3"><a href="codebook.html#cb325-3" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb325-4"><a href="codebook.html#cb325-4" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb325-5"><a href="codebook.html#cb325-5" tabindex="-1"></a>  <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb325-6"><a href="codebook.html#cb325-6" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb325-7"><a href="codebook.html#cb325-7" tabindex="-1"></a>  <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb325-8"><a href="codebook.html#cb325-8" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb325-9"><a href="codebook.html#cb325-9" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb325-10"><a href="codebook.html#cb325-10" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb325-11"><a href="codebook.html#cb325-11" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb325-12"><a href="codebook.html#cb325-12" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">&quot;fgsea&quot;</span></span>
<span id="cb325-13"><a href="codebook.html#cb325-13" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>## preparing geneSet collections...</code></pre>
<pre><code>## GSEA analysis...</code></pre>
<pre><code>## leading edge analysis...</code></pre>
<pre><code>## done...</code></pre>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="codebook.html#cb330-1" tabindex="-1"></a>gseaGO_results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(gseaGO)</span></code></pre></div>
</div>
<div id="visualizing-gsea-results-1" class="section level3 hasAnchor" number="9.10.4">
<h3><span class="header-section-number">9.10.4</span> Visualizing GSEA Results<a href="codebook.html#visualizing-gsea-results-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Visualize GSEA results with <code>dotplot()</code> for significant categories and <code>gseaplot2()</code> for gene set ranks.</p>
<div class="sourceCode" id="cb331"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb331-1"><a href="codebook.html#cb331-1" tabindex="-1"></a><span class="fu">dotplot</span>(gseaGO, <span class="at">showCategory =</span> <span class="dv">30</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-193-1.png" width="672" /></p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="codebook.html#cb332-1" tabindex="-1"></a><span class="fu">gseaplot2</span>(gseaGO, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-193-2.png" width="672" /></p>
<p>The emapplot clusters the 50 most significant (by padj) GO terms to visualize relationships between terms.</p>
<div class="sourceCode" id="cb333"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb333-1"><a href="codebook.html#cb333-1" tabindex="-1"></a><span class="co"># find the overlap between the top 50 terms</span></span>
<span id="cb333-2"><a href="codebook.html#cb333-2" tabindex="-1"></a>pwt <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(</span>
<span id="cb333-3"><a href="codebook.html#cb333-3" tabindex="-1"></a>  gseaGO,</span>
<span id="cb333-4"><a href="codebook.html#cb333-4" tabindex="-1"></a>  <span class="at">method =</span> <span class="st">&quot;JC&quot;</span>,</span>
<span id="cb333-5"><a href="codebook.html#cb333-5" tabindex="-1"></a>  <span class="at">semData =</span> <span class="cn">NULL</span></span>
<span id="cb333-6"><a href="codebook.html#cb333-6" tabindex="-1"></a>  )</span>
<span id="cb333-7"><a href="codebook.html#cb333-7" tabindex="-1"></a></span>
<span id="cb333-8"><a href="codebook.html#cb333-8" tabindex="-1"></a><span class="fu">emapplot</span>(pwt, <span class="at">showCategory =</span> <span class="dv">30</span>, <span class="at">node_label =</span> <span class="st">&quot;all&quot;</span>, </span>
<span id="cb333-9"><a href="codebook.html#cb333-9" tabindex="-1"></a>         <span class="at">cex_label_group =</span> <span class="dv">3</span>, </span>
<span id="cb333-10"><a href="codebook.html#cb333-10" tabindex="-1"></a>         <span class="at">layout.params =</span> <span class="fu">list</span>(<span class="st">&quot;mds&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb333-11"><a href="codebook.html#cb333-11" tabindex="-1"></a>         <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="2024_RNA-seq-analysis_files/figure-html/unnamed-chunk-194-1.png" width="672" /></p>
</div>
<div id="gsea-using-fgsea-1" class="section level3 hasAnchor" number="9.10.5">
<h3><span class="header-section-number">9.10.5</span> GSEA using fgsea<a href="codebook.html#gsea-using-fgsea-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>GO annotations are updated regularly, and the GO terms associated with genes can change over time. This can lead to inconsistencies in the results of GO enrichment analyses when using different versions of the GO annotations. To address this issue, the <code>GOenrichment</code> package provides <code>hGOBP.gmt</code> a recently downloaded version of <code>BP</code> GO annotations. This file is a list of GO terms and their associated genes, which is used as input for the enrichment analysis.</p>
<p>To run GSEA using the <code>fgsea</code> package, we need to load the gene sets from the <code>hGOBP.gmt</code> file and the gene-level statistics from the <code>foldchanges</code> vector. We can then run the GSEA analysis using the <code>fgseaSimple</code> function.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="codebook.html#cb334-1" tabindex="-1"></a><span class="co"># Let&#39;s look at the hGOBP.gmt file</span></span>
<span id="cb334-2"><a href="codebook.html#cb334-2" tabindex="-1"></a>hGOBP.gmt[<span class="dv">1</span>]</span></code></pre></div>
<pre><code>## $`MITOCHONDRIAL GENOME MAINTENANCE`
##  [1] &quot;AKT3&quot;     &quot;CHCHD4&quot;   &quot;DNA2&quot;     &quot;DNAJA3&quot;   &quot;ENDOG&quot;    &quot;FLCN&quot;    
##  [7] &quot;LIG3&quot;     &quot;LONP1&quot;    &quot;MEF2A&quot;    &quot;METTL4&quot;   &quot;MGME1&quot;    &quot;MPV17&quot;   
## [13] &quot;OPA1&quot;     &quot;PARP1&quot;    &quot;PIF1&quot;     &quot;POLG&quot;     &quot;POLG2&quot;    &quot;PPARGC1A&quot;
## [19] &quot;PRIMPOL&quot;  &quot;RRM1&quot;     &quot;RRM2B&quot;    &quot;SESN2&quot;    &quot;SLC25A33&quot; &quot;SLC25A36&quot;
## [25] &quot;SLC25A4&quot;  &quot;SSBP1&quot;    &quot;STOX1&quot;    &quot;TOP3A&quot;    &quot;TP53&quot;     &quot;TWNK&quot;    
## [31] &quot;TYMP&quot;</code></pre>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="codebook.html#cb336-1" tabindex="-1"></a>fgseaRes <span class="ot">&lt;-</span>  fgsea<span class="sc">::</span><span class="fu">fgseaSimple</span>(<span class="at">pathways =</span> hGOBP.gmt, </span>
<span id="cb336-2"><a href="codebook.html#cb336-2" tabindex="-1"></a>                                <span class="at">stats =</span> foldchanges, </span>
<span id="cb336-3"><a href="codebook.html#cb336-3" tabindex="-1"></a>                                <span class="at">nperm =</span> <span class="dv">1000</span>, <span class="at">maxSize =</span> <span class="dv">300</span>, </span>
<span id="cb336-4"><a href="codebook.html#cb336-4" tabindex="-1"></a>                                <span class="at">minSize =</span> <span class="dv">20</span>)</span>
<span id="cb336-5"><a href="codebook.html#cb336-5" tabindex="-1"></a></span>
<span id="cb336-6"><a href="codebook.html#cb336-6" tabindex="-1"></a>fgsea <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(fgseaRes,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb336-7"><a href="codebook.html#cb336-7" tabindex="-1"></a></span>
<span id="cb336-8"><a href="codebook.html#cb336-8" tabindex="-1"></a>w <span class="ot">=</span> <span class="fu">which</span>(fgsea<span class="sc">$</span>ES <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb336-9"><a href="codebook.html#cb336-9" tabindex="-1"></a></span>
<span id="cb336-10"><a href="codebook.html#cb336-10" tabindex="-1"></a>fposgsea <span class="ot">&lt;-</span> fgsea[w,]</span>
<span id="cb336-11"><a href="codebook.html#cb336-11" tabindex="-1"></a></span>
<span id="cb336-12"><a href="codebook.html#cb336-12" tabindex="-1"></a>fposgsea <span class="ot">&lt;-</span> fposgsea <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(padj)</span></code></pre></div>
<p>The <code>fgseaSimple</code> function returns a list of enriched gene sets and the enrichment score (ES).</p>
<p>We are going to compare these results to running the GO enrichment function <code>runGORESP</code>.</p>
<p>Let’s first load the <code>GOenrichment</code> package and check the available functions:</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="codebook.html#cb337-1" tabindex="-1"></a><span class="fu">library</span>(GOenrichment)</span>
<span id="cb337-2"><a href="codebook.html#cb337-2" tabindex="-1"></a></span>
<span id="cb337-3"><a href="codebook.html#cb337-3" tabindex="-1"></a><span class="fu">ls</span>(<span class="st">&quot;package:GOenrichment&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;compSCORE&quot;  &quot;hGOBP.gmt&quot;  &quot;hyperG&quot;     &quot;runGORESP&quot;  &quot;runNetwork&quot;
## [6] &quot;visSetup&quot;</code></pre>
</div>
<div id="goenrichment-package-analysis-1" class="section level3 hasAnchor" number="9.10.6">
<h3><span class="header-section-number">9.10.6</span> <code>GOenrichment</code> Package Analysis<a href="codebook.html#goenrichment-package-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><code>runGORESP</code> uses over-representation analysis to identify enriched GO terms and returns two <code>data.frames</code>; of enriched GO terms (nodes) and GO term relationships (edges).</p>
<p><code>compSCORE</code>evaluates a matrix of fitness scores to identify the <code>querySet</code> with scores above a specified significance threshold.</p>
<p>We’ll use a significance cutoff of 0.58, corresponding to a 1.5x change in expression.</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="codebook.html#cb339-1" tabindex="-1"></a><span class="fu">args</span>(runGORESP)</span></code></pre></div>
<pre><code>## function (scoreMat, curr_exp = &quot;test&quot;, fdrThresh = 0.2, bp_path = NULL, 
##     bp_input = NULL, go_path = NULL, go_input = NULL, minSetSize = 5, 
##     maxSetSize = 300) 
## NULL</code></pre>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="codebook.html#cb341-1" tabindex="-1"></a><span class="co"># ?runGORESP</span></span>
<span id="cb341-2"><a href="codebook.html#cb341-2" tabindex="-1"></a></span>
<span id="cb341-3"><a href="codebook.html#cb341-3" tabindex="-1"></a><span class="co"># Define the query set</span></span>
<span id="cb341-4"><a href="codebook.html#cb341-4" tabindex="-1"></a>matx <span class="ot">&lt;-</span> <span class="fu">cbind</span>(foldchanges,foldchanges)</span>
<span id="cb341-5"><a href="codebook.html#cb341-5" tabindex="-1"></a></span>
<span id="cb341-6"><a href="codebook.html#cb341-6" tabindex="-1"></a>scoreMat <span class="ot">=</span> <span class="fu">compSCORE</span>(matx, <span class="at">coln =</span> <span class="dv">1</span>, <span class="at">sig =</span>  <span class="fl">0.58</span>)</span>
<span id="cb341-7"><a href="codebook.html#cb341-7" tabindex="-1"></a></span>
<span id="cb341-8"><a href="codebook.html#cb341-8" tabindex="-1"></a><span class="fu">head</span>(scoreMat)</span></code></pre></div>
<div class="kable-table">
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">index</th>
<th align="right">score</th>
<th align="left">gene</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">HSPA6</td>
<td align="right">1</td>
<td align="right">6.246527</td>
<td align="left">HSPA6</td>
</tr>
<tr class="even">
<td align="left">MOV10</td>
<td align="right">1</td>
<td align="right">5.083302</td>
<td align="left">MOV10</td>
</tr>
<tr class="odd">
<td align="left">ASCL1</td>
<td align="right">1</td>
<td align="right">4.441095</td>
<td align="left">ASCL1</td>
</tr>
<tr class="even">
<td align="left">HSPA7</td>
<td align="right">1</td>
<td align="right">3.637043</td>
<td align="left">HSPA7</td>
</tr>
<tr class="odd">
<td align="left">SCRT1</td>
<td align="right">1</td>
<td align="right">2.925522</td>
<td align="left">SCRT1</td>
</tr>
<tr class="even">
<td align="left">SIGLEC14</td>
<td align="right">1</td>
<td align="right">2.614563</td>
<td align="left">SIGLEC14</td>
</tr>
</tbody>
</table>
</div>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="codebook.html#cb342-1" tabindex="-1"></a>hresp <span class="ot">=</span> <span class="fu">runGORESP</span>(<span class="at">fdrThresh =</span> <span class="fl">0.2</span>, <span class="at">scoreMat =</span> scoreMat,</span>
<span id="cb342-2"><a href="codebook.html#cb342-2" tabindex="-1"></a>  <span class="at">bp_input =</span> hGOBP.gmt, <span class="at">go_input =</span> <span class="cn">NULL</span>, <span class="at">minSetSize =</span> <span class="dv">20</span>,</span>
<span id="cb342-3"><a href="codebook.html#cb342-3" tabindex="-1"></a>  <span class="at">maxSetSize =</span> <span class="dv">300</span>)</span>
<span id="cb342-4"><a href="codebook.html#cb342-4" tabindex="-1"></a></span>
<span id="cb342-5"><a href="codebook.html#cb342-5" tabindex="-1"></a><span class="fu">names</span>(hresp<span class="sc">$</span>edgeMat)</span></code></pre></div>
<pre><code>## [1] &quot;source&quot;       &quot;target&quot;       &quot;overlapCoeff&quot; &quot;width&quot;       
## [5] &quot;label&quot;</code></pre>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="codebook.html#cb344-1" tabindex="-1"></a><span class="fu">names</span>(hresp<span class="sc">$</span>enrichInfo)</span></code></pre></div>
<pre><code>##  [1] &quot;filename&quot;            &quot;term&quot;                &quot;nGenes&quot;             
##  [4] &quot;nQuery&quot;              &quot;nOverlap&quot;            &quot;querySetFraction&quot;   
##  [7] &quot;geneSetFraction&quot;     &quot;foldEnrichment&quot;      &quot;P&quot;                  
## [10] &quot;FDR&quot;                 &quot;overlapGenes&quot;        &quot;maxOverlapGeneScore&quot;
## [13] &quot;cluster&quot;             &quot;id&quot;                  &quot;size&quot;               
## [16] &quot;formattedLabel&quot;</code></pre>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb346-1"><a href="codebook.html#cb346-1" tabindex="-1"></a><span class="fu">head</span>(hresp<span class="sc">$</span>enrichInfo[,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">10</span>)])</span></code></pre></div>
<div class="kable-table">
<table>
<colgroup>
<col width="59%" />
<col width="9%" />
<col width="9%" />
<col width="11%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">term</th>
<th align="right">nGenes</th>
<th align="right">nQuery</th>
<th align="right">nOverlap</th>
<th align="right">FDR</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">EXTRACELLULAR MATRIX ORGANIZATION</td>
<td align="right">293</td>
<td align="right">722</td>
<td align="right">39</td>
<td align="right">0.00000</td>
</tr>
<tr class="even">
<td align="left">EXTRACELLULAR STRUCTURE ORGANIZATION</td>
<td align="right">294</td>
<td align="right">724</td>
<td align="right">39</td>
<td align="right">0.00000</td>
</tr>
<tr class="odd">
<td align="left">EXTERNAL ENCAPSULATING STRUCTURE ORGANIZATION</td>
<td align="right">294</td>
<td align="right">724</td>
<td align="right">39</td>
<td align="right">0.00000</td>
</tr>
<tr class="even">
<td align="left">PROTEIN REFOLDING</td>
<td align="right">27</td>
<td align="right">727</td>
<td align="right">8</td>
<td align="right">0.00371</td>
</tr>
<tr class="odd">
<td align="left">REGULATION OF VASCULATURE DEVELOPMENT</td>
<td align="right">273</td>
<td align="right">739</td>
<td align="right">25</td>
<td align="right">0.02190</td>
</tr>
<tr class="even">
<td align="left">CELL ADHESION MEDIATED BY INTEGRIN</td>
<td align="right">82</td>
<td align="right">704</td>
<td align="right">12</td>
<td align="right">0.02420</td>
</tr>
</tbody>
</table>
</div>
<p>Let’s check the overlap between the enriched terms found using <code>runGORESP</code> and those found using <code>fgseaSimple</code> as they used the same GO term libraries:</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="codebook.html#cb347-1" tabindex="-1"></a>w <span class="ot">=</span> <span class="fu">which</span>(fposgsea<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.2</span>)</span>
<span id="cb347-2"><a href="codebook.html#cb347-2" tabindex="-1"></a></span>
<span id="cb347-3"><a href="codebook.html#cb347-3" tabindex="-1"></a>lens <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">intersect</span>(fposgsea<span class="sc">$</span>pathway[w],hresp<span class="sc">$</span>enrichInfo<span class="sc">$</span>term))</span>
<span id="cb347-4"><a href="codebook.html#cb347-4" tabindex="-1"></a></span>
<span id="cb347-5"><a href="codebook.html#cb347-5" tabindex="-1"></a><span class="fu">length</span>(w)</span></code></pre></div>
<pre><code>## [1] 555</code></pre>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="codebook.html#cb349-1" tabindex="-1"></a><span class="fu">dim</span>(hresp<span class="sc">$</span>enrichInfo)</span></code></pre></div>
<pre><code>## [1] 57 16</code></pre>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="codebook.html#cb351-1" tabindex="-1"></a>percent_overlap <span class="ot">&lt;-</span> lens<span class="sc">/</span><span class="fu">nrow</span>(hresp<span class="sc">$</span>enrichInfo)<span class="sc">*</span><span class="dv">100</span></span>
<span id="cb351-2"><a href="codebook.html#cb351-2" tabindex="-1"></a></span>
<span id="cb351-3"><a href="codebook.html#cb351-3" tabindex="-1"></a>percent_overlap</span></code></pre></div>
<pre><code>## [1] 77.19298</code></pre>
<p>80%, that’s very good because we are using two different GO enrichment methods, over-representation analysis and GSEA.</p>
<p>The <code>visNetwork</code> function allows us to create interactive network visualizations where enriched GO terms are nodes and the relationships between terms are edges.</p>
<p>The equivalent function in the <code>GOenrichment</code> package is <code>runNetwork</code>:</p>
<p>We use the<code>visSetup</code> function to prepare the data for network visualization. We then run the <code>runNetwork</code> function to generate the interactive network plot.</p>
<pre><code>## dplyr loaded: TRUE</code></pre>
<div id="htmlwidget-333a63a3d838c23c481b" style="width:100%;height:480px;" class="visNetwork html-widget"></div>
<script type="application/json" data-for="htmlwidget-333a63a3d838c23c481b">{"x":{"nodes":{"id":["1","2","3","4","5","6","7","9","8","10","11","12","13","14","15","16","18","17","19","20","22","21","23","24","25","26","27","28","29","32","31","30","44","36","38","37","40","34","41","46","42","43","33","45","39","35","48","52","51","47","50","54","53","49","55","57","56"],"filename":["test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test","test"],"term":["EXTRACELLULAR MATRIX ORGANIZATION","EXTRACELLULAR STRUCTURE ORGANIZATION","EXTERNAL ENCAPSULATING STRUCTURE ORGANIZATION","PROTEIN REFOLDING","REGULATION OF VASCULATURE DEVELOPMENT","CELL ADHESION MEDIATED BY INTEGRIN","RESPONSE TO GLUCOCORTICOID","RESPONSE TO CORTICOSTEROID","REGULATION OF ANGIOGENESIS","RESPONSE TO TOPOLOGICALLY INCORRECT PROTEIN","RESPONSE TO UNFOLDED PROTEIN","REGULATION OF ENDOPLASMIC RETICULUM STRESS-INDUCED INTRINSIC APOPTOTIC SIGNALING PATHWAY","INTRINSIC APOPTOTIC SIGNALING PATHWAY","REGULATION OF DOPAMINE SECRETION","CELLULAR RESPONSE TO CORTICOSTEROID STIMULUS","REGULATION OF CATECHOLAMINE SECRETION","BASEMENT MEMBRANE ORGANIZATION","CELLULAR RESPONSE TO GLUCOCORTICOID STIMULUS","'DE NOVO' PROTEIN FOLDING","DOPAMINE SECRETION","INTRINSIC APOPTOTIC SIGNALING PATHWAY BY P53 CLASS MEDIATOR","RESPONSE TO DEXAMETHASONE","CATECHOLAMINE SECRETION","REGULATION OF EXTRINSIC APOPTOTIC SIGNALING PATHWAY IN ABSENCE OF LIGAND","OVULATION CYCLE","CHAPERONE COFACTOR-DEPENDENT PROTEIN REFOLDING","POSITIVE REGULATION OF VASCULATURE DEVELOPMENT","VASCULAR ENDOTHELIAL GROWTH FACTOR SIGNALING PATHWAY","POSITIVE REGULATION OF DENDRITE EXTENSION","REGULATION OF INTRINSIC APOPTOTIC SIGNALING PATHWAY","POSTSYNAPTIC SPECIALIZATION ORGANIZATION","INTEGRIN-MEDIATED SIGNALING PATHWAY","NEPHRON DEVELOPMENT","ANIMAL ORGAN REGENERATION","POSITIVE REGULATION OF ANGIOGENESIS","CELLULAR RESPONSE TO VASCULAR ENDOTHELIAL GROWTH FACTOR STIMULUS","'DE NOVO' POST-TRANSLATIONAL PROTEIN FOLDING","CELL-MATRIX ADHESION","INTRINSIC APOPTOTIC SIGNALING PATHWAY IN RESPONSE TO ENDOPLASMIC RETICULUM STRESS","REGULATION OF DENDRITE EXTENSION","CELLULAR RESPONSE TO CAMP","CELLULAR RESPONSE TO DEXAMETHASONE STIMULUS","POLYSACCHARIDE METABOLIC PROCESS","EXTRINSIC APOPTOTIC SIGNALING PATHWAY IN ABSENCE OF LIGAND","NEUROMUSCULAR PROCESS CONTROLLING BALANCE","CALCIUM-ION REGULATED EXOCYTOSIS","POSITIVE REGULATION OF EPITHELIAL CELL PROLIFERATION","CELLULAR RESPONSE TO TOPOLOGICALLY INCORRECT PROTEIN","RESPONSE TO TEMPERATURE STIMULUS","SIGNAL TRANSDUCTION IN ABSENCE OF LIGAND","AMINO SUGAR METABOLIC PROCESS","REGULATION OF SPROUTING ANGIOGENESIS","REGULATION OF BLOOD CIRCULATION","REGULATION OF RESPONSE TO ENDOPLASMIC RETICULUM STRESS","DOPAMINE TRANSPORT","POSTSYNAPTIC DENSITY ORGANIZATION","REGULATION OF HEART RATE BY CARDIAC CONDUCTION"],"nGenes":[293,294,294,27,273,82,121,142,267,142,129,30,287,34,57,47,37,48,39,39,78,40,53,41,68,31,159,43,22,165,45,101,151,61,156,62,35,227,60,25,48,25,90,61,47,60,193,94,160,63,38,38,232,78,52,39,39],"nQuery":[722,724,724,727,739,704,714,729,728,729,713,699,731,700,750,726,699,729,698,698,713,700,728,701,748,752,712,701,713,715,702,734,739,726,739,727,748,717,725,714,701,714,714,726,700,725,728,712,741,727,750,750,724,748,702,751,751],"nOverlap":[39,39,39,8,25,12,15,16,24,16,15,7,24,7,9,8,7,8,7,7,10,7,8,7,9,6,15,7,5,15,7,11,14,8,14,8,6,18,8,5,7,5,10,8,7,8,16,10,14,8,6,6,18,9,7,6,6],"querySetFraction":[5.4,5.4,5.4,1.1,3.4,1.7,2.1,2.2,3.3,2.2,2.1,1,3.3,1,1.2,1.1,1,1.1,1,1,1.4,1,1.1,1,1.2,0.8,2.1,1,0.7,2.1,1,1.5,1.9,1.1,1.9,1.1,0.8,2.5,1.1,0.7,1,0.7,1.4,1.1,1,1.1,2.2,1.4,1.9,1.1,0.8,0.8,2.5,1.2,1,0.8,0.8],"geneSetFraction":[13.3,13.3,13.3,29.6,9.199999999999999,14.6,12.4,11.3,9,11.3,11.6,23.3,8.4,20.6,15.8,17,18.9,16.7,17.9,17.9,12.8,17.5,15.1,17.1,13.2,19.4,9.4,16.3,22.7,9.1,15.6,10.9,9.300000000000001,13.1,9,12.9,17.1,7.9,13.3,20,14.6,20,11.1,13.1,14.9,13.3,8.300000000000001,10.6,8.800000000000001,12.7,15.8,15.8,7.8,11.5,13.5,15.4,15.4],"foldEnrichment":[3.62,3.6,3.6,8.050000000000001,2.49,3.98,3.37,3.06,2.44,3.06,3.16,6.34,2.27,5.59,4.29,4.62,5.14,4.53,4.88,4.88,3.48,4.75,4.1,4.64,3.6,5.26,2.56,4.42,6.17,2.47,4.23,2.96,2.52,3.56,2.44,3.5,4.66,2.15,3.62,5.43,3.96,5.43,3.02,3.56,4.05,3.62,2.25,2.89,2.38,3.45,4.29,4.29,2.11,3.13,3.66,4.18,4.18],"P":[3.61e-12,4.02e-12,4.02e-12,3.87e-06,2.85e-05,4.41e-05,3.86e-05,6.999999999999999e-05,5.51e-05,6.999999999999999e-05,8.16e-05,8.64e-05,0.000169,0.000201,0.000218,0.000286,0.000349,0.000332,0.00049,0.00049,0.000564,0.000575,0.000664,0.000671,0.000835,0.000816,0.000803,0.000903,0.00104,0.00117,0.00119,0.00122,0.0014,0.00171,0.00191,0.0019,0.00159,0.00192,0.00153,0.00192,0.00176,0.00192,0.00173,0.00171,0.00155,0.00153,0.00214,0.0024,0.00241,0.0021,0.00246,0.00246,0.00244,0.00223,0.00282,0.00282,0.00282],"overlapGenes":["COL23A1(1.73)|OLFML2A(1.43)|ADAMTS7(1.28)|MMP28(1.25)|COL5A1(1.16)|ADAMTS2(1.13)|DPP4(1.07)|COL13A1(1.05)|ADAMTS1(1.03)|CREB3L1(1.01)|ADAMTSL3(1)|RECK(1)|ADAMTS10(0.98)|COL3A1(0.94)|LAMC1(0.93)|EGFLAM(0.91)|HAPLN2(0.89)|CSGALNACT1(0.87)|SMOC2(0.86)|MATN3(0.84)|COL2A1(0.83)|SULF1(0.82)|MMP2(0.82)|ITGA8(0.81)|THSD4(0.81)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|ABI3BP(0.75)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)|LUM(0.7)|ADAMTSL4(0.68)|NPNT(0.67)|KAZALD1(0.63)|SLC2A10(0.62)|ADAMTS4(0.62)|PDGFRA(0.58)","COL23A1(1.73)|OLFML2A(1.43)|ADAMTS7(1.28)|MMP28(1.25)|COL5A1(1.16)|ADAMTS2(1.13)|DPP4(1.07)|COL13A1(1.05)|ADAMTS1(1.03)|CREB3L1(1.01)|ADAMTSL3(1)|RECK(1)|ADAMTS10(0.98)|COL3A1(0.94)|LAMC1(0.93)|EGFLAM(0.91)|HAPLN2(0.89)|CSGALNACT1(0.87)|SMOC2(0.86)|MATN3(0.84)|COL2A1(0.83)|SULF1(0.82)|MMP2(0.82)|ITGA8(0.81)|THSD4(0.81)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|ABI3BP(0.75)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)|LUM(0.7)|ADAMTSL4(0.68)|NPNT(0.67)|KAZALD1(0.63)|SLC2A10(0.62)|ADAMTS4(0.62)|PDGFRA(0.58)","COL23A1(1.73)|OLFML2A(1.43)|ADAMTS7(1.28)|MMP28(1.25)|COL5A1(1.16)|ADAMTS2(1.13)|DPP4(1.07)|COL13A1(1.05)|ADAMTS1(1.03)|CREB3L1(1.01)|ADAMTSL3(1)|RECK(1)|ADAMTS10(0.98)|COL3A1(0.94)|LAMC1(0.93)|EGFLAM(0.91)|HAPLN2(0.89)|CSGALNACT1(0.87)|SMOC2(0.86)|MATN3(0.84)|COL2A1(0.83)|SULF1(0.82)|MMP2(0.82)|ITGA8(0.81)|THSD4(0.81)|PAPLN(0.77)|NTN4(0.77)|LAMA1(0.76)|ABI3BP(0.75)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)|LUM(0.7)|ADAMTSL4(0.68)|NPNT(0.67)|KAZALD1(0.63)|SLC2A10(0.62)|ADAMTS4(0.62)|PDGFRA(0.58)","HSPA6(6.25)|HSPA7(3.64)|CRYAB(1.65)|HSPA1A(0.89)|HSPB1(0.85)|HSPA1B(0.83)|HSPA1L(0.72)|FKBP1B(0.59)","ALOX5(1.42)|MDK(1.32)|PLK2(1.04)|ADAMTS1(1.03)|HSPG2(1.02)|CREB3L1(1.01)|RECK(1)|PLXND1(0.91)|RAPGEF3(0.9)|CXCL12(0.88)|FUT1(0.86)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|SULF1(0.82)|ITGA5(0.78)|TSPAN18(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|EFNA1(0.71)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","ITGA11(1.35)|PLAU(1.1)|DPP4(1.07)|SNAI2(0.84)|LPXN(0.82)|ITGA8(0.81)|ITGA5(0.78)|EFNA1(0.71)|EPHA8(0.68)|NPNT(0.67)|ITGA9(0.64)|ITGA4(0.64)","IGFBP7(1.45)|SLIT3(1.41)|PCK1(1.32)|PAPPA(1.22)|EGFR(1.09)|ZFP36L1(0.93)|ACE(0.93)|CDKN1A(0.81)|FOSL1(0.79)|DDIT4(0.72)|SSTR2(0.68)|PDX1(0.68)|AQP1(0.68)|ZFP36(0.68)|ASS1(0.62)","IGFBP7(1.45)|SLIT3(1.41)|PCK1(1.32)|PAPPA(1.22)|EGFR(1.09)|ZFP36L1(0.93)|ACE(0.93)|CDKN1A(0.81)|FOSL1(0.79)|DDIT4(0.72)|SSTR2(0.68)|PDX1(0.68)|AQP1(0.68)|ZFP36(0.68)|ASS1(0.62)|INHBA(0.59)","ALOX5(1.42)|MDK(1.32)|PLK2(1.04)|ADAMTS1(1.03)|HSPG2(1.02)|CREB3L1(1.01)|RECK(1)|PLXND1(0.91)|RAPGEF3(0.9)|FUT1(0.86)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|SULF1(0.82)|ITGA5(0.78)|TSPAN18(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|EFNA1(0.71)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","HSPA6(6.25)|CREB3L1(1.01)|CHAC1(0.92)|CTH(0.89)|HSPA1A(0.89)|HSPB1(0.85)|HSPB8(0.84)|DNAJB1(0.8)|HERPUD1(0.73)|HSPA1L(0.72)|ATF3(0.71)|CREB3L2(0.7)|EIF2AK3(0.66)|KLHL15(0.64)|CCND1(0.61)|XBP1(0.61)","HSPA6(6.25)|CREB3L1(1.01)|CHAC1(0.92)|CTH(0.89)|HSPA1A(0.89)|HSPB1(0.85)|HSPB8(0.84)|DNAJB1(0.8)|HERPUD1(0.73)|HSPA1L(0.72)|ATF3(0.71)|CREB3L2(0.7)|EIF2AK3(0.66)|CCND1(0.61)|XBP1(0.61)","CREB3L1(1.01)|HSPA1A(0.89)|HERPUD1(0.73)|PDX1(0.68)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","POU4F2(1.83)|BDKRB2(1.02)|CREB3L1(1.01)|CD44(0.94)|NUPR1(0.92)|CHAC1(0.92)|HSPA1A(0.89)|CXCL12(0.88)|HSPB1(0.85)|SNAI2(0.84)|MMP2(0.82)|CDKN1A(0.81)|CD24(0.81)|HERPUD1(0.73)|FHIT(0.72)|DDIT4(0.72)|RPS27L(0.69)|PDX1(0.68)|IER3(0.67)|ADCY10(0.66)|DYRK2(0.66)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|CXCL12(0.88)|SYT4(0.83)|SNCG(0.8)|SYT6(0.74)","PCK1(1.32)|EGFR(1.09)|ZFP36L1(0.93)|ACE(0.93)|DDIT4(0.72)|SSTR2(0.68)|AQP1(0.68)|ZFP36(0.68)|ASS1(0.62)","HRH3(2.52)|GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|CXCL12(0.88)|SYT4(0.83)|SNCG(0.8)|SYT6(0.74)","COL3A1(0.94)|LAMC1(0.93)|NTN4(0.77)|LAMA1(0.76)|NID2(0.73)|LAMB1(0.71)|RAMP2(0.7)","PCK1(1.32)|EGFR(1.09)|ZFP36L1(0.93)|DDIT4(0.72)|SSTR2(0.68)|AQP1(0.68)|ZFP36(0.68)|ASS1(0.62)","HSPA6(6.25)|HSPA7(3.64)|HSPA1A(0.89)|HSPA1B(0.83)|DNAJB1(0.8)|HSPA1L(0.72)|FKBP1B(0.59)","GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|CXCL12(0.88)|SYT4(0.83)|SNCG(0.8)|SYT6(0.74)","POU4F2(1.83)|BDKRB2(1.02)|CD44(0.94)|NUPR1(0.92)|CDKN1A(0.81)|FHIT(0.72)|DDIT4(0.72)|RPS27L(0.69)|DYRK2(0.66)|PMAIP1(0.59)","PCK1(1.32)|PAPPA(1.22)|EGFR(1.09)|ACE(0.93)|DDIT4(0.72)|AQP1(0.68)|ASS1(0.62)","HRH3(2.52)|GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|CXCL12(0.88)|SYT4(0.83)|SNCG(0.8)|SYT6(0.74)","GDNF(1.33)|SRPX(0.96)|HSPA1A(0.89)|SNAI2(0.84)|COL2A1(0.83)|HSPA1B(0.83)|INHBA(0.59)","SLIT3(1.41)|MDK(1.32)|EGFR(1.09)|ADAMTS1(1.03)|CA12(0.87)|MMP2(0.82)|NHLH2(0.78)|INHBA(0.59)|PDGFRA(0.58)","HSPA6(6.25)|HSPA7(3.64)|HSPA1A(0.89)|HSPA1B(0.83)|DNAJB1(0.8)|HSPA1L(0.72)","MDK(1.32)|RAPGEF3(0.9)|CXCL12(0.88)|FUT1(0.86)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","PTP4A3(0.89)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|DLL1(0.74)|IL12A(0.58)|PDGFRA(0.58)","SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|SYT4(0.83)","BDKRB2(1.02)|CREB3L1(1.01)|CD44(0.94)|NUPR1(0.92)|HSPA1A(0.89)|CXCL12(0.88)|HSPB1(0.85)|SNAI2(0.84)|MMP2(0.82)|HERPUD1(0.73)|PDX1(0.68)|ADCY10(0.66)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","SHANK1(1.46)|LRRTM2(1.25)|NTNG2(0.93)|LRRC4B(0.86)|SHANK2(0.75)|NPTX1(0.67)|CNTNAP1(0.64)","ITGA11(1.35)|ADAMTS1(1.03)|COL3A1(0.94)|LAMC1(0.93)|ITGA8(0.81)|ITGA5(0.78)|LAMA1(0.76)|LAMB1(0.71)|ITGA9(0.64)|ITGA4(0.64)|SLC2A10(0.62)","FOXJ1(2.31)|DCHS1(1.37)|GDNF(1.33)|TMEM59L(0.82)|SULF1(0.82)|CD24(0.81)|NOTCH3(0.78)|DLL1(0.74)|HES5(0.7)|AQP1(0.68)|NPNT(0.67)|MAGI2(0.64)|PLCE1(0.64)|PDGFRA(0.58)","EGFR(1.09)|HSPG2(1.02)|ACE(0.93)|CXCL12(0.88)|CDKN1A(0.81)|PDX1(0.68)|CCND1(0.61)|GLI1(0.61)","MDK(1.32)|RAPGEF3(0.9)|FUT1(0.86)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|ADM2(0.82)|ITGA5(0.78)|RRAS(0.77)|DLL1(0.74)|ZC3H12A(0.72)|RAMP2(0.7)|AQP1(0.68)|XBP1(0.61)","PTP4A3(0.89)|SMOC2(0.86)|HSPB1(0.85)|PGF(0.84)|DLL1(0.74)|RAMP2(0.7)|IL12A(0.58)|PDGFRA(0.58)","HSPA6(6.25)|HSPA7(3.64)|HSPA1A(0.89)|HSPA1B(0.83)|DNAJB1(0.8)|HSPA1L(0.72)","ITGA11(1.35)|PLAU(1.1)|COL13A1(1.05)|COL3A1(0.94)|CD44(0.94)|STON1(0.93)|FUT1(0.86)|SNED1(0.86)|ITGA8(0.81)|ITGA5(0.78)|BCAM(0.78)|RRAS(0.77)|NID2(0.73)|NPNT(0.67)|THBS3(0.65)|ITGA9(0.64)|ITGA4(0.64)|CORO2B(0.63)","CREB3L1(1.01)|CHAC1(0.92)|HSPA1A(0.89)|HERPUD1(0.73)|PDX1(0.68)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|RASAL1(0.91)|SYT4(0.83)","PCK1(1.32)|KCNQ1(1.28)|ZFP36L1(0.93)|RAPGEF3(0.9)|IGFBP5(0.7)|AQP1(0.68)|ASS1(0.62)","PCK1(1.32)|EGFR(1.09)|DDIT4(0.72)|AQP1(0.68)|ASS1(0.62)","NR1D1(1.59)|PPP1R3C(1.25)|CSGALNACT1(0.87)|PHLDA2(0.85)|STBD1(0.73)|CHST1(0.71)|ADCY10(0.66)|DYRK2(0.66)|CHST7(0.63)|PYGL(0.59)","GDNF(1.33)|SRPX(0.96)|HSPA1A(0.89)|SNAI2(0.84)|COL2A1(0.83)|HSPA1B(0.83)|ERBB3(0.66)|INHBA(0.59)","POU4F2(1.83)|USH1C(1.56)|SHANK1(1.46)|CDH23(1.39)|PDE8B(0.77)|ALDH1A3(0.64)|CNTNAP1(0.64)","SNAP25(1.99)|SYT2(0.94)|SYT17(0.94)|RIMS2(0.93)|SYT4(0.83)|TSPAN18(0.78)|SYT6(0.74)|SCIN(0.61)","NOD2(1.4)|SCG2(1.38)|MDK(1.32)|EGFR(1.09)|HSPG2(1.02)|LAMC1(0.93)|NEAT1(0.92)|CXCL12(0.88)|PGF(0.84)|CDH3(0.72)|LAMB1(0.71)|PDX1(0.68)|ITGA4(0.64)|CCND1(0.61)|EGFL7(0.61)|XBP1(0.61)","CREB3L1(1.01)|CTH(0.89)|HSPA1A(0.89)|HSPB8(0.84)|HERPUD1(0.73)|ATF3(0.71)|EIF2AK3(0.66)|KLHL15(0.64)|CCND1(0.61)|XBP1(0.61)","HSPA6(6.25)|CRYAB(1.65)|SCARA5(1.2)|HSPA1A(0.89)|CXCL12(0.88)|HSPB1(0.85)|HSPA1B(0.83)|CDKN1A(0.81)|DNAJB1(0.8)|HSBP1L1(0.71)|ACADVL(0.68)|METRNL(0.67)|EIF2AK3(0.66)|SLC25A27(0.63)","GDNF(1.33)|SRPX(0.96)|HSPA1A(0.89)|SNAI2(0.84)|COL2A1(0.83)|HSPA1B(0.83)|ERBB3(0.66)|INHBA(0.59)","CSGALNACT1(0.87)|MGAT3(0.81)|CHST1(0.71)|CHST7(0.63)|ST3GAL1(0.62)|CHST6(0.61)","ALOX5(1.42)|CREB3L1(1.01)|FUT1(0.86)|ITGA5(0.78)|TSPAN18(0.78)|DLL1(0.74)","KCNQ1(1.28)|EGFR(1.09)|BDKRB2(1.02)|ACE(0.93)|CACNA1G(0.9)|ADM2(0.82)|MMP2(0.82)|ATP1B2(0.79)|ZC3H12A(0.72)|SCN1B(0.67)|ADCY10(0.66)|ITGA9(0.64)|ITGA4(0.64)|CACNA1H(0.61)|CACNA1C(0.6)|FKBP1B(0.59)|KCNJ2(0.59)|KCNIP2(0.58)","ALOX5(1.42)|CREB3L1(1.01)|NUPR1(0.92)|HSPA1A(0.89)|HERPUD1(0.73)|PDX1(0.68)|EIF2AK3(0.66)|XBP1(0.61)|PMAIP1(0.59)","GDNF(1.33)|SYT2(0.94)|SYT17(0.94)|CXCL12(0.88)|SYT4(0.83)|SNCG(0.8)|SYT6(0.74)","SHANK1(1.46)|LRRTM2(1.25)|LRRC4B(0.86)|SHANK2(0.75)|NPTX1(0.67)|CNTNAP1(0.64)","KCNQ1(1.28)|ACE(0.93)|CACNA1G(0.9)|SCN1B(0.67)|CACNA1C(0.6)|KCNJ2(0.59)"],"maxOverlapGeneScore":[1.73,1.73,1.73,6.25,1.42,1.35,1.45,1.45,1.42,6.25,6.25,1.01,1.83,1.33,1.32,2.52,0.9399999999999999,1.32,6.25,1.33,1.83,1.32,2.52,1.33,1.41,6.25,1.32,0.89,0.9399999999999999,1.02,1.46,1.35,2.31,1.09,1.32,0.89,6.25,1.35,1.01,0.9399999999999999,1.32,1.32,1.59,1.33,1.83,1.99,1.4,1.01,6.25,1.33,0.87,1.42,1.28,1.42,1.33,1.46,1.28],"size":[40,40,40,16.77,13.72,13.55,13.55,13.37,13.39,13.37,13.32,13.32,12.3,12.13,12.11,11.75,11.61,11.61,11.21,11.21,11.1,11.1,10.98,10.98,10.81,10.81,10.81,10.74,10.55,10.44,10.44,10.44,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.29,10.18,10.14,10.14,10.18,10.14,10.14,10.14,10.14,10,10,10],"formattedLabel":["EXTRACELLULAR\nMATRIX ORGANIZATION","EXTRACELLULAR\nSTRUCTURE\nORGANIZATION","EXTERNAL\nENCAPSULATING\nSTRUCTURE\nORGANIZATION","PROTEIN REFOLDING","REGULATION OF\nVASCULATURE\nDEVELOPMENT","CELL ADHESION\nMEDIATED BY\nINTEGRIN","RESPONSE TO\nGLUCOCORTICOID","RESPONSE TO\nCORTICOSTEROID","REGULATION OF\nANGIOGENESIS","RESPONSE TO\nTOPOLOGICALLY\nINCORRECT PROTEIN","RESPONSE TO\nUNFOLDED PROTEIN","REGULATION OF\nENDOPLASMIC\nRETICULUM\nSTRESS-INDUCED\nINTRINSIC APOPTOTIC\nSIGNALING PATHWAY","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY","REGULATION OF\nDOPAMINE SECRETION","CELLULAR RESPONSE\nTO CORTICOSTEROID\nSTIMULUS","REGULATION OF\nCATECHOLAMINE\nSECRETION","BASEMENT MEMBRANE\nORGANIZATION","CELLULAR RESPONSE\nTO GLUCOCORTICOID\nSTIMULUS","'DE NOVO' PROTEIN\nFOLDING","DOPAMINE SECRETION","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nBY P53 CLASS\nMEDIATOR","RESPONSE TO\nDEXAMETHASONE","CATECHOLAMINE\nSECRETION","REGULATION OF\nEXTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN ABSENCE OF\nLIGAND","OVULATION CYCLE","CHAPERONE\nCOFACTOR-DEPENDENT\nPROTEIN REFOLDING","POSITIVE REGULATION\nOF VASCULATURE\nDEVELOPMENT","VASCULAR\nENDOTHELIAL GROWTH\nFACTOR SIGNALING\nPATHWAY","POSITIVE REGULATION\nOF DENDRITE\nEXTENSION","REGULATION OF\nINTRINSIC APOPTOTIC\nSIGNALING PATHWAY","POSTSYNAPTIC\nSPECIALIZATION\nORGANIZATION","INTEGRIN-MEDIATED\nSIGNALING PATHWAY","NEPHRON DEVELOPMENT","ANIMAL ORGAN\nREGENERATION","POSITIVE REGULATION\nOF ANGIOGENESIS","CELLULAR RESPONSE\nTO VASCULAR\nENDOTHELIAL GROWTH\nFACTOR STIMULUS","'DE NOVO'\nPOST-TRANSLATIONAL\nPROTEIN FOLDING","CELL-MATRIX\nADHESION","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN RESPONSE TO\nENDOPLASMIC\nRETICULUM STRESS","REGULATION OF\nDENDRITE EXTENSION","CELLULAR RESPONSE\nTO CAMP","CELLULAR RESPONSE\nTO DEXAMETHASONE\nSTIMULUS","POLYSACCHARIDE\nMETABOLIC PROCESS","EXTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN ABSENCE OF\nLIGAND","NEUROMUSCULAR\nPROCESS CONTROLLING\nBALANCE","CALCIUM-ION\nREGULATED\nEXOCYTOSIS","POSITIVE REGULATION\nOF EPITHELIAL CELL\nPROLIFERATION","CELLULAR RESPONSE\nTO TOPOLOGICALLY\nINCORRECT PROTEIN","RESPONSE TO\nTEMPERATURE\nSTIMULUS","SIGNAL TRANSDUCTION\nIN ABSENCE OF\nLIGAND","AMINO SUGAR\nMETABOLIC PROCESS","REGULATION OF\nSPROUTING\nANGIOGENESIS","REGULATION OF BLOOD\nCIRCULATION","REGULATION OF\nRESPONSE TO\nENDOPLASMIC\nRETICULUM STRESS","DOPAMINE TRANSPORT","POSTSYNAPTIC\nDENSITY\nORGANIZATION","REGULATION OF HEART\nRATE BY CARDIAC\nCONDUCTION"],"label":["EXTRACELLULAR\nMATRIX ORGANIZATION","EXTRACELLULAR\nSTRUCTURE\nORGANIZATION","EXTERNAL\nENCAPSULATING\nSTRUCTURE\nORGANIZATION","PROTEIN REFOLDING","REGULATION OF\nVASCULATURE\nDEVELOPMENT","CELL ADHESION\nMEDIATED BY\nINTEGRIN","RESPONSE TO\nGLUCOCORTICOID","RESPONSE TO\nCORTICOSTEROID","REGULATION OF\nANGIOGENESIS","RESPONSE TO\nTOPOLOGICALLY\nINCORRECT PROTEIN","RESPONSE TO\nUNFOLDED PROTEIN","REGULATION OF\nENDOPLASMIC\nRETICULUM\nSTRESS-INDUCED\nINTRINSIC APOPTOTIC\nSIGNALING PATHWAY","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY","REGULATION OF\nDOPAMINE SECRETION","CELLULAR RESPONSE\nTO CORTICOSTEROID\nSTIMULUS","REGULATION OF\nCATECHOLAMINE\nSECRETION","BASEMENT MEMBRANE\nORGANIZATION","CELLULAR RESPONSE\nTO GLUCOCORTICOID\nSTIMULUS","'DE NOVO' PROTEIN\nFOLDING","DOPAMINE SECRETION","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nBY P53 CLASS\nMEDIATOR","RESPONSE TO\nDEXAMETHASONE","CATECHOLAMINE\nSECRETION","REGULATION OF\nEXTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN ABSENCE OF\nLIGAND","OVULATION CYCLE","CHAPERONE\nCOFACTOR-DEPENDENT\nPROTEIN REFOLDING","POSITIVE REGULATION\nOF VASCULATURE\nDEVELOPMENT","VASCULAR\nENDOTHELIAL GROWTH\nFACTOR SIGNALING\nPATHWAY","POSITIVE REGULATION\nOF DENDRITE\nEXTENSION","REGULATION OF\nINTRINSIC APOPTOTIC\nSIGNALING PATHWAY","POSTSYNAPTIC\nSPECIALIZATION\nORGANIZATION","INTEGRIN-MEDIATED\nSIGNALING PATHWAY","NEPHRON DEVELOPMENT","ANIMAL ORGAN\nREGENERATION","POSITIVE REGULATION\nOF ANGIOGENESIS","CELLULAR RESPONSE\nTO VASCULAR\nENDOTHELIAL GROWTH\nFACTOR STIMULUS","'DE NOVO'\nPOST-TRANSLATIONAL\nPROTEIN FOLDING","CELL-MATRIX\nADHESION","INTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN RESPONSE TO\nENDOPLASMIC\nRETICULUM STRESS","REGULATION OF\nDENDRITE EXTENSION","CELLULAR RESPONSE\nTO CAMP","CELLULAR RESPONSE\nTO DEXAMETHASONE\nSTIMULUS","POLYSACCHARIDE\nMETABOLIC PROCESS","EXTRINSIC APOPTOTIC\nSIGNALING PATHWAY\nIN ABSENCE OF\nLIGAND","NEUROMUSCULAR\nPROCESS CONTROLLING\nBALANCE","CALCIUM-ION\nREGULATED\nEXOCYTOSIS","POSITIVE REGULATION\nOF EPITHELIAL CELL\nPROLIFERATION","CELLULAR RESPONSE\nTO TOPOLOGICALLY\nINCORRECT PROTEIN","RESPONSE TO\nTEMPERATURE\nSTIMULUS","SIGNAL TRANSDUCTION\nIN ABSENCE OF\nLIGAND","AMINO SUGAR\nMETABOLIC PROCESS","REGULATION OF\nSPROUTING\nANGIOGENESIS","REGULATION OF BLOOD\nCIRCULATION","REGULATION OF\nRESPONSE TO\nENDOPLASMIC\nRETICULUM STRESS","DOPAMINE TRANSPORT","POSTSYNAPTIC\nDENSITY\nORGANIZATION","REGULATION OF HEART\nRATE BY CARDIAC\nCONDUCTION"],"color.background":["#FF00CC","#FF00CC","#FF00CC","#33CCFF","#33CC00","#9900FF","#FF9900","#FF9900","#33CC00","#FFFF00","#FFFF00","#FFFF00","#FFFF00","#FFCCFF","#FF9900","#FFCCFF","#FF00CC","#FF9900","#33CCFF","#FFCCFF","#FFFF00","#FF9900","#FFCCFF","#FF0000","#006600","#33CCFF","#33CC00","#33CC00","#009999","#FFFF00","#CCCC00","#9900FF","#9966FF","#6699CC","#33CC00","#33CC00","#33CCFF","#9900FF","#FFFF00","#009999","#FFCC99","#FF9900","#993300","#FF0000","#CCCCFF","#CC99CC","#CC6600","#FFFF00","#33CCFF","#FF0000","#CCFFFF","#33CC00","#99CC00","#FFFF00","#FFCCFF","#CCCC00","#99CC00"],"FDR":[5.19384e-09,5.1958e-09,5.22645e-09,0.00371,0.0219,0.0242971,0.0245492,0.0263462,0.0264,0.0272652,0.0276027,0.0280679,0.0498,0.055,0.0558,0.06850000000000001,0.0732525,0.0758215,0.09481580000000001,0.0956212,0.09820379999999999,0.101897,0.105752,0.105993,0.117584,0.118598,0.120378,0.124,0.138,0.14706,0.147243,0.149598,0.157275,0.157341,0.15762,0.157851,0.158111,0.158328,0.158527,0.15876,0.159298,0.161246,0.161619,0.162414,0.162475,0.162821,0.16762,0.17231,0.173034,0.173615,0.174473,0.177354,0.177489,0.17767,0.188384,0.18981,0.190318],"color.border":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"],"borderWidthSelected":[4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4],"color.highlight.border":["#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066"],"color.highlight.background":["#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff"],"color.hover.background":["#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066","#000066"],"color.hover.border":["#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff","#c0b3ff"],"font.face":["Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial","Arial"],"shape":["dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot","dot"],"font.size":[26,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22,22],"font.bold":[true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false],"borderWidth":[4,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2],"labelHighlightBold":[true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true,true],"x":[-0.4402260894100755,-0.4668442634678753,-0.6013426686524119,-0.3736770107770968,-0.8759724971842837,0.4602460320573487,-0.005611071934103151,0.1329458524810967,-0.9846627088811939,0.4088435681608875,0.3246409246579132,0.7029353149386772,0.7913042874268068,0.06651190993881029,0.1584426997165755,-0.07033244129014926,-0.5706544342972034,0.02963057490149268,-0.2830746215853023,0.1178257291371949,0.9022121285919162,0.1965555588164345,-0.0489014173157446,0.6188553837374191,-0.4198908153637704,-0.2683873817776452,-0.9980375360524608,-0.9627555543852127,-0.5693045565124202,0.9025180136429047,-0.5647316551548784,0.7683913797815847,-0.8534334193090648,0.3295738009304294,-0.8946140503405908,-1,-0.1355503972205246,0.6137938162448289,0.7790868555068684,-0.6398985803678765,0.0939749594160344,0.05906660578924372,-0.2334176181355033,0.7315158853918471,0.6575331889721374,-0.9189158309163562,0.4526024052667352,0.2353927902695314,-0.4206568625562772,0.5883475534523204,-0.1472101465200996,-0.8183046446527534,0.9412764301705618,0.602728603818848,0.03199086963412734,-0.6989916598456269,1],"y":[0.9001511175485724,0.7616490725255958,0.7875811570318936,-0.1667478930567456,-0.07526272816154911,-0.6496981614237984,0.923892985801483,0.8089286360251975,-0.09405484821042254,-0.08505984738286587,-0.252191461360157,0.1704074788761196,0.2387804482197908,-0.8149627452384347,0.9868986242228717,-0.9479111249933263,0.9186051303894751,0.8247589220652991,-0.3245215094286826,-0.9215130178846165,0.4120735453926514,0.8883674161897366,-0.8313032782287858,0.8242502355601609,-0.9889525296585406,-0.4931165203563708,-0.2295495493944856,0.1047992737191084,-0.6912635131453684,0.1349166932631765,0.3522075253377805,-0.7100448475047831,-0.643901647823196,0.4324208507226066,-0.2777676128557248,0.2862325459565007,-0.4050873261603957,-0.7010434272631735,0.051770440562529,-0.8260146105298184,0.2747275964316376,1,0.4624406332027984,0.7207592347653129,-0.3951871766486854,0.6120300454300003,-0.9649818412634918,-0.1294884935328473,-0.003210837905617714,0.6709794912877021,0.1621610384564141,-0.1936138211495424,-0.3858178476284763,0.06638262912649995,-1,0.4260498774580594,-0.2448634100629082]},"edges":{"from":["1","1","1","2","2","3","4","4","5","5","5","5","5","6","7","7","7","7","7","8","8","8","8","9","9","9","9","10","10","10","11","12","12","12","12","13","13","13","13","14","14","14","14","15","15","15","16","16","16","17","17","19","19","20","20","21","23","24","24","26","27","27","28","29","30","31","32","38","41","45","53"],"to":["2","3","18","3","18","18","19","51","8","27","28","38","54","34","9","15","17","21","43","27","28","38","54","15","17","21","43","11","49","52","52","13","32","41","49","22","32","41","49","16","20","23","55","17","21","43","20","23","55","21","43","26","40","23","55","43","55","45","47","40","38","54","37","46","34","57","41","54","49","47","56"],"overlapCoeff":[1,1,1,1,1,1,0.52,0.52,1,1,0.6,1,1,0.54,1,0.86,1,1,1,0.98,0.6,1,1,1,1,1,1,1,0.51,1,0.88,1,1,1,1,1,1,1,0.55,1,1,1,1,1,0.72,1,0.87,1,0.72,0.7,1,1,1,1,1,1,0.75,1,1,1,1,0.53,1,1,0.53,1,0.6,0.53,0.6,1,1],"width":[5,5,5,4.97,5,5,1.15,1.15,5,5,1.84,5,5,1.29,5,3.88,5,5,5,4.85,1.84,5,5,5,5,5,5,5,1.1,5,4.06,5,5,5,5,5,5,5,1.41,5,5,5,5,5,2.8,5,3.97,5,2.79,2.6,5,5,5,5,5,5,3,5,5,5,5,1.21,5,5,1.28,5,1.8,1.21,1.8,5,5],"color":["black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black","black"]},"nodesToDataframe":true,"edgesToDataframe":true,"options":{"width":"100%","height":"100%","nodes":{"shape":"dot","shadow":{"enabled":true,"size":25},"physics":true},"manipulation":{"enabled":false},"interaction":{"hover":true,"zoomSpeed":1},"edges":{"smooth":false},"physics":{"stabilization":false}},"groups":null,"width":"100%","height":null,"idselection":{"enabled":true,"style":"width: 400px; height = 31px;\n        font-size: 18px;\n        color: #000066;border: 3px solid #4d88ff;","useLabels":true,"main":"Select by id","values":["40","19","50","36","18","35","23","6","34","42","15","43","17","52","37","26","20","55","3","1","2","45","30","13","22","41","44","39","25","33","38","29","48","27","57","31","4","8","53","16","46","14","12","24","56","32","49","54","5","9","21","7","51","10","11","47","28"]},"byselection":{"enabled":true,"style":"width: 400px; height = 31px;\n        font-size: 18px;\n        color: #000066;border: 3px solid #4d88ff;","multiple":false,"hideColor":"rgba(200,200,200,0.5)","highlight":false,"variable":"FDR","main":"Select by FDR","values":[5.19384e-09,5.1958e-09,5.22645e-09,0.00371,0.0219,0.0242971,0.0245492,0.0263462,0.0264,0.0272652,0.0276027,0.0280679,0.0498,0.055,0.0558,0.06850000000000001,0.0732525,0.0758215,0.09481580000000001,0.0956212,0.09820379999999999,0.101897,0.105752,0.105993,0.117584,0.118598,0.120378,0.124,0.138,0.14706,0.147243,0.149598,0.157275,0.157341,0.15762,0.157851,0.158111,0.158328,0.158527,0.15876,0.159298,0.161246,0.161619,0.162414,0.162475,0.162821,0.16762,0.17231,0.173034,0.173615,0.174473,0.177354,0.177489,0.17767,0.188384,0.18981,0.190318]},"main":null,"submain":null,"footer":null,"background":"rgba(0, 0, 0, 0)","tooltipStay":300,"tooltipStyle":"position: fixed;visibility:hidden;padding: 5px;white-space: nowrap;font-family: verdana;font-size:14px;font-color:#000000;background-color: #f5f4ed;-moz-border-radius: 3px;-webkit-border-radius: 3px;border-radius: 3px;border: 1px solid #808074;box-shadow: 3px 3px 10px rgba(0, 0, 0, 0.2);","highlight":{"enabled":true,"hoverNearest":true,"degree":5,"algorithm":"all","hideColor":"rgba(200,200,200,0.5)","labelOnly":true},"collapse":{"enabled":false,"fit":false,"resetHighlight":true,"clusterOptions":null,"keepCoord":true,"labelSuffix":"(cluster)"},"igraphlayout":{"type":"full"}},"evals":[],"jsHooks":[]}</script>
<p>This network analysis is based on <a href="https://cytoscape.org/">Cytoscape</a>, an open source bioinformatics software platform for visualizing molecular interaction networks. out of all the GO packages.</p>
<p><strong>Homework: fill in the missing code for the DGE_analysis_exercises.Rmd in the exercises folder.</strong> points = +30.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="functional-analysis-of-rnaseq-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["2024_RNA-seq-analysis.pdf", "2024_RNA-seq-analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
