<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 10 Functional analysis | 08-GO_enrichment_analysis</title>
  <meta name="description" content="RNA-seq analysis workshop" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 10 Functional analysis | 08-GO_enrichment_analysis" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bookdown.org/ggiaever/RNA-seq-analysis/img/football22.png" />
  <meta property="og:description" content="RNA-seq analysis workshop" />
  <meta name="github-repo" content="gurinina/RNA-seq-analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 10 Functional analysis | 08-GO_enrichment_analysis" />
  
  <meta name="twitter:description" content="RNA-seq analysis workshop" />
  <meta name="twitter:image" content="https://bookdown.org/ggiaever/RNA-seq-analysis/img/football22.png" />



<meta name="date" content="2024-08-31" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/icons8-biotech-64.png" type="image/x-icon" />
<link rel="prev" href="codebook.html"/>
<link rel="next" href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/vis-9.1.0/vis-network.min.css" rel="stylesheet" />
<script src="libs/vis-9.1.0/vis-network.min.js"></script>
<script src="libs/visNetwork-binding-2.1.2/visNetwork.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-QV7GCCXQ4R"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-QV7GCCXQ4R');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="css/git.css" type="text/css" />
<link rel="stylesheet" href="css/toc.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Authoring Books with R Markdown</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>RNA-seq analysis workshop</a></li>
<li class="chapter" data-level="1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html"><i class="fa fa-check"></i><b>1</b> Differential gene expression (DGE) analysis overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#review-of-the-dataset"><i class="fa fa-check"></i><b>1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="1.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#setting-up"><i class="fa fa-check"></i><b>1.2</b> Setting up</a>
<ul>
<li class="chapter" data-level="1.2.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-libraries"><i class="fa fa-check"></i><b>1.2.1</b> Loading libraries</a></li>
<li class="chapter" data-level="1.2.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#loading-data"><i class="fa fa-check"></i><b>1.2.2</b> Loading data</a></li>
<li class="chapter" data-level="1.2.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#viewing-data"><i class="fa fa-check"></i><b>1.2.3</b> Viewing data</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-gene-expression-analysis-overview"><i class="fa fa-check"></i><b>1.3</b> Differential gene expression analysis overview</a>
<ul>
<li class="chapter" data-level="1.3.1" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#rna-seq-count-distribution"><i class="fa fa-check"></i><b>1.3.1</b> RNA-seq count distribution</a></li>
<li class="chapter" data-level="1.3.2" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#modeling-count-data"><i class="fa fa-check"></i><b>1.3.2</b> Modeling count data</a></li>
<li class="chapter" data-level="1.3.3" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#improving-mean-estimates-i.e.-reducing-variance-with-biological-replicates"><i class="fa fa-check"></i><b>1.3.3</b> Improving mean estimates (i.e. reducing variance) with biological replicates</a></li>
<li class="chapter" data-level="1.3.4" data-path="differential-gene-expression-dge-analysis-overview.html"><a href="differential-gene-expression-dge-analysis-overview.html#differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>1.3.4</b> Differential expression analysis workflow</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="count-normalization.html"><a href="count-normalization.html"><i class="fa fa-check"></i><b>2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="2.1" data-path="count-normalization.html"><a href="count-normalization.html#normalization"><i class="fa fa-check"></i><b>2.1</b> Normalization</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="count-normalization.html"><a href="count-normalization.html#common-normalization-methods"><i class="fa fa-check"></i><b>2.1.1</b> Common normalization methods</a></li>
<li class="chapter" data-level="2.1.2" data-path="count-normalization.html"><a href="count-normalization.html#rpkmfpkm-not-recommended"><i class="fa fa-check"></i><b>2.1.2</b> RPKM/FPKM (not recommended)</a></li>
<li class="chapter" data-level="2.1.3" data-path="count-normalization.html"><a href="count-normalization.html#deseq2-normalized-counts-median-of-ratios-method"><i class="fa fa-check"></i><b>2.1.3</b> DESeq2-normalized counts: Median of ratios method</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="count-normalization.html"><a href="count-normalization.html#geometric-mean"><i class="fa fa-check"></i><b>2.2</b> Geometric Mean</a></li>
<li class="chapter" data-level="2.3" data-path="count-normalization.html"><a href="count-normalization.html#count-normalization-of-mov10-dataset"><i class="fa fa-check"></i><b>2.3</b> Count normalization of Mov10 dataset</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="count-normalization.html"><a href="count-normalization.html#match-the-metadata-and-counts-data"><i class="fa fa-check"></i><b>2.3.1</b> 1. Match the metadata and counts data</a></li>
<li class="chapter" data-level="2.3.2" data-path="count-normalization.html"><a href="count-normalization.html#create-deseq2-object"><i class="fa fa-check"></i><b>2.3.2</b> 2. Create DESEq2 object</a></li>
<li class="chapter" data-level="2.3.3" data-path="count-normalization.html"><a href="count-normalization.html#generate-the-mov10-normalized-counts"><i class="fa fa-check"></i><b>2.3.3</b> 3. Generate the Mov10 normalized counts</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="quality-control.html"><a href="quality-control.html"><i class="fa fa-check"></i><b>3</b> Quality Control</a>
<ul>
<li class="chapter" data-level="3.1" data-path="quality-control.html"><a href="quality-control.html#sample-level-qc"><i class="fa fa-check"></i><b>3.1</b> Sample-level QC</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="quality-control.html"><a href="quality-control.html#principal-component-analysis-pca"><i class="fa fa-check"></i><b>3.1.1</b> Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="3.1.2" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering-heatmap"><i class="fa fa-check"></i><b>3.1.2</b> Hierarchical Clustering Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="quality-control.html"><a href="quality-control.html#gene-level-qc"><i class="fa fa-check"></i><b>3.2</b> Gene-level QC</a></li>
<li class="chapter" data-level="3.3" data-path="quality-control.html"><a href="quality-control.html#mov10-quality-assessment-and-exploratory-analysis-using-deseq2"><i class="fa fa-check"></i><b>3.3</b> Mov10 quality assessment and exploratory analysis using DESeq2</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="quality-control.html"><a href="quality-control.html#transform-normalized-counts-using-the-rlog-transformation"><i class="fa fa-check"></i><b>3.3.1</b> Transform normalized counts using the rlog transformation</a></li>
<li class="chapter" data-level="3.3.2" data-path="quality-control.html"><a href="quality-control.html#principal-components-analysis-pca"><i class="fa fa-check"></i><b>3.3.2</b> Principal components analysis (PCA)</a></li>
<li class="chapter" data-level="3.3.3" data-path="quality-control.html"><a href="quality-control.html#hierarchical-clustering"><i class="fa fa-check"></i><b>3.3.3</b> Hierarchical Clustering</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html"><i class="fa fa-check"></i><b>4</b> DGE analysis workflow</a>
<ul>
<li class="chapter" data-level="4.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#running-deseq2"><i class="fa fa-check"></i><b>4.1</b> Running DESeq2</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#design-formula"><i class="fa fa-check"></i><b>4.1.1</b> Design formula</a></li>
<li class="chapter" data-level="4.1.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#mov10-de-analysis"><i class="fa fa-check"></i><b>4.1.2</b> MOV10 DE analysis</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#deseq2-differential-gene-expression-analysis-workflow"><i class="fa fa-check"></i><b>4.2</b> DESeq2 differential gene expression analysis workflow</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-1-estimate-size-factors"><i class="fa fa-check"></i><b>4.2.1</b> Step 1: Estimate size factors</a></li>
<li class="chapter" data-level="4.2.2" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-2-estimate-gene-wise-dispersion"><i class="fa fa-check"></i><b>4.2.2</b> Step 2: Estimate gene-wise dispersion</a></li>
<li class="chapter" data-level="4.2.3" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-3-fit-curve-to-gene-wise-dispersion-estimates"><i class="fa fa-check"></i><b>4.2.3</b> Step 3: Fit curve to gene-wise dispersion estimates</a></li>
<li class="chapter" data-level="4.2.4" data-path="dge-analysis-workflow.html"><a href="dge-analysis-workflow.html#step-4-shrink-gene-wise-dispersion-estimates-toward-the-values-predicted-by-the-curve"><i class="fa fa-check"></i><b>4.2.4</b> Step 4: Shrink gene-wise dispersion estimates toward the values predicted by the curve</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html"><i class="fa fa-check"></i><b>5</b> Model and hypothesis testing</a>
<ul>
<li class="chapter" data-level="5.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#generalized-linear-model-fit-for-each-gene"><i class="fa fa-check"></i><b>5.1</b> Generalized Linear Model fit for each gene</a></li>
<li class="chapter" data-level="5.2" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#shrunken-log2-foldchanges-lfc"><i class="fa fa-check"></i><b>5.2</b> Shrunken log2 foldchanges (LFC)</a></li>
<li class="chapter" data-level="5.3" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#statistical-test-for-lfc-estimates-wald-test"><i class="fa fa-check"></i><b>5.3</b> Statistical test for LFC estimates: Wald test</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#overall-deseq2-workflow"><i class="fa fa-check"></i><b>5.3.1</b> Overall DESeq2 workflow:</a></li>
<li class="chapter" data-level="5.3.2" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#creating-contrasts"><i class="fa fa-check"></i><b>5.3.2</b> Creating contrasts</a></li>
<li class="chapter" data-level="5.3.3" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#building-the-results-table"><i class="fa fa-check"></i><b>5.3.3</b> Building the results table</a></li>
<li class="chapter" data-level="5.3.4" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#ma-plot"><i class="fa fa-check"></i><b>5.3.4</b> MA Plot</a></li>
<li class="chapter" data-level="5.3.5" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#mov10-de-analysis-results-exploration"><i class="fa fa-check"></i><b>5.3.5</b> MOV10 DE analysis: results exploration</a></li>
<li class="chapter" data-level="5.3.6" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#multiple-test-correction"><i class="fa fa-check"></i><b>5.3.6</b> Multiple test correction</a></li>
<li class="chapter" data-level="5.3.7" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#mov10-de-analysis-control-versus-knockdown"><i class="fa fa-check"></i><b>5.3.7</b> MOV10 DE analysis: Control versus Knockdown</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#summarizing-results"><i class="fa fa-check"></i><b>5.4</b> Summarizing results</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="model-and-hypothesis-testing.html"><a href="model-and-hypothesis-testing.html#extracting-significant-differentially-expressed-genes"><i class="fa fa-check"></i><b>5.4.1</b> Extracting significant differentially expressed genes</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html"><i class="fa fa-check"></i><b>6</b> Visualizing RNA-seq results</a>
<ul>
<li class="chapter" data-level="6.0.1" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#plotting-signicant-de-genes"><i class="fa fa-check"></i><b>6.0.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="6.0.2" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#heatmap"><i class="fa fa-check"></i><b>6.0.2</b> Heatmap</a></li>
<li class="chapter" data-level="6.0.3" data-path="visualizing-rna-seq-results.html"><a href="visualizing-rna-seq-results.html#volcano-plot"><i class="fa fa-check"></i><b>6.0.3</b> Volcano plot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="summary-of-dge-workflow.html"><a href="summary-of-dge-workflow.html"><i class="fa fa-check"></i><b>7</b> Summary of DGE workflow</a></li>
<li class="chapter" data-level="8" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html"><i class="fa fa-check"></i><b>8</b> Functional analysis of RNAseq data</a>
<ul>
<li class="chapter" data-level="8.0.1" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#over-representation-analysis"><i class="fa fa-check"></i><b>8.0.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="8.0.2" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#clusterprofiler"><i class="fa fa-check"></i><b>8.0.2</b> clusterProfiler</a></li>
<li class="chapter" data-level="8.0.3" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#gene-set-enrichment-analysis-gsea"><i class="fa fa-check"></i><b>8.0.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="8.0.4" data-path="functional-analysis-of-rnaseq-data.html"><a href="functional-analysis-of-rnaseq-data.html#other-tools-and-resources"><i class="fa fa-check"></i><b>8.0.4</b> Other tools and resources</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="codebook.html"><a href="codebook.html"><i class="fa fa-check"></i><b>9</b> Codebook</a>
<ul>
<li class="chapter" data-level="9.1" data-path="codebook.html"><a href="codebook.html#dge-analysis-overview"><i class="fa fa-check"></i><b>9.1</b> DGE analysis overview</a>
<ul>
<li class="chapter" data-level="9.1.1" data-path="codebook.html"><a href="codebook.html#review-of-the-dataset-1"><i class="fa fa-check"></i><b>9.1.1</b> Review of the dataset</a></li>
<li class="chapter" data-level="9.1.2" data-path="codebook.html"><a href="codebook.html#setting-up-1"><i class="fa fa-check"></i><b>9.1.2</b> Setting up</a></li>
<li class="chapter" data-level="9.1.3" data-path="codebook.html"><a href="codebook.html#dge-analysis-workflow-1"><i class="fa fa-check"></i><b>9.1.3</b> DGE analysis workflow</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="codebook.html"><a href="codebook.html#count-normalization-1"><i class="fa fa-check"></i><b>9.2</b> Count normalization</a>
<ul>
<li class="chapter" data-level="9.2.1" data-path="codebook.html"><a href="codebook.html#normalization-1"><i class="fa fa-check"></i><b>9.2.1</b> Normalization</a></li>
<li class="chapter" data-level="9.2.2" data-path="codebook.html"><a href="codebook.html#count-normalization-of-mov10-dataset-1"><i class="fa fa-check"></i><b>9.2.2</b> Count normalization of Mov10 dataset</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="codebook.html"><a href="codebook.html#dge-qc-analysis"><i class="fa fa-check"></i><b>9.3</b> DGE QC analysis</a>
<ul>
<li class="chapter" data-level="9.3.1" data-path="codebook.html"><a href="codebook.html#sample-level-qc-1"><i class="fa fa-check"></i><b>9.3.1</b> Sample-level QC</a></li>
<li class="chapter" data-level="9.3.2" data-path="codebook.html"><a href="codebook.html#mov10-quality-assessment-and-exploratory-analysis"><i class="fa fa-check"></i><b>9.3.2</b> Mov10 quality assessment and exploratory analysis</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="codebook.html"><a href="codebook.html#deg-analysis-workflow"><i class="fa fa-check"></i><b>9.4</b> DEG analysis workflow</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="codebook.html"><a href="codebook.html#running-deseq2-1"><i class="fa fa-check"></i><b>9.4.1</b> Running DESeq2</a></li>
<li class="chapter" data-level="9.4.2" data-path="codebook.html"><a href="codebook.html#deseq2-differential-gene-expression-analysis-workflow-1"><i class="fa fa-check"></i><b>9.4.2</b> DESeq2 differential gene expression analysis workflow</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="codebook.html"><a href="codebook.html#model-fitting"><i class="fa fa-check"></i><b>9.5</b> Model fitting</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="codebook.html"><a href="codebook.html#generalized-linear-model-fit-for-each-gene-1"><i class="fa fa-check"></i><b>9.5.1</b> Generalized Linear Model fit for each gene</a></li>
<li class="chapter" data-level="9.5.2" data-path="codebook.html"><a href="codebook.html#shrunken-log2-foldchanges-lfc-1"><i class="fa fa-check"></i><b>9.5.2</b> Shrunken log2 foldchanges (LFC)</a></li>
<li class="chapter" data-level="9.5.3" data-path="codebook.html"><a href="codebook.html#hypothesis-testing-using-the-wald-test"><i class="fa fa-check"></i><b>9.5.3</b> Hypothesis testing using the Wald test</a></li>
<li class="chapter" data-level="9.5.4" data-path="codebook.html"><a href="codebook.html#summarizing-results-1"><i class="fa fa-check"></i><b>9.5.4</b> Summarizing results</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="codebook.html"><a href="codebook.html#visualizing-rna-seq-results-1"><i class="fa fa-check"></i><b>9.6</b> Visualizing rna-seq results</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="codebook.html"><a href="codebook.html#plotting-signicant-de-genes-1"><i class="fa fa-check"></i><b>9.6.1</b> Plotting signicant DE genes</a></li>
<li class="chapter" data-level="9.6.2" data-path="codebook.html"><a href="codebook.html#heatmap-1"><i class="fa fa-check"></i><b>9.6.2</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="codebook.html"><a href="codebook.html#summary-of-differential-expression-analysis-workflow"><i class="fa fa-check"></i><b>9.7</b> Summary of differential expression analysis workflow</a>
<ul>
<li class="chapter" data-level="9.7.1" data-path="codebook.html"><a href="codebook.html#import-data-into-dds-object"><i class="fa fa-check"></i><b>9.7.1</b> 1. Import data into dds object:</a></li>
<li class="chapter" data-level="9.7.2" data-path="codebook.html"><a href="codebook.html#exploratory-data-analysis-pca-heirarchical-clustering---identifying-outliers-and-sources-of-variation-in-the-data"><i class="fa fa-check"></i><b>9.7.2</b> 2. Exploratory data analysis (PCA &amp; heirarchical clustering) - identifying outliers and sources of variation in the data:</a></li>
<li class="chapter" data-level="9.7.3" data-path="codebook.html"><a href="codebook.html#run-deseq2"><i class="fa fa-check"></i><b>9.7.3</b> 3. Run DESeq2:</a></li>
<li class="chapter" data-level="9.7.4" data-path="codebook.html"><a href="codebook.html#check-the-fit-of-the-dispersion-estimates"><i class="fa fa-check"></i><b>9.7.4</b> 4. Check the fit of the dispersion estimates:</a></li>
<li class="chapter" data-level="9.7.5" data-path="codebook.html"><a href="codebook.html#create-contrasts-to-perform-wald-testing-on-the-shrunken-log2-foldchanges-between-specific-conditions"><i class="fa fa-check"></i><b>9.7.5</b> 5. Create contrasts to perform Wald testing on the shrunken log2 foldchanges between specific conditions:</a></li>
<li class="chapter" data-level="9.7.6" data-path="codebook.html"><a href="codebook.html#output-significant-results"><i class="fa fa-check"></i><b>9.7.6</b> 6. Output significant results:</a></li>
<li class="chapter" data-level="9.7.7" data-path="codebook.html"><a href="codebook.html#visualize-results-volcano-plots-heatmaps-normalized-counts-plots-of-top-genes-etc."><i class="fa fa-check"></i><b>9.7.7</b> 7. Visualize results: volcano plots, heatmaps, normalized counts plots of top genes, etc.</a></li>
<li class="chapter" data-level="9.7.8" data-path="codebook.html"><a href="codebook.html#make-sure-to-output-the-versions-of-all-tools-used-in-the-de-analysis"><i class="fa fa-check"></i><b>9.7.8</b> 8. Make sure to output the versions of all tools used in the DE analysis:</a></li>
</ul></li>
<li class="chapter" data-level="9.8" data-path="codebook.html"><a href="codebook.html#functional-analysis"><i class="fa fa-check"></i><b>9.8</b> Functional analysis</a>
<ul>
<li class="chapter" data-level="9.8.1" data-path="codebook.html"><a href="codebook.html#over-representation-analysis-1"><i class="fa fa-check"></i><b>9.8.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="9.8.2" data-path="codebook.html"><a href="codebook.html#clusterprofiler-1"><i class="fa fa-check"></i><b>9.8.2</b> clusterProfiler</a></li>
<li class="chapter" data-level="9.8.3" data-path="codebook.html"><a href="codebook.html#gene-set-enrichment-analysis-gsea-1"><i class="fa fa-check"></i><b>9.8.3</b> Gene set enrichment analysis (GSEA)</a></li>
<li class="chapter" data-level="9.8.4" data-path="codebook.html"><a href="codebook.html#other-tools-and-resources-1"><i class="fa fa-check"></i><b>9.8.4</b> Other tools and resources</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html"><i class="fa fa-check"></i><b>10</b> Functional analysis</a>
<ul>
<li class="chapter" data-level="10.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#over-representation-analysis-2"><i class="fa fa-check"></i><b>10.1</b> Over-representation analysis</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#gene-ontology-project"><i class="fa fa-check"></i><b>10.1.1</b> Gene Ontology project</a></li>
<li class="chapter" data-level="10.1.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#hypergeometric-testing"><i class="fa fa-check"></i><b>10.1.2</b> Hypergeometric testing</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#clusterprofiler-2"><i class="fa fa-check"></i><b>10.2</b> clusterProfiler</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#running-clusterprofiler-2"><i class="fa fa-check"></i><b>10.2.1</b> Running clusterProfiler</a></li>
<li class="chapter" data-level="10.2.2" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#visualizing-clusterprofiler-results-2"><i class="fa fa-check"></i><b>10.2.2</b> Visualizing clusterProfiler results</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#gene-set-enrichment-analysis-gsea-2"><i class="fa fa-check"></i><b>10.3</b> Gene set enrichment analysis (GSEA)</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="functional-analysis-1.html"><a href="functional-analysis-1.html#gsea-using-clusterprofiler-2"><i class="fa fa-check"></i><b>10.3.1</b> GSEA using clusterProfiler</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"><a href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"><i class="fa fa-check"></i><b>11</b> this has to be commented out in epub and pdf formats because the dynamic network cannot be displayed</a>
<ul>
<li class="chapter" data-level="11.1" data-path="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"><a href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html#other-tools"><i class="fa fa-check"></i><b>11.1</b> Other Tools</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"><a href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html#genemania"><i class="fa fa-check"></i><b>11.1.1</b> GeneMANIA</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html"><a href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html#resources-for-functional-analysis"><i class="fa fa-check"></i><b>11.2</b> Resources for functional analysis</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bookdown.org" target="_blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">08-GO_enrichment_analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-analysis-1" class="section level1 hasAnchor" number="10">
<h1><span class="header-section-number">Chapter 10</span> Functional analysis<a href="functional-analysis-1.html#functional-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The output of RNA-seq differential expression analysis is a list of significant differentially expressed genes (DEGs). To gain greater biological insight on the differentially expressed genes there are various analyses that can be done. While the tools for these functional analyses span a wide variety of techniques, they can loosely be categorized into three main types: over-representation analysis, functional class scoring, and pathway topology [<a href="https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course/raw/master/resources/pathway_tools.pdf">1</a>].</p>
<div class="float">
<img src="img/pathway_analysis.png" alt="Pathway analysis tools" />
<div class="figcaption">Pathway analysis tools</div>
</div>
<div id="over-representation-analysis-2" class="section level2 hasAnchor" number="10.1">
<h2><span class="header-section-number">10.1</span> Over-representation analysis<a href="functional-analysis-1.html#over-representation-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are a plethora of functional enrichment tools that perform some type of "over-representation" analysis by querying databases containing information about gene function and interactions.</p>
<p>These databases typically categorize genes into groups based on shared function, or involvement in a pathway, or presence in a specific cellular location, or other categorizations, e.g. functional pathways, etc. These categories have been consistently named (controlled vocabulary) based on how the gene has been annotated functionally.</p>
<div id="gene-ontology-project" class="section level3 hasAnchor" number="10.1.1">
<h3><span class="header-section-number">10.1.1</span> Gene Ontology project<a href="functional-analysis-1.html#gene-ontology-project" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One of the most widely-used categorizations is the <strong>Gene Ontology (GO)</strong> established by the Gene Ontology project.</p>
<p>"The Gene Ontology project is a collaborative effort to address the need for consistent descriptions of gene products across databases" [<a href="http://geneontology.org/page/introduction-go-resource">2</a>]. The <a href="http://geneontology.org/page/go-consortium-contributors-list">Gene Ontology Consortium</a> maintains the GO terms, and these GO terms are incorporated into gene annotations in many of the popular repositories for animal, plant, and microbial genomes.</p>
<p>Tools that investigate enrichment of biological functions or interactions often use the Gene Ontology (GO) categorizations, i.e. the GO terms to determine whether any have significantly modified representation in a given list of genes.</p>
<p>To determine whether any GO terms are over-represented, you can determine the probability of having the observed proportion of genes associated with a specific GO term in your gene list based on the proportion of genes associated with the same GO term in the background set.</p>
<div class="float">
<img src="img/go_proportions.png" width="600" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<div class="float">
<img src="img/go_proportions_table3.png" width="600" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<p>The statistical test that will determine whether something is actually over-represented is the <em>Hypergeometric test</em>.</p>
</div>
<div id="hypergeometric-testing" class="section level3 hasAnchor" number="10.1.2">
<h3><span class="header-section-number">10.1.2</span> Hypergeometric testing<a href="functional-analysis-1.html#hypergeometric-testing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Using the example of the first functional category above, hypergeometric distribution is a probability distribution that describes the probability of 25 genes (k) being associated with "Functional category 1", for all genes in our gene list (n=1000), from a population of all of the genes in entire genome (N=13,000) which contains 35 genes (K) associated with "Functional category 1" [<a href="https://en.wikipedia.org/wiki/Hypergeometric_distribution">4</a>].</p>
<p>The calculation of probability of k successes follows the formula:</p>
<div class="float">
<img src="img/hypergeo.png" alt="hypergeo" />
<div class="figcaption">hypergeo</div>
</div>
<p>This test will result in an adjusted p-value (after multiple test correction) for each category tested.</p>
<div id="go-ontologies" class="section level4 hasAnchor" number="10.1.2.1">
<h4><span class="header-section-number">10.1.2.1</span> GO Ontologies<a href="functional-analysis-1.html#go-ontologies" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To best use and interpret the results from these functional analysis tools, it is helpful to have a good understanding of the GO terms themselves and their organization. To describe the roles of genes and gene products, GO terms are organized into three independent controlled vocabularies (ontologies) in a species-independent manner:</p>
<ul>
<li><p><strong>Biological process:</strong> refers to the biological role involving the gene or gene product, and could include "transcription", "signal transduction", and "apoptosis". A biological process generally involves a chemical or physical change of the starting material or input.</p></li>
<li><p><strong>Molecular function:</strong> represents the biochemical activity of the gene product, such activities could include "ligand", "GTPase", and "transporter".</p></li>
<li><p><strong>Cellular component:</strong> refers to the location in the cell of the gene product. Cellular components could include "nucleus", "lysosome", and "plasma membrane".</p></li>
</ul>
<p>Each GO term has a term name (e.g. <strong>DNA repair</strong>) and a unique term accession number (<strong><a href="GO:0005125" class="uri">GO:0005125</a></strong>), and a single gene product can be associated with many GO terms, since a single gene product "may function in several processes, contain domains that carry out diverse molecular functions, and participate in multiple alternative interactions with other proteins, organelles or locations in the cell" [<a href="https://github.com/hbctraining/In-depth-NGS-Data-Analysis-Course/raw/master/resources/go_terms.pdf">3</a>].</p>
</div>
<div id="go-term-hierarchy" class="section level4 hasAnchor" number="10.1.2.2">
<h4><span class="header-section-number">10.1.2.2</span> GO term hierarchy<a href="functional-analysis-1.html#go-term-hierarchy" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The GO ontologies were developed to describe and query biological knowledge with differing levels of information available. To do this, GO ontologies are loosely hierarchical, ranging from general, 'parent', terms to more specific, 'child' terms. The GO ontologies are "loosely" hierarchical since 'child' terms can have multiple 'parent' terms.</p>
<p>Some genes with less information may only be associated with general 'parent' terms or no terms at all, while other genes with a lot of information be associated with many terms.</p>
<div class="float">
<img src="img/go_heirarchy.jpg" alt="Nature Reviews Cancer 7, 23-34 (January 2007)" />
<div class="figcaption">Nature Reviews Cancer 7, 23-34 (January 2007)</div>
</div>
<p><a href="http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003343">Tips for working with GO terms</a></p>
</div>
</div>
</div>
<div id="clusterprofiler-2" class="section level2 hasAnchor" number="10.2">
<h2><span class="header-section-number">10.2</span> clusterProfiler<a href="functional-analysis-1.html#clusterprofiler-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will be using <a href="http://bioconductor.org/packages/release/bioc/html/clusterProfiler.html">clusterProfiler</a> to perform over-representation analysis on GO terms associated with our list of significant genes. The tool takes as input a significant gene list and a background gene list and performs statistical enrichment analysis using hypergeometric testing. The basic arguments allow the user to select the appropriate organism and GO ontology (BP, CC, MF) to test.</p>
<div id="running-clusterprofiler-2" class="section level3 hasAnchor" number="10.2.1">
<h3><span class="header-section-number">10.2.1</span> Running clusterProfiler<a href="functional-analysis-1.html#running-clusterprofiler-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Load the following libraries:</p>
<div class="sourceCode" id="cb300"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb300-1"><a href="functional-analysis-1.html#cb300-1" tabindex="-1"></a></span>
<span id="cb300-2"><a href="functional-analysis-1.html#cb300-2" tabindex="-1"></a><span class="co"># you may have to install some of these libraries; use </span></span>
<span id="cb300-3"><a href="functional-analysis-1.html#cb300-3" tabindex="-1"></a><span class="co"># BiocManager::install(c(&quot;org.Hs.eg.db&quot;,&quot;clusterProfiler&quot;,&quot;enrichplot&quot;,&quot;fgsea&quot;))</span></span>
<span id="cb300-4"><a href="functional-analysis-1.html#cb300-4" tabindex="-1"></a></span>
<span id="cb300-5"><a href="functional-analysis-1.html#cb300-5" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb300-6"><a href="functional-analysis-1.html#cb300-6" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb300-7"><a href="functional-analysis-1.html#cb300-7" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb300-8"><a href="functional-analysis-1.html#cb300-8" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb300-9"><a href="functional-analysis-1.html#cb300-9" tabindex="-1"></a><span class="fu">library</span>(fgsea)</span></code></pre></div>
<div class="sourceCode" id="cb301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb301-1"><a href="functional-analysis-1.html#cb301-1" tabindex="-1"></a></span>
<span id="cb301-2"><a href="functional-analysis-1.html#cb301-2" tabindex="-1"></a>res_tableOE <span class="ot">=</span> <span class="fu">readRDS</span>(<span class="st">&quot;data/res_tableOE.rds&quot;</span>)</span>
<span id="cb301-3"><a href="functional-analysis-1.html#cb301-3" tabindex="-1"></a></span>
<span id="cb301-4"><a href="functional-analysis-1.html#cb301-4" tabindex="-1"></a>res_tableOE_tb <span class="ot">&lt;-</span> res_tableOE <span class="sc">%&gt;%</span></span>
<span id="cb301-5"><a href="functional-analysis-1.html#cb301-5" tabindex="-1"></a><span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb301-6"><a href="functional-analysis-1.html#cb301-6" tabindex="-1"></a><span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;gene&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb301-7"><a href="functional-analysis-1.html#cb301-7" tabindex="-1"></a>dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange))  <span class="sc">%&gt;%</span> <span class="fu">as_tibble</span>()</span></code></pre></div>
<p>To perform the over-representation analysis, we need a list of background genes and a list of significant genes. For our background dataset we will use all genes tested for differential expression (all genes in our results table). For our significant gene list we will use genes with p-adjusted values less than 0.05 (we could include a fold change threshold too if we have many DE genes).</p>
<div class="sourceCode" id="cb302"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb302-1"><a href="functional-analysis-1.html#cb302-1" tabindex="-1"></a><span class="do">## background set of genes</span></span>
<span id="cb302-2"><a href="functional-analysis-1.html#cb302-2" tabindex="-1"></a>allOE_genes <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span>
<span id="cb302-3"><a href="functional-analysis-1.html#cb302-3" tabindex="-1"></a></span>
<span id="cb302-4"><a href="functional-analysis-1.html#cb302-4" tabindex="-1"></a>sigOE <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">filter</span>(res_tableOE_tb, padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb302-5"><a href="functional-analysis-1.html#cb302-5" tabindex="-1"></a>sigOE_genes <span class="ot">=</span> sigOE<span class="sc">$</span>gene</span></code></pre></div>
<p>Now we can perform the GO enrichment analysis and save the results:</p>
<div class="sourceCode" id="cb303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb303-1"><a href="functional-analysis-1.html#cb303-1" tabindex="-1"></a><span class="do">## Run GO enrichment analysis </span></span>
<span id="cb303-2"><a href="functional-analysis-1.html#cb303-2" tabindex="-1"></a>ego <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(<span class="at">gene =</span> sigOE_genes, </span>
<span id="cb303-3"><a href="functional-analysis-1.html#cb303-3" tabindex="-1"></a>                <span class="at">universe =</span> allOE_genes,</span>
<span id="cb303-4"><a href="functional-analysis-1.html#cb303-4" tabindex="-1"></a>                <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb303-5"><a href="functional-analysis-1.html#cb303-5" tabindex="-1"></a>                <span class="at">OrgDb =</span> org.Hs.eg.db, </span>
<span id="cb303-6"><a href="functional-analysis-1.html#cb303-6" tabindex="-1"></a>                <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb303-7"><a href="functional-analysis-1.html#cb303-7" tabindex="-1"></a>                <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb303-8"><a href="functional-analysis-1.html#cb303-8" tabindex="-1"></a>                <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb303-9"><a href="functional-analysis-1.html#cb303-9" tabindex="-1"></a>                <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, </span>
<span id="cb303-10"><a href="functional-analysis-1.html#cb303-10" tabindex="-1"></a>                <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb303-11"><a href="functional-analysis-1.html#cb303-11" tabindex="-1"></a>                <span class="at">readable =</span> <span class="cn">TRUE</span>)</span>
<span id="cb303-12"><a href="functional-analysis-1.html#cb303-12" tabindex="-1"></a>                </span>
<span id="cb303-13"><a href="functional-analysis-1.html#cb303-13" tabindex="-1"></a><span class="do">## Output results from GO analysis to a table</span></span>
<span id="cb303-14"><a href="functional-analysis-1.html#cb303-14" tabindex="-1"></a>cluster_summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ego)</span>
<span id="cb303-15"><a href="functional-analysis-1.html#cb303-15" tabindex="-1"></a><span class="do">## Note: folder results needs to exist here, add it if it doesn&#39;t</span></span>
<span id="cb303-16"><a href="functional-analysis-1.html#cb303-16" tabindex="-1"></a><span class="fu">write.csv</span>(cluster_summary, <span class="st">&quot;results/clusterProfiler_Mov10oe.csv&quot;</span>)</span></code></pre></div>
<blockquote>
<p><strong>NOTE:</strong> The different organisms with annotation databases available to use with clusterProfiler can be found <a href="img/orgdb_annotation_databases.png">here</a></p>
</blockquote>
<div class="float">
<img src="img/cluster_summary.png" alt="cluster_summary" />
<div class="figcaption">cluster_summary</div>
</div>
</div>
<div id="visualizing-clusterprofiler-results-2" class="section level3 hasAnchor" number="10.2.2">
<h3><span class="header-section-number">10.2.2</span> Visualizing clusterProfiler results<a href="functional-analysis-1.html#visualizing-clusterprofiler-results-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>clusterProfiler has a variety of options for viewing the over-represented GO terms. We will explore the dotplot, enrichment plot, and the category netplot.</p>
<p>The <strong>dotplot</strong> shows the number of genes associated with the first 50 terms (size) and the p-adjusted values for these terms (color). This plot displays the top 50 genes by gene ratio (# genes related to GO term / total number of sig genes), not p-adjusted value.</p>
<div class="sourceCode" id="cb304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb304-1"><a href="functional-analysis-1.html#cb304-1" tabindex="-1"></a><span class="do">## Dotplot </span></span>
<span id="cb304-2"><a href="functional-analysis-1.html#cb304-2" tabindex="-1"></a><span class="fu">dotplot</span>(ego, <span class="at">showCategory =</span> <span class="dv">50</span>)</span></code></pre></div>
<p><strong>To save the figure,</strong> click on the <code>Export</code> button in the RStudio <code>Plots</code> tab and <code>Save as PDF...</code>. In the pop-up window, change:
- <code>PDF size</code> to <code>8 x 14</code> to give a figure of appropriate size for the text labels</p>
<div class="float">
<img src="img/mov10oe_dotplot.png" width="600" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<p>The next plot is the <strong>enrichment GO plot</strong>, which shows the relationship between the top 50 most significantly enriched GO terms (padj.), by grouping similar terms together. The color represents the p-values relative to the other displayed terms (brighter red is more significant) and the size of the terms represents the number of genes that are significant from our list.</p>
<div class="sourceCode" id="cb305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb305-1"><a href="functional-analysis-1.html#cb305-1" tabindex="-1"></a><span class="do">## Enrichmap clusters the 50 most significant (by padj) GO terms to visualize relationships between terms</span></span>
<span id="cb305-2"><a href="functional-analysis-1.html#cb305-2" tabindex="-1"></a></span>
<span id="cb305-3"><a href="functional-analysis-1.html#cb305-3" tabindex="-1"></a>pwt <span class="ot">&lt;-</span> <span class="fu">pairwise_termsim</span>(</span>
<span id="cb305-4"><a href="functional-analysis-1.html#cb305-4" tabindex="-1"></a>ego,</span>
<span id="cb305-5"><a href="functional-analysis-1.html#cb305-5" tabindex="-1"></a><span class="at">method =</span> <span class="st">&quot;JC&quot;</span>,</span>
<span id="cb305-6"><a href="functional-analysis-1.html#cb305-6" tabindex="-1"></a><span class="at">semData =</span> <span class="cn">NULL</span>,</span>
<span id="cb305-7"><a href="functional-analysis-1.html#cb305-7" tabindex="-1"></a><span class="at">showCategory =</span> <span class="dv">50</span></span>
<span id="cb305-8"><a href="functional-analysis-1.html#cb305-8" tabindex="-1"></a>)</span>
<span id="cb305-9"><a href="functional-analysis-1.html#cb305-9" tabindex="-1"></a></span>
<span id="cb305-10"><a href="functional-analysis-1.html#cb305-10" tabindex="-1"></a><span class="fu">emapplot</span>(pwt, <span class="at">showCategory =</span> <span class="dv">50</span>)</span></code></pre></div>
<p><strong>To save the figure,</strong> click on the <code>Export</code> button in the RStudio <code>Plots</code> tab and <code>Save as PDF...</code>. In the pop-up window, change the <code>PDF size</code> to <code>24 x 32</code> to give a figure of appropriate size for the text labels.</p>
<div class="float">
<img src="img/mov10oe_enrichmap.png" width="800" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<p>Finally, the <strong>category netplot</strong> shows the relationships between the genes associated with the top five most significant GO terms and the fold changes of the significant genes associated with these terms (color). The size of the GO terms reflects the pvalues of the terms, with the more significant terms being larger. This plot is particularly useful for hypothesis generation in identifying genes that may be important to several of the most affected processes.</p>
<div class="sourceCode" id="cb306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb306-1"><a href="functional-analysis-1.html#cb306-1" tabindex="-1"></a><span class="do">## To color genes by log2 fold changes, we need to extract the log2 fold changes from our results table creating a named vector</span></span>
<span id="cb306-2"><a href="functional-analysis-1.html#cb306-2" tabindex="-1"></a></span>
<span id="cb306-3"><a href="functional-analysis-1.html#cb306-3" tabindex="-1"></a>OE_foldchanges <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>log2FoldChange</span>
<span id="cb306-4"><a href="functional-analysis-1.html#cb306-4" tabindex="-1"></a></span>
<span id="cb306-5"><a href="functional-analysis-1.html#cb306-5" tabindex="-1"></a><span class="fu">names</span>(OE_foldchanges) <span class="ot">&lt;-</span> sigOE<span class="sc">$</span>gene</span>
<span id="cb306-6"><a href="functional-analysis-1.html#cb306-6" tabindex="-1"></a></span>
<span id="cb306-7"><a href="functional-analysis-1.html#cb306-7" tabindex="-1"></a><span class="do">## Cnetplot details the genes associated with one or more terms - by default gives the top 5 significant terms (by padj)</span></span>
<span id="cb306-8"><a href="functional-analysis-1.html#cb306-8" tabindex="-1"></a><span class="fu">cnetplot</span>(ego, </span>
<span id="cb306-9"><a href="functional-analysis-1.html#cb306-9" tabindex="-1"></a>         <span class="at">categorySize=</span><span class="st">&quot;pvalue&quot;</span>, </span>
<span id="cb306-10"><a href="functional-analysis-1.html#cb306-10" tabindex="-1"></a>         <span class="at">showCategory =</span> <span class="dv">5</span>, </span>
<span id="cb306-11"><a href="functional-analysis-1.html#cb306-11" tabindex="-1"></a>         <span class="at">foldChange =</span> OE_foldchanges, </span>
<span id="cb306-12"><a href="functional-analysis-1.html#cb306-12" tabindex="-1"></a>         <span class="at">vertex.label.font=</span><span class="dv">6</span>)</span>
<span id="cb306-13"><a href="functional-analysis-1.html#cb306-13" tabindex="-1"></a>         </span></code></pre></div>
<p><strong>Again, to save the figure,</strong> click on the <code>Export</code> button in the RStudio <code>Plots</code> tab and <code>Save as PDF...</code>. Change the <code>PDF size</code> to <code>24 x 32</code> to give a figure of appropriate size for the text labels.</p>
<div class="float">
<img src="img/mov10oe_cnetplot.png" width="800" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<p>Over-representation analysis is only a single type of functional analysis method. Let's look GeneSet Enrichment Analysis (GSEA), a functional class scoring method.</p>
<div class="float">
<img src="img/pathway_analysis.png" alt="Pathway analysis tools" />
<div class="figcaption">Pathway analysis tools</div>
</div>
</div>
</div>
<div id="gene-set-enrichment-analysis-gsea-2" class="section level2 hasAnchor" number="10.3">
<h2><span class="header-section-number">10.3</span> Gene set enrichment analysis (GSEA)<a href="functional-analysis-1.html#gene-set-enrichment-analysis-gsea-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://software.broadinstitute.org/gsea/index.jsp">GSEA</a>, most often uses the gene-level statistics or log2 fold changes for all genes from the differential expression results, then look to see whether gene sets for particular biological pathways are enriched among the large positive or negative fold changes.</p>
<div class="float">
<img src="img/gsea_theory.png" width="600" alt="Alt text" />
<div class="figcaption">Alt text</div>
</div>
<p>The hypothesis of GSEA is that although large changes in individual genes can have significant effects on pathways (and will be detected via ORA methods), weaker but coordinated changes in sets of functionally related genes (i.e., pathways) can also have significant effects. Thus, rather than setting an arbitrary threshold to identify 'significant genes', <strong>all genes are considered</strong> in the analysis. The gene-level statistics from the dataset are aggregated to generate a single pathway-level statistic and statistical significance of each pathway is reported. This type of analysis can be particularly helpful if the differential expression analysis only outputs a small list of significant DE genes.</p>
<div id="gsea-using-clusterprofiler-2" class="section level3 hasAnchor" number="10.3.1">
<h3><span class="header-section-number">10.3.1</span> GSEA using clusterProfiler<a href="functional-analysis-1.html#gsea-using-clusterprofiler-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Using the log2 fold changes obtained from the differential expression analysis for every gene, gene set enrichment analysis and pathway analysis can be performed using clusterProfiler tools.</p>
<p>For gene set or pathway analysis using clusterProfiler, coordinated differential expression over gene sets is tested instead of changes of individual genes. "Gene sets are pre-defined groups of genes, which are functionally related. Commonly used gene sets include those derived from KEGG pathways, Gene Ontology terms, MSigDB, Reactome, or gene groups that share some other functional annotations, etc. Consistent perturbations over such gene sets frequently suggest mechanistic changes" [<a href="pathway_tools.pdf">1</a>].</p>
<p>Extract and name the fold changes:</p>
<div class="sourceCode" id="cb307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb307-1"><a href="functional-analysis-1.html#cb307-1" tabindex="-1"></a><span class="do">## Extract the foldchanges</span></span>
<span id="cb307-2"><a href="functional-analysis-1.html#cb307-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>log2FoldChange</span>
<span id="cb307-3"><a href="functional-analysis-1.html#cb307-3" tabindex="-1"></a></span>
<span id="cb307-4"><a href="functional-analysis-1.html#cb307-4" tabindex="-1"></a><span class="do">## Name each fold change with the corresponding Entrez ID</span></span>
<span id="cb307-5"><a href="functional-analysis-1.html#cb307-5" tabindex="-1"></a><span class="fu">names</span>(foldchanges) <span class="ot">&lt;-</span> res_tableOE_tb<span class="sc">$</span>gene</span></code></pre></div>
<p>Next we need to order the fold changes in decreasing order. To do this we'll use the <code>sort()</code> function, which takes a vector as input. This is in contrast to Tidyverse's <code>arrange()</code>, which requires a data frame.</p>
<div class="sourceCode" id="cb308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb308-1"><a href="functional-analysis-1.html#cb308-1" tabindex="-1"></a><span class="do">## Sort fold changes in decreasing order</span></span>
<span id="cb308-2"><a href="functional-analysis-1.html#cb308-2" tabindex="-1"></a>foldchanges <span class="ot">&lt;-</span> <span class="fu">sort</span>(foldchanges, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)</span>
<span id="cb308-3"><a href="functional-analysis-1.html#cb308-3" tabindex="-1"></a></span>
<span id="cb308-4"><a href="functional-analysis-1.html#cb308-4" tabindex="-1"></a><span class="fu">head</span>(foldchanges)</span></code></pre></div>
<p>We can explore the enrichment of BP Gene Ontology terms using gene set enrichment analysis:</p>
<div class="sourceCode" id="cb309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb309-1"><a href="functional-analysis-1.html#cb309-1" tabindex="-1"></a></span>
<span id="cb309-2"><a href="functional-analysis-1.html#cb309-2" tabindex="-1"></a><span class="co"># GSEA using gene sets associated with BP Gene Ontology terms</span></span>
<span id="cb309-3"><a href="functional-analysis-1.html#cb309-3" tabindex="-1"></a></span>
<span id="cb309-4"><a href="functional-analysis-1.html#cb309-4" tabindex="-1"></a>gseaGO <span class="ot">&lt;-</span> clusterProfiler<span class="sc">::</span><span class="fu">gseGO</span>(</span>
<span id="cb309-5"><a href="functional-analysis-1.html#cb309-5" tabindex="-1"></a>  <span class="at">geneList =</span> foldchanges,</span>
<span id="cb309-6"><a href="functional-analysis-1.html#cb309-6" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb309-7"><a href="functional-analysis-1.html#cb309-7" tabindex="-1"></a>  <span class="at">keyType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb309-8"><a href="functional-analysis-1.html#cb309-8" tabindex="-1"></a>  <span class="at">eps =</span> <span class="dv">0</span>,</span>
<span id="cb309-9"><a href="functional-analysis-1.html#cb309-9" tabindex="-1"></a>  <span class="at">minGSSize =</span> <span class="dv">20</span>,</span>
<span id="cb309-10"><a href="functional-analysis-1.html#cb309-10" tabindex="-1"></a>  <span class="at">maxGSSize =</span> <span class="dv">300</span>,</span>
<span id="cb309-11"><a href="functional-analysis-1.html#cb309-11" tabindex="-1"></a>  <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,</span>
<span id="cb309-12"><a href="functional-analysis-1.html#cb309-12" tabindex="-1"></a>  <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb309-13"><a href="functional-analysis-1.html#cb309-13" tabindex="-1"></a>  <span class="at">verbose =</span> <span class="cn">TRUE</span>,</span>
<span id="cb309-14"><a href="functional-analysis-1.html#cb309-14" tabindex="-1"></a>  <span class="at">OrgDb =</span> <span class="st">&quot;org.Hs.eg.db&quot;</span>,</span>
<span id="cb309-15"><a href="functional-analysis-1.html#cb309-15" tabindex="-1"></a>  <span class="at">by =</span> <span class="st">&quot;fgsea&quot;</span></span>
<span id="cb309-16"><a href="functional-analysis-1.html#cb309-16" tabindex="-1"></a>)</span>
<span id="cb309-17"><a href="functional-analysis-1.html#cb309-17" tabindex="-1"></a></span>
<span id="cb309-18"><a href="functional-analysis-1.html#cb309-18" tabindex="-1"></a>gseaGO_results <span class="ot">&lt;-</span> gseaGO<span class="sc">@</span>result</span>
<span id="cb309-19"><a href="functional-analysis-1.html#cb309-19" tabindex="-1"></a></span>
<span id="cb309-20"><a href="functional-analysis-1.html#cb309-20" tabindex="-1"></a><span class="fu">goplot</span>(gseaGO)</span>
<span id="cb309-21"><a href="functional-analysis-1.html#cb309-21" tabindex="-1"></a></span>
<span id="cb309-22"><a href="functional-analysis-1.html#cb309-22" tabindex="-1"></a><span class="fu">gseaplot2</span>(gseaGO, <span class="at">geneSetID =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)</span></code></pre></div>
<p>We can also use our homemade GO enrichment analysis. To do this we need to load the library GOenrichment:</p>
<pre><code>
# Uncomment the following if you haven&#39;t yet installed GOenrichment.

# devtools::install_github(&quot;gurinina/GOenrichment&quot;)

library(GOenrichment)
library(visNetwork)
library(igraph)

ls(&quot;package:GOenrichment&quot;)</code></pre>
<p>One of the problems with GO enrichment analysis is that the GO annotations are in constant flux.</p>
<p>Here we can use the GO annotations in <code>hGOBP.gmt</code> (downloaded recently) to run GSEA using the <code>fgsea</code> package to run GSEA:</p>
<pre><code>fgseaRes &lt;-  fgsea::fgseaSimple(pathways = hGOBP.gmt,stats=foldchanges,nperm=1000,maxSize = 300,minSize = 20)

fgsea &lt;- data.frame(fgseaRes,stringsAsFactors = F)

w = which(fgsea$ES &gt; 0)

fposgsea &lt;- fgsea[w,]

fposgsea &lt;- fposgsea %&gt;% arrange(padj)
</code></pre>
<p>We are going to compare these results to runing the GO enrichment function <code>runGORESP</code>. <code>runGORESP</code> uses over-representation analysis to identify enriched GO terms, so we need to define a significance cutoff for the <code>querySet</code>.</p>
<pre><code>args(runGORESP)

?runGORESP

# we&#39;ll define our significance cutoff as 0.58, corresponding to 1.5x change.

# `runGORESP` requires a matrix, so we can turn foldchanges into a matrix using `cbind`:
matx &lt;- cbind(foldchanges,foldchanges)

hresp = GOenrichment::runGORESP(fdrThresh = 0.2,mat=matx,
coln=1,curr_exp = colnames(matx)[1], sig = 0.58,
bp_input = hGOBP.gmt,go_input = NULL,minSetSize = 20,
maxSetSize = 300)

names(hresp$edgeMat)
names(hresp$enrichInfo)
# head(hresp$enrichInfo[,c(2,3,4,5,10)])</code></pre>
<p>Let's check the overlap between the enriched terms found using <code>runGORESP</code> and those found using <code>fgseaSimple</code> as they used the same GO term libraries:</p>
<pre><code>w = which(fposgsea$padj &lt;= 0.2)

lens &lt;- length(intersect(fposgsea$pathway[w],hresp$enrichInfo$term))

length(w)
dim(hresp$enrichInfo)

percent_overlap &lt;- lens/nrow(hresp$enrichInfo)*100

percent_overlap</code></pre>
<p>80%, that's very good, especially because we are using two different GO enrichment methods, over-representation analysis and GSEA. The overlap between these enrichment and the ones using the other GO enrichment tools will be very small because of the differences in the GO annotation libraries.</p>
<p>Now to set up the results for viewing in a network, we use the function <code>visSetup</code>, which creates a set of nodes and edges in the network, where nodes are GO terms (node size proportional to FDR score) and edges represent the overlap between GO terms (proportional to edge width). This network analysis is based on <a href="https://cytoscape.org/">Cytoscape</a>, an open source bioinformatics software platform for visualizing molecular interaction networks.</p>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb314-1"><a href="functional-analysis-1.html#cb314-1" tabindex="-1"></a>vis <span class="ot">=</span> <span class="fu">visSetup</span>(hresp<span class="sc">$</span>enrichInfo,hresp<span class="sc">$</span>edgeMat)</span>
<span id="cb314-2"><a href="functional-analysis-1.html#cb314-2" tabindex="-1"></a><span class="fu">names</span>(vis)</span></code></pre></div>
<pre><code>## [1] &quot;nodes&quot; &quot;edges&quot;</code></pre>
<p>Now we use runNetwork to view the map:</p>
</div>
</div>
</div>
<div class="intro-to-R_footer">
  
</div>
<script>

/* update total correct if #webex-total_correct exists */
update_total_correct = function() {
  if (t = document.getElementById("webex-total_correct")) {
    t.innerHTML =
      document.getElementsByClassName("webex-correct").length + " of " +
      document.getElementsByClassName("webex-solveme").length + " correct";
  }
}

/* webex-solution button toggling function */
b_func = function() {
  var cl = this.parentElement.classList;
  if (cl.contains('open')) {
    cl.remove("open");
  } else {
    cl.add("open");
  }
}

/* function for checking solveme answers */
solveme_func = function(e) {
  var real_answers = JSON.parse(this.dataset.answer);
  var my_answer = this.value;
  var cl = this.classList;
  if (cl.contains("ignorecase")) {
    my_answer = my_answer.toLowerCase();
  }
  if (cl.contains("nospaces")) {
    my_answer = my_answer.replace(/ /g, "")
  }
  
  if (my_answer !== "" & real_answers.includes(my_answer)) {
    cl.add("webex-correct");
  } else {
    cl.remove("webex-correct");
  }

  // match numeric answers within a specified tolerance
  if(this.dataset.tol > 0){
    var tol = JSON.parse(this.dataset.tol);  
    var matches = real_answers.map(x => Math.abs(x - my_answer) < tol)
    if (matches.reduce((a, b) => a + b, 0) > 0) {
      cl.add("webex-correct");
    } else {
      cl.remove("webex-correct");
    }  
  }

  // added regex bit
  if (cl.contains("regex")){
    answer_regex = RegExp(real_answers.join("|"))
    if (answer_regex.test(my_answer)) {
      cl.add("webex-correct");
    }  
  }
  
  update_total_correct();
}

window.onload = function() {
  /* set up solution buttons */
  var buttons = document.getElementsByTagName("button");

  for (var i = 0; i < buttons.length; i++) {
    if (buttons[i].parentElement.classList.contains('webex-solution')) {
      buttons[i].onclick = b_func;
    }
  }
  
  /* set up webex-solveme inputs */
  var solveme = document.getElementsByClassName("webex-solveme");

  for (var i = 0; i < solveme.length; i++) {
    /* make sure input boxes don't auto-anything */
    solveme[i].setAttribute("autocomplete","off");
    solveme[i].setAttribute("autocorrect", "off");
    solveme[i].setAttribute("autocapitalize", "off"); 
    solveme[i].setAttribute("spellcheck", "false");
    solveme[i].value = "";
    
    /* adjust answer for ignorecase or nospaces */
    var cl = solveme[i].classList;
    var real_answer = solveme[i].dataset.answer;
    if (cl.contains("ignorecase")) {
      real_answer = real_answer.toLowerCase();
    }
    if (cl.contains("nospaces")) {
      real_answer = real_answer.replace(/ /g, "");
    }
    solveme[i].dataset.answer = real_answer;
    
    /* attach checking function */
    solveme[i].onkeyup = solveme_func;
    solveme[i].onchange = solveme_func;
  }
  
  update_total_correct();
}

</script>
            </section>

          </div>
        </div>
      </div>
<a href="codebook.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="this-has-to-be-commented-out-in-epub-and-pdf-formats-because-the-dynamic-network-cannot-be-displayed.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown/edit/main/inst/examples/lessons/08-GO_enrichment_analysis.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["RNA-seq-analysis.pdf", "RNA-seq-analysis.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
},
"fig_height": 5.3,
"fig_width": 8.5
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
