---
title: "DGE_analysis_exercises_answers"
output: html_document
date: "2023-08-07"
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
total points = 30

# Using DESeq2 for gene-level differential expression analysis

- The metadata below describes an RNA-seq analysis experiment, in which the metadata table below and associated count matrix have been loaded into R as `meta` and `counts`, respectively. Additionally, all of the appropriate libraries have been loaded for you. Use the information in the table to answer the following questions.  

	**meta**
	
	| |Genotype	|Celltype	|Batch|
	| ------ | ------- | -------- | --- |
	|sample1	|Wt	|typeA	|second |
	|sample2	|Wt	|typeA	|second|
	|sample3	|Wt	|typeA	|first|
	|sample4	|KO	|typeA	|first|
	|sample5	|KO	|typeA	|first|
	|sample6	|KO	|typeA	|second|
	|sample7	|Wt	|typeB	|second|
	|sample8	|Wt	|typeB	|first|
	|sample9	|Wt	|typeB	|second|
	|sample10	|KO	|typeB	|first|
	|sample11	|KO	|typeB	|first|
	|sample12	|KO	|typeB	|second|


	**NOTE: This is an exercise in thinking about running DESeq2. You do not need to run any code in R/RStudio. Refer to the materials/lessons from class to answer the following questions.** 

	**a.** Reorder the columns of the `counts` dataset such that `rownames(meta) == colnames(counts)`. points = +2
	
```{r, eval=FALSE}
idx <- match(rownames(meta), colnames(counts))

counts <- counts[,idx]

all(colnames(counts)==rownames(meta))

```
	
	**b.** Provide the line of code used to create a DESeqDataSet object called `dds` in which `Genotype` is the factor of interest and `Celltype` and `Batch` are other contributing sources of variation in your data. points = +1
	
```{r, eval=FALSE}
dds <- DESeqDataSetFromMatrix(countData = counts, colData = meta, design = ~ Celltype + Batch + Genotype)
```

	**c.** Provide the line of code required to run DESeq2 on `dds`. points = +1
	
```{r, eval=FALSE}
dds <- DESeq(dds)
```
	
	**d.** Provide the line of code to create a dispersion plot. points = +1
	
```{r, eval=FALSE}
plotDispEsts(dds)
```
	
	**e.** Provide the line of code to return the results of a Wald test comparison for `Celltype` categories `typeA` versus `typeB` (i.e the fold changes reported should reflect gene expression changes relative to `typeB`). points = +2
	 	
```{r, eval=FALSE}

contrast <- c("Celltype","typeA","typeB")

res <- results(dds, contrast = contrast)
```
	
	**f.** Provide the line of code to return the results with log2 fold change shrinkage performed. points = +1
	
```{r, eval=FALSE}

res_shrunken <- lfcShrink(dds = dds, coef = "Celltype_typeA_vs_typeB", res = res,type = "apeglm")

```
	
	**g.** Provide the line of code to write the results of the Wald test with shrunken log2 fold changes to file. points = +1
	
```{r, eval=FALSE}

write.table(res_shrunken, file="data/res_shrunkeg.txt", sep="\t")

```
	
	**h.** Provide the line of code to subset the results to return those genes with adjusted p-value < 0.05 and logFold2Change > 1. points = +2
	
```{r, eval=FALSE}

padj.cutoff <- 0.05
lfc.cutoff <- 1

library(tidyverse)

res_shrunken_tb <- res_shrunken %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble()


sig <- res_shrunken_tb %>%
        dplyr::filter(padj < padj.cutoff & log2FoldChange > lfc.cutoff)

```
	
# Working with the DESeq2 results table

- Using the DESeq2_expression_analysis.Rmd script change the thresholds for adjusted p-value and log2 fold change to the following values:
 
```{r}
	padj.cutoff <- 0.01
	
	lfc.cutoff <- 1.5
```
	
	Using these new cutoffs, perform the following steps:

	**a.** Subset `res_tableOE` to only return those rows that meet the criteria we specified above (adjusted p-values < 0.01 and log fold changes >1.5). Save the subsetted table to a data frame called `sig_table_hw_oe`. You'll have to load the res_tableOE first. Write the code below: points = +1
	
```{r}

res_tableOE <-  readRDS("data/res_tableOE.rds")

res_tableOE_tb <- res_tableOE %>%
  data.frame() %>%
  rownames_to_column(var="gene") %>%
  as_tibble()


sig_table_hw_oe <- res_tableOE_tb %>%
        dplyr::filter(padj < padj.cutoff & abs(log2FoldChange) > lfc.cutoff)
```
	
	
	**b.** Get the gene names from `sig_table_hw_oe` and save them to a vector called `sigOE_hw`. Write the code below: points = +1
	
```{r}

sigOE_hw <- sig_table_hw_oe$gene
```
	
	**c.** Write the `sigOE_hw` vector of gene names to a file called `sigOE_hw.txt` using the `write()` function. Ensure the genes are listed in a single column. Write the code below. points = +2
	
```{r}

write(sigOE_hw ,file = "data/sigOE_hw.txt",
ncolumns = 1,
append = FALSE, sep = "\t")


```

	 
# Visualizing Results

- For the genes that are differentially expressed in the knockdown versus control comparison (`sigKD`), plot an expression heatmap using normalized counts and `pheatmap()` following the instructions below. Write the code you would use to create the heatmap.

	**a.** The heatmap should only include control and knockdown samples. points = +4
	
```{r}
normalized_counts <- read.delim("data/normalized_counts.txt",row.names = 1)

normalized_counts <- normalized_counts %>% 
  data.frame() %>%
  rownames_to_column(var="gene") %>% 
  as_tibble()


meta <- read.table("meta/Mov10_full_meta.txt", header=T, row.names=1)

mov10_meta <- meta %>% 
  rownames_to_column(var="samplename") %>% 
  as_tibble()

sigKD <- readRDS("data/sigKD.rds")

colnames(normalized_counts)

## Extract normalized expression for significant genes


norm_KDsig <- normalized_counts[,c(1,2:4,5:6)] %>% 
              filter(gene %in% sigKD$gene) %>% 
	      data.frame() %>%
	      column_to_rownames(var = "gene") 


```
	
```{r}
### Annotate our heatmap (optional)


annotation <- mov10_meta %>% 
	dplyr::select(samplename, sampletype) %>% 
	data.frame(row.names = "samplename")


```


	**b.** Set up a heat.colors vector using a palette of your choice from brewer.pal (make sure it is different from the one used in class). points = +1
	
```{r}

### Set a color palette
library(RColorBrewer)
heat_colors <- brewer.pal(6, "BuGn")

```
	
	
	**c.** Plot the heatmap and scale expression values by row. points = +2
	
```{r}

library(pheatmap)

pheatmap(norm_KDsig, 
         color = heat_colors, 
         cluster_rows = T, 
         cluster_cols = F,
         show_rownames = F,
         scale = "row",
         annotation = annotation, 
         border_color = NA, 
         fontsize = 10,
         fontsize_row = 10, 
         height = 20)

```
	
# Use significant gene lists to find overlaps between the two comparisons 

- Using the original cutoff values, perform the following steps:

```{r}
	padj.cutoff <- 0.05

	lfc.cutoff <- 0.58
```

	**a.** Create separate vectors with gene names for up-regulated genes and down-regulated genes from `res_tableOE` and save as `up_OE` and `down_OE`, respectively. Write the code below:
points = +2
```{r}

w <-  which(res_tableOE$log2FoldChange > lfc.cutoff & res_tableOE$padj < padj.cutoff)
up_OE <- rownames(res_tableOE)[w]

w <-  which(res_tableOE$log2FoldChange < -lfc.cutoff & res_tableOE$padj < padj.cutoff)
down_OE <- rownames(res_tableOE)[w]

```
	
	**b.** Create separate vectors with gene names for up-regulated genes and down-regulated genes from `res_tableKD` and save as `up_KD` and `down_KD`, respectively. Write the code below:
points = +2	
```{r}

res_tableKD <- readRDS("data/res_tableKD.rds")
w <-  which(res_tableKD$log2FoldChange > lfc.cutoff)
up_KD <- rownames(res_tableKD)[w]

w <-  which(res_tableKD$log2FoldChange < -lfc.cutoff)
down_KD <- rownames(res_tableKD)[w]

```
	
	**c.** Test for overlaps between the lists:
	
	- How many, and which genes in `up_OE` are also in `down_KD`? points = +2
		
```{r}
int <- intersect(up_OE,down_KD)

int

length(int) #17
```

	- How many, and which genes in `up_KD` are also in `down_OE`? points = +2
		
```{r}

int <- intersect(up_KD,down_OE)

int

length(int) #13
```

	 

	

