# DGE analysis workflow

Differential expression analysis with DESeq2 involves multiple steps as displayed in the flowchart below in blue. Briefly, DESeq2 will:

**Full Workflow in DESeq2:**

Load Data: Start with raw count data and sample metadata.

Estimate Size Factors: Normalize counts across samples.

Estimate Raw Dispersions: Calculate the raw dispersion for each gene.

Fit the Global Trend: Fit a parametric or non-parametric global trend to model the relationship between mean counts and dispersion.

Shrink the Dispersion Estimates: Shrink the gene-wise dispersion estimates toward the values predicted by the global trend to generate more accurate values.

Fit the negative binomial using the final gene-wise dispersion estimates to obtain the log-fold changes in gene expression between conditions.


![Alt text](img/DESeq2_workflow_2018.png){ width=600 }

This final step in the differential expression analysis workflow of fitting the raw counts to the NB model and performing the statistical test for differentially expressed genes, is the step we care about. This is the step that determines whether the mean expression levels of different sample groups are significantly different.


![Alt text](img/de_theory.png){ width=600 }

The [DESeq2 paper](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8) was published in 2014, but the package is continually updated and available for use in R through Bioconductor. 

## Running DESeq2

Prior to performing the differential expression analysis, it is a good idea to know what **sources of variation** are present in your data, either by exploration during the QC and/or prior knowledge. Once you know the major sources of variation, you can remove them prior to analysis or control for them in the statistical model by including them in your **design formula**.

### Design formula

A design formula tells the statistical software the known sources of variation to control for, as well as, the factor of interest to test for during differential expression testing. For example, if you know that sex is a significant source of variation in your data, then sex should be included in your model. The design formula should have all of the factors in your metadata that account for major sources of variation in your data. The last factor entered in the formula should be the condition of interest. 

For example, suppose you have the following metadata:

![Alt text](img/meta_example.png){ width=300 }

If you want to examine the expression differences between treatments, and you know that major sources of variation include sex and age, then your design formula would be:

`design <- ~ sex + age + treatment`

The tilde (`~`) should always proceed your factors and tells DESeq2 to model the counts using the following formula. Note the **factors included in the design formula need to match the column names in the metadata**. In this case, the comparison between treatment and control in the results() function will still compare the two levels of condition, but after adjusting for sex and age effects. This helps ensure that any differences in gene expression are due to the biological condition (e.g., treatment vs. control) and not sex and age effects.

***

**Exercise** points = +3

1. Suppose you wanted to study the expression differences between the two age groups in the metadata shown above, and major sources of variation were `sex` and `treatment`, how would the design formula be written?

```{r, eval = FALSE}

# Your code here

```

```{r, results = "asis", echo = FALSE}

solutions(
  '# `DESeq2::results` defaults to the last factor in the formula, so your condition of interest should come last:\n
     ~ sex + treatment + age'
)

```

2. Based on our Mov10 metadata dataframe, which factors could we include in our design formula?

  - Ans: 

```{r, results = "asis", echo = FALSE}
solutions("sampletype, you could also use MOVexpr, but it is 
           redundant with sampletype. Sampletype is also 
           more descriptive.")
```


3. What would you do if you wanted to include a factor in your design formula that is not in your metadata?

  - Ans: 
  
```{r, results = "asis", echo = FALSE}

solutions(
  "Add the factor as an additional column in the metadata points"
)

```


***

### MOV10 DE analysis 

Now that we know how to specify the model to DESeq2, we can run the differential expression pipeline on the **raw counts**. 

**To get our differential expression results from our raw count data, we only need to run 2 lines of code!**

First we create a DESeqDataSet as we did in the ['Count normalization'](https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html#2-create-deseq2-object) lesson and specify the location of our raw counts and metadata, and input our design formula:

```{r}
## Create DESeq object
dds <- DESeqDataSetFromMatrix(countData = data, colData = meta, design = ~ sampletype)
```

Then, to run the actual differential expression analysis, we use a single call to the function `DESeq()`. 

```{r}
## Run analysis
dds <- DESeq(dds)
```

By re-assigning the results of the function back to the same variable name (`dds`), we can fill in the `slots` of our `DESeqDataSet` object.

![deseq1](img/deseq_obj2.png)

**Everything from normalization to linear modeling was carried out by the use of a single function!** This function will print out a message for the various steps it performs: 

```
estimating size factors
estimating dispersions
gene-wise dispersion estimates
mean-dispersion relationship
final dispersion estimates
fitting model and testing
``` 

## DESeq2 differential gene expression analysis workflow

With the 2 lines of code above, we just completed the workflow for the differential gene expression analysis with DESeq2. The steps in the analysis are output below:

![Alt text](img/deseq2_workflow_separate.png){ width=200 }

We will be taking a detailed look at each of these steps to better understand how DESeq2 is performing the statistical analysis and what metrics we should examine to explore the quality of our analysis.

### Step 1: Estimate size factors

The first step in the differential expression analysis is to estimate the size factors, which is exactly what we already did to normalize the raw counts. 

![Alt text](img/deseq2_workflow_separate_sf.png){ width=200 }

DESeq2 will automatically estimate the size factors when performing the differential expression analysis. However, if you have already generated the size factors using `estimateSizeFactors()`, as we did earlier, then DESeq2 will use these values.

To normalize the count data, DESeq2 calculates size factors for each sample using the *median of ratios method* discussed previously in the Count normalization (Chapter \@ref(count-normalization)) lesson.

#### MOV10 DE analysis: examining the size factors

Let's take a quick look at size factor values we have for each sample:

```{r}
## Check the size factors
sizeFactors(dds)

```
 
Take a look at the total number of reads for each sample:

```{r}
## Total number of raw counts per sample
colSums(counts(dds))
```

*How do the numbers correlate with the size factor?*


### Step 2: Estimate gene-wise dispersion

The next step in the differential expression analysis is the estimation of gene-wise dispersions. Before we get into the details, we should have a good idea about what dispersion is referring to in DESeq2.

![Alt text](img/deseq2_workflow_separate_dis.png){ width=200 }

**What is dispersion?** 

Dispersion is a measure of spread or variability in the data. Variance, standard deviation, IQR, among other measures, can all be used to measure dispersion. However, DESeq2 uses a specific measure of dispersion (α) related to the mean (μ) and variance of the data: `Var = μ + α*μ^2`.  For genes with moderate to high count values, the square root of dispersion will be equal to the coefficient of variation (`σ / μ`). So 0.01 dispersion means 10% variation around the mean expected across biological replicates. 

**What does the DESeq2 dispersion represent?** 

The DESeq2 dispersion estimates are **inversely related to the mean** and **directly related to variance**. **Based on this relationship, the dispersion is higher for small mean counts and lower for large mean counts.** The dispersion estimates for genes with the same mean will differ only based on their variance. **Therefore, the dispersion estimates reflect the variance in gene expression for a given mean value.** 

The plot of mean versus variance in count data below shows the variance in gene expression increases with the mean expression (each black dot is a gene). Notice that the relationship between mean and variance is linear on the log scale, and for higher means, we could predict the variance relatively accurately given the mean. However, **for low mean counts, the variance estimates have a much larger spread; therefore, the dispersion estimates will differ much more between genes with small means**. 

![Alt text](img/deseq_mean_vs_variance.png){ width=600 }

**How does the dispersion relate to our model?** 

To accurately model sequencing counts, we need to generate accurate estimates of within-group variation (variation between replicates of the same sample group) for each gene. With only a few (3-6) replicates per group, the **estimates of variation for each gene are often unreliable** (due to the large differences in dispersion for genes with similar means). 

To address this problem, DESeq2 **shares information across genes** to generate more accurate estimates of variation based on the mean expression level of the gene using a method called 'shrinkage'. **DESeq2 assumes that genes with similar expression levels have similar dispersion.** 

**Estimating the dispersion for each gene separately:**

To estimate dispersion, DESeq2 uses the negative binomial (NB) distribution, which is a probability distribution that models the count data. The negative binomial distribution is a two-parameter distribution that describes the distribution of counts in RNA-Seq data. The two parameters are the mean expression level and the dispersion. The negative binomial distribution is used in DESeq2 because it is well-suited for modeling count data, which is often overdispersed (variance is greater than the mean:\

$$
Y_{ij} \sim \text{NB}\left(\mu_{ij} = \text{sizeFactor}_{j} \times \text{normalizedCounts}_{ij}, \alpha_i = \text{dispersion}\right)
$$

Where:

- \( Y_{ij}\) is the observed count for gene \( i \) in sample \( j \)

- \( \mu_{ij}\) is the mean normalized count for gene \( i \) in sample \( j \)

- \( \alpha_i \) is the dispersion parameter for gene \( i \) 

- \( sizeFactor{j} \) is the size factor for sample \( j \)


The first step in estimating dispersion is to get the dispersion estimates for each gene. The dispersion for each gene is estimated using maximum likelihood estimation (MLE) based on the negative binomial distribution. MLE can be thought about as **given the count values of the replicates, the most likely estimate of dispersion is calculated**, more specifically:

Log-Likelihood Function of the Negative Binomial Distribution
The log-likelihood function for the negative binomial distribution, which is used to estimate both the mean ($\mu$) and dispersion ($\alpha$) parameters, is given by:

$$
\log L(Y_{ij}| \mu_ij, \alpha_i) = \log \left( \frac{\Gamma(Y_{ij}+ \frac{1}{\alpha_i})}{\Gamma(Y_ij+1) \Gamma(\frac{1}{\alpha_i})} \right) + Y_{ij}\log \left( \frac{\alpha_i \mu_{ij}}{1 + \alpha_i \mu_{ij}} \right) - \frac{1}{\alpha_i} \log(1 + \alpha_i \mu_{ij})
$$
Where:

- \( Y_{ij}\) is the observed count for gene \( i \) in sample \( j \)

- \( \mu_{ij}\) is the mean normalized count for gene \( i \) in sample \( j \)

- \( \alpha_i \) is the dispersion parameter for gene i

- \(⋅\Gamma \): Gamma function, which arises from the negative binomial distribution..

The dispersion is estimated by maximizing the likelihood of observing the counts $Y_{ij}$ based on the mean ($\mu_{ij}$) and variance $\mu_{ij} + \alpha \mu_{ij}^2$).

**The initial raw dispersion estimate:** 

The variance of $Y_{ij}$ in the Negative Binomial model is given by:

$$
V{ij} = \mu_{ij} + \alpha_{i} \mu_{ij}^2
$$
Where:
- \( V_{ij} \) is the observed variance in counts.for gene \( i \) in sample \( j \)
- \( \mu_{ij} \) is the mean normalized count for gene \( i \) in sample \( j \)
- \( \alpha_i\) is the dispersion parameter for gene \( i \).

This equation shows how the variance increases as the mean increases in the presence of dispersion $ \alpha $. Rearranging the equation, we can solve for the initial dispersion estimate for the maximum likelihood estimation (MLE) of the dispersion:

$$
\alpha_{i} = \dfrac{V_{ij} - \mu_{ij}}{\mu_{ij}^2 \vphantom{\dfrac{A}{B}}}
$$


Where:
- \( V_{ij} \) is the observed variance of gene \( i \) in sample \( j \)
- \( \mu_{ij} \) the expected (or mean) normalized count for gene \( i \) in sample \( j \)
- \( \alpha_{i} \) is the initial estimate of the dispersion parameter for gene \( i \).which represents the degree of overdispersion (the extent to which the variance exceeds the mean).


![Alt text](img/deseq2_workflow_separate_dis.png){ width=200 }

### Step 3: Fit curve to gene-wise dispersion estimates

**Fitting the Global Dispersion Trend:** 

Once the initial gene-wise dispersion estimates ($\alpha_{i}$) are obtained, a global trend is fit to model how dispersion typically changes as a function of the mean expression ($\mu_{i}$) across all samples. This trend is used to account for the general pattern that genes with low mean expression tend to have higher dispersion, while highly expressed genes have lower dispersion.

The global trend represents the expected value of the dispersion given the expression values of the genes. The idea behind fitting a curve to the data is that different genes will have different scales of biological variability, but, over all genes, there will be a distribution of reasonable estimates of dispersion.

This trend is used to account for the general pattern that genes with low mean expression tend to have higher dispersion, while highly expressed genes have lower dispersion. This step is crucial because lowly expressed genes tend to have more variability, and highly expressed genes tend to have less variability.

![Alt text](img/deseq2_workflow_separate_fit.png){ width=200 }

This global trend is displayed as a red line in the figure below, which plots the estimate for the **expected dispersion value for genes of a given expression strength**. Each black dot is a gene with an associated mean expression level and maximum likelihood estimation (MLE) of the dispersion (Step 1).

![Alt text](img/deseq_dispersion1.png){ width=400 }


![Alt text](img/deseq2_workflow_separate_fit.png){ width=200 }

The parametric trend describes the relationship between the mean count and dispersion in RNA-Seq data. It is given by the following formula:

$$\alpha(\mu) = \alpha_0 + \frac{\alpha_1}{\mu}$$


Where:
- \( \alpha(\mu) \) is the dispersion as a function of the mean count \( \mu \).
- \( \alpha_0 \) is the baseline dispersion.
- \( \alpha_1 \) controls how the dispersion changes as the mean count decreases.

This formula assumes that dispersion decreases as the mean count increases, which is typically seen in RNA-Seq data. The dispersion trend is fit to the raw dispersion estimates for each gene. Fitting the trend involves minimizing the differences between the raw dispersion estimates and the modeled values from the trend:

The parametric trend is fitted using non-linear regression to estimate the parameters $\alpha_0$ and $\alpha_1$ that best describe the relationship between the raw dispersion estimates and the mean counts.

Non-linear least squares regression is used to minimize the difference between the observed raw dispersion estimates and the values predicted by the parametric model. The objective is to fit the curve

The trend helps capture the overall relationship between dispersion and mean counts across all genes. This is important because it allows DESeq2 to borrow information across genes to estimate the dispersion more accurately. The dispersion estimates for each gene are then shrunken towards the fitted curve to generate the final dispersion estimates. 

### Step 4: Shrink gene-wise dispersion estimates toward the values predicted by the curve

To further improve the reliability of dispersion estimates, DESeq2 applies **empirical Bayes shrinkage**, which shrinks the gene-specific dispersion estimates toward the global trend to get the final dispersion estimates. This ensures that genes with low or moderate counts, where estimates are less reliable, have more stable dispersions.

![Alt text](img/deseq2_workflow_separate_shr.png){ width=200 }

The global trend allows for more accurate identification of differentially expressed genes when sample sizes are small, and the strength of the shrinkage for each gene depends on :
	
- how close gene dispersions are from the curve 
- sample size (more samples = less shrinkage)

**This shrinkage method is particularly important to reduce false positives in the differential expression analysis.** Genes with low dispersion estimates are shrunken towards the curve, and the more accurate, higher shrunken values are output for fitting of the model and differential expression testing.

Dispersion estimates that are slightly above the curve are also shrunk toward the curve for better dispersion estimation; however, genes with extremely high dispersion values are not. This is due to the likelihood that the gene does not follow the modeling assumptions and has higher variability than others for biological or technical reasons and therefore is likely to be an outlier. Shrinking the values toward the curve could result in false positives, so these values are not shrunken. These dispersion outliers are shown surrounded by blue circles below. 

![Alt text](img/deseq_dispersion2.png){ width=600 }

**This is a good plot to examine to ensure your data is a good fit for the DESeq2 model.** You expect your data to generally scatter around the curve, with the dispersion decreasing with increasing mean expression levels. If you see a cloud or different shapes, then you might want to explore your data more to see if you have contamination (mitochondrial, etc.) or outlier samples.  Note how much shrinkage you get across the whole range of means in the `plotDispEsts()` plot for any experiment with low degrees of freedom.

Examples of **worrisome dispersion plots** are shown below:

![Alt text](img/bad_dispersion1.png){ width=600 }

![Alt text](img/bad_dispersion2.png){ width=600 }


#### MOV10 DE analysis: exploring the dispersion estimates and assessing model fit

Let's take a look at the dispersion estimates for our MOV10 data:

```{r}
## Plot dispersion estimates
plotDispEsts(dds)
```

 
**Since we have a small sample size, for many genes we see quite a bit of shrinkage. Do you think our data are a good fit for the model?**

***







